// oci-ons/lib/request/list-topics-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListTopicsRequest = void 0;
var ListTopicsRequest;
(function (ListTopicsRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["Timecreated"] = "TIMECREATED";
        SortBy["Lifecyclestate"] = "LIFECYCLESTATE";
    })(SortBy = ListTopicsRequest.SortBy || (ListTopicsRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListTopicsRequest.SortOrder || (ListTopicsRequest.SortOrder = {}));
})(ListTopicsRequest = exports.ListTopicsRequest || (exports.ListTopicsRequest = {}));
