// oci-objectstorage/lib/model/bucket.js
"use strict";
/**
 * Object Storage Service API
 * Common set of Object Storage and Archive Storage APIs for managing buckets, objects, and related resources.
For more information, see [Overview of Object Storage](/Content/Object/Concepts/objectstorageoverview.htm) and
[Overview of Archive Storage](/Content/Archive/Concepts/archivestorageoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: opc_casper_users_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Bucket = void 0;
var Bucket;
(function (Bucket) {
    let PublicAccessType;
    (function (PublicAccessType) {
        PublicAccessType["NoPublicAccess"] = "NoPublicAccess";
        PublicAccessType["ObjectRead"] = "ObjectRead";
        PublicAccessType["ObjectReadWithoutList"] = "ObjectReadWithoutList";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        PublicAccessType["UnknownValue"] = "UNKNOWN_VALUE";
    })(PublicAccessType = Bucket.PublicAccessType || (Bucket.PublicAccessType = {}));
    let StorageTier;
    (function (StorageTier) {
        StorageTier["Standard"] = "Standard";
        StorageTier["Archive"] = "Archive";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        StorageTier["UnknownValue"] = "UNKNOWN_VALUE";
    })(StorageTier = Bucket.StorageTier || (Bucket.StorageTier = {}));
    let Versioning;
    (function (Versioning) {
        Versioning["Enabled"] = "Enabled";
        Versioning["Suspended"] = "Suspended";
        Versioning["Disabled"] = "Disabled";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Versioning["UnknownValue"] = "UNKNOWN_VALUE";
    })(Versioning = Bucket.Versioning || (Bucket.Versioning = {}));
    let AutoTiering;
    (function (AutoTiering) {
        AutoTiering["Disabled"] = "Disabled";
        AutoTiering["InfrequentAccess"] = "InfrequentAccess";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        AutoTiering["UnknownValue"] = "UNKNOWN_VALUE";
    })(AutoTiering = Bucket.AutoTiering || (Bucket.AutoTiering = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    Bucket.getJsonObj = getJsonObj;
})(Bucket = exports.Bucket || (exports.Bucket = {}));
