// oci-objectstorage/lib/model/create-bucket-details.js
"use strict";
/**
 * Object Storage Service API
 * Common set of Object Storage and Archive Storage APIs for managing buckets, objects, and related resources.
For more information, see [Overview of Object Storage](/Content/Object/Concepts/objectstorageoverview.htm) and
[Overview of Archive Storage](/Content/Archive/Concepts/archivestorageoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: opc_casper_users_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateBucketDetails = void 0;
var CreateBucketDetails;
(function (CreateBucketDetails) {
    let PublicAccessType;
    (function (PublicAccessType) {
        PublicAccessType["NoPublicAccess"] = "NoPublicAccess";
        PublicAccessType["ObjectRead"] = "ObjectRead";
        PublicAccessType["ObjectReadWithoutList"] = "ObjectReadWithoutList";
    })(PublicAccessType = CreateBucketDetails.PublicAccessType || (CreateBucketDetails.PublicAccessType = {}));
    let StorageTier;
    (function (StorageTier) {
        StorageTier["Standard"] = "Standard";
        StorageTier["Archive"] = "Archive";
    })(StorageTier = CreateBucketDetails.StorageTier || (CreateBucketDetails.StorageTier = {}));
    let Versioning;
    (function (Versioning) {
        Versioning["Enabled"] = "Enabled";
        Versioning["Disabled"] = "Disabled";
    })(Versioning = CreateBucketDetails.Versioning || (CreateBucketDetails.Versioning = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CreateBucketDetails.getJsonObj = getJsonObj;
})(CreateBucketDetails = exports.CreateBucketDetails || (exports.CreateBucketDetails = {}));
