// oci-objectstorage/lib/model/index.js
"use strict";
/**
 * Object Storage Service API
 * Common set of Object Storage and Archive Storage APIs for managing buckets, objects, and related resources.
For more information, see [Overview of Object Storage](/Content/Object/Concepts/objectstorageoverview.htm) and
[Overview of Archive Storage](/Content/Archive/Concepts/archivestorageoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: opc_casper_users_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestSummary = exports.WorkRequestResourceMetadataKey = exports.WorkRequestResource = exports.WorkRequestLogEntry = exports.WorkRequestError = exports.WorkRequest = exports.UpdateRetentionRuleDetails = exports.UpdateObjectStorageTierDetails = exports.UpdateNamespaceMetadataDetails = exports.UpdateBucketDetails = exports.StorageTier = exports.SSECustomerKeyDetails = exports.RetentionRuleSummary = exports.RetentionRuleDetails = exports.RetentionRuleCollection = exports.RetentionRule = exports.RestoreObjectsDetails = exports.ReplicationSource = exports.ReplicationPolicySummary = exports.ReplicationPolicy = exports.RenameObjectDetails = exports.ReencryptObjectDetails = exports.PutObjectLifecyclePolicyDetails = exports.PreauthenticatedRequestSummary = exports.PreauthenticatedRequest = exports.PatternDetails = exports.ObjectVersionSummary = exports.ObjectVersionCollection = exports.ObjectSummary = exports.ObjectNameFilter = exports.ObjectLifecycleRule = exports.ObjectLifecyclePolicy = exports.NamespaceMetadata = exports.MultipartUploadPartSummary = exports.MultipartUpload = exports.ListObjects = exports.Duration = exports.CreateRetentionRuleDetails = exports.CreateReplicationPolicyDetails = exports.CreatePreauthenticatedRequestDetails = exports.CreateMultipartUploadDetails = exports.CreateBucketDetails = exports.CopyObjectDetails = exports.CommitMultipartUploadPartDetails = exports.CommitMultipartUploadDetails = exports.BucketSummary = exports.Bucket = exports.ArchivalState = void 0;
const ArchivalState = __importStar(require("./archival-state"));
exports.ArchivalState = ArchivalState.ArchivalState;
const Bucket = __importStar(require("./bucket"));
exports.Bucket = Bucket.Bucket;
const BucketSummary = __importStar(require("./bucket-summary"));
exports.BucketSummary = BucketSummary.BucketSummary;
const CommitMultipartUploadDetails = __importStar(require("./commit-multipart-upload-details"));
exports.CommitMultipartUploadDetails = CommitMultipartUploadDetails.CommitMultipartUploadDetails;
const CommitMultipartUploadPartDetails = __importStar(require("./commit-multipart-upload-part-details"));
exports.CommitMultipartUploadPartDetails = CommitMultipartUploadPartDetails.CommitMultipartUploadPartDetails;
const CopyObjectDetails = __importStar(require("./copy-object-details"));
exports.CopyObjectDetails = CopyObjectDetails.CopyObjectDetails;
const CreateBucketDetails = __importStar(require("./create-bucket-details"));
exports.CreateBucketDetails = CreateBucketDetails.CreateBucketDetails;
const CreateMultipartUploadDetails = __importStar(require("./create-multipart-upload-details"));
exports.CreateMultipartUploadDetails = CreateMultipartUploadDetails.CreateMultipartUploadDetails;
const CreatePreauthenticatedRequestDetails = __importStar(require("./create-preauthenticated-request-details"));
exports.CreatePreauthenticatedRequestDetails = CreatePreauthenticatedRequestDetails.CreatePreauthenticatedRequestDetails;
const CreateReplicationPolicyDetails = __importStar(require("./create-replication-policy-details"));
exports.CreateReplicationPolicyDetails = CreateReplicationPolicyDetails.CreateReplicationPolicyDetails;
const CreateRetentionRuleDetails = __importStar(require("./create-retention-rule-details"));
exports.CreateRetentionRuleDetails = CreateRetentionRuleDetails.CreateRetentionRuleDetails;
const Duration = __importStar(require("./duration"));
exports.Duration = Duration.Duration;
const ListObjects = __importStar(require("./list-objects"));
exports.ListObjects = ListObjects.ListObjects;
const MultipartUpload = __importStar(require("./multipart-upload"));
exports.MultipartUpload = MultipartUpload.MultipartUpload;
const MultipartUploadPartSummary = __importStar(require("./multipart-upload-part-summary"));
exports.MultipartUploadPartSummary = MultipartUploadPartSummary.MultipartUploadPartSummary;
const NamespaceMetadata = __importStar(require("./namespace-metadata"));
exports.NamespaceMetadata = NamespaceMetadata.NamespaceMetadata;
const ObjectLifecyclePolicy = __importStar(require("./object-lifecycle-policy"));
exports.ObjectLifecyclePolicy = ObjectLifecyclePolicy.ObjectLifecyclePolicy;
const ObjectLifecycleRule = __importStar(require("./object-lifecycle-rule"));
exports.ObjectLifecycleRule = ObjectLifecycleRule.ObjectLifecycleRule;
const ObjectNameFilter = __importStar(require("./object-name-filter"));
exports.ObjectNameFilter = ObjectNameFilter.ObjectNameFilter;
const ObjectSummary = __importStar(require("./object-summary"));
exports.ObjectSummary = ObjectSummary.ObjectSummary;
const ObjectVersionCollection = __importStar(require("./object-version-collection"));
exports.ObjectVersionCollection = ObjectVersionCollection.ObjectVersionCollection;
const ObjectVersionSummary = __importStar(require("./object-version-summary"));
exports.ObjectVersionSummary = ObjectVersionSummary.ObjectVersionSummary;
const PatternDetails = __importStar(require("./pattern-details"));
exports.PatternDetails = PatternDetails.PatternDetails;
const PreauthenticatedRequest = __importStar(require("./preauthenticated-request"));
exports.PreauthenticatedRequest = PreauthenticatedRequest.PreauthenticatedRequest;
const PreauthenticatedRequestSummary = __importStar(require("./preauthenticated-request-summary"));
exports.PreauthenticatedRequestSummary = PreauthenticatedRequestSummary.PreauthenticatedRequestSummary;
const PutObjectLifecyclePolicyDetails = __importStar(require("./put-object-lifecycle-policy-details"));
exports.PutObjectLifecyclePolicyDetails = PutObjectLifecyclePolicyDetails.PutObjectLifecyclePolicyDetails;
const ReencryptObjectDetails = __importStar(require("./reencrypt-object-details"));
exports.ReencryptObjectDetails = ReencryptObjectDetails.ReencryptObjectDetails;
const RenameObjectDetails = __importStar(require("./rename-object-details"));
exports.RenameObjectDetails = RenameObjectDetails.RenameObjectDetails;
const ReplicationPolicy = __importStar(require("./replication-policy"));
exports.ReplicationPolicy = ReplicationPolicy.ReplicationPolicy;
const ReplicationPolicySummary = __importStar(require("./replication-policy-summary"));
exports.ReplicationPolicySummary = ReplicationPolicySummary.ReplicationPolicySummary;
const ReplicationSource = __importStar(require("./replication-source"));
exports.ReplicationSource = ReplicationSource.ReplicationSource;
const RestoreObjectsDetails = __importStar(require("./restore-objects-details"));
exports.RestoreObjectsDetails = RestoreObjectsDetails.RestoreObjectsDetails;
const RetentionRule = __importStar(require("./retention-rule"));
exports.RetentionRule = RetentionRule.RetentionRule;
const RetentionRuleCollection = __importStar(require("./retention-rule-collection"));
exports.RetentionRuleCollection = RetentionRuleCollection.RetentionRuleCollection;
const RetentionRuleDetails = __importStar(require("./retention-rule-details"));
exports.RetentionRuleDetails = RetentionRuleDetails.RetentionRuleDetails;
const RetentionRuleSummary = __importStar(require("./retention-rule-summary"));
exports.RetentionRuleSummary = RetentionRuleSummary.RetentionRuleSummary;
const SSECustomerKeyDetails = __importStar(require("./s-se-customer-key-details"));
exports.SSECustomerKeyDetails = SSECustomerKeyDetails.SSECustomerKeyDetails;
const StorageTier = __importStar(require("./storage-tier"));
exports.StorageTier = StorageTier.StorageTier;
const UpdateBucketDetails = __importStar(require("./update-bucket-details"));
exports.UpdateBucketDetails = UpdateBucketDetails.UpdateBucketDetails;
const UpdateNamespaceMetadataDetails = __importStar(require("./update-namespace-metadata-details"));
exports.UpdateNamespaceMetadataDetails = UpdateNamespaceMetadataDetails.UpdateNamespaceMetadataDetails;
const UpdateObjectStorageTierDetails = __importStar(require("./update-object-storage-tier-details"));
exports.UpdateObjectStorageTierDetails = UpdateObjectStorageTierDetails.UpdateObjectStorageTierDetails;
const UpdateRetentionRuleDetails = __importStar(require("./update-retention-rule-details"));
exports.UpdateRetentionRuleDetails = UpdateRetentionRuleDetails.UpdateRetentionRuleDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestResourceMetadataKey = __importStar(require("./work-request-resource-metadata-key"));
exports.WorkRequestResourceMetadataKey = WorkRequestResourceMetadataKey.WorkRequestResourceMetadataKey;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
