// oci-functions/lib/model/index.js
"use strict";
/**
 * Functions Service API
 * API for the Functions service.

 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateFunctionDetails = exports.UpdateApplicationDetails = exports.FunctionTraceConfig = exports.FunctionSummary = exports.Function = exports.CreateFunctionDetails = exports.CreateApplicationDetails = exports.ChangeApplicationCompartmentDetails = exports.ApplicationTraceConfig = exports.ApplicationSummary = exports.Application = void 0;
const Application = __importStar(require("./application"));
exports.Application = Application.Application;
const ApplicationSummary = __importStar(require("./application-summary"));
exports.ApplicationSummary = ApplicationSummary.ApplicationSummary;
const ApplicationTraceConfig = __importStar(require("./application-trace-config"));
exports.ApplicationTraceConfig = ApplicationTraceConfig.ApplicationTraceConfig;
const ChangeApplicationCompartmentDetails = __importStar(require("./change-application-compartment-details"));
exports.ChangeApplicationCompartmentDetails = ChangeApplicationCompartmentDetails.ChangeApplicationCompartmentDetails;
const CreateApplicationDetails = __importStar(require("./create-application-details"));
exports.CreateApplicationDetails = CreateApplicationDetails.CreateApplicationDetails;
const CreateFunctionDetails = __importStar(require("./create-function-details"));
exports.CreateFunctionDetails = CreateFunctionDetails.CreateFunctionDetails;
const Function = __importStar(require("./function"));
exports.Function = Function.Function;
const FunctionSummary = __importStar(require("./function-summary"));
exports.FunctionSummary = FunctionSummary.FunctionSummary;
const FunctionTraceConfig = __importStar(require("./function-trace-config"));
exports.FunctionTraceConfig = FunctionTraceConfig.FunctionTraceConfig;
const UpdateApplicationDetails = __importStar(require("./update-application-details"));
exports.UpdateApplicationDetails = UpdateApplicationDetails.UpdateApplicationDetails;
const UpdateFunctionDetails = __importStar(require("./update-function-details"));
exports.UpdateFunctionDetails = UpdateFunctionDetails.UpdateFunctionDetails;
