// oci-datacatalog/lib/request/get-term-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190325
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetTermRequest = void 0;
var GetTermRequest;
(function (GetTermRequest) {
    let Fields;
    (function (Fields) {
        Fields["Key"] = "key";
        Fields["DisplayName"] = "displayName";
        Fields["Description"] = "description";
        Fields["GlossaryKey"] = "glossaryKey";
        Fields["ParentTermKey"] = "parentTermKey";
        Fields["IsAllowedToHaveChildTerms"] = "isAllowedToHaveChildTerms";
        Fields["Path"] = "path";
        Fields["LifecycleState"] = "lifecycleState";
        Fields["TimeCreated"] = "timeCreated";
        Fields["TimeUpdated"] = "timeUpdated";
        Fields["CreatedById"] = "createdById";
        Fields["UpdatedById"] = "updatedById";
        Fields["Owner"] = "owner";
        Fields["WorkflowStatus"] = "workflowStatus";
        Fields["Uri"] = "uri";
        Fields["RelatedTerms"] = "relatedTerms";
        Fields["AssociatedObjectCount"] = "associatedObjectCount";
        Fields["AssociatedObjects"] = "associatedObjects";
    })(Fields = GetTermRequest.Fields || (GetTermRequest.Fields = {}));
})(GetTermRequest = exports.GetTermRequest || (exports.GetTermRequest = {}));
