// oci-datacatalog/lib/request/get-job-definition-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190325
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetJobDefinitionRequest = void 0;
var GetJobDefinitionRequest;
(function (GetJobDefinitionRequest) {
    let Fields;
    (function (Fields) {
        Fields["Key"] = "key";
        Fields["DisplayName"] = "displayName";
        Fields["Description"] = "description";
        Fields["CatalogId"] = "catalogId";
        Fields["JobType"] = "jobType";
        Fields["IsIncremental"] = "isIncremental";
        Fields["DataAssetKey"] = "dataAssetKey";
        Fields["ConnectionKey"] = "connectionKey";
        Fields["InternalVersion"] = "internalVersion";
        Fields["LifecycleState"] = "lifecycleState";
        Fields["TimeCreated"] = "timeCreated";
        Fields["TimeUpdated"] = "timeUpdated";
        Fields["CreatedById"] = "createdById";
        Fields["UpdatedById"] = "updatedById";
        Fields["Uri"] = "uri";
        Fields["IsSampleDataExtracted"] = "isSampleDataExtracted";
        Fields["SampleDataSizeInMBs"] = "sampleDataSizeInMBs";
        Fields["TimeLatestExecutionStarted"] = "timeLatestExecutionStarted";
        Fields["TimeLatestExecutionEnded"] = "timeLatestExecutionEnded";
        Fields["JobExecutionState"] = "jobExecutionState";
        Fields["ScheduleType"] = "scheduleType";
        Fields["Properties"] = "properties";
    })(Fields = GetJobDefinitionRequest.Fields || (GetJobDefinitionRequest.Fields = {}));
})(GetJobDefinitionRequest = exports.GetJobDefinitionRequest || (exports.GetJobDefinitionRequest = {}));
