// oci-datacatalog/lib/request/get-type-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190325
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetTypeRequest = void 0;
var GetTypeRequest;
(function (GetTypeRequest) {
    let Fields;
    (function (Fields) {
        Fields["Key"] = "key";
        Fields["Description"] = "description";
        Fields["Name"] = "name";
        Fields["CatalogId"] = "catalogId";
        Fields["Properties"] = "properties";
        Fields["IsInternal"] = "isInternal";
        Fields["IsTag"] = "isTag";
        Fields["IsApproved"] = "isApproved";
        Fields["TypeCategory"] = "typeCategory";
        Fields["ExternalTypeName"] = "externalTypeName";
        Fields["LifecycleState"] = "lifecycleState";
        Fields["Uri"] = "uri";
    })(Fields = GetTypeRequest.Fields || (GetTypeRequest.Fields = {}));
})(GetTypeRequest = exports.GetTypeRequest || (exports.GetTypeRequest = {}));
