// oci-datacatalog/lib/request/get-attribute-tag-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190325
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetAttributeTagRequest = void 0;
var GetAttributeTagRequest;
(function (GetAttributeTagRequest) {
    let Fields;
    (function (Fields) {
        Fields["Key"] = "key";
        Fields["Name"] = "name";
        Fields["TermKey"] = "termKey";
        Fields["TermPath"] = "termPath";
        Fields["TermDescription"] = "termDescription";
        Fields["LifecycleState"] = "lifecycleState";
        Fields["TimeCreated"] = "timeCreated";
        Fields["CreatedById"] = "createdById";
        Fields["Uri"] = "uri";
        Fields["AttributeKey"] = "attributeKey";
    })(Fields = GetAttributeTagRequest.Fields || (GetAttributeTagRequest.Fields = {}));
})(GetAttributeTagRequest = exports.GetAttributeTagRequest || (exports.GetAttributeTagRequest = {}));
