// oci-datacatalog/lib/request/get-job-metrics-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190325
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetJobMetricsRequest = void 0;
var GetJobMetricsRequest;
(function (GetJobMetricsRequest) {
    let Fields;
    (function (Fields) {
        Fields["Key"] = "key";
        Fields["Description"] = "description";
        Fields["DisplayName"] = "displayName";
        Fields["TimeInserted"] = "timeInserted";
        Fields["Category"] = "category";
        Fields["SubCategory"] = "subCategory";
        Fields["Unit"] = "unit";
        Fields["Value"] = "value";
        Fields["BatchKey"] = "batchKey";
        Fields["JobExecutionKey"] = "jobExecutionKey";
        Fields["CreatedById"] = "createdById";
        Fields["UpdatedById"] = "updatedById";
        Fields["TimeUpdated"] = "timeUpdated";
        Fields["TimeCreated"] = "timeCreated";
        Fields["Uri"] = "uri";
    })(Fields = GetJobMetricsRequest.Fields || (GetJobMetricsRequest.Fields = {}));
})(GetJobMetricsRequest = exports.GetJobMetricsRequest || (exports.GetJobMetricsRequest = {}));
