// oci-datacatalog/lib/request/get-job-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190325
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetJobRequest = void 0;
var GetJobRequest;
(function (GetJobRequest) {
    let Fields;
    (function (Fields) {
        Fields["Key"] = "key";
        Fields["DisplayName"] = "displayName";
        Fields["Description"] = "description";
        Fields["CatalogId"] = "catalogId";
        Fields["LifecycleState"] = "lifecycleState";
        Fields["TimeCreated"] = "timeCreated";
        Fields["TimeUpdated"] = "timeUpdated";
        Fields["JobType"] = "jobType";
        Fields["ScheduleCronExpression"] = "scheduleCronExpression";
        Fields["TimeScheduleBegin"] = "timeScheduleBegin";
        Fields["TimeScheduleEnd"] = "timeScheduleEnd";
        Fields["ScheduleType"] = "scheduleType";
        Fields["ConnectionKey"] = "connectionKey";
        Fields["JobDefinitionKey"] = "jobDefinitionKey";
        Fields["InternalVersion"] = "internalVersion";
        Fields["ExecutionCount"] = "executionCount";
        Fields["TimeOfLatestExecution"] = "timeOfLatestExecution";
        Fields["Executions"] = "executions";
        Fields["CreatedById"] = "createdById";
        Fields["UpdatedById"] = "updatedById";
        Fields["Uri"] = "uri";
        Fields["JobDefinitionName"] = "jobDefinitionName";
        Fields["ErrorCode"] = "errorCode";
        Fields["ErrorMessage"] = "errorMessage";
    })(Fields = GetJobRequest.Fields || (GetJobRequest.Fields = {}));
})(GetJobRequest = exports.GetJobRequest || (exports.GetJobRequest = {}));
