// oci-datacatalog/lib/request/get-data-asset-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190325
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetDataAssetRequest = void 0;
var GetDataAssetRequest;
(function (GetDataAssetRequest) {
    let Fields;
    (function (Fields) {
        Fields["Key"] = "key";
        Fields["DisplayName"] = "displayName";
        Fields["Description"] = "description";
        Fields["CatalogId"] = "catalogId";
        Fields["ExternalKey"] = "externalKey";
        Fields["TypeKey"] = "typeKey";
        Fields["LifecycleState"] = "lifecycleState";
        Fields["TimeCreated"] = "timeCreated";
        Fields["TimeUpdated"] = "timeUpdated";
        Fields["CreatedById"] = "createdById";
        Fields["UpdatedById"] = "updatedById";
        Fields["Uri"] = "uri";
        Fields["Properties"] = "properties";
    })(Fields = GetDataAssetRequest.Fields || (GetDataAssetRequest.Fields = {}));
})(GetDataAssetRequest = exports.GetDataAssetRequest || (exports.GetDataAssetRequest = {}));
