// oci-datacatalog/lib/request/index.js
"use strict";
/**
 * Data Catalog API
 * Use the Data Catalog APIs to collect, organize, find, access, understand, enrich, and activate technical, business, and operational metadata.
 * OpenAPI spec version: 20190325
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWorkRequestLogsRequest = exports.ListWorkRequestErrorsRequest = exports.ListTypesRequest = exports.ListTermsRequest = exports.ListTermRelationshipsRequest = exports.ListTagsRequest = exports.ListRulesRequest = exports.ListPatternsRequest = exports.ListNamespacesRequest = exports.ListJobsRequest = exports.ListJobMetricsRequest = exports.ListJobLogsRequest = exports.ListJobExecutionsRequest = exports.ListJobDefinitionsRequest = exports.ListGlossariesRequest = exports.ListFoldersRequest = exports.ListFolderTagsRequest = exports.ListEntityTagsRequest = exports.ListEntitiesRequest = exports.ListDerivedLogicalEntitiesRequest = exports.ListDataAssetsRequest = exports.ListDataAssetTagsRequest = exports.ListCustomPropertiesRequest = exports.ListConnectionsRequest = exports.ListCatalogsRequest = exports.ListCatalogPrivateEndpointsRequest = exports.ListAttributesRequest = exports.ListAttributeTagsRequest = exports.ListAggregatedPhysicalEntitiesRequest = exports.GetTypeRequest = exports.GetTermRelationshipRequest = exports.GetTermRequest = exports.GetPatternRequest = exports.GetNamespaceRequest = exports.GetJobMetricsRequest = exports.GetJobLogRequest = exports.GetJobExecutionRequest = exports.GetJobDefinitionRequest = exports.GetJobRequest = exports.GetGlossaryRequest = exports.GetFolderTagRequest = exports.GetFolderRequest = exports.GetEntityTagRequest = exports.GetEntityRequest = exports.GetDataAssetTagRequest = exports.GetDataAssetRequest = exports.GetCustomPropertyRequest = exports.GetConnectionRequest = exports.GetAttributeTagRequest = exports.GetAttributeRequest = void 0;
exports.UsersRequest = exports.SearchCriteriaRequest = exports.ObjectStatsRequest = void 0;
const GetAttributeRequest = __importStar(require("./get-attribute-request"));
exports.GetAttributeRequest = GetAttributeRequest.GetAttributeRequest;
const GetAttributeTagRequest = __importStar(require("./get-attribute-tag-request"));
exports.GetAttributeTagRequest = GetAttributeTagRequest.GetAttributeTagRequest;
const GetConnectionRequest = __importStar(require("./get-connection-request"));
exports.GetConnectionRequest = GetConnectionRequest.GetConnectionRequest;
const GetCustomPropertyRequest = __importStar(require("./get-custom-property-request"));
exports.GetCustomPropertyRequest = GetCustomPropertyRequest.GetCustomPropertyRequest;
const GetDataAssetRequest = __importStar(require("./get-data-asset-request"));
exports.GetDataAssetRequest = GetDataAssetRequest.GetDataAssetRequest;
const GetDataAssetTagRequest = __importStar(require("./get-data-asset-tag-request"));
exports.GetDataAssetTagRequest = GetDataAssetTagRequest.GetDataAssetTagRequest;
const GetEntityRequest = __importStar(require("./get-entity-request"));
exports.GetEntityRequest = GetEntityRequest.GetEntityRequest;
const GetEntityTagRequest = __importStar(require("./get-entity-tag-request"));
exports.GetEntityTagRequest = GetEntityTagRequest.GetEntityTagRequest;
const GetFolderRequest = __importStar(require("./get-folder-request"));
exports.GetFolderRequest = GetFolderRequest.GetFolderRequest;
const GetFolderTagRequest = __importStar(require("./get-folder-tag-request"));
exports.GetFolderTagRequest = GetFolderTagRequest.GetFolderTagRequest;
const GetGlossaryRequest = __importStar(require("./get-glossary-request"));
exports.GetGlossaryRequest = GetGlossaryRequest.GetGlossaryRequest;
const GetJobRequest = __importStar(require("./get-job-request"));
exports.GetJobRequest = GetJobRequest.GetJobRequest;
const GetJobDefinitionRequest = __importStar(require("./get-job-definition-request"));
exports.GetJobDefinitionRequest = GetJobDefinitionRequest.GetJobDefinitionRequest;
const GetJobExecutionRequest = __importStar(require("./get-job-execution-request"));
exports.GetJobExecutionRequest = GetJobExecutionRequest.GetJobExecutionRequest;
const GetJobLogRequest = __importStar(require("./get-job-log-request"));
exports.GetJobLogRequest = GetJobLogRequest.GetJobLogRequest;
const GetJobMetricsRequest = __importStar(require("./get-job-metrics-request"));
exports.GetJobMetricsRequest = GetJobMetricsRequest.GetJobMetricsRequest;
const GetNamespaceRequest = __importStar(require("./get-namespace-request"));
exports.GetNamespaceRequest = GetNamespaceRequest.GetNamespaceRequest;
const GetPatternRequest = __importStar(require("./get-pattern-request"));
exports.GetPatternRequest = GetPatternRequest.GetPatternRequest;
const GetTermRequest = __importStar(require("./get-term-request"));
exports.GetTermRequest = GetTermRequest.GetTermRequest;
const GetTermRelationshipRequest = __importStar(require("./get-term-relationship-request"));
exports.GetTermRelationshipRequest = GetTermRelationshipRequest.GetTermRelationshipRequest;
const GetTypeRequest = __importStar(require("./get-type-request"));
exports.GetTypeRequest = GetTypeRequest.GetTypeRequest;
const ListAggregatedPhysicalEntitiesRequest = __importStar(require("./list-aggregated-physical-entities-request"));
exports.ListAggregatedPhysicalEntitiesRequest = ListAggregatedPhysicalEntitiesRequest.ListAggregatedPhysicalEntitiesRequest;
const ListAttributeTagsRequest = __importStar(require("./list-attribute-tags-request"));
exports.ListAttributeTagsRequest = ListAttributeTagsRequest.ListAttributeTagsRequest;
const ListAttributesRequest = __importStar(require("./list-attributes-request"));
exports.ListAttributesRequest = ListAttributesRequest.ListAttributesRequest;
const ListCatalogPrivateEndpointsRequest = __importStar(require("./list-catalog-private-endpoints-request"));
exports.ListCatalogPrivateEndpointsRequest = ListCatalogPrivateEndpointsRequest.ListCatalogPrivateEndpointsRequest;
const ListCatalogsRequest = __importStar(require("./list-catalogs-request"));
exports.ListCatalogsRequest = ListCatalogsRequest.ListCatalogsRequest;
const ListConnectionsRequest = __importStar(require("./list-connections-request"));
exports.ListConnectionsRequest = ListConnectionsRequest.ListConnectionsRequest;
const ListCustomPropertiesRequest = __importStar(require("./list-custom-properties-request"));
exports.ListCustomPropertiesRequest = ListCustomPropertiesRequest.ListCustomPropertiesRequest;
const ListDataAssetTagsRequest = __importStar(require("./list-data-asset-tags-request"));
exports.ListDataAssetTagsRequest = ListDataAssetTagsRequest.ListDataAssetTagsRequest;
const ListDataAssetsRequest = __importStar(require("./list-data-assets-request"));
exports.ListDataAssetsRequest = ListDataAssetsRequest.ListDataAssetsRequest;
const ListDerivedLogicalEntitiesRequest = __importStar(require("./list-derived-logical-entities-request"));
exports.ListDerivedLogicalEntitiesRequest = ListDerivedLogicalEntitiesRequest.ListDerivedLogicalEntitiesRequest;
const ListEntitiesRequest = __importStar(require("./list-entities-request"));
exports.ListEntitiesRequest = ListEntitiesRequest.ListEntitiesRequest;
const ListEntityTagsRequest = __importStar(require("./list-entity-tags-request"));
exports.ListEntityTagsRequest = ListEntityTagsRequest.ListEntityTagsRequest;
const ListFolderTagsRequest = __importStar(require("./list-folder-tags-request"));
exports.ListFolderTagsRequest = ListFolderTagsRequest.ListFolderTagsRequest;
const ListFoldersRequest = __importStar(require("./list-folders-request"));
exports.ListFoldersRequest = ListFoldersRequest.ListFoldersRequest;
const ListGlossariesRequest = __importStar(require("./list-glossaries-request"));
exports.ListGlossariesRequest = ListGlossariesRequest.ListGlossariesRequest;
const ListJobDefinitionsRequest = __importStar(require("./list-job-definitions-request"));
exports.ListJobDefinitionsRequest = ListJobDefinitionsRequest.ListJobDefinitionsRequest;
const ListJobExecutionsRequest = __importStar(require("./list-job-executions-request"));
exports.ListJobExecutionsRequest = ListJobExecutionsRequest.ListJobExecutionsRequest;
const ListJobLogsRequest = __importStar(require("./list-job-logs-request"));
exports.ListJobLogsRequest = ListJobLogsRequest.ListJobLogsRequest;
const ListJobMetricsRequest = __importStar(require("./list-job-metrics-request"));
exports.ListJobMetricsRequest = ListJobMetricsRequest.ListJobMetricsRequest;
const ListJobsRequest = __importStar(require("./list-jobs-request"));
exports.ListJobsRequest = ListJobsRequest.ListJobsRequest;
const ListNamespacesRequest = __importStar(require("./list-namespaces-request"));
exports.ListNamespacesRequest = ListNamespacesRequest.ListNamespacesRequest;
const ListPatternsRequest = __importStar(require("./list-patterns-request"));
exports.ListPatternsRequest = ListPatternsRequest.ListPatternsRequest;
const ListRulesRequest = __importStar(require("./list-rules-request"));
exports.ListRulesRequest = ListRulesRequest.ListRulesRequest;
const ListTagsRequest = __importStar(require("./list-tags-request"));
exports.ListTagsRequest = ListTagsRequest.ListTagsRequest;
const ListTermRelationshipsRequest = __importStar(require("./list-term-relationships-request"));
exports.ListTermRelationshipsRequest = ListTermRelationshipsRequest.ListTermRelationshipsRequest;
const ListTermsRequest = __importStar(require("./list-terms-request"));
exports.ListTermsRequest = ListTermsRequest.ListTermsRequest;
const ListTypesRequest = __importStar(require("./list-types-request"));
exports.ListTypesRequest = ListTypesRequest.ListTypesRequest;
const ListWorkRequestErrorsRequest = __importStar(require("./list-work-request-errors-request"));
exports.ListWorkRequestErrorsRequest = ListWorkRequestErrorsRequest.ListWorkRequestErrorsRequest;
const ListWorkRequestLogsRequest = __importStar(require("./list-work-request-logs-request"));
exports.ListWorkRequestLogsRequest = ListWorkRequestLogsRequest.ListWorkRequestLogsRequest;
const ObjectStatsRequest = __importStar(require("./object-stats-request"));
exports.ObjectStatsRequest = ObjectStatsRequest.ObjectStatsRequest;
const SearchCriteriaRequest = __importStar(require("./search-criteria-request"));
exports.SearchCriteriaRequest = SearchCriteriaRequest.SearchCriteriaRequest;
const UsersRequest = __importStar(require("./users-request"));
exports.UsersRequest = UsersRequest.UsersRequest;
