// oci-datacatalog/lib/request/list-job-executions-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190325
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListJobExecutionsRequest = void 0;
var ListJobExecutionsRequest;
(function (ListJobExecutionsRequest) {
    let Fields;
    (function (Fields) {
        Fields["Key"] = "key";
        Fields["JobKey"] = "jobKey";
        Fields["JobType"] = "jobType";
        Fields["ParentKey"] = "parentKey";
        Fields["ScheduleInstanceKey"] = "scheduleInstanceKey";
        Fields["LifecycleState"] = "lifecycleState";
        Fields["TimeCreated"] = "timeCreated";
        Fields["TimeStarted"] = "timeStarted";
        Fields["TimeEnded"] = "timeEnded";
        Fields["Uri"] = "uri";
    })(Fields = ListJobExecutionsRequest.Fields || (ListJobExecutionsRequest.Fields = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["Timecreated"] = "TIMECREATED";
        SortBy["Displayname"] = "DISPLAYNAME";
    })(SortBy = ListJobExecutionsRequest.SortBy || (ListJobExecutionsRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListJobExecutionsRequest.SortOrder || (ListJobExecutionsRequest.SortOrder = {}));
})(ListJobExecutionsRequest = exports.ListJobExecutionsRequest || (exports.ListJobExecutionsRequest = {}));
