// oci-datacatalog/lib/request/list-jobs-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190325
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListJobsRequest = void 0;
var ListJobsRequest;
(function (ListJobsRequest) {
    let Fields;
    (function (Fields) {
        Fields["Key"] = "key";
        Fields["DisplayName"] = "displayName";
        Fields["Description"] = "description";
        Fields["CatalogId"] = "catalogId";
        Fields["JobDefinitionKey"] = "jobDefinitionKey";
        Fields["LifecycleState"] = "lifecycleState";
        Fields["TimeCreated"] = "timeCreated";
        Fields["TimeUpdated"] = "timeUpdated";
        Fields["CreatedById"] = "createdById";
        Fields["UpdatedById"] = "updatedById";
        Fields["JobType"] = "jobType";
        Fields["ScheduleCronExpression"] = "scheduleCronExpression";
        Fields["TimeScheduleBegin"] = "timeScheduleBegin";
        Fields["ScheduleType"] = "scheduleType";
        Fields["ExecutionCount"] = "executionCount";
        Fields["TimeOfLatestExecution"] = "timeOfLatestExecution";
        Fields["Executions"] = "executions";
        Fields["Uri"] = "uri";
        Fields["JobDefinitionName"] = "jobDefinitionName";
        Fields["ErrorCode"] = "errorCode";
        Fields["ErrorMessage"] = "errorMessage";
    })(Fields = ListJobsRequest.Fields || (ListJobsRequest.Fields = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["Timecreated"] = "TIMECREATED";
        SortBy["Displayname"] = "DISPLAYNAME";
    })(SortBy = ListJobsRequest.SortBy || (ListJobsRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListJobsRequest.SortOrder || (ListJobsRequest.SortOrder = {}));
})(ListJobsRequest = exports.ListJobsRequest || (exports.ListJobsRequest = {}));
