// oci-datacatalog/lib/request/list-folder-tags-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190325
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListFolderTagsRequest = void 0;
var ListFolderTagsRequest;
(function (ListFolderTagsRequest) {
    let Fields;
    (function (Fields) {
        Fields["Key"] = "key";
        Fields["Name"] = "name";
        Fields["TermKey"] = "termKey";
        Fields["TermPath"] = "termPath";
        Fields["TermDescription"] = "termDescription";
        Fields["LifecycleState"] = "lifecycleState";
        Fields["TimeCreated"] = "timeCreated";
        Fields["Uri"] = "uri";
        Fields["GlossaryKey"] = "glossaryKey";
        Fields["FolderKey"] = "folderKey";
    })(Fields = ListFolderTagsRequest.Fields || (ListFolderTagsRequest.Fields = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["Timecreated"] = "TIMECREATED";
        SortBy["Displayname"] = "DISPLAYNAME";
    })(SortBy = ListFolderTagsRequest.SortBy || (ListFolderTagsRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListFolderTagsRequest.SortOrder || (ListFolderTagsRequest.SortOrder = {}));
})(ListFolderTagsRequest = exports.ListFolderTagsRequest || (exports.ListFolderTagsRequest = {}));
