// oci-datacatalog/lib/request/list-attributes-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190325
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListAttributesRequest = void 0;
var ListAttributesRequest;
(function (ListAttributesRequest) {
    let Fields;
    (function (Fields) {
        Fields["Key"] = "key";
        Fields["DisplayName"] = "displayName";
        Fields["Description"] = "description";
        Fields["EntityKey"] = "entityKey";
        Fields["LifecycleState"] = "lifecycleState";
        Fields["TimeCreated"] = "timeCreated";
        Fields["ExternalDataType"] = "externalDataType";
        Fields["ExternalKey"] = "externalKey";
        Fields["Length"] = "length";
        Fields["Precision"] = "precision";
        Fields["Scale"] = "scale";
        Fields["IsNullable"] = "isNullable";
        Fields["Uri"] = "uri";
        Fields["Path"] = "path";
        Fields["MinCollectionCount"] = "minCollectionCount";
        Fields["MaxCollectionCount"] = "maxCollectionCount";
        Fields["DatatypeEntityKey"] = "datatypeEntityKey";
        Fields["ExternalDatatypeEntityKey"] = "externalDatatypeEntityKey";
        Fields["ParentAttributeKey"] = "parentAttributeKey";
        Fields["ExternalParentAttributeKey"] = "externalParentAttributeKey";
    })(Fields = ListAttributesRequest.Fields || (ListAttributesRequest.Fields = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["Timecreated"] = "TIMECREATED";
        SortBy["Displayname"] = "DISPLAYNAME";
    })(SortBy = ListAttributesRequest.SortBy || (ListAttributesRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListAttributesRequest.SortOrder || (ListAttributesRequest.SortOrder = {}));
})(ListAttributesRequest = exports.ListAttributesRequest || (exports.ListAttributesRequest = {}));
