// oci-datacatalog/lib/request/list-folders-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190325
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListFoldersRequest = void 0;
var ListFoldersRequest;
(function (ListFoldersRequest) {
    let Fields;
    (function (Fields) {
        Fields["Key"] = "key";
        Fields["DisplayName"] = "displayName";
        Fields["Description"] = "description";
        Fields["ParentFolderKey"] = "parentFolderKey";
        Fields["Path"] = "path";
        Fields["DataAssetKey"] = "dataAssetKey";
        Fields["ExternalKey"] = "externalKey";
        Fields["TimeExternal"] = "timeExternal";
        Fields["TimeCreated"] = "timeCreated";
        Fields["LifecycleState"] = "lifecycleState";
        Fields["Uri"] = "uri";
    })(Fields = ListFoldersRequest.Fields || (ListFoldersRequest.Fields = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["Timecreated"] = "TIMECREATED";
        SortBy["Displayname"] = "DISPLAYNAME";
    })(SortBy = ListFoldersRequest.SortBy || (ListFoldersRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListFoldersRequest.SortOrder || (ListFoldersRequest.SortOrder = {}));
})(ListFoldersRequest = exports.ListFoldersRequest || (exports.ListFoldersRequest = {}));
