// oci-datacatalog/lib/request/list-job-metrics-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190325
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListJobMetricsRequest = void 0;
var ListJobMetricsRequest;
(function (ListJobMetricsRequest) {
    let Fields;
    (function (Fields) {
        Fields["Key"] = "key";
        Fields["Description"] = "description";
        Fields["DisplayName"] = "displayName";
        Fields["TimeInserted"] = "timeInserted";
        Fields["Category"] = "category";
        Fields["SubCategory"] = "subCategory";
        Fields["Unit"] = "unit";
        Fields["Value"] = "value";
        Fields["BatchKey"] = "batchKey";
        Fields["JobExecutionKey"] = "jobExecutionKey";
        Fields["TimeCreated"] = "timeCreated";
        Fields["Uri"] = "uri";
    })(Fields = ListJobMetricsRequest.Fields || (ListJobMetricsRequest.Fields = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["Timecreated"] = "TIMECREATED";
        SortBy["Displayname"] = "DISPLAYNAME";
    })(SortBy = ListJobMetricsRequest.SortBy || (ListJobMetricsRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListJobMetricsRequest.SortOrder || (ListJobMetricsRequest.SortOrder = {}));
})(ListJobMetricsRequest = exports.ListJobMetricsRequest || (exports.ListJobMetricsRequest = {}));
