// oci-datacatalog/lib/request/list-job-logs-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190325
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListJobLogsRequest = void 0;
var ListJobLogsRequest;
(function (ListJobLogsRequest) {
    let Fields;
    (function (Fields) {
        Fields["Key"] = "key";
        Fields["JobExecutionKey"] = "jobExecutionKey";
        Fields["Severity"] = "severity";
        Fields["TimeCreated"] = "timeCreated";
        Fields["LogMessage"] = "logMessage";
        Fields["Uri"] = "uri";
    })(Fields = ListJobLogsRequest.Fields || (ListJobLogsRequest.Fields = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["Timecreated"] = "TIMECREATED";
        SortBy["Displayname"] = "DISPLAYNAME";
    })(SortBy = ListJobLogsRequest.SortBy || (ListJobLogsRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListJobLogsRequest.SortOrder || (ListJobLogsRequest.SortOrder = {}));
})(ListJobLogsRequest = exports.ListJobLogsRequest || (exports.ListJobLogsRequest = {}));
