// oci-datacatalog/lib/request/list-entities-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190325
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListEntitiesRequest = void 0;
var ListEntitiesRequest;
(function (ListEntitiesRequest) {
    let Fields;
    (function (Fields) {
        Fields["Key"] = "key";
        Fields["DisplayName"] = "displayName";
        Fields["Description"] = "description";
        Fields["DataAssetKey"] = "dataAssetKey";
        Fields["TimeCreated"] = "timeCreated";
        Fields["TimeUpdated"] = "timeUpdated";
        Fields["UpdatedById"] = "updatedById";
        Fields["LifecycleState"] = "lifecycleState";
        Fields["FolderKey"] = "folderKey";
        Fields["FolderName"] = "folderName";
        Fields["ExternalKey"] = "externalKey";
        Fields["Path"] = "path";
        Fields["Uri"] = "uri";
    })(Fields = ListEntitiesRequest.Fields || (ListEntitiesRequest.Fields = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["Timecreated"] = "TIMECREATED";
        SortBy["Displayname"] = "DISPLAYNAME";
    })(SortBy = ListEntitiesRequest.SortBy || (ListEntitiesRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListEntitiesRequest.SortOrder || (ListEntitiesRequest.SortOrder = {}));
})(ListEntitiesRequest = exports.ListEntitiesRequest || (exports.ListEntitiesRequest = {}));
