// oci-datacatalog/lib/request/list-term-relationships-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190325
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListTermRelationshipsRequest = void 0;
var ListTermRelationshipsRequest;
(function (ListTermRelationshipsRequest) {
    let Fields;
    (function (Fields) {
        Fields["Key"] = "key";
        Fields["DisplayName"] = "displayName";
        Fields["Description"] = "description";
        Fields["RelatedTermKey"] = "relatedTermKey";
        Fields["RelatedTermDisplayName"] = "relatedTermDisplayName";
        Fields["ParentTermKey"] = "parentTermKey";
        Fields["ParentTermDisplayName"] = "parentTermDisplayName";
        Fields["LifecycleState"] = "lifecycleState";
        Fields["TimeCreated"] = "timeCreated";
        Fields["Uri"] = "uri";
    })(Fields = ListTermRelationshipsRequest.Fields || (ListTermRelationshipsRequest.Fields = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["Timecreated"] = "TIMECREATED";
        SortBy["Displayname"] = "DISPLAYNAME";
    })(SortBy = ListTermRelationshipsRequest.SortBy || (ListTermRelationshipsRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListTermRelationshipsRequest.SortOrder || (ListTermRelationshipsRequest.SortOrder = {}));
})(ListTermRelationshipsRequest = exports.ListTermRelationshipsRequest || (exports.ListTermRelationshipsRequest = {}));
