// oci-datacatalog/lib/request/list-job-definitions-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190325
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListJobDefinitionsRequest = void 0;
var ListJobDefinitionsRequest;
(function (ListJobDefinitionsRequest) {
    let Fields;
    (function (Fields) {
        Fields["Key"] = "key";
        Fields["DisplayName"] = "displayName";
        Fields["Description"] = "description";
        Fields["CatalogId"] = "catalogId";
        Fields["JobType"] = "jobType";
        Fields["ConnectionKey"] = "connectionKey";
        Fields["LifecycleState"] = "lifecycleState";
        Fields["TimeCreated"] = "timeCreated";
        Fields["IsSampleDataExtracted"] = "isSampleDataExtracted";
        Fields["Uri"] = "uri";
        Fields["TimeLatestExecutionStarted"] = "timeLatestExecutionStarted";
        Fields["TimeLatestExecutionEnded"] = "timeLatestExecutionEnded";
        Fields["JobExecutionState"] = "jobExecutionState";
        Fields["ScheduleType"] = "scheduleType";
    })(Fields = ListJobDefinitionsRequest.Fields || (ListJobDefinitionsRequest.Fields = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["Timecreated"] = "TIMECREATED";
        SortBy["Displayname"] = "DISPLAYNAME";
        SortBy["Timelatestexecutionstarted"] = "TIMELATESTEXECUTIONSTARTED";
    })(SortBy = ListJobDefinitionsRequest.SortBy || (ListJobDefinitionsRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListJobDefinitionsRequest.SortOrder || (ListJobDefinitionsRequest.SortOrder = {}));
})(ListJobDefinitionsRequest = exports.ListJobDefinitionsRequest || (exports.ListJobDefinitionsRequest = {}));
