// oci-mysql/lib/request/list-backups-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190415
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListBackupsRequest = void 0;
var ListBackupsRequest;
(function (ListBackupsRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["TimeUpdated"] = "timeUpdated";
        SortBy["DisplayName"] = "displayName";
    })(SortBy = ListBackupsRequest.SortBy || (ListBackupsRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListBackupsRequest.SortOrder || (ListBackupsRequest.SortOrder = {}));
})(ListBackupsRequest = exports.ListBackupsRequest || (exports.ListBackupsRequest = {}));
