// oci-resourcemanager/lib/request/index.js
"use strict";
/**
 * Resource Manager API
 * API for the Resource Manager service.
Use this API to install, configure, and manage resources via the "infrastructure-as-code" model.
For more information, see
[Overview of Resource Manager](/iaas/Content/ResourceManager/Concepts/resourcemanager.htm).

 * OpenAPI spec version: 20180917
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWorkRequestLogsRequest = exports.ListWorkRequestErrorsRequest = exports.ListTemplatesRequest = exports.ListStacksRequest = exports.ListJobsRequest = exports.ListConfigurationSourceProvidersRequest = exports.GetJobLogsRequest = void 0;
const GetJobLogsRequest = __importStar(require("./get-job-logs-request"));
exports.GetJobLogsRequest = GetJobLogsRequest.GetJobLogsRequest;
const ListConfigurationSourceProvidersRequest = __importStar(require("./list-configuration-source-providers-request"));
exports.ListConfigurationSourceProvidersRequest = ListConfigurationSourceProvidersRequest.ListConfigurationSourceProvidersRequest;
const ListJobsRequest = __importStar(require("./list-jobs-request"));
exports.ListJobsRequest = ListJobsRequest.ListJobsRequest;
const ListStacksRequest = __importStar(require("./list-stacks-request"));
exports.ListStacksRequest = ListStacksRequest.ListStacksRequest;
const ListTemplatesRequest = __importStar(require("./list-templates-request"));
exports.ListTemplatesRequest = ListTemplatesRequest.ListTemplatesRequest;
const ListWorkRequestErrorsRequest = __importStar(require("./list-work-request-errors-request"));
exports.ListWorkRequestErrorsRequest = ListWorkRequestErrorsRequest.ListWorkRequestErrorsRequest;
const ListWorkRequestLogsRequest = __importStar(require("./list-work-request-logs-request"));
exports.ListWorkRequestLogsRequest = ListWorkRequestLogsRequest.ListWorkRequestLogsRequest;
