// oci-healthchecks/lib/request/index.js
"use strict";
/**
 * Health Checks API
 * API for the Health Checks service. Use this API to manage endpoint probes and monitors.
For more information, see
[Overview of the Health Checks Service](/iaas/Content/HealthChecks/Concepts/healthchecks.htm).

 * OpenAPI spec version: 20180501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListPingProbeResultsRequest = exports.ListPingMonitorsRequest = exports.ListHttpProbeResultsRequest = exports.ListHttpMonitorsRequest = exports.ListHealthChecksVantagePointsRequest = void 0;
const ListHealthChecksVantagePointsRequest = __importStar(require("./list-health-checks-vantage-points-request"));
exports.ListHealthChecksVantagePointsRequest = ListHealthChecksVantagePointsRequest.ListHealthChecksVantagePointsRequest;
const ListHttpMonitorsRequest = __importStar(require("./list-http-monitors-request"));
exports.ListHttpMonitorsRequest = ListHttpMonitorsRequest.ListHttpMonitorsRequest;
const ListHttpProbeResultsRequest = __importStar(require("./list-http-probe-results-request"));
exports.ListHttpProbeResultsRequest = ListHttpProbeResultsRequest.ListHttpProbeResultsRequest;
const ListPingMonitorsRequest = __importStar(require("./list-ping-monitors-request"));
exports.ListPingMonitorsRequest = ListPingMonitorsRequest.ListPingMonitorsRequest;
const ListPingProbeResultsRequest = __importStar(require("./list-ping-probe-results-request"));
exports.ListPingProbeResultsRequest = ListPingProbeResultsRequest.ListPingProbeResultsRequest;
