// oci-healthchecks/lib/request/list-http-monitors-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20180501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListHttpMonitorsRequest = void 0;
var ListHttpMonitorsRequest;
(function (ListHttpMonitorsRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["Id"] = "id";
        SortBy["DisplayName"] = "displayName";
        SortBy["TimeCreated"] = "timeCreated";
    })(SortBy = ListHttpMonitorsRequest.SortBy || (ListHttpMonitorsRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListHttpMonitorsRequest.SortOrder || (ListHttpMonitorsRequest.SortOrder = {}));
})(ListHttpMonitorsRequest = exports.ListHttpMonitorsRequest || (exports.ListHttpMonitorsRequest = {}));
