// oci-databasemanagement/lib/model/database-type.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to perform tasks such as obtaining performance and resource usage metrics
for a fleet of Managed Databases or a specific Managed Database, creating Managed Database Groups, and
running a SQL job on a Managed Database or Managed Database Group.

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatabaseType = void 0;
/**
 * The type of Oracle Database installation.
 **/
var DatabaseType;
(function (DatabaseType) {
    DatabaseType["ExternalSidb"] = "EXTERNAL_SIDB";
    DatabaseType["ExternalRac"] = "EXTERNAL_RAC";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DatabaseType["UnknownValue"] = "UNKNOWN_VALUE";
})(DatabaseType = exports.DatabaseType || (exports.DatabaseType = {}));
(function (DatabaseType) {
    function getJsonObj(obj) {
        return { obj };
    }
    DatabaseType.getJsonObj = getJsonObj;
})(DatabaseType = exports.DatabaseType || (exports.DatabaseType = {}));
