// oci-databasemanagement/lib/model/index.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to perform tasks such as obtaining performance and resource usage metrics
for a fleet of Managed Databases or a specific Managed Database, creating Managed Database Groups, and
running a SQL job on a Managed Database or Managed Database Group.

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagedDatabaseGroup = exports.ManagedDatabaseCollection = exports.ManagedDatabase = exports.LifecycleStates = exports.JobTypes = exports.JobSummary = exports.JobRunSummary = exports.JobRunCollection = exports.JobRun = exports.JobExecutionSummary = exports.JobExecutionResultLocation = exports.JobExecutionResultDetails = exports.JobExecutionCollection = exports.JobExecution = exports.JobDatabase = exports.JobCollection = exports.Job = exports.InstanceDetails = exports.FleetSummary = exports.FleetStatusByCategory = exports.FleetMetricSummaryDefinition = exports.FleetMetricDefinition = exports.Datafile = exports.DatabaseUsageMetrics = exports.DatabaseType = exports.DatabaseTimeAggregateMetrics = exports.DatabaseSubType = exports.DatabaseStorageAggregateMetrics = exports.DatabaseStatus = exports.DatabaseParametersCollection = exports.DatabaseParameterUpdateStatus = exports.DatabaseParameterSummary = exports.DatabaseInstanceHomeMetricsDefinition = exports.DatabaseIOAggregateMetrics = exports.DatabaseHomeMetrics = exports.DatabaseHomeMetricDefinition = exports.DatabaseFleetHealthMetrics = exports.DatabaseCredentials = exports.CreateManagedDatabaseGroupDetails = exports.CreateJobDetails = exports.CompareType = exports.ClusterCacheMetric = exports.ChildDatabase = exports.ChangeManagedDatabaseGroupCompartmentDetails = exports.ChangeJobCompartmentDetails = exports.ChangeDatabaseParametersDetails = exports.ChangeDatabaseParameterDetails = exports.AllowedParameterValue = exports.AddManagedDatabaseToManagedDatabaseGroupDetails = exports.ActivityTimeSeriesMetrics = void 0;
exports.SqlJob = exports.ObjectStorageJobExecutionResultLocation = exports.ObjectStorageJobExecutionResultDetails = exports.CreateSqlJobDetails = exports.UpdateManagedDatabaseGroupDetails = exports.UpdateDatabaseParametersResult = exports.TimeSeriesMetricDefinition = exports.TimeSeriesMetricDataPoint = exports.TablespaceSummary = exports.TablespaceCollection = exports.Tablespace = exports.SortOrders = exports.ResetDatabaseParametersDetails = exports.RemoveManagedDatabaseFromManagedDatabaseGroupDetails = exports.PdbStatusDetails = exports.ParentGroup = exports.ParameterScope = exports.MetricDimensionDefinition = exports.MetricDataPoint = exports.MemoryAggregateMetrics = exports.ManagedDatabaseSummary = exports.ManagedDatabaseGroupSummary = exports.ManagedDatabaseGroupCollection = void 0;
const ActivityTimeSeriesMetrics = __importStar(require("./activity-time-series-metrics"));
exports.ActivityTimeSeriesMetrics = ActivityTimeSeriesMetrics.ActivityTimeSeriesMetrics;
const AddManagedDatabaseToManagedDatabaseGroupDetails = __importStar(require("./add-managed-database-to-managed-database-group-details"));
exports.AddManagedDatabaseToManagedDatabaseGroupDetails = AddManagedDatabaseToManagedDatabaseGroupDetails.AddManagedDatabaseToManagedDatabaseGroupDetails;
const AllowedParameterValue = __importStar(require("./allowed-parameter-value"));
exports.AllowedParameterValue = AllowedParameterValue.AllowedParameterValue;
const ChangeDatabaseParameterDetails = __importStar(require("./change-database-parameter-details"));
exports.ChangeDatabaseParameterDetails = ChangeDatabaseParameterDetails.ChangeDatabaseParameterDetails;
const ChangeDatabaseParametersDetails = __importStar(require("./change-database-parameters-details"));
exports.ChangeDatabaseParametersDetails = ChangeDatabaseParametersDetails.ChangeDatabaseParametersDetails;
const ChangeJobCompartmentDetails = __importStar(require("./change-job-compartment-details"));
exports.ChangeJobCompartmentDetails = ChangeJobCompartmentDetails.ChangeJobCompartmentDetails;
const ChangeManagedDatabaseGroupCompartmentDetails = __importStar(require("./change-managed-database-group-compartment-details"));
exports.ChangeManagedDatabaseGroupCompartmentDetails = ChangeManagedDatabaseGroupCompartmentDetails.ChangeManagedDatabaseGroupCompartmentDetails;
const ChildDatabase = __importStar(require("./child-database"));
exports.ChildDatabase = ChildDatabase.ChildDatabase;
const ClusterCacheMetric = __importStar(require("./cluster-cache-metric"));
exports.ClusterCacheMetric = ClusterCacheMetric.ClusterCacheMetric;
const CompareType = __importStar(require("./compare-type"));
exports.CompareType = CompareType.CompareType;
const CreateJobDetails = __importStar(require("./create-job-details"));
exports.CreateJobDetails = CreateJobDetails.CreateJobDetails;
const CreateManagedDatabaseGroupDetails = __importStar(require("./create-managed-database-group-details"));
exports.CreateManagedDatabaseGroupDetails = CreateManagedDatabaseGroupDetails.CreateManagedDatabaseGroupDetails;
const DatabaseCredentials = __importStar(require("./database-credentials"));
exports.DatabaseCredentials = DatabaseCredentials.DatabaseCredentials;
const DatabaseFleetHealthMetrics = __importStar(require("./database-fleet-health-metrics"));
exports.DatabaseFleetHealthMetrics = DatabaseFleetHealthMetrics.DatabaseFleetHealthMetrics;
const DatabaseHomeMetricDefinition = __importStar(require("./database-home-metric-definition"));
exports.DatabaseHomeMetricDefinition = DatabaseHomeMetricDefinition.DatabaseHomeMetricDefinition;
const DatabaseHomeMetrics = __importStar(require("./database-home-metrics"));
exports.DatabaseHomeMetrics = DatabaseHomeMetrics.DatabaseHomeMetrics;
const DatabaseIOAggregateMetrics = __importStar(require("./database-io-aggregate-metrics"));
exports.DatabaseIOAggregateMetrics = DatabaseIOAggregateMetrics.DatabaseIOAggregateMetrics;
const DatabaseInstanceHomeMetricsDefinition = __importStar(require("./database-instance-home-metrics-definition"));
exports.DatabaseInstanceHomeMetricsDefinition = DatabaseInstanceHomeMetricsDefinition.DatabaseInstanceHomeMetricsDefinition;
const DatabaseParameterSummary = __importStar(require("./database-parameter-summary"));
exports.DatabaseParameterSummary = DatabaseParameterSummary.DatabaseParameterSummary;
const DatabaseParameterUpdateStatus = __importStar(require("./database-parameter-update-status"));
exports.DatabaseParameterUpdateStatus = DatabaseParameterUpdateStatus.DatabaseParameterUpdateStatus;
const DatabaseParametersCollection = __importStar(require("./database-parameters-collection"));
exports.DatabaseParametersCollection = DatabaseParametersCollection.DatabaseParametersCollection;
const DatabaseStatus = __importStar(require("./database-status"));
exports.DatabaseStatus = DatabaseStatus.DatabaseStatus;
const DatabaseStorageAggregateMetrics = __importStar(require("./database-storage-aggregate-metrics"));
exports.DatabaseStorageAggregateMetrics = DatabaseStorageAggregateMetrics.DatabaseStorageAggregateMetrics;
const DatabaseSubType = __importStar(require("./database-sub-type"));
exports.DatabaseSubType = DatabaseSubType.DatabaseSubType;
const DatabaseTimeAggregateMetrics = __importStar(require("./database-time-aggregate-metrics"));
exports.DatabaseTimeAggregateMetrics = DatabaseTimeAggregateMetrics.DatabaseTimeAggregateMetrics;
const DatabaseType = __importStar(require("./database-type"));
exports.DatabaseType = DatabaseType.DatabaseType;
const DatabaseUsageMetrics = __importStar(require("./database-usage-metrics"));
exports.DatabaseUsageMetrics = DatabaseUsageMetrics.DatabaseUsageMetrics;
const Datafile = __importStar(require("./datafile"));
exports.Datafile = Datafile.Datafile;
const FleetMetricDefinition = __importStar(require("./fleet-metric-definition"));
exports.FleetMetricDefinition = FleetMetricDefinition.FleetMetricDefinition;
const FleetMetricSummaryDefinition = __importStar(require("./fleet-metric-summary-definition"));
exports.FleetMetricSummaryDefinition = FleetMetricSummaryDefinition.FleetMetricSummaryDefinition;
const FleetStatusByCategory = __importStar(require("./fleet-status-by-category"));
exports.FleetStatusByCategory = FleetStatusByCategory.FleetStatusByCategory;
const FleetSummary = __importStar(require("./fleet-summary"));
exports.FleetSummary = FleetSummary.FleetSummary;
const InstanceDetails = __importStar(require("./instance-details"));
exports.InstanceDetails = InstanceDetails.InstanceDetails;
const Job = __importStar(require("./job"));
exports.Job = Job.Job;
const JobCollection = __importStar(require("./job-collection"));
exports.JobCollection = JobCollection.JobCollection;
const JobDatabase = __importStar(require("./job-database"));
exports.JobDatabase = JobDatabase.JobDatabase;
const JobExecution = __importStar(require("./job-execution"));
exports.JobExecution = JobExecution.JobExecution;
const JobExecutionCollection = __importStar(require("./job-execution-collection"));
exports.JobExecutionCollection = JobExecutionCollection.JobExecutionCollection;
const JobExecutionResultDetails = __importStar(require("./job-execution-result-details"));
exports.JobExecutionResultDetails = JobExecutionResultDetails.JobExecutionResultDetails;
const JobExecutionResultLocation = __importStar(require("./job-execution-result-location"));
exports.JobExecutionResultLocation = JobExecutionResultLocation.JobExecutionResultLocation;
const JobExecutionSummary = __importStar(require("./job-execution-summary"));
exports.JobExecutionSummary = JobExecutionSummary.JobExecutionSummary;
const JobRun = __importStar(require("./job-run"));
exports.JobRun = JobRun.JobRun;
const JobRunCollection = __importStar(require("./job-run-collection"));
exports.JobRunCollection = JobRunCollection.JobRunCollection;
const JobRunSummary = __importStar(require("./job-run-summary"));
exports.JobRunSummary = JobRunSummary.JobRunSummary;
const JobSummary = __importStar(require("./job-summary"));
exports.JobSummary = JobSummary.JobSummary;
const JobTypes = __importStar(require("./job-types"));
exports.JobTypes = JobTypes.JobTypes;
const LifecycleStates = __importStar(require("./lifecycle-states"));
exports.LifecycleStates = LifecycleStates.LifecycleStates;
const ManagedDatabase = __importStar(require("./managed-database"));
exports.ManagedDatabase = ManagedDatabase.ManagedDatabase;
const ManagedDatabaseCollection = __importStar(require("./managed-database-collection"));
exports.ManagedDatabaseCollection = ManagedDatabaseCollection.ManagedDatabaseCollection;
const ManagedDatabaseGroup = __importStar(require("./managed-database-group"));
exports.ManagedDatabaseGroup = ManagedDatabaseGroup.ManagedDatabaseGroup;
const ManagedDatabaseGroupCollection = __importStar(require("./managed-database-group-collection"));
exports.ManagedDatabaseGroupCollection = ManagedDatabaseGroupCollection.ManagedDatabaseGroupCollection;
const ManagedDatabaseGroupSummary = __importStar(require("./managed-database-group-summary"));
exports.ManagedDatabaseGroupSummary = ManagedDatabaseGroupSummary.ManagedDatabaseGroupSummary;
const ManagedDatabaseSummary = __importStar(require("./managed-database-summary"));
exports.ManagedDatabaseSummary = ManagedDatabaseSummary.ManagedDatabaseSummary;
const MemoryAggregateMetrics = __importStar(require("./memory-aggregate-metrics"));
exports.MemoryAggregateMetrics = MemoryAggregateMetrics.MemoryAggregateMetrics;
const MetricDataPoint = __importStar(require("./metric-data-point"));
exports.MetricDataPoint = MetricDataPoint.MetricDataPoint;
const MetricDimensionDefinition = __importStar(require("./metric-dimension-definition"));
exports.MetricDimensionDefinition = MetricDimensionDefinition.MetricDimensionDefinition;
const ParameterScope = __importStar(require("./parameter-scope"));
exports.ParameterScope = ParameterScope.ParameterScope;
const ParentGroup = __importStar(require("./parent-group"));
exports.ParentGroup = ParentGroup.ParentGroup;
const PdbStatusDetails = __importStar(require("./pdb-status-details"));
exports.PdbStatusDetails = PdbStatusDetails.PdbStatusDetails;
const RemoveManagedDatabaseFromManagedDatabaseGroupDetails = __importStar(require("./remove-managed-database-from-managed-database-group-details"));
exports.RemoveManagedDatabaseFromManagedDatabaseGroupDetails = RemoveManagedDatabaseFromManagedDatabaseGroupDetails.RemoveManagedDatabaseFromManagedDatabaseGroupDetails;
const ResetDatabaseParametersDetails = __importStar(require("./reset-database-parameters-details"));
exports.ResetDatabaseParametersDetails = ResetDatabaseParametersDetails.ResetDatabaseParametersDetails;
const SortOrders = __importStar(require("./sort-orders"));
exports.SortOrders = SortOrders.SortOrders;
const Tablespace = __importStar(require("./tablespace"));
exports.Tablespace = Tablespace.Tablespace;
const TablespaceCollection = __importStar(require("./tablespace-collection"));
exports.TablespaceCollection = TablespaceCollection.TablespaceCollection;
const TablespaceSummary = __importStar(require("./tablespace-summary"));
exports.TablespaceSummary = TablespaceSummary.TablespaceSummary;
const TimeSeriesMetricDataPoint = __importStar(require("./time-series-metric-data-point"));
exports.TimeSeriesMetricDataPoint = TimeSeriesMetricDataPoint.TimeSeriesMetricDataPoint;
const TimeSeriesMetricDefinition = __importStar(require("./time-series-metric-definition"));
exports.TimeSeriesMetricDefinition = TimeSeriesMetricDefinition.TimeSeriesMetricDefinition;
const UpdateDatabaseParametersResult = __importStar(require("./update-database-parameters-result"));
exports.UpdateDatabaseParametersResult = UpdateDatabaseParametersResult.UpdateDatabaseParametersResult;
const UpdateManagedDatabaseGroupDetails = __importStar(require("./update-managed-database-group-details"));
exports.UpdateManagedDatabaseGroupDetails = UpdateManagedDatabaseGroupDetails.UpdateManagedDatabaseGroupDetails;
const CreateSqlJobDetails = __importStar(require("./create-sql-job-details"));
exports.CreateSqlJobDetails = CreateSqlJobDetails.CreateSqlJobDetails;
const ObjectStorageJobExecutionResultDetails = __importStar(require("./object-storage-job-execution-result-details"));
exports.ObjectStorageJobExecutionResultDetails = ObjectStorageJobExecutionResultDetails.ObjectStorageJobExecutionResultDetails;
const ObjectStorageJobExecutionResultLocation = __importStar(require("./object-storage-job-execution-result-location"));
exports.ObjectStorageJobExecutionResultLocation = ObjectStorageJobExecutionResultLocation.ObjectStorageJobExecutionResultLocation;
const SqlJob = __importStar(require("./sql-job"));
exports.SqlJob = SqlJob.SqlJob;
