// oci-databasemanagement/lib/model/parameter-scope.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to perform tasks such as obtaining performance and resource usage metrics
for a fleet of Managed Databases or a specific Managed Database, creating Managed Database Groups, and
running a SQL job on a Managed Database or Managed Database Group.

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ParameterScope = void 0;
/**
 * The database parameter scope.
 **/
var ParameterScope;
(function (ParameterScope) {
    ParameterScope["Memory"] = "MEMORY";
    ParameterScope["Spfile"] = "SPFILE";
    ParameterScope["Both"] = "BOTH";
})(ParameterScope = exports.ParameterScope || (exports.ParameterScope = {}));
(function (ParameterScope) {
    function getJsonObj(obj) {
        return { obj };
    }
    ParameterScope.getJsonObj = getJsonObj;
})(ParameterScope = exports.ParameterScope || (exports.ParameterScope = {}));
