// oci-databasemanagement/lib/model/tablespace-summary.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to perform tasks such as obtaining performance and resource usage metrics
for a fleet of Managed Databases or a specific Managed Database, creating Managed Database Groups, and
running a SQL job on a Managed Database or Managed Database Group.

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TablespaceSummary = void 0;
const model = __importStar(require("../model"));
var TablespaceSummary;
(function (TablespaceSummary) {
    let Type;
    (function (Type) {
        Type["Undo"] = "UNDO";
        Type["LostWriteProtection"] = "LOST_WRITE_PROTECTION";
        Type["Permanent"] = "PERMANENT";
        Type["Temporary"] = "TEMPORARY";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Type["UnknownValue"] = "UNKNOWN_VALUE";
    })(Type = TablespaceSummary.Type || (TablespaceSummary.Type = {}));
    let Status;
    (function (Status) {
        Status["Online"] = "ONLINE";
        Status["Offline"] = "OFFLINE";
        Status["ReadOnly"] = "READ_ONLY";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Status["UnknownValue"] = "UNKNOWN_VALUE";
    })(Status = TablespaceSummary.Status || (TablespaceSummary.Status = {}));
    let Logging;
    (function (Logging) {
        Logging["Logging"] = "LOGGING";
        Logging["Nologging"] = "NOLOGGING";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Logging["UnknownValue"] = "UNKNOWN_VALUE";
    })(Logging = TablespaceSummary.Logging || (TablespaceSummary.Logging = {}));
    let ExtentManagement;
    (function (ExtentManagement) {
        ExtentManagement["Local"] = "LOCAL";
        ExtentManagement["Dictionary"] = "DICTIONARY";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ExtentManagement["UnknownValue"] = "UNKNOWN_VALUE";
    })(ExtentManagement = TablespaceSummary.ExtentManagement || (TablespaceSummary.ExtentManagement = {}));
    let AllocationType;
    (function (AllocationType) {
        AllocationType["System"] = "SYSTEM";
        AllocationType["Uniform"] = "UNIFORM";
        AllocationType["User"] = "USER";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        AllocationType["UnknownValue"] = "UNKNOWN_VALUE";
    })(AllocationType = TablespaceSummary.AllocationType || (TablespaceSummary.AllocationType = {}));
    let SegmentSpaceManagement;
    (function (SegmentSpaceManagement) {
        SegmentSpaceManagement["Manual"] = "MANUAL";
        SegmentSpaceManagement["Auto"] = "AUTO";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        SegmentSpaceManagement["UnknownValue"] = "UNKNOWN_VALUE";
    })(SegmentSpaceManagement = TablespaceSummary.SegmentSpaceManagement || (TablespaceSummary.SegmentSpaceManagement = {}));
    let DefaultTableCompression;
    (function (DefaultTableCompression) {
        DefaultTableCompression["Enabled"] = "ENABLED";
        DefaultTableCompression["Disabled"] = "DISABLED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        DefaultTableCompression["UnknownValue"] = "UNKNOWN_VALUE";
    })(DefaultTableCompression = TablespaceSummary.DefaultTableCompression || (TablespaceSummary.DefaultTableCompression = {}));
    let Retention;
    (function (Retention) {
        Retention["Guarantee"] = "GUARANTEE";
        Retention["Noguarantee"] = "NOGUARANTEE";
        Retention["NotApply"] = "NOT_APPLY";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Retention["UnknownValue"] = "UNKNOWN_VALUE";
    })(Retention = TablespaceSummary.Retention || (TablespaceSummary.Retention = {}));
    let PredicateEvaluation;
    (function (PredicateEvaluation) {
        PredicateEvaluation["Host"] = "HOST";
        PredicateEvaluation["Storage"] = "STORAGE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        PredicateEvaluation["UnknownValue"] = "UNKNOWN_VALUE";
    })(PredicateEvaluation = TablespaceSummary.PredicateEvaluation || (TablespaceSummary.PredicateEvaluation = {}));
    let CompressFor;
    (function (CompressFor) {
        CompressFor["Basic"] = "BASIC";
        CompressFor["Advanced"] = "ADVANCED";
        CompressFor["QueryLow"] = "QUERY_LOW";
        CompressFor["QueryHigh"] = "QUERY_HIGH";
        CompressFor["ArchiveLow"] = "ARCHIVE_LOW";
        CompressFor["ArchiveHigh"] = "ARCHIVE_HIGH";
        CompressFor["DirectLoadOnly"] = "DIRECT_LOAD_ONLY";
        CompressFor["ForAllOperations"] = "FOR_ALL_OPERATIONS";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        CompressFor["UnknownValue"] = "UNKNOWN_VALUE";
    })(CompressFor = TablespaceSummary.CompressFor || (TablespaceSummary.CompressFor = {}));
    let DefaultInMemory;
    (function (DefaultInMemory) {
        DefaultInMemory["Enabled"] = "ENABLED";
        DefaultInMemory["Disabled"] = "DISABLED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        DefaultInMemory["UnknownValue"] = "UNKNOWN_VALUE";
    })(DefaultInMemory = TablespaceSummary.DefaultInMemory || (TablespaceSummary.DefaultInMemory = {}));
    let DefaultInMemoryPriority;
    (function (DefaultInMemoryPriority) {
        DefaultInMemoryPriority["Low"] = "LOW";
        DefaultInMemoryPriority["Medium"] = "MEDIUM";
        DefaultInMemoryPriority["High"] = "HIGH";
        DefaultInMemoryPriority["Critical"] = "CRITICAL";
        DefaultInMemoryPriority["None"] = "NONE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        DefaultInMemoryPriority["UnknownValue"] = "UNKNOWN_VALUE";
    })(DefaultInMemoryPriority = TablespaceSummary.DefaultInMemoryPriority || (TablespaceSummary.DefaultInMemoryPriority = {}));
    let DefaultInMemoryDistribute;
    (function (DefaultInMemoryDistribute) {
        DefaultInMemoryDistribute["Auto"] = "AUTO";
        DefaultInMemoryDistribute["ByRowidRange"] = "BY_ROWID_RANGE";
        DefaultInMemoryDistribute["ByPartition"] = "BY_PARTITION";
        DefaultInMemoryDistribute["BySubpartition"] = "BY_SUBPARTITION";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        DefaultInMemoryDistribute["UnknownValue"] = "UNKNOWN_VALUE";
    })(DefaultInMemoryDistribute = TablespaceSummary.DefaultInMemoryDistribute || (TablespaceSummary.DefaultInMemoryDistribute = {}));
    let DefaultInMemoryCompression;
    (function (DefaultInMemoryCompression) {
        DefaultInMemoryCompression["NoMemcompress"] = "NO_MEMCOMPRESS";
        DefaultInMemoryCompression["ForDml"] = "FOR_DML";
        DefaultInMemoryCompression["ForQueryLow"] = "FOR_QUERY_LOW";
        DefaultInMemoryCompression["ForQueryHigh"] = "FOR_QUERY_HIGH";
        DefaultInMemoryCompression["ForCapacityLow"] = "FOR_CAPACITY_LOW";
        DefaultInMemoryCompression["ForCapacityHigh"] = "FOR_CAPACITY_HIGH";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        DefaultInMemoryCompression["UnknownValue"] = "UNKNOWN_VALUE";
    })(DefaultInMemoryCompression = TablespaceSummary.DefaultInMemoryCompression || (TablespaceSummary.DefaultInMemoryCompression = {}));
    let DefaultInMemoryDuplicate;
    (function (DefaultInMemoryDuplicate) {
        DefaultInMemoryDuplicate["NoDuplicate"] = "NO_DUPLICATE";
        DefaultInMemoryDuplicate["Duplicate"] = "DUPLICATE";
        DefaultInMemoryDuplicate["DuplicateAll"] = "DUPLICATE_ALL";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        DefaultInMemoryDuplicate["UnknownValue"] = "UNKNOWN_VALUE";
    })(DefaultInMemoryDuplicate = TablespaceSummary.DefaultInMemoryDuplicate || (TablespaceSummary.DefaultInMemoryDuplicate = {}));
    let Shared;
    (function (Shared) {
        Shared["Shared"] = "SHARED";
        Shared["LocalOnLeaf"] = "LOCAL_ON_LEAF";
        Shared["LocalOnAll"] = "LOCAL_ON_ALL";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Shared["UnknownValue"] = "UNKNOWN_VALUE";
    })(Shared = TablespaceSummary.Shared || (TablespaceSummary.Shared = {}));
    let DefaultIndexCompression;
    (function (DefaultIndexCompression) {
        DefaultIndexCompression["Enabled"] = "ENABLED";
        DefaultIndexCompression["Disabled"] = "DISABLED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        DefaultIndexCompression["UnknownValue"] = "UNKNOWN_VALUE";
    })(DefaultIndexCompression = TablespaceSummary.DefaultIndexCompression || (TablespaceSummary.DefaultIndexCompression = {}));
    let IndexCompressFor;
    (function (IndexCompressFor) {
        IndexCompressFor["AdvancedLow"] = "ADVANCED_LOW";
        IndexCompressFor["AdvancedHigh"] = "ADVANCED_HIGH";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        IndexCompressFor["UnknownValue"] = "UNKNOWN_VALUE";
    })(IndexCompressFor = TablespaceSummary.IndexCompressFor || (TablespaceSummary.IndexCompressFor = {}));
    let DefaultInMemoryService;
    (function (DefaultInMemoryService) {
        DefaultInMemoryService["Default"] = "DEFAULT";
        DefaultInMemoryService["None"] = "NONE";
        DefaultInMemoryService["All"] = "ALL";
        DefaultInMemoryService["UserDefined"] = "USER_DEFINED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        DefaultInMemoryService["UnknownValue"] = "UNKNOWN_VALUE";
    })(DefaultInMemoryService = TablespaceSummary.DefaultInMemoryService || (TablespaceSummary.DefaultInMemoryService = {}));
    let LostWriteProtect;
    (function (LostWriteProtect) {
        LostWriteProtect["Enabled"] = "ENABLED";
        LostWriteProtect["ProtectOff"] = "PROTECT_OFF";
        LostWriteProtect["Suspend"] = "SUSPEND";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LostWriteProtect["UnknownValue"] = "UNKNOWN_VALUE";
    })(LostWriteProtect = TablespaceSummary.LostWriteProtect || (TablespaceSummary.LostWriteProtect = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "datafiles": obj.datafiles
                ? obj.datafiles.map(item => {
                    return model.Datafile.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    TablespaceSummary.getJsonObj = getJsonObj;
})(TablespaceSummary = exports.TablespaceSummary || (exports.TablespaceSummary = {}));
