// oci-announcementsservice/lib/model/base-create-announcements-preferences-details.js
"use strict";
/**
 * Announcements Service API
 * Manage Oracle Cloud Infrastructure console announcements.
 * OpenAPI spec version: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BaseCreateAnnouncementsPreferencesDetails = void 0;
const model = __importStar(require("../model"));
var BaseCreateAnnouncementsPreferencesDetails;
(function (BaseCreateAnnouncementsPreferencesDetails) {
    let PreferenceType;
    (function (PreferenceType) {
        PreferenceType["OptInTenantAnnouncements"] = "OPT_IN_TENANT_ANNOUNCEMENTS";
        PreferenceType["OptInTenantAndInformationalAnnouncements"] = "OPT_IN_TENANT_AND_INFORMATIONAL_ANNOUNCEMENTS";
        PreferenceType["OptOutAllAnnouncements"] = "OPT_OUT_ALL_ANNOUNCEMENTS";
    })(PreferenceType = BaseCreateAnnouncementsPreferencesDetails.PreferenceType || (BaseCreateAnnouncementsPreferencesDetails.PreferenceType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        if ("type" in obj && obj.type) {
            switch (obj.type) {
                case "CreateAnnouncementsPreferencesDetails":
                    return model.CreateAnnouncementsPreferencesDetails.getJsonObj(jsonObj, true);
                case "UpdateAnnouncementsPreferencesDetails":
                    return model.UpdateAnnouncementsPreferencesDetails.getJsonObj(jsonObj, true);
                default:
                    throw Error("Unknown value for: " + obj.type);
            }
        }
        return jsonObj;
    }
    BaseCreateAnnouncementsPreferencesDetails.getJsonObj = getJsonObj;
})(BaseCreateAnnouncementsPreferencesDetails = exports.BaseCreateAnnouncementsPreferencesDetails || (exports.BaseCreateAnnouncementsPreferencesDetails = {}));
