// oci-blockchain/lib/model/availability-domain.js
"use strict";
/**
 * Blockchain Platform Control Plane API
 * Blockchain Platform Control Plane API
 * OpenAPI spec version: 20191010
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AvailabilityDomain = void 0;
var AvailabilityDomain;
(function (AvailabilityDomain) {
    let Ads;
    (function (Ads) {
        Ads["Ad1"] = "AD1";
        Ads["Ad2"] = "AD2";
        Ads["Ad3"] = "AD3";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Ads["UnknownValue"] = "UNKNOWN_VALUE";
    })(Ads = AvailabilityDomain.Ads || (AvailabilityDomain.Ads = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AvailabilityDomain.getJsonObj = getJsonObj;
})(AvailabilityDomain = exports.AvailabilityDomain || (exports.AvailabilityDomain = {}));
