// oci-blockchain/lib/model/work-request-resource-sub-type-detail.js
"use strict";
/**
 * Blockchain Platform Control Plane API
 * Blockchain Platform Control Plane API
 * OpenAPI spec version: 20191010
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestResourceSubTypeDetail = void 0;
var WorkRequestResourceSubTypeDetail;
(function (WorkRequestResourceSubTypeDetail) {
    let SubTypeStatus;
    (function (SubTypeStatus) {
        SubTypeStatus["Created"] = "CREATED";
        SubTypeStatus["Updated"] = "UPDATED";
        SubTypeStatus["Deleted"] = "DELETED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        SubTypeStatus["UnknownValue"] = "UNKNOWN_VALUE";
    })(SubTypeStatus = WorkRequestResourceSubTypeDetail.SubTypeStatus || (WorkRequestResourceSubTypeDetail.SubTypeStatus = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    WorkRequestResourceSubTypeDetail.getJsonObj = getJsonObj;
})(WorkRequestResourceSubTypeDetail = exports.WorkRequestResourceSubTypeDetail || (exports.WorkRequestResourceSubTypeDetail = {}));
