// oci-artifacts/lib/request/list-container-image-signatures-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListContainerImageSignaturesRequest = void 0;
var ListContainerImageSignaturesRequest;
(function (ListContainerImageSignaturesRequest) {
    let SigningAlgorithm;
    (function (SigningAlgorithm) {
        SigningAlgorithm["Sha224RsaPkcsPss"] = "SHA_224_RSA_PKCS_PSS";
        SigningAlgorithm["Sha256RsaPkcsPss"] = "SHA_256_RSA_PKCS_PSS";
        SigningAlgorithm["Sha384RsaPkcsPss"] = "SHA_384_RSA_PKCS_PSS";
        SigningAlgorithm["Sha512RsaPkcsPss"] = "SHA_512_RSA_PKCS_PSS";
    })(SigningAlgorithm = ListContainerImageSignaturesRequest.SigningAlgorithm || (ListContainerImageSignaturesRequest.SigningAlgorithm = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["Timecreated"] = "TIMECREATED";
        SortBy["Displayname"] = "DISPLAYNAME";
    })(SortBy = ListContainerImageSignaturesRequest.SortBy || (ListContainerImageSignaturesRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListContainerImageSignaturesRequest.SortOrder || (ListContainerImageSignaturesRequest.SortOrder = {}));
})(ListContainerImageSignaturesRequest = exports.ListContainerImageSignaturesRequest || (exports.ListContainerImageSignaturesRequest = {}));
