// oci-opsi/lib/model/operation-type.js
"use strict";
/**
 * Operations Insights API
 * Use the Operations Insights API to perform data extraction operations to obtain database
resource utilization, performance statistics, and reference information. For more information,
see [About Oracle Cloud Infrastructure Operations Insights](https://docs.cloud.oracle.com/en-us/iaas/operations-insights/doc/operations-insights.html).

 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationType = void 0;
/**
 * Possible operation types.
 **/
var OperationType;
(function (OperationType) {
    OperationType["EnableDatabaseInsight"] = "ENABLE_DATABASE_INSIGHT";
    OperationType["DisableDatabaseInsight"] = "DISABLE_DATABASE_INSIGHT";
    OperationType["UpdateDatabaseInsight"] = "UPDATE_DATABASE_INSIGHT";
    OperationType["CreateDatabaseInsight"] = "CREATE_DATABASE_INSIGHT";
    OperationType["MoveDatabaseInsight"] = "MOVE_DATABASE_INSIGHT";
    OperationType["DeleteDatabaseInsight"] = "DELETE_DATABASE_INSIGHT";
    OperationType["CreateEnterpriseManagerBridge"] = "CREATE_ENTERPRISE_MANAGER_BRIDGE";
    OperationType["UdpateEnterpriseManagerBridge"] = "UDPATE_ENTERPRISE_MANAGER_BRIDGE";
    OperationType["MoveEnterpriseManagerBridge"] = "MOVE_ENTERPRISE_MANAGER_BRIDGE";
    OperationType["DeleteEnterpriseManagerBridge"] = "DELETE_ENTERPRISE_MANAGER_BRIDGE";
    OperationType["EnableHostInsight"] = "ENABLE_HOST_INSIGHT";
    OperationType["DisableHostInsight"] = "DISABLE_HOST_INSIGHT";
    OperationType["UpdateHostInsight"] = "UPDATE_HOST_INSIGHT";
    OperationType["CreateHostInsight"] = "CREATE_HOST_INSIGHT";
    OperationType["MoveHostInsight"] = "MOVE_HOST_INSIGHT";
    OperationType["DeleteHostInsight"] = "DELETE_HOST_INSIGHT";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationType = exports.OperationType || (exports.OperationType = {}));
(function (OperationType) {
    function getJsonObj(obj) {
        return { obj };
    }
    OperationType.getJsonObj = getJsonObj;
})(OperationType = exports.OperationType || (exports.OperationType = {}));
