// oci-nosql/lib/request/list-tables-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190828
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListTablesRequest = void 0;
var ListTablesRequest;
(function (ListTablesRequest) {
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListTablesRequest.SortOrder || (ListTablesRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["Name"] = "name";
    })(SortBy = ListTablesRequest.SortBy || (ListTablesRequest.SortBy = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["All"] = "ALL";
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        LifecycleState["Inactive"] = "INACTIVE";
    })(LifecycleState = ListTablesRequest.LifecycleState || (ListTablesRequest.LifecycleState = {}));
})(ListTablesRequest = exports.ListTablesRequest || (exports.ListTablesRequest = {}));
