// oci-nosql/lib/request/list-indexes-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190828
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListIndexesRequest = void 0;
var ListIndexesRequest;
(function (ListIndexesRequest) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["All"] = "ALL";
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        LifecycleState["Inactive"] = "INACTIVE";
    })(LifecycleState = ListIndexesRequest.LifecycleState || (ListIndexesRequest.LifecycleState = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListIndexesRequest.SortOrder || (ListIndexesRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["Name"] = "name";
    })(SortBy = ListIndexesRequest.SortBy || (ListIndexesRequest.SortBy = {}));
})(ListIndexesRequest = exports.ListIndexesRequest || (exports.ListIndexesRequest = {}));
