// oci-oda/lib/model/create-oda-instance-details.js
"use strict";
/**
 * Digital Assistant Service Instance API
 * API to create and maintain Oracle Digital Assistant service instances.
 * OpenAPI spec version: 20190506
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateOdaInstanceDetails = void 0;
var CreateOdaInstanceDetails;
(function (CreateOdaInstanceDetails) {
    let ShapeName;
    (function (ShapeName) {
        ShapeName["Development"] = "DEVELOPMENT";
        ShapeName["Production"] = "PRODUCTION";
    })(ShapeName = CreateOdaInstanceDetails.ShapeName || (CreateOdaInstanceDetails.ShapeName = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CreateOdaInstanceDetails.getJsonObj = getJsonObj;
})(CreateOdaInstanceDetails = exports.CreateOdaInstanceDetails || (exports.CreateOdaInstanceDetails = {}));
