// oci-oda/lib/model/oda-instance-summary.js
"use strict";
/**
 * Digital Assistant Service Instance API
 * API to create and maintain Oracle Digital Assistant service instances.
 * OpenAPI spec version: 20190506
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OdaInstanceSummary = void 0;
var OdaInstanceSummary;
(function (OdaInstanceSummary) {
    let ShapeName;
    (function (ShapeName) {
        ShapeName["Development"] = "DEVELOPMENT";
        ShapeName["Production"] = "PRODUCTION";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ShapeName["UnknownValue"] = "UNKNOWN_VALUE";
    })(ShapeName = OdaInstanceSummary.ShapeName || (OdaInstanceSummary.ShapeName = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Inactive"] = "INACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = OdaInstanceSummary.LifecycleState || (OdaInstanceSummary.LifecycleState = {}));
    let LifecycleSubState;
    (function (LifecycleSubState) {
        LifecycleSubState["Creating"] = "CREATING";
        LifecycleSubState["Starting"] = "STARTING";
        LifecycleSubState["Stopping"] = "STOPPING";
        LifecycleSubState["ChangingCompartment"] = "CHANGING_COMPARTMENT";
        LifecycleSubState["Deleting"] = "DELETING";
        LifecycleSubState["DeletePending"] = "DELETE_PENDING";
        LifecycleSubState["Recovering"] = "RECOVERING";
        LifecycleSubState["Purging"] = "PURGING";
        LifecycleSubState["Queued"] = "QUEUED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleSubState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleSubState = OdaInstanceSummary.LifecycleSubState || (OdaInstanceSummary.LifecycleSubState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    OdaInstanceSummary.getJsonObj = getJsonObj;
})(OdaInstanceSummary = exports.OdaInstanceSummary || (exports.OdaInstanceSummary = {}));
