// oci-limits/lib/model/index.js
"use strict";
/**
 * Service Limits APIs
 * APIs that interact with the resource limits of a specific resource type
 * OpenAPI spec version: 20181025
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateQuotaDetails = exports.ServiceSummary = exports.ResourceAvailability = exports.QuotaSummary = exports.Quota = exports.LimitValueSummary = exports.LimitDefinitionSummary = exports.CreateQuotaDetails = void 0;
const CreateQuotaDetails = __importStar(require("./create-quota-details"));
exports.CreateQuotaDetails = CreateQuotaDetails.CreateQuotaDetails;
const LimitDefinitionSummary = __importStar(require("./limit-definition-summary"));
exports.LimitDefinitionSummary = LimitDefinitionSummary.LimitDefinitionSummary;
const LimitValueSummary = __importStar(require("./limit-value-summary"));
exports.LimitValueSummary = LimitValueSummary.LimitValueSummary;
const Quota = __importStar(require("./quota"));
exports.Quota = Quota.Quota;
const QuotaSummary = __importStar(require("./quota-summary"));
exports.QuotaSummary = QuotaSummary.QuotaSummary;
const ResourceAvailability = __importStar(require("./resource-availability"));
exports.ResourceAvailability = ResourceAvailability.ResourceAvailability;
const ServiceSummary = __importStar(require("./service-summary"));
exports.ServiceSummary = ServiceSummary.ServiceSummary;
const UpdateQuotaDetails = __importStar(require("./update-quota-details"));
exports.UpdateQuotaDetails = UpdateQuotaDetails.UpdateQuotaDetails;
