// oci-limits/lib/model/limit-value-summary.js
"use strict";
/**
 * Service Limits APIs
 * APIs that interact with the resource limits of a specific resource type
 * OpenAPI spec version: 20181025
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LimitValueSummary = void 0;
var LimitValueSummary;
(function (LimitValueSummary) {
    let ScopeType;
    (function (ScopeType) {
        ScopeType["Global"] = "GLOBAL";
        ScopeType["Region"] = "REGION";
        ScopeType["Ad"] = "AD";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ScopeType["UnknownValue"] = "UNKNOWN_VALUE";
    })(ScopeType = LimitValueSummary.ScopeType || (LimitValueSummary.ScopeType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    LimitValueSummary.getJsonObj = getJsonObj;
})(LimitValueSummary = exports.LimitValueSummary || (exports.LimitValueSummary = {}));
