// oci-streaming/lib/model/create-cursor-details.js
"use strict";
/**
 * Streaming Service API
 * The API for the Streaming Service.
 * OpenAPI spec version: 20180418
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateCursorDetails = void 0;
var CreateCursorDetails;
(function (CreateCursorDetails) {
    let Type;
    (function (Type) {
        Type["AfterOffset"] = "AFTER_OFFSET";
        Type["AtOffset"] = "AT_OFFSET";
        Type["AtTime"] = "AT_TIME";
        Type["Latest"] = "LATEST";
        Type["TrimHorizon"] = "TRIM_HORIZON";
    })(Type = CreateCursorDetails.Type || (CreateCursorDetails.Type = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CreateCursorDetails.getJsonObj = getJsonObj;
})(CreateCursorDetails = exports.CreateCursorDetails || (exports.CreateCursorDetails = {}));
