// oci-streaming/lib/model/custom-encryption-key.js
"use strict";
/**
 * Streaming Service API
 * The API for the Streaming Service.
 * OpenAPI spec version: 20180418
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomEncryptionKey = void 0;
var CustomEncryptionKey;
(function (CustomEncryptionKey) {
    let KeyState;
    (function (KeyState) {
        KeyState["Active"] = "ACTIVE";
        KeyState["Creating"] = "CREATING";
        KeyState["Deleting"] = "DELETING";
        KeyState["None"] = "NONE";
        KeyState["Failed"] = "FAILED";
        KeyState["Updating"] = "UPDATING";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        KeyState["UnknownValue"] = "UNKNOWN_VALUE";
    })(KeyState = CustomEncryptionKey.KeyState || (CustomEncryptionKey.KeyState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CustomEncryptionKey.getJsonObj = getJsonObj;
})(CustomEncryptionKey = exports.CustomEncryptionKey || (exports.CustomEncryptionKey = {}));
