// oci-streaming/lib/model/index.js
"use strict";
/**
 * Streaming Service API
 * The API for the Streaming Service.
 * OpenAPI spec version: 20180418
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateStreamPoolDetails = exports.UpdateStreamDetails = exports.UpdateGroupDetails = exports.UpdateConnectHarnessDetails = exports.StreamSummary = exports.StreamPoolSummary = exports.StreamPool = exports.Stream = exports.PutMessagesResultEntry = exports.PutMessagesResult = exports.PutMessagesDetailsEntry = exports.PutMessagesDetails = exports.PrivateEndpointSettings = exports.PrivateEndpointDetails = exports.PartitionReservation = exports.Message = exports.KafkaSettings = exports.Group = exports.CustomEncryptionKeyDetails = exports.CustomEncryptionKey = exports.Cursor = exports.CreateStreamPoolDetails = exports.CreateStreamDetails = exports.CreateGroupCursorDetails = exports.CreateCursorDetails = exports.CreateConnectHarnessDetails = exports.ConnectHarnessSummary = exports.ConnectHarness = exports.ChangeStreamPoolCompartmentDetails = exports.ChangeStreamCompartmentDetails = exports.ChangeConnectHarnessCompartmentDetails = void 0;
const ChangeConnectHarnessCompartmentDetails = __importStar(require("./change-connect-harness-compartment-details"));
exports.ChangeConnectHarnessCompartmentDetails = ChangeConnectHarnessCompartmentDetails.ChangeConnectHarnessCompartmentDetails;
const ChangeStreamCompartmentDetails = __importStar(require("./change-stream-compartment-details"));
exports.ChangeStreamCompartmentDetails = ChangeStreamCompartmentDetails.ChangeStreamCompartmentDetails;
const ChangeStreamPoolCompartmentDetails = __importStar(require("./change-stream-pool-compartment-details"));
exports.ChangeStreamPoolCompartmentDetails = ChangeStreamPoolCompartmentDetails.ChangeStreamPoolCompartmentDetails;
const ConnectHarness = __importStar(require("./connect-harness"));
exports.ConnectHarness = ConnectHarness.ConnectHarness;
const ConnectHarnessSummary = __importStar(require("./connect-harness-summary"));
exports.ConnectHarnessSummary = ConnectHarnessSummary.ConnectHarnessSummary;
const CreateConnectHarnessDetails = __importStar(require("./create-connect-harness-details"));
exports.CreateConnectHarnessDetails = CreateConnectHarnessDetails.CreateConnectHarnessDetails;
const CreateCursorDetails = __importStar(require("./create-cursor-details"));
exports.CreateCursorDetails = CreateCursorDetails.CreateCursorDetails;
const CreateGroupCursorDetails = __importStar(require("./create-group-cursor-details"));
exports.CreateGroupCursorDetails = CreateGroupCursorDetails.CreateGroupCursorDetails;
const CreateStreamDetails = __importStar(require("./create-stream-details"));
exports.CreateStreamDetails = CreateStreamDetails.CreateStreamDetails;
const CreateStreamPoolDetails = __importStar(require("./create-stream-pool-details"));
exports.CreateStreamPoolDetails = CreateStreamPoolDetails.CreateStreamPoolDetails;
const Cursor = __importStar(require("./cursor"));
exports.Cursor = Cursor.Cursor;
const CustomEncryptionKey = __importStar(require("./custom-encryption-key"));
exports.CustomEncryptionKey = CustomEncryptionKey.CustomEncryptionKey;
const CustomEncryptionKeyDetails = __importStar(require("./custom-encryption-key-details"));
exports.CustomEncryptionKeyDetails = CustomEncryptionKeyDetails.CustomEncryptionKeyDetails;
const Group = __importStar(require("./group"));
exports.Group = Group.Group;
const KafkaSettings = __importStar(require("./kafka-settings"));
exports.KafkaSettings = KafkaSettings.KafkaSettings;
const Message = __importStar(require("./message"));
exports.Message = Message.Message;
const PartitionReservation = __importStar(require("./partition-reservation"));
exports.PartitionReservation = PartitionReservation.PartitionReservation;
const PrivateEndpointDetails = __importStar(require("./private-endpoint-details"));
exports.PrivateEndpointDetails = PrivateEndpointDetails.PrivateEndpointDetails;
const PrivateEndpointSettings = __importStar(require("./private-endpoint-settings"));
exports.PrivateEndpointSettings = PrivateEndpointSettings.PrivateEndpointSettings;
const PutMessagesDetails = __importStar(require("./put-messages-details"));
exports.PutMessagesDetails = PutMessagesDetails.PutMessagesDetails;
const PutMessagesDetailsEntry = __importStar(require("./put-messages-details-entry"));
exports.PutMessagesDetailsEntry = PutMessagesDetailsEntry.PutMessagesDetailsEntry;
const PutMessagesResult = __importStar(require("./put-messages-result"));
exports.PutMessagesResult = PutMessagesResult.PutMessagesResult;
const PutMessagesResultEntry = __importStar(require("./put-messages-result-entry"));
exports.PutMessagesResultEntry = PutMessagesResultEntry.PutMessagesResultEntry;
const Stream = __importStar(require("./stream"));
exports.Stream = Stream.Stream;
const StreamPool = __importStar(require("./stream-pool"));
exports.StreamPool = StreamPool.StreamPool;
const StreamPoolSummary = __importStar(require("./stream-pool-summary"));
exports.StreamPoolSummary = StreamPoolSummary.StreamPoolSummary;
const StreamSummary = __importStar(require("./stream-summary"));
exports.StreamSummary = StreamSummary.StreamSummary;
const UpdateConnectHarnessDetails = __importStar(require("./update-connect-harness-details"));
exports.UpdateConnectHarnessDetails = UpdateConnectHarnessDetails.UpdateConnectHarnessDetails;
const UpdateGroupDetails = __importStar(require("./update-group-details"));
exports.UpdateGroupDetails = UpdateGroupDetails.UpdateGroupDetails;
const UpdateStreamDetails = __importStar(require("./update-stream-details"));
exports.UpdateStreamDetails = UpdateStreamDetails.UpdateStreamDetails;
const UpdateStreamPoolDetails = __importStar(require("./update-stream-pool-details"));
exports.UpdateStreamPoolDetails = UpdateStreamPoolDetails.UpdateStreamPoolDetails;
