// oci-oda/lib/request/list-work-requests-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190506
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWorkRequestsRequest = void 0;
var ListWorkRequestsRequest;
(function (ListWorkRequestsRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["OperationType"] = "OPERATION_TYPE";
        SortBy["Status"] = "STATUS";
        SortBy["TimeAccepted"] = "TIME_ACCEPTED";
        SortBy["TimeStarted"] = "TIME_STARTED";
        SortBy["TimeFinished"] = "TIME_FINISHED";
    })(SortBy = ListWorkRequestsRequest.SortBy || (ListWorkRequestsRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListWorkRequestsRequest.SortOrder || (ListWorkRequestsRequest.SortOrder = {}));
})(ListWorkRequestsRequest = exports.ListWorkRequestsRequest || (exports.ListWorkRequestsRequest = {}));
