// oci-optimizer/lib/request/index.js
"use strict";
/**
 * Cloud Advisor API
 * APIs for managing Cloud Advisor. Cloud Advisor provides recommendations that help you maximize cost savings and improve the security posture of your tenancy.
 * OpenAPI spec version: 20200606
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListResourceActionsRequest = exports.ListRecommendationsRequest = exports.ListRecommendationStrategiesRequest = exports.ListProfilesRequest = exports.ListHistoriesRequest = exports.ListEnrollmentStatusesRequest = exports.ListCategoriesRequest = void 0;
const ListCategoriesRequest = __importStar(require("./list-categories-request"));
exports.ListCategoriesRequest = ListCategoriesRequest.ListCategoriesRequest;
const ListEnrollmentStatusesRequest = __importStar(require("./list-enrollment-statuses-request"));
exports.ListEnrollmentStatusesRequest = ListEnrollmentStatusesRequest.ListEnrollmentStatusesRequest;
const ListHistoriesRequest = __importStar(require("./list-histories-request"));
exports.ListHistoriesRequest = ListHistoriesRequest.ListHistoriesRequest;
const ListProfilesRequest = __importStar(require("./list-profiles-request"));
exports.ListProfilesRequest = ListProfilesRequest.ListProfilesRequest;
const ListRecommendationStrategiesRequest = __importStar(require("./list-recommendation-strategies-request"));
exports.ListRecommendationStrategiesRequest = ListRecommendationStrategiesRequest.ListRecommendationStrategiesRequest;
const ListRecommendationsRequest = __importStar(require("./list-recommendations-request"));
exports.ListRecommendationsRequest = ListRecommendationsRequest.ListRecommendationsRequest;
const ListResourceActionsRequest = __importStar(require("./list-resource-actions-request"));
exports.ListResourceActionsRequest = ListResourceActionsRequest.ListResourceActionsRequest;
