// oci-functions/lib/request/invoke-function-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvokeFunctionRequest = void 0;
var InvokeFunctionRequest;
(function (InvokeFunctionRequest) {
    let FnIntent;
    (function (FnIntent) {
        FnIntent["Httprequest"] = "httprequest";
        FnIntent["Cloudevent"] = "cloudevent";
    })(FnIntent = InvokeFunctionRequest.FnIntent || (InvokeFunctionRequest.FnIntent = {}));
    let FnInvokeType;
    (function (FnInvokeType) {
        FnInvokeType["Detached"] = "detached";
        FnInvokeType["Sync"] = "sync";
    })(FnInvokeType = InvokeFunctionRequest.FnInvokeType || (InvokeFunctionRequest.FnInvokeType = {}));
})(InvokeFunctionRequest = exports.InvokeFunctionRequest || (exports.InvokeFunctionRequest = {}));
