// oci-healthchecks/lib/model/http-probe-method.js
"use strict";
/**
 * Health Checks API
 * API for the Health Checks service. Use this API to manage endpoint probes and monitors.
For more information, see
[Overview of the Health Checks Service](/iaas/Content/HealthChecks/Concepts/healthchecks.htm).

 * OpenAPI spec version: 20180501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.HttpProbeMethod = void 0;
/**
 * The supported HTTP methods available for probes.
 **/
var HttpProbeMethod;
(function (HttpProbeMethod) {
    HttpProbeMethod["Get"] = "GET";
    HttpProbeMethod["Head"] = "HEAD";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    HttpProbeMethod["UnknownValue"] = "UNKNOWN_VALUE";
})(HttpProbeMethod = exports.HttpProbeMethod || (exports.HttpProbeMethod = {}));
(function (HttpProbeMethod) {
    function getJsonObj(obj) {
        return { obj };
    }
    HttpProbeMethod.getJsonObj = getJsonObj;
})(HttpProbeMethod = exports.HttpProbeMethod || (exports.HttpProbeMethod = {}));
