// oci-healthchecks/lib/model/http-probe-result-summary.js
"use strict";
/**
 * Health Checks API
 * API for the Health Checks service. Use this API to manage endpoint probes and monitors.
For more information, see
[Overview of the Health Checks Service](/iaas/Content/HealthChecks/Concepts/healthchecks.htm).

 * OpenAPI spec version: 20180501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.HttpProbeResultSummary = void 0;
const model = __importStar(require("../model"));
var HttpProbeResultSummary;
(function (HttpProbeResultSummary) {
    let ErrorCategory;
    (function (ErrorCategory) {
        ErrorCategory["None"] = "NONE";
        ErrorCategory["Dns"] = "DNS";
        ErrorCategory["Transport"] = "TRANSPORT";
        ErrorCategory["Network"] = "NETWORK";
        ErrorCategory["System"] = "SYSTEM";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ErrorCategory["UnknownValue"] = "UNKNOWN_VALUE";
    })(ErrorCategory = HttpProbeResultSummary.ErrorCategory || (HttpProbeResultSummary.ErrorCategory = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "connection": obj.connection ? model.TcpConnection.getJsonObj(obj.connection) : undefined,
            "dns": obj.dns ? model.DNS.getJsonObj(obj.dns) : undefined
        });
        return jsonObj;
    }
    HttpProbeResultSummary.getJsonObj = getJsonObj;
})(HttpProbeResultSummary = exports.HttpProbeResultSummary || (exports.HttpProbeResultSummary = {}));
