// oci-tenantmanagercontrolplane/lib/model/index.js
"use strict";
/**
 * Organizations API
 * The Organizations API allows you to consolidate multiple OCI tenancies into an organization, and centrally manage your tenancies and its resources.
 * OpenAPI spec version: 20200801
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestSummary = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequestCollection = exports.WorkRequest = exports.UpdateSenderInvitationDetails = exports.UpdateRecipientInvitationDetails = exports.UpdateDomainGovernanceDetails = exports.UpdateDomainDetails = exports.SortOrder = exports.SenderInvitationSummary = exports.SenderInvitationStatus = exports.SenderInvitationCollection = exports.SenderInvitation = exports.RecipientInvitationSummary = exports.RecipientInvitationStatus = exports.RecipientInvitationCollection = exports.RecipientInvitation = exports.OperationType = exports.OperationStatus = exports.LinkSummary = exports.LinkCollection = exports.Link = exports.LifecycleState = exports.DomainSummary = exports.DomainGovernanceSummary = exports.DomainGovernanceCollection = exports.DomainGovernance = exports.DomainCollection = exports.Domain = exports.CreateSenderInvitationDetails = exports.CreateDomainGovernanceDetails = exports.CreateDomainDetails = exports.ActionType = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const CreateDomainDetails = __importStar(require("./create-domain-details"));
exports.CreateDomainDetails = CreateDomainDetails.CreateDomainDetails;
const CreateDomainGovernanceDetails = __importStar(require("./create-domain-governance-details"));
exports.CreateDomainGovernanceDetails = CreateDomainGovernanceDetails.CreateDomainGovernanceDetails;
const CreateSenderInvitationDetails = __importStar(require("./create-sender-invitation-details"));
exports.CreateSenderInvitationDetails = CreateSenderInvitationDetails.CreateSenderInvitationDetails;
const Domain = __importStar(require("./domain"));
exports.Domain = Domain.Domain;
const DomainCollection = __importStar(require("./domain-collection"));
exports.DomainCollection = DomainCollection.DomainCollection;
const DomainGovernance = __importStar(require("./domain-governance"));
exports.DomainGovernance = DomainGovernance.DomainGovernance;
const DomainGovernanceCollection = __importStar(require("./domain-governance-collection"));
exports.DomainGovernanceCollection = DomainGovernanceCollection.DomainGovernanceCollection;
const DomainGovernanceSummary = __importStar(require("./domain-governance-summary"));
exports.DomainGovernanceSummary = DomainGovernanceSummary.DomainGovernanceSummary;
const DomainSummary = __importStar(require("./domain-summary"));
exports.DomainSummary = DomainSummary.DomainSummary;
const LifecycleState = __importStar(require("./lifecycle-state"));
exports.LifecycleState = LifecycleState.LifecycleState;
const Link = __importStar(require("./link"));
exports.Link = Link.Link;
const LinkCollection = __importStar(require("./link-collection"));
exports.LinkCollection = LinkCollection.LinkCollection;
const LinkSummary = __importStar(require("./link-summary"));
exports.LinkSummary = LinkSummary.LinkSummary;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const RecipientInvitation = __importStar(require("./recipient-invitation"));
exports.RecipientInvitation = RecipientInvitation.RecipientInvitation;
const RecipientInvitationCollection = __importStar(require("./recipient-invitation-collection"));
exports.RecipientInvitationCollection = RecipientInvitationCollection.RecipientInvitationCollection;
const RecipientInvitationStatus = __importStar(require("./recipient-invitation-status"));
exports.RecipientInvitationStatus = RecipientInvitationStatus.RecipientInvitationStatus;
const RecipientInvitationSummary = __importStar(require("./recipient-invitation-summary"));
exports.RecipientInvitationSummary = RecipientInvitationSummary.RecipientInvitationSummary;
const SenderInvitation = __importStar(require("./sender-invitation"));
exports.SenderInvitation = SenderInvitation.SenderInvitation;
const SenderInvitationCollection = __importStar(require("./sender-invitation-collection"));
exports.SenderInvitationCollection = SenderInvitationCollection.SenderInvitationCollection;
const SenderInvitationStatus = __importStar(require("./sender-invitation-status"));
exports.SenderInvitationStatus = SenderInvitationStatus.SenderInvitationStatus;
const SenderInvitationSummary = __importStar(require("./sender-invitation-summary"));
exports.SenderInvitationSummary = SenderInvitationSummary.SenderInvitationSummary;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const UpdateDomainDetails = __importStar(require("./update-domain-details"));
exports.UpdateDomainDetails = UpdateDomainDetails.UpdateDomainDetails;
const UpdateDomainGovernanceDetails = __importStar(require("./update-domain-governance-details"));
exports.UpdateDomainGovernanceDetails = UpdateDomainGovernanceDetails.UpdateDomainGovernanceDetails;
const UpdateRecipientInvitationDetails = __importStar(require("./update-recipient-invitation-details"));
exports.UpdateRecipientInvitationDetails = UpdateRecipientInvitationDetails.UpdateRecipientInvitationDetails;
const UpdateSenderInvitationDetails = __importStar(require("./update-sender-invitation-details"));
exports.UpdateSenderInvitationDetails = UpdateSenderInvitationDetails.UpdateSenderInvitationDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestCollection = __importStar(require("./work-request-collection"));
exports.WorkRequestCollection = WorkRequestCollection.WorkRequestCollection;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
