// oci-tenantmanagercontrolplane/lib/model/sender-invitation-status.js
"use strict";
/**
 * Organizations API
 * The Organizations API allows you to consolidate multiple OCI tenancies into an organization, and centrally manage your tenancies and its resources.
 * OpenAPI spec version: 20200801
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SenderInvitationStatus = void 0;
/**
 * Possible statuses for a sender invitation.
 **/
var SenderInvitationStatus;
(function (SenderInvitationStatus) {
    SenderInvitationStatus["Pending"] = "PENDING";
    SenderInvitationStatus["Canceled"] = "CANCELED";
    SenderInvitationStatus["Accepted"] = "ACCEPTED";
    SenderInvitationStatus["Expired"] = "EXPIRED";
    SenderInvitationStatus["Failed"] = "FAILED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    SenderInvitationStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(SenderInvitationStatus = exports.SenderInvitationStatus || (exports.SenderInvitationStatus = {}));
(function (SenderInvitationStatus) {
    function getJsonObj(obj) {
        return { obj };
    }
    SenderInvitationStatus.getJsonObj = getJsonObj;
})(SenderInvitationStatus = exports.SenderInvitationStatus || (exports.SenderInvitationStatus = {}));
