// oci-opsi/lib/request/summarize-database-insight-resource-statistics-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SummarizeDatabaseInsightResourceStatisticsRequest = void 0;
var SummarizeDatabaseInsightResourceStatisticsRequest;
(function (SummarizeDatabaseInsightResourceStatisticsRequest) {
    let DatabaseType;
    (function (DatabaseType) {
        DatabaseType["AdwS"] = "ADW-S";
        DatabaseType["AtpS"] = "ATP-S";
        DatabaseType["AdwD"] = "ADW-D";
        DatabaseType["AtpD"] = "ATP-D";
        DatabaseType["ExternalPdb"] = "EXTERNAL-PDB";
        DatabaseType["ExternalNoncdb"] = "EXTERNAL-NONCDB";
    })(DatabaseType = SummarizeDatabaseInsightResourceStatisticsRequest.DatabaseType || (SummarizeDatabaseInsightResourceStatisticsRequest.DatabaseType = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["UtilizationPercent"] = "utilizationPercent";
        SortBy["Usage"] = "usage";
        SortBy["UsageChangePercent"] = "usageChangePercent";
        SortBy["DatabaseName"] = "databaseName";
        SortBy["DatabaseType"] = "databaseType";
    })(SortBy = SummarizeDatabaseInsightResourceStatisticsRequest.SortBy || (SummarizeDatabaseInsightResourceStatisticsRequest.SortBy = {}));
})(SummarizeDatabaseInsightResourceStatisticsRequest = exports.SummarizeDatabaseInsightResourceStatisticsRequest || (exports.SummarizeDatabaseInsightResourceStatisticsRequest = {}));
