// oci-opsi/lib/request/summarize-database-insight-resource-forecast-trend-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SummarizeDatabaseInsightResourceForecastTrendRequest = void 0;
var SummarizeDatabaseInsightResourceForecastTrendRequest;
(function (SummarizeDatabaseInsightResourceForecastTrendRequest) {
    let DatabaseType;
    (function (DatabaseType) {
        DatabaseType["AdwS"] = "ADW-S";
        DatabaseType["AtpS"] = "ATP-S";
        DatabaseType["AdwD"] = "ADW-D";
        DatabaseType["AtpD"] = "ATP-D";
        DatabaseType["ExternalPdb"] = "EXTERNAL-PDB";
        DatabaseType["ExternalNoncdb"] = "EXTERNAL-NONCDB";
    })(DatabaseType = SummarizeDatabaseInsightResourceForecastTrendRequest.DatabaseType || (SummarizeDatabaseInsightResourceForecastTrendRequest.DatabaseType = {}));
    let Statistic;
    (function (Statistic) {
        Statistic["Avg"] = "AVG";
        Statistic["Max"] = "MAX";
    })(Statistic = SummarizeDatabaseInsightResourceForecastTrendRequest.Statistic || (SummarizeDatabaseInsightResourceForecastTrendRequest.Statistic = {}));
    let ForecastModel;
    (function (ForecastModel) {
        ForecastModel["Linear"] = "LINEAR";
        ForecastModel["MlAuto"] = "ML_AUTO";
        ForecastModel["MlNoAuto"] = "ML_NO_AUTO";
    })(ForecastModel = SummarizeDatabaseInsightResourceForecastTrendRequest.ForecastModel || (SummarizeDatabaseInsightResourceForecastTrendRequest.ForecastModel = {}));
    let UtilizationLevel;
    (function (UtilizationLevel) {
        UtilizationLevel["HighUtilization"] = "HIGH_UTILIZATION";
        UtilizationLevel["LowUtilization"] = "LOW_UTILIZATION";
        UtilizationLevel["MediumHighUtilization"] = "MEDIUM_HIGH_UTILIZATION";
        UtilizationLevel["MediumLowUtilization"] = "MEDIUM_LOW_UTILIZATION";
    })(UtilizationLevel = SummarizeDatabaseInsightResourceForecastTrendRequest.UtilizationLevel || (SummarizeDatabaseInsightResourceForecastTrendRequest.UtilizationLevel = {}));
})(SummarizeDatabaseInsightResourceForecastTrendRequest = exports.SummarizeDatabaseInsightResourceForecastTrendRequest || (exports.SummarizeDatabaseInsightResourceForecastTrendRequest = {}));
