// oci-opsi/lib/request/summarize-host-insight-resource-capacity-trend-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SummarizeHostInsightResourceCapacityTrendRequest = void 0;
var SummarizeHostInsightResourceCapacityTrendRequest;
(function (SummarizeHostInsightResourceCapacityTrendRequest) {
    let PlatformType;
    (function (PlatformType) {
        PlatformType["Linux"] = "LINUX";
    })(PlatformType = SummarizeHostInsightResourceCapacityTrendRequest.PlatformType || (SummarizeHostInsightResourceCapacityTrendRequest.PlatformType = {}));
    let UtilizationLevel;
    (function (UtilizationLevel) {
        UtilizationLevel["HighUtilization"] = "HIGH_UTILIZATION";
        UtilizationLevel["LowUtilization"] = "LOW_UTILIZATION";
        UtilizationLevel["MediumHighUtilization"] = "MEDIUM_HIGH_UTILIZATION";
        UtilizationLevel["MediumLowUtilization"] = "MEDIUM_LOW_UTILIZATION";
    })(UtilizationLevel = SummarizeHostInsightResourceCapacityTrendRequest.UtilizationLevel || (SummarizeHostInsightResourceCapacityTrendRequest.UtilizationLevel = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["EndTimestamp"] = "endTimestamp";
        SortBy["Capacity"] = "capacity";
    })(SortBy = SummarizeHostInsightResourceCapacityTrendRequest.SortBy || (SummarizeHostInsightResourceCapacityTrendRequest.SortBy = {}));
})(SummarizeHostInsightResourceCapacityTrendRequest = exports.SummarizeHostInsightResourceCapacityTrendRequest || (exports.SummarizeHostInsightResourceCapacityTrendRequest = {}));
