// oci-opsi/lib/request/summarize-host-insight-resource-forecast-trend-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SummarizeHostInsightResourceForecastTrendRequest = void 0;
var SummarizeHostInsightResourceForecastTrendRequest;
(function (SummarizeHostInsightResourceForecastTrendRequest) {
    let PlatformType;
    (function (PlatformType) {
        PlatformType["Linux"] = "LINUX";
    })(PlatformType = SummarizeHostInsightResourceForecastTrendRequest.PlatformType || (SummarizeHostInsightResourceForecastTrendRequest.PlatformType = {}));
    let Statistic;
    (function (Statistic) {
        Statistic["Avg"] = "AVG";
        Statistic["Max"] = "MAX";
    })(Statistic = SummarizeHostInsightResourceForecastTrendRequest.Statistic || (SummarizeHostInsightResourceForecastTrendRequest.Statistic = {}));
    let ForecastModel;
    (function (ForecastModel) {
        ForecastModel["Linear"] = "LINEAR";
        ForecastModel["MlAuto"] = "ML_AUTO";
        ForecastModel["MlNoAuto"] = "ML_NO_AUTO";
    })(ForecastModel = SummarizeHostInsightResourceForecastTrendRequest.ForecastModel || (SummarizeHostInsightResourceForecastTrendRequest.ForecastModel = {}));
    let UtilizationLevel;
    (function (UtilizationLevel) {
        UtilizationLevel["HighUtilization"] = "HIGH_UTILIZATION";
        UtilizationLevel["LowUtilization"] = "LOW_UTILIZATION";
        UtilizationLevel["MediumHighUtilization"] = "MEDIUM_HIGH_UTILIZATION";
        UtilizationLevel["MediumLowUtilization"] = "MEDIUM_LOW_UTILIZATION";
    })(UtilizationLevel = SummarizeHostInsightResourceForecastTrendRequest.UtilizationLevel || (SummarizeHostInsightResourceForecastTrendRequest.UtilizationLevel = {}));
})(SummarizeHostInsightResourceForecastTrendRequest = exports.SummarizeHostInsightResourceForecastTrendRequest || (exports.SummarizeHostInsightResourceForecastTrendRequest = {}));
