// oci-opsi/lib/request/summarize-host-insight-resource-statistics-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SummarizeHostInsightResourceStatisticsRequest = void 0;
var SummarizeHostInsightResourceStatisticsRequest;
(function (SummarizeHostInsightResourceStatisticsRequest) {
    let PlatformType;
    (function (PlatformType) {
        PlatformType["Linux"] = "LINUX";
    })(PlatformType = SummarizeHostInsightResourceStatisticsRequest.PlatformType || (SummarizeHostInsightResourceStatisticsRequest.PlatformType = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["UtilizationPercent"] = "utilizationPercent";
        SortBy["Usage"] = "usage";
        SortBy["UsageChangePercent"] = "usageChangePercent";
        SortBy["HostName"] = "hostName";
        SortBy["PlatformType"] = "platformType";
    })(SortBy = SummarizeHostInsightResourceStatisticsRequest.SortBy || (SummarizeHostInsightResourceStatisticsRequest.SortBy = {}));
})(SummarizeHostInsightResourceStatisticsRequest = exports.SummarizeHostInsightResourceStatisticsRequest || (exports.SummarizeHostInsightResourceStatisticsRequest = {}));
