// oci-opsi/lib/request/summarize-host-insight-resource-usage-trend-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SummarizeHostInsightResourceUsageTrendRequest = void 0;
var SummarizeHostInsightResourceUsageTrendRequest;
(function (SummarizeHostInsightResourceUsageTrendRequest) {
    let PlatformType;
    (function (PlatformType) {
        PlatformType["Linux"] = "LINUX";
    })(PlatformType = SummarizeHostInsightResourceUsageTrendRequest.PlatformType || (SummarizeHostInsightResourceUsageTrendRequest.PlatformType = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["EndTimestamp"] = "endTimestamp";
        SortBy["Usage"] = "usage";
        SortBy["Capacity"] = "capacity";
    })(SortBy = SummarizeHostInsightResourceUsageTrendRequest.SortBy || (SummarizeHostInsightResourceUsageTrendRequest.SortBy = {}));
})(SummarizeHostInsightResourceUsageTrendRequest = exports.SummarizeHostInsightResourceUsageTrendRequest || (exports.SummarizeHostInsightResourceUsageTrendRequest = {}));
