// oci-datasafe/lib/model/index.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestSummary = exports.WorkRequestResource = exports.WorkRequestLogEntry = exports.WorkRequestError = exports.WorkRequest = exports.UpdateOnPremConnectorWalletDetails = exports.UpdateOnPremConnectorDetails = exports.UpdateDataSafePrivateEndpointDetails = exports.ServiceList = exports.OnPremConnectorSummary = exports.OnPremConnectorLifecycleState = exports.OnPremConnector = exports.LifecycleState = exports.GenerateOnPremConnectorConfigurationDetails = exports.EnableDataSafeConfigurationDetails = exports.DataSafePrivateEndpointSummary = exports.DataSafePrivateEndpoint = exports.DataSafeConfiguration = exports.CreateOnPremConnectorDetails = exports.CreateDataSafePrivateEndpointDetails = exports.ChangeOnPremConnectorCompartmentDetails = exports.ChangeDataSafePrivateEndpointCompartmentDetails = void 0;
const ChangeDataSafePrivateEndpointCompartmentDetails = __importStar(require("./change-data-safe-private-endpoint-compartment-details"));
exports.ChangeDataSafePrivateEndpointCompartmentDetails = ChangeDataSafePrivateEndpointCompartmentDetails.ChangeDataSafePrivateEndpointCompartmentDetails;
const ChangeOnPremConnectorCompartmentDetails = __importStar(require("./change-on-prem-connector-compartment-details"));
exports.ChangeOnPremConnectorCompartmentDetails = ChangeOnPremConnectorCompartmentDetails.ChangeOnPremConnectorCompartmentDetails;
const CreateDataSafePrivateEndpointDetails = __importStar(require("./create-data-safe-private-endpoint-details"));
exports.CreateDataSafePrivateEndpointDetails = CreateDataSafePrivateEndpointDetails.CreateDataSafePrivateEndpointDetails;
const CreateOnPremConnectorDetails = __importStar(require("./create-on-prem-connector-details"));
exports.CreateOnPremConnectorDetails = CreateOnPremConnectorDetails.CreateOnPremConnectorDetails;
const DataSafeConfiguration = __importStar(require("./data-safe-configuration"));
exports.DataSafeConfiguration = DataSafeConfiguration.DataSafeConfiguration;
const DataSafePrivateEndpoint = __importStar(require("./data-safe-private-endpoint"));
exports.DataSafePrivateEndpoint = DataSafePrivateEndpoint.DataSafePrivateEndpoint;
const DataSafePrivateEndpointSummary = __importStar(require("./data-safe-private-endpoint-summary"));
exports.DataSafePrivateEndpointSummary = DataSafePrivateEndpointSummary.DataSafePrivateEndpointSummary;
const EnableDataSafeConfigurationDetails = __importStar(require("./enable-data-safe-configuration-details"));
exports.EnableDataSafeConfigurationDetails = EnableDataSafeConfigurationDetails.EnableDataSafeConfigurationDetails;
const GenerateOnPremConnectorConfigurationDetails = __importStar(require("./generate-on-prem-connector-configuration-details"));
exports.GenerateOnPremConnectorConfigurationDetails = GenerateOnPremConnectorConfigurationDetails.GenerateOnPremConnectorConfigurationDetails;
const LifecycleState = __importStar(require("./lifecycle-state"));
exports.LifecycleState = LifecycleState.LifecycleState;
const OnPremConnector = __importStar(require("./on-prem-connector"));
exports.OnPremConnector = OnPremConnector.OnPremConnector;
const OnPremConnectorLifecycleState = __importStar(require("./on-prem-connector-lifecycle-state"));
exports.OnPremConnectorLifecycleState = OnPremConnectorLifecycleState.OnPremConnectorLifecycleState;
const OnPremConnectorSummary = __importStar(require("./on-prem-connector-summary"));
exports.OnPremConnectorSummary = OnPremConnectorSummary.OnPremConnectorSummary;
const ServiceList = __importStar(require("./service-list"));
exports.ServiceList = ServiceList.ServiceList;
const UpdateDataSafePrivateEndpointDetails = __importStar(require("./update-data-safe-private-endpoint-details"));
exports.UpdateDataSafePrivateEndpointDetails = UpdateDataSafePrivateEndpointDetails.UpdateDataSafePrivateEndpointDetails;
const UpdateOnPremConnectorDetails = __importStar(require("./update-on-prem-connector-details"));
exports.UpdateOnPremConnectorDetails = UpdateOnPremConnectorDetails.UpdateOnPremConnectorDetails;
const UpdateOnPremConnectorWalletDetails = __importStar(require("./update-on-prem-connector-wallet-details"));
exports.UpdateOnPremConnectorWalletDetails = UpdateOnPremConnectorWalletDetails.UpdateOnPremConnectorWalletDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
