// oci-datasafe/lib/model/on-prem-connector-lifecycle-state.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OnPremConnectorLifecycleState = void 0;
/**
 * The current state of the on-premises connector.
 **/
var OnPremConnectorLifecycleState;
(function (OnPremConnectorLifecycleState) {
    OnPremConnectorLifecycleState["Creating"] = "CREATING";
    OnPremConnectorLifecycleState["Updating"] = "UPDATING";
    OnPremConnectorLifecycleState["Active"] = "ACTIVE";
    OnPremConnectorLifecycleState["Inactive"] = "INACTIVE";
    OnPremConnectorLifecycleState["Deleting"] = "DELETING";
    OnPremConnectorLifecycleState["Deleted"] = "DELETED";
    OnPremConnectorLifecycleState["Failed"] = "FAILED";
})(OnPremConnectorLifecycleState = exports.OnPremConnectorLifecycleState || (exports.OnPremConnectorLifecycleState = {}));
(function (OnPremConnectorLifecycleState) {
    function getJsonObj(obj) {
        return { obj };
    }
    OnPremConnectorLifecycleState.getJsonObj = getJsonObj;
})(OnPremConnectorLifecycleState = exports.OnPremConnectorLifecycleState || (exports.OnPremConnectorLifecycleState = {}));
