// oci-datasafe/lib/model/service-list.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceList = void 0;
/**
 * List of (Oracle) Data Safe services, the only production service is DataSafe, the other services are used in devtest.
 **/
var ServiceList;
(function (ServiceList) {
    ServiceList["DataSafe"] = "DataSafe";
    ServiceList["DataSafeDev"] = "DataSafe-dev";
    ServiceList["DataSafeDev1"] = "DataSafe-dev1";
    ServiceList["DataSafeDev2"] = "DataSafe-dev2";
    ServiceList["DataSafeDev3"] = "DataSafe-dev3";
    ServiceList["DataSafeDev4"] = "DataSafe-dev4";
    ServiceList["DataSafeLrg1"] = "DataSafe-lrg1";
    ServiceList["DataSafeLrg2"] = "DataSafe-lrg2";
    ServiceList["DataSafeLrg3"] = "DataSafe-lrg3";
    ServiceList["DataSafeLrg4"] = "DataSafe-lrg4";
    ServiceList["DataSafePtest"] = "DataSafe-ptest";
    ServiceList["DataSafeStest"] = "DataSafe-stest";
    ServiceList["DataSafeStage"] = "DataSafe-stage";
})(ServiceList = exports.ServiceList || (exports.ServiceList = {}));
(function (ServiceList) {
    function getJsonObj(obj) {
        return { obj };
    }
    ServiceList.getJsonObj = getJsonObj;
})(ServiceList = exports.ServiceList || (exports.ServiceList = {}));
