// oci-datasafe/lib/model/work-request-summary.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestSummary = void 0;
const model = __importStar(require("../model"));
var WorkRequestSummary;
(function (WorkRequestSummary) {
    let OperationType;
    (function (OperationType) {
        OperationType["EnableDataSafeConfiguration"] = "ENABLE_DATA_SAFE_CONFIGURATION";
        OperationType["CreatePrivateEndpoint"] = "CREATE_PRIVATE_ENDPOINT";
        OperationType["UpdatePrivateEndpoint"] = "UPDATE_PRIVATE_ENDPOINT";
        OperationType["DeletePrivateEndpoint"] = "DELETE_PRIVATE_ENDPOINT";
        OperationType["ChangePrivateEndpointCompartment"] = "CHANGE_PRIVATE_ENDPOINT_COMPARTMENT";
        OperationType["CreateOnpremConnector"] = "CREATE_ONPREM_CONNECTOR";
        OperationType["UpdateOnpremConnector"] = "UPDATE_ONPREM_CONNECTOR";
        OperationType["DeleteOnpremConnector"] = "DELETE_ONPREM_CONNECTOR";
        OperationType["UpdateOnpremConnectorWallet"] = "UPDATE_ONPREM_CONNECTOR_WALLET";
        OperationType["ChangeOnpremConnectorCompartment"] = "CHANGE_ONPREM_CONNECTOR_COMPARTMENT";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        OperationType["UnknownValue"] = "UNKNOWN_VALUE";
    })(OperationType = WorkRequestSummary.OperationType || (WorkRequestSummary.OperationType = {}));
    let Status;
    (function (Status) {
        Status["Accepted"] = "ACCEPTED";
        Status["InProgress"] = "IN_PROGRESS";
        Status["Failed"] = "FAILED";
        Status["Succeeded"] = "SUCCEEDED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Status["UnknownValue"] = "UNKNOWN_VALUE";
    })(Status = WorkRequestSummary.Status || (WorkRequestSummary.Status = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "resources": obj.resources
                ? obj.resources.map(item => {
                    return model.WorkRequestResource.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    WorkRequestSummary.getJsonObj = getJsonObj;
})(WorkRequestSummary = exports.WorkRequestSummary || (exports.WorkRequestSummary = {}));
