// oci-loggingsearch/lib/model/field-info.js
"use strict";
/**
 * Logging Search API
 * Search for logs in your compartments, log groups, and log objects.
 * OpenAPI spec version: 20190909
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FieldInfo = void 0;
var FieldInfo;
(function (FieldInfo) {
    let FieldType;
    (function (FieldType) {
        FieldType["String"] = "STRING";
        FieldType["Number"] = "NUMBER";
        FieldType["Boolean"] = "BOOLEAN";
        FieldType["Array"] = "ARRAY";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        FieldType["UnknownValue"] = "UNKNOWN_VALUE";
    })(FieldType = FieldInfo.FieldType || (FieldInfo.FieldType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    FieldInfo.getJsonObj = getJsonObj;
})(FieldInfo = exports.FieldInfo || (exports.FieldInfo = {}));
