// oci-containerengine/lib/model/cluster-lifecycle-state.js
"use strict";
/**
 * Container Engine for Kubernetes API
 * API for the Container Engine for Kubernetes service. Use this API to build, deploy,
and manage cloud-native applications. For more information, see
[Overview of Container Engine for Kubernetes](/iaas/Content/ContEng/Concepts/contengoverview.htm).

 * OpenAPI spec version: 20180222
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ClusterLifecycleState = void 0;
/**
 * The lifecycle state of a cluster.
 **/
var ClusterLifecycleState;
(function (ClusterLifecycleState) {
    ClusterLifecycleState["Creating"] = "CREATING";
    ClusterLifecycleState["Active"] = "ACTIVE";
    ClusterLifecycleState["Failed"] = "FAILED";
    ClusterLifecycleState["Deleting"] = "DELETING";
    ClusterLifecycleState["Deleted"] = "DELETED";
    ClusterLifecycleState["Updating"] = "UPDATING";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ClusterLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(ClusterLifecycleState = exports.ClusterLifecycleState || (exports.ClusterLifecycleState = {}));
(function (ClusterLifecycleState) {
    function getJsonObj(obj) {
        return { obj };
    }
    ClusterLifecycleState.getJsonObj = getJsonObj;
})(ClusterLifecycleState = exports.ClusterLifecycleState || (exports.ClusterLifecycleState = {}));
