// oci-containerengine/lib/model/create-cluster-kubeconfig-content-details.js
"use strict";
/**
 * Container Engine for Kubernetes API
 * API for the Container Engine for Kubernetes service. Use this API to build, deploy,
and manage cloud-native applications. For more information, see
[Overview of Container Engine for Kubernetes](/iaas/Content/ContEng/Concepts/contengoverview.htm).

 * OpenAPI spec version: 20180222
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateClusterKubeconfigContentDetails = void 0;
var CreateClusterKubeconfigContentDetails;
(function (CreateClusterKubeconfigContentDetails) {
    let Endpoint;
    (function (Endpoint) {
        Endpoint["LegacyKubernetes"] = "LEGACY_KUBERNETES";
        Endpoint["PublicEndpoint"] = "PUBLIC_ENDPOINT";
        Endpoint["PrivateEndpoint"] = "PRIVATE_ENDPOINT";
    })(Endpoint = CreateClusterKubeconfigContentDetails.Endpoint || (CreateClusterKubeconfigContentDetails.Endpoint = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CreateClusterKubeconfigContentDetails.getJsonObj = getJsonObj;
})(CreateClusterKubeconfigContentDetails = exports.CreateClusterKubeconfigContentDetails || (exports.CreateClusterKubeconfigContentDetails = {}));
