// oci-containerengine/lib/model/index.js
"use strict";
/**
 * Container Engine for Kubernetes API
 * API for the Container Engine for Kubernetes service. Use this API to build, deploy,
and manage cloud-native applications. For more information, see
[Overview of Container Engine for Kubernetes](/iaas/Content/ContEng/Concepts/contengoverview.htm).

 * OpenAPI spec version: 20180222
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NodeSourceViaImageOption = exports.NodeSourceViaImageDetails = exports.WorkRequestSummary = exports.WorkRequestStatus = exports.WorkRequestResource = exports.WorkRequestOperationType = exports.WorkRequestLogEntry = exports.WorkRequestError = exports.WorkRequest = exports.UpdateNodeShapeConfigDetails = exports.UpdateNodePoolNodeConfigDetails = exports.UpdateNodePoolDetails = exports.UpdateImagePolicyConfigDetails = exports.UpdateClusterOptionsDetails = exports.UpdateClusterEndpointConfigDetails = exports.UpdateClusterDetails = exports.SortOrder = exports.NodeSourceType = exports.NodeSourceOption = exports.NodeSourceDetails = exports.NodeShapeConfig = exports.NodePoolSummary = exports.NodePoolPlacementConfigDetails = exports.NodePoolOptions = exports.NodePoolNodeConfigDetails = exports.NodePool = exports.NodeError = exports.Node = exports.KubernetesNetworkConfig = exports.KeyValue = exports.KeyDetails = exports.ImagePolicyConfig = exports.CreateNodeShapeConfigDetails = exports.CreateNodePoolNodeConfigDetails = exports.CreateNodePoolDetails = exports.CreateImagePolicyConfigDetails = exports.CreateClusterKubeconfigContentDetails = exports.CreateClusterEndpointConfigDetails = exports.CreateClusterDetails = exports.ClusterSummary = exports.ClusterOptions = exports.ClusterMetadata = exports.ClusterLifecycleState = exports.ClusterEndpoints = exports.ClusterEndpointConfig = exports.ClusterCreateOptions = exports.Cluster = exports.AdmissionControllerOptions = exports.AddOnOptions = void 0;
const AddOnOptions = __importStar(require("./add-on-options"));
exports.AddOnOptions = AddOnOptions.AddOnOptions;
const AdmissionControllerOptions = __importStar(require("./admission-controller-options"));
exports.AdmissionControllerOptions = AdmissionControllerOptions.AdmissionControllerOptions;
const Cluster = __importStar(require("./cluster"));
exports.Cluster = Cluster.Cluster;
const ClusterCreateOptions = __importStar(require("./cluster-create-options"));
exports.ClusterCreateOptions = ClusterCreateOptions.ClusterCreateOptions;
const ClusterEndpointConfig = __importStar(require("./cluster-endpoint-config"));
exports.ClusterEndpointConfig = ClusterEndpointConfig.ClusterEndpointConfig;
const ClusterEndpoints = __importStar(require("./cluster-endpoints"));
exports.ClusterEndpoints = ClusterEndpoints.ClusterEndpoints;
const ClusterLifecycleState = __importStar(require("./cluster-lifecycle-state"));
exports.ClusterLifecycleState = ClusterLifecycleState.ClusterLifecycleState;
const ClusterMetadata = __importStar(require("./cluster-metadata"));
exports.ClusterMetadata = ClusterMetadata.ClusterMetadata;
const ClusterOptions = __importStar(require("./cluster-options"));
exports.ClusterOptions = ClusterOptions.ClusterOptions;
const ClusterSummary = __importStar(require("./cluster-summary"));
exports.ClusterSummary = ClusterSummary.ClusterSummary;
const CreateClusterDetails = __importStar(require("./create-cluster-details"));
exports.CreateClusterDetails = CreateClusterDetails.CreateClusterDetails;
const CreateClusterEndpointConfigDetails = __importStar(require("./create-cluster-endpoint-config-details"));
exports.CreateClusterEndpointConfigDetails = CreateClusterEndpointConfigDetails.CreateClusterEndpointConfigDetails;
const CreateClusterKubeconfigContentDetails = __importStar(require("./create-cluster-kubeconfig-content-details"));
exports.CreateClusterKubeconfigContentDetails = CreateClusterKubeconfigContentDetails.CreateClusterKubeconfigContentDetails;
const CreateImagePolicyConfigDetails = __importStar(require("./create-image-policy-config-details"));
exports.CreateImagePolicyConfigDetails = CreateImagePolicyConfigDetails.CreateImagePolicyConfigDetails;
const CreateNodePoolDetails = __importStar(require("./create-node-pool-details"));
exports.CreateNodePoolDetails = CreateNodePoolDetails.CreateNodePoolDetails;
const CreateNodePoolNodeConfigDetails = __importStar(require("./create-node-pool-node-config-details"));
exports.CreateNodePoolNodeConfigDetails = CreateNodePoolNodeConfigDetails.CreateNodePoolNodeConfigDetails;
const CreateNodeShapeConfigDetails = __importStar(require("./create-node-shape-config-details"));
exports.CreateNodeShapeConfigDetails = CreateNodeShapeConfigDetails.CreateNodeShapeConfigDetails;
const ImagePolicyConfig = __importStar(require("./image-policy-config"));
exports.ImagePolicyConfig = ImagePolicyConfig.ImagePolicyConfig;
const KeyDetails = __importStar(require("./key-details"));
exports.KeyDetails = KeyDetails.KeyDetails;
const KeyValue = __importStar(require("./key-value"));
exports.KeyValue = KeyValue.KeyValue;
const KubernetesNetworkConfig = __importStar(require("./kubernetes-network-config"));
exports.KubernetesNetworkConfig = KubernetesNetworkConfig.KubernetesNetworkConfig;
const Node = __importStar(require("./node"));
exports.Node = Node.Node;
const NodeError = __importStar(require("./node-error"));
exports.NodeError = NodeError.NodeError;
const NodePool = __importStar(require("./node-pool"));
exports.NodePool = NodePool.NodePool;
const NodePoolNodeConfigDetails = __importStar(require("./node-pool-node-config-details"));
exports.NodePoolNodeConfigDetails = NodePoolNodeConfigDetails.NodePoolNodeConfigDetails;
const NodePoolOptions = __importStar(require("./node-pool-options"));
exports.NodePoolOptions = NodePoolOptions.NodePoolOptions;
const NodePoolPlacementConfigDetails = __importStar(require("./node-pool-placement-config-details"));
exports.NodePoolPlacementConfigDetails = NodePoolPlacementConfigDetails.NodePoolPlacementConfigDetails;
const NodePoolSummary = __importStar(require("./node-pool-summary"));
exports.NodePoolSummary = NodePoolSummary.NodePoolSummary;
const NodeShapeConfig = __importStar(require("./node-shape-config"));
exports.NodeShapeConfig = NodeShapeConfig.NodeShapeConfig;
const NodeSourceDetails = __importStar(require("./node-source-details"));
exports.NodeSourceDetails = NodeSourceDetails.NodeSourceDetails;
const NodeSourceOption = __importStar(require("./node-source-option"));
exports.NodeSourceOption = NodeSourceOption.NodeSourceOption;
const NodeSourceType = __importStar(require("./node-source-type"));
exports.NodeSourceType = NodeSourceType.NodeSourceType;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const UpdateClusterDetails = __importStar(require("./update-cluster-details"));
exports.UpdateClusterDetails = UpdateClusterDetails.UpdateClusterDetails;
const UpdateClusterEndpointConfigDetails = __importStar(require("./update-cluster-endpoint-config-details"));
exports.UpdateClusterEndpointConfigDetails = UpdateClusterEndpointConfigDetails.UpdateClusterEndpointConfigDetails;
const UpdateClusterOptionsDetails = __importStar(require("./update-cluster-options-details"));
exports.UpdateClusterOptionsDetails = UpdateClusterOptionsDetails.UpdateClusterOptionsDetails;
const UpdateImagePolicyConfigDetails = __importStar(require("./update-image-policy-config-details"));
exports.UpdateImagePolicyConfigDetails = UpdateImagePolicyConfigDetails.UpdateImagePolicyConfigDetails;
const UpdateNodePoolDetails = __importStar(require("./update-node-pool-details"));
exports.UpdateNodePoolDetails = UpdateNodePoolDetails.UpdateNodePoolDetails;
const UpdateNodePoolNodeConfigDetails = __importStar(require("./update-node-pool-node-config-details"));
exports.UpdateNodePoolNodeConfigDetails = UpdateNodePoolNodeConfigDetails.UpdateNodePoolNodeConfigDetails;
const UpdateNodeShapeConfigDetails = __importStar(require("./update-node-shape-config-details"));
exports.UpdateNodeShapeConfigDetails = UpdateNodeShapeConfigDetails.UpdateNodeShapeConfigDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestOperationType = __importStar(require("./work-request-operation-type"));
exports.WorkRequestOperationType = WorkRequestOperationType.WorkRequestOperationType;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestStatus = __importStar(require("./work-request-status"));
exports.WorkRequestStatus = WorkRequestStatus.WorkRequestStatus;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const NodeSourceViaImageDetails = __importStar(require("./node-source-via-image-details"));
exports.NodeSourceViaImageDetails = NodeSourceViaImageDetails.NodeSourceViaImageDetails;
const NodeSourceViaImageOption = __importStar(require("./node-source-via-image-option"));
exports.NodeSourceViaImageOption = NodeSourceViaImageOption.NodeSourceViaImageOption;
