// oci-containerengine/lib/model/work-request-resource.js
"use strict";
/**
 * Container Engine for Kubernetes API
 * API for the Container Engine for Kubernetes service. Use this API to build, deploy,
and manage cloud-native applications. For more information, see
[Overview of Container Engine for Kubernetes](/iaas/Content/ContEng/Concepts/contengoverview.htm).

 * OpenAPI spec version: 20180222
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestResource = void 0;
var WorkRequestResource;
(function (WorkRequestResource) {
    let ActionType;
    (function (ActionType) {
        ActionType["Created"] = "CREATED";
        ActionType["Updated"] = "UPDATED";
        ActionType["Deleted"] = "DELETED";
        ActionType["Related"] = "RELATED";
        ActionType["InProgress"] = "IN_PROGRESS";
        ActionType["Failed"] = "FAILED";
        ActionType["CanceledCreate"] = "CANCELED_CREATE";
        ActionType["CanceledUpdate"] = "CANCELED_UPDATE";
        ActionType["CanceledDelete"] = "CANCELED_DELETE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ActionType["UnknownValue"] = "UNKNOWN_VALUE";
    })(ActionType = WorkRequestResource.ActionType || (WorkRequestResource.ActionType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    WorkRequestResource.getJsonObj = getJsonObj;
})(WorkRequestResource = exports.WorkRequestResource || (exports.WorkRequestResource = {}));
