// oci-apigateway/lib/model/execution-log-policy.js
"use strict";
/**
 * API Gateway API
 * API for the API Gateway service. Use this API to manage gateways, deployments, and related items.
For more information, see
[Overview of API Gateway](/iaas/Content/APIGateway/Concepts/apigatewayoverview.htm).

 * OpenAPI spec version: 20190501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExecutionLogPolicy = void 0;
var ExecutionLogPolicy;
(function (ExecutionLogPolicy) {
    let LogLevel;
    (function (LogLevel) {
        LogLevel["Info"] = "INFO";
        LogLevel["Warn"] = "WARN";
        LogLevel["Error"] = "ERROR";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LogLevel["UnknownValue"] = "UNKNOWN_VALUE";
    })(LogLevel = ExecutionLogPolicy.LogLevel || (ExecutionLogPolicy.LogLevel = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ExecutionLogPolicy.getJsonObj = getJsonObj;
})(ExecutionLogPolicy = exports.ExecutionLogPolicy || (exports.ExecutionLogPolicy = {}));
