// oci-apigateway/lib/model/set-query-parameter-policy-item.js
"use strict";
/**
 * API Gateway API
 * API for the API Gateway service. Use this API to manage gateways, deployments, and related items.
For more information, see
[Overview of API Gateway](/iaas/Content/APIGateway/Concepts/apigatewayoverview.htm).

 * OpenAPI spec version: 20190501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SetQueryParameterPolicyItem = void 0;
var SetQueryParameterPolicyItem;
(function (SetQueryParameterPolicyItem) {
    let IfExists;
    (function (IfExists) {
        IfExists["Overwrite"] = "OVERWRITE";
        IfExists["Append"] = "APPEND";
        IfExists["Skip"] = "SKIP";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        IfExists["UnknownValue"] = "UNKNOWN_VALUE";
    })(IfExists = SetQueryParameterPolicyItem.IfExists || (SetQueryParameterPolicyItem.IfExists = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SetQueryParameterPolicyItem.getJsonObj = getJsonObj;
})(SetQueryParameterPolicyItem = exports.SetQueryParameterPolicyItem || (exports.SetQueryParameterPolicyItem = {}));
