// oci-dns/lib/request/index.js
"use strict";
/**
 * DNS API
 * API for the DNS service. Use this API to manage DNS zones, records, and other DNS resources.
For more information, see [Overview of the DNS Service](/iaas/Content/DNS/Concepts/dnszonemanagement.htm).

 * OpenAPI spec version: 20180115
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListZonesRequest = exports.ListViewsRequest = exports.ListTsigKeysRequest = exports.ListSteeringPolicyAttachmentsRequest = exports.ListSteeringPoliciesRequest = exports.ListResolversRequest = exports.ListResolverEndpointsRequest = exports.GetZoneRecordsRequest = exports.GetDomainRecordsRequest = void 0;
const GetDomainRecordsRequest = __importStar(require("./get-domain-records-request"));
exports.GetDomainRecordsRequest = GetDomainRecordsRequest.GetDomainRecordsRequest;
const GetZoneRecordsRequest = __importStar(require("./get-zone-records-request"));
exports.GetZoneRecordsRequest = GetZoneRecordsRequest.GetZoneRecordsRequest;
const ListResolverEndpointsRequest = __importStar(require("./list-resolver-endpoints-request"));
exports.ListResolverEndpointsRequest = ListResolverEndpointsRequest.ListResolverEndpointsRequest;
const ListResolversRequest = __importStar(require("./list-resolvers-request"));
exports.ListResolversRequest = ListResolversRequest.ListResolversRequest;
const ListSteeringPoliciesRequest = __importStar(require("./list-steering-policies-request"));
exports.ListSteeringPoliciesRequest = ListSteeringPoliciesRequest.ListSteeringPoliciesRequest;
const ListSteeringPolicyAttachmentsRequest = __importStar(require("./list-steering-policy-attachments-request"));
exports.ListSteeringPolicyAttachmentsRequest = ListSteeringPolicyAttachmentsRequest.ListSteeringPolicyAttachmentsRequest;
const ListTsigKeysRequest = __importStar(require("./list-tsig-keys-request"));
exports.ListTsigKeysRequest = ListTsigKeysRequest.ListTsigKeysRequest;
const ListViewsRequest = __importStar(require("./list-views-request"));
exports.ListViewsRequest = ListViewsRequest.ListViewsRequest;
const ListZonesRequest = __importStar(require("./list-zones-request"));
exports.ListZonesRequest = ListZonesRequest.ListZonesRequest;
