// oci-dns/lib/request/list-zones-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20180115
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListZonesRequest = void 0;
var ListZonesRequest;
(function (ListZonesRequest) {
    let ZoneType;
    (function (ZoneType) {
        ZoneType["Primary"] = "PRIMARY";
        ZoneType["Secondary"] = "SECONDARY";
    })(ZoneType = ListZonesRequest.ZoneType || (ListZonesRequest.ZoneType = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Failed"] = "FAILED";
    })(LifecycleState = ListZonesRequest.LifecycleState || (ListZonesRequest.LifecycleState = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["Name"] = "name";
        SortBy["ZoneType"] = "zoneType";
        SortBy["TimeCreated"] = "timeCreated";
    })(SortBy = ListZonesRequest.SortBy || (ListZonesRequest.SortBy = {}));
})(ListZonesRequest = exports.ListZonesRequest || (exports.ListZonesRequest = {}));
