// oci-dns/lib/request/list-steering-policies-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20180115
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListSteeringPoliciesRequest = void 0;
var ListSteeringPoliciesRequest;
(function (ListSteeringPoliciesRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["DisplayName"] = "displayName";
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["Template"] = "template";
    })(SortBy = ListSteeringPoliciesRequest.SortBy || (ListSteeringPoliciesRequest.SortBy = {}));
})(ListSteeringPoliciesRequest = exports.ListSteeringPoliciesRequest || (exports.ListSteeringPoliciesRequest = {}));
