// oci-objectstorage/lib/request/get-bucket-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetBucketRequest = void 0;
var GetBucketRequest;
(function (GetBucketRequest) {
    let Fields;
    (function (Fields) {
        Fields["ApproximateCount"] = "approximateCount";
        Fields["ApproximateSize"] = "approximateSize";
        Fields["AutoTiering"] = "autoTiering";
    })(Fields = GetBucketRequest.Fields || (GetBucketRequest.Fields = {}));
})(GetBucketRequest = exports.GetBucketRequest || (exports.GetBucketRequest = {}));
