// oci-sch/lib/model/action-type.js
"use strict";
/**
 * Service Connector Hub API
 * Use the Service Connector Hub API to transfer data between services in Oracle Cloud Infrastructure.
For more information about Service Connector Hub, see
[Service Connector Hub Overview](/iaas/Content/service-connector-hub/overview.htm).

 * OpenAPI spec version: 20200909
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ActionType = void 0;
/**
 * Possible types of actions.
 *
 **/
var ActionType;
(function (ActionType) {
    ActionType["Created"] = "CREATED";
    ActionType["Updated"] = "UPDATED";
    ActionType["Deleted"] = "DELETED";
    ActionType["InProgress"] = "IN_PROGRESS";
    ActionType["Related"] = "RELATED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ActionType["UnknownValue"] = "UNKNOWN_VALUE";
})(ActionType = exports.ActionType || (exports.ActionType = {}));
(function (ActionType) {
    function getJsonObj(obj) {
        return { obj };
    }
    ActionType.getJsonObj = getJsonObj;
})(ActionType = exports.ActionType || (exports.ActionType = {}));
