// oci-sch/lib/model/operation-type.js
"use strict";
/**
 * Service Connector Hub API
 * Use the Service Connector Hub API to transfer data between services in Oracle Cloud Infrastructure.
For more information about Service Connector Hub, see
[Service Connector Hub Overview](/iaas/Content/service-connector-hub/overview.htm).

 * OpenAPI spec version: 20200909
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationType = void 0;
/**
 * Possible operation types.
 *
 **/
var OperationType;
(function (OperationType) {
    OperationType["CreateServiceConnector"] = "CREATE_SERVICE_CONNECTOR";
    OperationType["UpdateServiceConnector"] = "UPDATE_SERVICE_CONNECTOR";
    OperationType["DeleteServiceConnector"] = "DELETE_SERVICE_CONNECTOR";
    OperationType["ActivateServiceConnector"] = "ACTIVATE_SERVICE_CONNECTOR";
    OperationType["DeactivateServiceConnector"] = "DEACTIVATE_SERVICE_CONNECTOR";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationType = exports.OperationType || (exports.OperationType = {}));
(function (OperationType) {
    function getJsonObj(obj) {
        return { obj };
    }
    OperationType.getJsonObj = getJsonObj;
})(OperationType = exports.OperationType || (exports.OperationType = {}));
