// oci-databasemanagement/lib/request/index.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to perform tasks such as obtaining performance and resource usage metrics
for a fleet of Managed Databases or a specific Managed Database, creating Managed Database Groups, and
running a SQL job on a Managed Database or Managed Database Group.

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListTablespacesRequest = exports.ListManagedDatabasesRequest = exports.ListManagedDatabaseGroupsRequest = exports.ListJobsRequest = exports.ListJobRunsRequest = exports.ListJobExecutionsRequest = exports.ListDatabaseParametersRequest = void 0;
const ListDatabaseParametersRequest = __importStar(require("./list-database-parameters-request"));
exports.ListDatabaseParametersRequest = ListDatabaseParametersRequest.ListDatabaseParametersRequest;
const ListJobExecutionsRequest = __importStar(require("./list-job-executions-request"));
exports.ListJobExecutionsRequest = ListJobExecutionsRequest.ListJobExecutionsRequest;
const ListJobRunsRequest = __importStar(require("./list-job-runs-request"));
exports.ListJobRunsRequest = ListJobRunsRequest.ListJobRunsRequest;
const ListJobsRequest = __importStar(require("./list-jobs-request"));
exports.ListJobsRequest = ListJobsRequest.ListJobsRequest;
const ListManagedDatabaseGroupsRequest = __importStar(require("./list-managed-database-groups-request"));
exports.ListManagedDatabaseGroupsRequest = ListManagedDatabaseGroupsRequest.ListManagedDatabaseGroupsRequest;
const ListManagedDatabasesRequest = __importStar(require("./list-managed-databases-request"));
exports.ListManagedDatabasesRequest = ListManagedDatabasesRequest.ListManagedDatabasesRequest;
const ListTablespacesRequest = __importStar(require("./list-tablespaces-request"));
exports.ListTablespacesRequest = ListTablespacesRequest.ListTablespacesRequest;
