// oci-databasemanagement/lib/request/list-database-parameters-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListDatabaseParametersRequest = void 0;
var ListDatabaseParametersRequest;
(function (ListDatabaseParametersRequest) {
    let Source;
    (function (Source) {
        Source["Current"] = "CURRENT";
        Source["Spfile"] = "SPFILE";
    })(Source = ListDatabaseParametersRequest.Source || (ListDatabaseParametersRequest.Source = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["Name"] = "NAME";
    })(SortBy = ListDatabaseParametersRequest.SortBy || (ListDatabaseParametersRequest.SortBy = {}));
})(ListDatabaseParametersRequest = exports.ListDatabaseParametersRequest || (exports.ListDatabaseParametersRequest = {}));
