// oci-apmtraces/lib/model/index.js
"use strict";
/**
 * Apm Traces API
 * API for APM Trace service. Use this API to query the Traces and associated Spans.

 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TraceSpanSummary = exports.TraceServiceSummary = exports.Trace = exports.Tag = exports.SpanLogCollection = exports.SpanLog = exports.Span = exports.QuickPickSummary = exports.QueryResultsOrderedBySummary = exports.QueryResultsGroupedBySummary = exports.QueryResultRowTypeSummary = exports.QueryResultRow = exports.QueryResultResponse = exports.QueryResultMetadataSummary = exports.QueryDetails = void 0;
const QueryDetails = __importStar(require("./query-details"));
exports.QueryDetails = QueryDetails.QueryDetails;
const QueryResultMetadataSummary = __importStar(require("./query-result-metadata-summary"));
exports.QueryResultMetadataSummary = QueryResultMetadataSummary.QueryResultMetadataSummary;
const QueryResultResponse = __importStar(require("./query-result-response"));
exports.QueryResultResponse = QueryResultResponse.QueryResultResponse;
const QueryResultRow = __importStar(require("./query-result-row"));
exports.QueryResultRow = QueryResultRow.QueryResultRow;
const QueryResultRowTypeSummary = __importStar(require("./query-result-row-type-summary"));
exports.QueryResultRowTypeSummary = QueryResultRowTypeSummary.QueryResultRowTypeSummary;
const QueryResultsGroupedBySummary = __importStar(require("./query-results-grouped-by-summary"));
exports.QueryResultsGroupedBySummary = QueryResultsGroupedBySummary.QueryResultsGroupedBySummary;
const QueryResultsOrderedBySummary = __importStar(require("./query-results-ordered-by-summary"));
exports.QueryResultsOrderedBySummary = QueryResultsOrderedBySummary.QueryResultsOrderedBySummary;
const QuickPickSummary = __importStar(require("./quick-pick-summary"));
exports.QuickPickSummary = QuickPickSummary.QuickPickSummary;
const Span = __importStar(require("./span"));
exports.Span = Span.Span;
const SpanLog = __importStar(require("./span-log"));
exports.SpanLog = SpanLog.SpanLog;
const SpanLogCollection = __importStar(require("./span-log-collection"));
exports.SpanLogCollection = SpanLogCollection.SpanLogCollection;
const Tag = __importStar(require("./tag"));
exports.Tag = Tag.Tag;
const Trace = __importStar(require("./trace"));
exports.Trace = Trace.Trace;
const TraceServiceSummary = __importStar(require("./trace-service-summary"));
exports.TraceServiceSummary = TraceServiceSummary.TraceServiceSummary;
const TraceSpanSummary = __importStar(require("./trace-span-summary"));
exports.TraceSpanSummary = TraceSpanSummary.TraceSpanSummary;
