// oci-monitoring/lib/request/list-alarms-status-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20180401
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListAlarmsStatusRequest = void 0;
var ListAlarmsStatusRequest;
(function (ListAlarmsStatusRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["DisplayName"] = "displayName";
        SortBy["Severity"] = "severity";
    })(SortBy = ListAlarmsStatusRequest.SortBy || (ListAlarmsStatusRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListAlarmsStatusRequest.SortOrder || (ListAlarmsStatusRequest.SortOrder = {}));
})(ListAlarmsStatusRequest = exports.ListAlarmsStatusRequest || (exports.ListAlarmsStatusRequest = {}));
