// oci-database/lib/database-waiter.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatabaseWaiter = void 0;
const models = __importStar(require("./model"));
const oci_common_1 = require("oci-common");
const oci_workrequests_1 = require("oci-workrequests");
class DatabaseWaiter {
    constructor(client, workRequestClient, config) {
        this.client = client;
        this.workRequestClient = workRequestClient;
        this.config = config;
    }
    /**
     * Waits forActivateExadataInfrastructure
     *
     * @param request the request to send
     * @return response returns ActivateExadataInfrastructureResponse, GetWorkRequestResponse tuple
     */
    forActivateExadataInfrastructure(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const activateExadataInfrastructureResponse = yield this.client.activateExadataInfrastructure(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, activateExadataInfrastructureResponse.opcWorkRequestId);
            return {
                response: activateExadataInfrastructureResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forAutonomousDatabaseManualRefresh
     *
     * @param request the request to send
     * @return response returns AutonomousDatabaseManualRefreshResponse, GetWorkRequestResponse tuple
     */
    forAutonomousDatabaseManualRefresh(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const autonomousDatabaseManualRefreshResponse = yield this.client.autonomousDatabaseManualRefresh(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, autonomousDatabaseManualRefreshResponse.opcWorkRequestId);
            return {
                response: autonomousDatabaseManualRefreshResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forChangeAutonomousContainerDatabaseCompartment
     *
     * @param request the request to send
     * @return response returns ChangeAutonomousContainerDatabaseCompartmentResponse, GetWorkRequestResponse tuple
     */
    forChangeAutonomousContainerDatabaseCompartment(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const changeAutonomousContainerDatabaseCompartmentResponse = yield this.client.changeAutonomousContainerDatabaseCompartment(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, changeAutonomousContainerDatabaseCompartmentResponse.opcWorkRequestId);
            return {
                response: changeAutonomousContainerDatabaseCompartmentResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forChangeAutonomousDatabaseCompartment
     *
     * @param request the request to send
     * @return response returns ChangeAutonomousDatabaseCompartmentResponse, GetWorkRequestResponse tuple
     */
    forChangeAutonomousDatabaseCompartment(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const changeAutonomousDatabaseCompartmentResponse = yield this.client.changeAutonomousDatabaseCompartment(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, changeAutonomousDatabaseCompartmentResponse.opcWorkRequestId);
            return {
                response: changeAutonomousDatabaseCompartmentResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forChangeAutonomousExadataInfrastructureCompartment
     *
     * @param request the request to send
     * @return response returns ChangeAutonomousExadataInfrastructureCompartmentResponse, GetWorkRequestResponse tuple
     */
    forChangeAutonomousExadataInfrastructureCompartment(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const changeAutonomousExadataInfrastructureCompartmentResponse = yield this.client.changeAutonomousExadataInfrastructureCompartment(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, changeAutonomousExadataInfrastructureCompartmentResponse.opcWorkRequestId);
            return {
                response: changeAutonomousExadataInfrastructureCompartmentResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forChangeAutonomousVmClusterCompartment
     *
     * @param request the request to send
     * @return response returns ChangeAutonomousVmClusterCompartmentResponse, GetWorkRequestResponse tuple
     */
    forChangeAutonomousVmClusterCompartment(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const changeAutonomousVmClusterCompartmentResponse = yield this.client.changeAutonomousVmClusterCompartment(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, changeAutonomousVmClusterCompartmentResponse.opcWorkRequestId);
            return {
                response: changeAutonomousVmClusterCompartmentResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forChangeBackupDestinationCompartment
     *
     * @param request the request to send
     * @return response returns ChangeBackupDestinationCompartmentResponse, GetWorkRequestResponse tuple
     */
    forChangeBackupDestinationCompartment(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const changeBackupDestinationCompartmentResponse = yield this.client.changeBackupDestinationCompartment(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, changeBackupDestinationCompartmentResponse.opcWorkRequestId);
            return {
                response: changeBackupDestinationCompartmentResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forChangeCloudExadataInfrastructureCompartment
     *
     * @param request the request to send
     * @return response returns ChangeCloudExadataInfrastructureCompartmentResponse, GetWorkRequestResponse tuple
     */
    forChangeCloudExadataInfrastructureCompartment(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const changeCloudExadataInfrastructureCompartmentResponse = yield this.client.changeCloudExadataInfrastructureCompartment(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, changeCloudExadataInfrastructureCompartmentResponse.opcWorkRequestId);
            return {
                response: changeCloudExadataInfrastructureCompartmentResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forChangeCloudVmClusterCompartment
     *
     * @param request the request to send
     * @return response returns ChangeCloudVmClusterCompartmentResponse, GetWorkRequestResponse tuple
     */
    forChangeCloudVmClusterCompartment(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const changeCloudVmClusterCompartmentResponse = yield this.client.changeCloudVmClusterCompartment(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, changeCloudVmClusterCompartmentResponse.opcWorkRequestId);
            return {
                response: changeCloudVmClusterCompartmentResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forChangeDatabaseSoftwareImageCompartment
     *
     * @param request the request to send
     * @return response returns ChangeDatabaseSoftwareImageCompartmentResponse, GetWorkRequestResponse tuple
     */
    forChangeDatabaseSoftwareImageCompartment(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const changeDatabaseSoftwareImageCompartmentResponse = yield this.client.changeDatabaseSoftwareImageCompartment(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, changeDatabaseSoftwareImageCompartmentResponse.opcWorkRequestId);
            return {
                response: changeDatabaseSoftwareImageCompartmentResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forChangeDbSystemCompartment
     *
     * @param request the request to send
     * @return response returns ChangeDbSystemCompartmentResponse, GetWorkRequestResponse tuple
     */
    forChangeDbSystemCompartment(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const changeDbSystemCompartmentResponse = yield this.client.changeDbSystemCompartment(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, changeDbSystemCompartmentResponse.opcWorkRequestId);
            return {
                response: changeDbSystemCompartmentResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forChangeExadataInfrastructureCompartment
     *
     * @param request the request to send
     * @return response returns ChangeExadataInfrastructureCompartmentResponse, GetWorkRequestResponse tuple
     */
    forChangeExadataInfrastructureCompartment(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const changeExadataInfrastructureCompartmentResponse = yield this.client.changeExadataInfrastructureCompartment(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, changeExadataInfrastructureCompartmentResponse.opcWorkRequestId);
            return {
                response: changeExadataInfrastructureCompartmentResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forChangeExternalContainerDatabaseCompartment
     *
     * @param request the request to send
     * @return response returns ChangeExternalContainerDatabaseCompartmentResponse, GetWorkRequestResponse tuple
     */
    forChangeExternalContainerDatabaseCompartment(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const changeExternalContainerDatabaseCompartmentResponse = yield this.client.changeExternalContainerDatabaseCompartment(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, changeExternalContainerDatabaseCompartmentResponse.opcWorkRequestId);
            return {
                response: changeExternalContainerDatabaseCompartmentResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forChangeExternalNonContainerDatabaseCompartment
     *
     * @param request the request to send
     * @return response returns ChangeExternalNonContainerDatabaseCompartmentResponse, GetWorkRequestResponse tuple
     */
    forChangeExternalNonContainerDatabaseCompartment(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const changeExternalNonContainerDatabaseCompartmentResponse = yield this.client.changeExternalNonContainerDatabaseCompartment(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, changeExternalNonContainerDatabaseCompartmentResponse.opcWorkRequestId);
            return {
                response: changeExternalNonContainerDatabaseCompartmentResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forChangeExternalPluggableDatabaseCompartment
     *
     * @param request the request to send
     * @return response returns ChangeExternalPluggableDatabaseCompartmentResponse, GetWorkRequestResponse tuple
     */
    forChangeExternalPluggableDatabaseCompartment(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const changeExternalPluggableDatabaseCompartmentResponse = yield this.client.changeExternalPluggableDatabaseCompartment(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, changeExternalPluggableDatabaseCompartmentResponse.opcWorkRequestId);
            return {
                response: changeExternalPluggableDatabaseCompartmentResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forChangeKeyStoreCompartment
     *
     * @param request the request to send
     * @return response returns ChangeKeyStoreCompartmentResponse, GetWorkRequestResponse tuple
     */
    forChangeKeyStoreCompartment(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const changeKeyStoreCompartmentResponse = yield this.client.changeKeyStoreCompartment(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, changeKeyStoreCompartmentResponse.opcWorkRequestId);
            return {
                response: changeKeyStoreCompartmentResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forChangeVmClusterCompartment
     *
     * @param request the request to send
     * @return response returns ChangeVmClusterCompartmentResponse, GetWorkRequestResponse tuple
     */
    forChangeVmClusterCompartment(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const changeVmClusterCompartmentResponse = yield this.client.changeVmClusterCompartment(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, changeVmClusterCompartmentResponse.opcWorkRequestId);
            return {
                response: changeVmClusterCompartmentResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forCheckExternalDatabaseConnectorConnectionStatus
     *
     * @param request the request to send
     * @return response returns CheckExternalDatabaseConnectorConnectionStatusResponse, GetWorkRequestResponse tuple
     */
    forCheckExternalDatabaseConnectorConnectionStatus(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const checkExternalDatabaseConnectorConnectionStatusResponse = yield this.client.checkExternalDatabaseConnectorConnectionStatus(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, checkExternalDatabaseConnectorConnectionStatusResponse.opcWorkRequestId);
            return {
                response: checkExternalDatabaseConnectorConnectionStatusResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forCompleteExternalBackupJob
     *
     * @param request the request to send
     * @return response returns CompleteExternalBackupJobResponse, GetWorkRequestResponse tuple
     */
    forCompleteExternalBackupJob(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const completeExternalBackupJobResponse = yield this.client.completeExternalBackupJob(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, completeExternalBackupJobResponse.opcWorkRequestId);
            return {
                response: completeExternalBackupJobResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forCreateAutonomousContainerDatabase
     *
     * @param request the request to send
     * @return response returns CreateAutonomousContainerDatabaseResponse, GetWorkRequestResponse tuple
     */
    forCreateAutonomousContainerDatabase(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const createAutonomousContainerDatabaseResponse = yield this.client.createAutonomousContainerDatabase(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, createAutonomousContainerDatabaseResponse.opcWorkRequestId);
            return {
                response: createAutonomousContainerDatabaseResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forCreateAutonomousDatabase
     *
     * @param request the request to send
     * @return response returns CreateAutonomousDatabaseResponse, GetWorkRequestResponse tuple
     */
    forCreateAutonomousDatabase(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const createAutonomousDatabaseResponse = yield this.client.createAutonomousDatabase(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, createAutonomousDatabaseResponse.opcWorkRequestId);
            return {
                response: createAutonomousDatabaseResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forCreateAutonomousDatabaseBackup
     *
     * @param request the request to send
     * @return response returns CreateAutonomousDatabaseBackupResponse, GetWorkRequestResponse tuple
     */
    forCreateAutonomousDatabaseBackup(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const createAutonomousDatabaseBackupResponse = yield this.client.createAutonomousDatabaseBackup(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, createAutonomousDatabaseBackupResponse.opcWorkRequestId);
            return {
                response: createAutonomousDatabaseBackupResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forCreateAutonomousVmCluster
     *
     * @param request the request to send
     * @return response returns CreateAutonomousVmClusterResponse, GetWorkRequestResponse tuple
     */
    forCreateAutonomousVmCluster(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const createAutonomousVmClusterResponse = yield this.client.createAutonomousVmCluster(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, createAutonomousVmClusterResponse.opcWorkRequestId);
            return {
                response: createAutonomousVmClusterResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forCreateBackup
     *
     * @param request the request to send
     * @return response returns CreateBackupResponse, GetWorkRequestResponse tuple
     */
    forCreateBackup(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const createBackupResponse = yield this.client.createBackup(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, createBackupResponse.opcWorkRequestId);
            return { response: createBackupResponse, workRequestResponse: getWorkRequestResponse };
        });
    }
    /**
     * Waits forCreateCloudExadataInfrastructure
     *
     * @param request the request to send
     * @return response returns CreateCloudExadataInfrastructureResponse, GetWorkRequestResponse tuple
     */
    forCreateCloudExadataInfrastructure(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const createCloudExadataInfrastructureResponse = yield this.client.createCloudExadataInfrastructure(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, createCloudExadataInfrastructureResponse.opcWorkRequestId);
            return {
                response: createCloudExadataInfrastructureResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forCreateCloudVmCluster
     *
     * @param request the request to send
     * @return response returns CreateCloudVmClusterResponse, GetWorkRequestResponse tuple
     */
    forCreateCloudVmCluster(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const createCloudVmClusterResponse = yield this.client.createCloudVmCluster(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, createCloudVmClusterResponse.opcWorkRequestId);
            return { response: createCloudVmClusterResponse, workRequestResponse: getWorkRequestResponse };
        });
    }
    /**
     * Waits forCreateDataGuardAssociation
     *
     * @param request the request to send
     * @return response returns CreateDataGuardAssociationResponse, GetWorkRequestResponse tuple
     */
    forCreateDataGuardAssociation(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const createDataGuardAssociationResponse = yield this.client.createDataGuardAssociation(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, createDataGuardAssociationResponse.opcWorkRequestId);
            return {
                response: createDataGuardAssociationResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forCreateDatabase
     *
     * @param request the request to send
     * @return response returns CreateDatabaseResponse, GetWorkRequestResponse tuple
     */
    forCreateDatabase(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const createDatabaseResponse = yield this.client.createDatabase(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, createDatabaseResponse.opcWorkRequestId);
            return { response: createDatabaseResponse, workRequestResponse: getWorkRequestResponse };
        });
    }
    /**
     * Waits forCreateDatabaseSoftwareImage
     *
     * @param request the request to send
     * @return response returns CreateDatabaseSoftwareImageResponse, GetWorkRequestResponse tuple
     */
    forCreateDatabaseSoftwareImage(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const createDatabaseSoftwareImageResponse = yield this.client.createDatabaseSoftwareImage(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, createDatabaseSoftwareImageResponse.opcWorkRequestId);
            return {
                response: createDatabaseSoftwareImageResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forCreateDbHome
     *
     * @param request the request to send
     * @return response returns CreateDbHomeResponse, GetWorkRequestResponse tuple
     */
    forCreateDbHome(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const createDbHomeResponse = yield this.client.createDbHome(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, createDbHomeResponse.opcWorkRequestId);
            return { response: createDbHomeResponse, workRequestResponse: getWorkRequestResponse };
        });
    }
    /**
     * Waits forCreateExadataInfrastructure
     *
     * @param request the request to send
     * @return response returns CreateExadataInfrastructureResponse, GetWorkRequestResponse tuple
     */
    forCreateExadataInfrastructure(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const createExadataInfrastructureResponse = yield this.client.createExadataInfrastructure(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, createExadataInfrastructureResponse.opcWorkRequestId);
            return {
                response: createExadataInfrastructureResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forCreateExternalBackupJob
     *
     * @param request the request to send
     * @return response returns CreateExternalBackupJobResponse, GetWorkRequestResponse tuple
     */
    forCreateExternalBackupJob(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const createExternalBackupJobResponse = yield this.client.createExternalBackupJob(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, createExternalBackupJobResponse.opcWorkRequestId);
            return {
                response: createExternalBackupJobResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forCreateExternalContainerDatabase
     *
     * @param request the request to send
     * @return response returns CreateExternalContainerDatabaseResponse, GetWorkRequestResponse tuple
     */
    forCreateExternalContainerDatabase(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const createExternalContainerDatabaseResponse = yield this.client.createExternalContainerDatabase(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, createExternalContainerDatabaseResponse.opcWorkRequestId);
            return {
                response: createExternalContainerDatabaseResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forCreateExternalDatabaseConnector
     *
     * @param request the request to send
     * @return response returns CreateExternalDatabaseConnectorResponse, GetWorkRequestResponse tuple
     */
    forCreateExternalDatabaseConnector(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const createExternalDatabaseConnectorResponse = yield this.client.createExternalDatabaseConnector(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, createExternalDatabaseConnectorResponse.opcWorkRequestId);
            return {
                response: createExternalDatabaseConnectorResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forCreateExternalNonContainerDatabase
     *
     * @param request the request to send
     * @return response returns CreateExternalNonContainerDatabaseResponse, GetWorkRequestResponse tuple
     */
    forCreateExternalNonContainerDatabase(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const createExternalNonContainerDatabaseResponse = yield this.client.createExternalNonContainerDatabase(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, createExternalNonContainerDatabaseResponse.opcWorkRequestId);
            return {
                response: createExternalNonContainerDatabaseResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forCreateExternalPluggableDatabase
     *
     * @param request the request to send
     * @return response returns CreateExternalPluggableDatabaseResponse, GetWorkRequestResponse tuple
     */
    forCreateExternalPluggableDatabase(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const createExternalPluggableDatabaseResponse = yield this.client.createExternalPluggableDatabase(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, createExternalPluggableDatabaseResponse.opcWorkRequestId);
            return {
                response: createExternalPluggableDatabaseResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forCreateVmCluster
     *
     * @param request the request to send
     * @return response returns CreateVmClusterResponse, GetWorkRequestResponse tuple
     */
    forCreateVmCluster(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const createVmClusterResponse = yield this.client.createVmCluster(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, createVmClusterResponse.opcWorkRequestId);
            return { response: createVmClusterResponse, workRequestResponse: getWorkRequestResponse };
        });
    }
    /**
     * Waits forCreateVmClusterNetwork
     *
     * @param request the request to send
     * @return response returns CreateVmClusterNetworkResponse, GetWorkRequestResponse tuple
     */
    forCreateVmClusterNetwork(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const createVmClusterNetworkResponse = yield this.client.createVmClusterNetwork(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, createVmClusterNetworkResponse.opcWorkRequestId);
            return {
                response: createVmClusterNetworkResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forDbNodeAction
     *
     * @param request the request to send
     * @return response returns DbNodeActionResponse, GetWorkRequestResponse tuple
     */
    forDbNodeAction(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const dbNodeActionResponse = yield this.client.dbNodeAction(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, dbNodeActionResponse.opcWorkRequestId);
            return { response: dbNodeActionResponse, workRequestResponse: getWorkRequestResponse };
        });
    }
    /**
     * Waits forDeleteAutonomousDatabase
     *
     * @param request the request to send
     * @return response returns DeleteAutonomousDatabaseResponse, GetWorkRequestResponse tuple
     */
    forDeleteAutonomousDatabase(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const deleteAutonomousDatabaseResponse = yield this.client.deleteAutonomousDatabase(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, deleteAutonomousDatabaseResponse.opcWorkRequestId);
            return {
                response: deleteAutonomousDatabaseResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forDeleteAutonomousVmCluster
     *
     * @param request the request to send
     * @return response returns DeleteAutonomousVmClusterResponse, GetWorkRequestResponse tuple
     */
    forDeleteAutonomousVmCluster(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const deleteAutonomousVmClusterResponse = yield this.client.deleteAutonomousVmCluster(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, deleteAutonomousVmClusterResponse.opcWorkRequestId);
            return {
                response: deleteAutonomousVmClusterResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forDeleteBackup
     *
     * @param request the request to send
     * @return response returns DeleteBackupResponse, GetWorkRequestResponse tuple
     */
    forDeleteBackup(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const deleteBackupResponse = yield this.client.deleteBackup(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, deleteBackupResponse.opcWorkRequestId);
            return { response: deleteBackupResponse, workRequestResponse: getWorkRequestResponse };
        });
    }
    /**
     * Waits forDeleteCloudExadataInfrastructure
     *
     * @param request the request to send
     * @return response returns DeleteCloudExadataInfrastructureResponse, GetWorkRequestResponse tuple
     */
    forDeleteCloudExadataInfrastructure(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const deleteCloudExadataInfrastructureResponse = yield this.client.deleteCloudExadataInfrastructure(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, deleteCloudExadataInfrastructureResponse.opcWorkRequestId);
            return {
                response: deleteCloudExadataInfrastructureResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forDeleteCloudVmCluster
     *
     * @param request the request to send
     * @return response returns DeleteCloudVmClusterResponse, GetWorkRequestResponse tuple
     */
    forDeleteCloudVmCluster(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const deleteCloudVmClusterResponse = yield this.client.deleteCloudVmCluster(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, deleteCloudVmClusterResponse.opcWorkRequestId);
            return { response: deleteCloudVmClusterResponse, workRequestResponse: getWorkRequestResponse };
        });
    }
    /**
     * Waits forDeleteDatabase
     *
     * @param request the request to send
     * @return response returns DeleteDatabaseResponse, GetWorkRequestResponse tuple
     */
    forDeleteDatabase(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const deleteDatabaseResponse = yield this.client.deleteDatabase(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, deleteDatabaseResponse.opcWorkRequestId);
            return { response: deleteDatabaseResponse, workRequestResponse: getWorkRequestResponse };
        });
    }
    /**
     * Waits forDeleteDatabaseSoftwareImage
     *
     * @param request the request to send
     * @return response returns DeleteDatabaseSoftwareImageResponse, GetWorkRequestResponse tuple
     */
    forDeleteDatabaseSoftwareImage(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const deleteDatabaseSoftwareImageResponse = yield this.client.deleteDatabaseSoftwareImage(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, deleteDatabaseSoftwareImageResponse.opcWorkRequestId);
            return {
                response: deleteDatabaseSoftwareImageResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forDeleteDbHome
     *
     * @param request the request to send
     * @return response returns DeleteDbHomeResponse, GetWorkRequestResponse tuple
     */
    forDeleteDbHome(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const deleteDbHomeResponse = yield this.client.deleteDbHome(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, deleteDbHomeResponse.opcWorkRequestId);
            return { response: deleteDbHomeResponse, workRequestResponse: getWorkRequestResponse };
        });
    }
    /**
     * Waits forDeleteExadataInfrastructure
     *
     * @param request the request to send
     * @return response returns DeleteExadataInfrastructureResponse, GetWorkRequestResponse tuple
     */
    forDeleteExadataInfrastructure(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const deleteExadataInfrastructureResponse = yield this.client.deleteExadataInfrastructure(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, deleteExadataInfrastructureResponse.opcWorkRequestId);
            return {
                response: deleteExadataInfrastructureResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forDeleteExternalContainerDatabase
     *
     * @param request the request to send
     * @return response returns DeleteExternalContainerDatabaseResponse, GetWorkRequestResponse tuple
     */
    forDeleteExternalContainerDatabase(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const deleteExternalContainerDatabaseResponse = yield this.client.deleteExternalContainerDatabase(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, deleteExternalContainerDatabaseResponse.opcWorkRequestId);
            return {
                response: deleteExternalContainerDatabaseResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forDeleteExternalDatabaseConnector
     *
     * @param request the request to send
     * @return response returns DeleteExternalDatabaseConnectorResponse, GetWorkRequestResponse tuple
     */
    forDeleteExternalDatabaseConnector(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const deleteExternalDatabaseConnectorResponse = yield this.client.deleteExternalDatabaseConnector(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, deleteExternalDatabaseConnectorResponse.opcWorkRequestId);
            return {
                response: deleteExternalDatabaseConnectorResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forDeleteExternalNonContainerDatabase
     *
     * @param request the request to send
     * @return response returns DeleteExternalNonContainerDatabaseResponse, GetWorkRequestResponse tuple
     */
    forDeleteExternalNonContainerDatabase(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const deleteExternalNonContainerDatabaseResponse = yield this.client.deleteExternalNonContainerDatabase(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, deleteExternalNonContainerDatabaseResponse.opcWorkRequestId);
            return {
                response: deleteExternalNonContainerDatabaseResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forDeleteExternalPluggableDatabase
     *
     * @param request the request to send
     * @return response returns DeleteExternalPluggableDatabaseResponse, GetWorkRequestResponse tuple
     */
    forDeleteExternalPluggableDatabase(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const deleteExternalPluggableDatabaseResponse = yield this.client.deleteExternalPluggableDatabase(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, deleteExternalPluggableDatabaseResponse.opcWorkRequestId);
            return {
                response: deleteExternalPluggableDatabaseResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forDeleteVmCluster
     *
     * @param request the request to send
     * @return response returns DeleteVmClusterResponse, GetWorkRequestResponse tuple
     */
    forDeleteVmCluster(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const deleteVmClusterResponse = yield this.client.deleteVmCluster(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, deleteVmClusterResponse.opcWorkRequestId);
            return { response: deleteVmClusterResponse, workRequestResponse: getWorkRequestResponse };
        });
    }
    /**
     * Waits forDeleteVmClusterNetwork
     *
     * @param request the request to send
     * @return response returns DeleteVmClusterNetworkResponse, GetWorkRequestResponse tuple
     */
    forDeleteVmClusterNetwork(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const deleteVmClusterNetworkResponse = yield this.client.deleteVmClusterNetwork(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, deleteVmClusterNetworkResponse.opcWorkRequestId);
            return {
                response: deleteVmClusterNetworkResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forDeregisterAutonomousDatabaseDataSafe
     *
     * @param request the request to send
     * @return response returns DeregisterAutonomousDatabaseDataSafeResponse, GetWorkRequestResponse tuple
     */
    forDeregisterAutonomousDatabaseDataSafe(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const deregisterAutonomousDatabaseDataSafeResponse = yield this.client.deregisterAutonomousDatabaseDataSafe(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, deregisterAutonomousDatabaseDataSafeResponse.opcWorkRequestId);
            return {
                response: deregisterAutonomousDatabaseDataSafeResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forDisableAutonomousDatabaseOperationsInsights
     *
     * @param request the request to send
     * @return response returns DisableAutonomousDatabaseOperationsInsightsResponse, GetWorkRequestResponse tuple
     */
    forDisableAutonomousDatabaseOperationsInsights(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const disableAutonomousDatabaseOperationsInsightsResponse = yield this.client.disableAutonomousDatabaseOperationsInsights(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, disableAutonomousDatabaseOperationsInsightsResponse.opcWorkRequestId);
            return {
                response: disableAutonomousDatabaseOperationsInsightsResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forDisableExternalContainerDatabaseDatabaseManagement
     *
     * @param request the request to send
     * @return response returns DisableExternalContainerDatabaseDatabaseManagementResponse, GetWorkRequestResponse tuple
     */
    forDisableExternalContainerDatabaseDatabaseManagement(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const disableExternalContainerDatabaseDatabaseManagementResponse = yield this.client.disableExternalContainerDatabaseDatabaseManagement(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, disableExternalContainerDatabaseDatabaseManagementResponse.opcWorkRequestId);
            return {
                response: disableExternalContainerDatabaseDatabaseManagementResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forDisableExternalNonContainerDatabaseDatabaseManagement
     *
     * @param request the request to send
     * @return response returns DisableExternalNonContainerDatabaseDatabaseManagementResponse, GetWorkRequestResponse tuple
     */
    forDisableExternalNonContainerDatabaseDatabaseManagement(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const disableExternalNonContainerDatabaseDatabaseManagementResponse = yield this.client.disableExternalNonContainerDatabaseDatabaseManagement(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, disableExternalNonContainerDatabaseDatabaseManagementResponse.opcWorkRequestId);
            return {
                response: disableExternalNonContainerDatabaseDatabaseManagementResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forDisableExternalNonContainerDatabaseOperationsInsights
     *
     * @param request the request to send
     * @return response returns DisableExternalNonContainerDatabaseOperationsInsightsResponse, GetWorkRequestResponse tuple
     */
    forDisableExternalNonContainerDatabaseOperationsInsights(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const disableExternalNonContainerDatabaseOperationsInsightsResponse = yield this.client.disableExternalNonContainerDatabaseOperationsInsights(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, disableExternalNonContainerDatabaseOperationsInsightsResponse.opcWorkRequestId);
            return {
                response: disableExternalNonContainerDatabaseOperationsInsightsResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forDisableExternalPluggableDatabaseDatabaseManagement
     *
     * @param request the request to send
     * @return response returns DisableExternalPluggableDatabaseDatabaseManagementResponse, GetWorkRequestResponse tuple
     */
    forDisableExternalPluggableDatabaseDatabaseManagement(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const disableExternalPluggableDatabaseDatabaseManagementResponse = yield this.client.disableExternalPluggableDatabaseDatabaseManagement(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, disableExternalPluggableDatabaseDatabaseManagementResponse.opcWorkRequestId);
            return {
                response: disableExternalPluggableDatabaseDatabaseManagementResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forDisableExternalPluggableDatabaseOperationsInsights
     *
     * @param request the request to send
     * @return response returns DisableExternalPluggableDatabaseOperationsInsightsResponse, GetWorkRequestResponse tuple
     */
    forDisableExternalPluggableDatabaseOperationsInsights(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const disableExternalPluggableDatabaseOperationsInsightsResponse = yield this.client.disableExternalPluggableDatabaseOperationsInsights(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, disableExternalPluggableDatabaseOperationsInsightsResponse.opcWorkRequestId);
            return {
                response: disableExternalPluggableDatabaseOperationsInsightsResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forEnableAutonomousDatabaseOperationsInsights
     *
     * @param request the request to send
     * @return response returns EnableAutonomousDatabaseOperationsInsightsResponse, GetWorkRequestResponse tuple
     */
    forEnableAutonomousDatabaseOperationsInsights(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const enableAutonomousDatabaseOperationsInsightsResponse = yield this.client.enableAutonomousDatabaseOperationsInsights(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, enableAutonomousDatabaseOperationsInsightsResponse.opcWorkRequestId);
            return {
                response: enableAutonomousDatabaseOperationsInsightsResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forEnableExternalContainerDatabaseDatabaseManagement
     *
     * @param request the request to send
     * @return response returns EnableExternalContainerDatabaseDatabaseManagementResponse, GetWorkRequestResponse tuple
     */
    forEnableExternalContainerDatabaseDatabaseManagement(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const enableExternalContainerDatabaseDatabaseManagementResponse = yield this.client.enableExternalContainerDatabaseDatabaseManagement(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, enableExternalContainerDatabaseDatabaseManagementResponse.opcWorkRequestId);
            return {
                response: enableExternalContainerDatabaseDatabaseManagementResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forEnableExternalNonContainerDatabaseDatabaseManagement
     *
     * @param request the request to send
     * @return response returns EnableExternalNonContainerDatabaseDatabaseManagementResponse, GetWorkRequestResponse tuple
     */
    forEnableExternalNonContainerDatabaseDatabaseManagement(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const enableExternalNonContainerDatabaseDatabaseManagementResponse = yield this.client.enableExternalNonContainerDatabaseDatabaseManagement(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, enableExternalNonContainerDatabaseDatabaseManagementResponse.opcWorkRequestId);
            return {
                response: enableExternalNonContainerDatabaseDatabaseManagementResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forEnableExternalNonContainerDatabaseOperationsInsights
     *
     * @param request the request to send
     * @return response returns EnableExternalNonContainerDatabaseOperationsInsightsResponse, GetWorkRequestResponse tuple
     */
    forEnableExternalNonContainerDatabaseOperationsInsights(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const enableExternalNonContainerDatabaseOperationsInsightsResponse = yield this.client.enableExternalNonContainerDatabaseOperationsInsights(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, enableExternalNonContainerDatabaseOperationsInsightsResponse.opcWorkRequestId);
            return {
                response: enableExternalNonContainerDatabaseOperationsInsightsResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forEnableExternalPluggableDatabaseDatabaseManagement
     *
     * @param request the request to send
     * @return response returns EnableExternalPluggableDatabaseDatabaseManagementResponse, GetWorkRequestResponse tuple
     */
    forEnableExternalPluggableDatabaseDatabaseManagement(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const enableExternalPluggableDatabaseDatabaseManagementResponse = yield this.client.enableExternalPluggableDatabaseDatabaseManagement(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, enableExternalPluggableDatabaseDatabaseManagementResponse.opcWorkRequestId);
            return {
                response: enableExternalPluggableDatabaseDatabaseManagementResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forEnableExternalPluggableDatabaseOperationsInsights
     *
     * @param request the request to send
     * @return response returns EnableExternalPluggableDatabaseOperationsInsightsResponse, GetWorkRequestResponse tuple
     */
    forEnableExternalPluggableDatabaseOperationsInsights(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const enableExternalPluggableDatabaseOperationsInsightsResponse = yield this.client.enableExternalPluggableDatabaseOperationsInsights(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, enableExternalPluggableDatabaseOperationsInsightsResponse.opcWorkRequestId);
            return {
                response: enableExternalPluggableDatabaseOperationsInsightsResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forFailOverAutonomousDatabase
     *
     * @param request the request to send
     * @return response returns FailOverAutonomousDatabaseResponse, GetWorkRequestResponse tuple
     */
    forFailOverAutonomousDatabase(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const failOverAutonomousDatabaseResponse = yield this.client.failOverAutonomousDatabase(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, failOverAutonomousDatabaseResponse.opcWorkRequestId);
            return {
                response: failOverAutonomousDatabaseResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forFailoverAutonomousContainerDatabaseDataguardAssociation
     *
     * @param request the request to send
     * @return response returns FailoverAutonomousContainerDatabaseDataguardAssociationResponse, GetWorkRequestResponse tuple
     */
    forFailoverAutonomousContainerDatabaseDataguardAssociation(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const failoverAutonomousContainerDatabaseDataguardAssociationResponse = yield this.client.failoverAutonomousContainerDatabaseDataguardAssociation(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, failoverAutonomousContainerDatabaseDataguardAssociationResponse.opcWorkRequestId);
            return {
                response: failoverAutonomousContainerDatabaseDataguardAssociationResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forFailoverDataGuardAssociation
     *
     * @param request the request to send
     * @return response returns FailoverDataGuardAssociationResponse, GetWorkRequestResponse tuple
     */
    forFailoverDataGuardAssociation(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const failoverDataGuardAssociationResponse = yield this.client.failoverDataGuardAssociation(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, failoverDataGuardAssociationResponse.opcWorkRequestId);
            return {
                response: failoverDataGuardAssociationResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forAutonomousContainerDatabase till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetAutonomousContainerDatabaseResponse | null (null in case of 404 response)
     */
    forAutonomousContainerDatabase(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getAutonomousContainerDatabase(request), response => targetStates.includes(response.autonomousContainerDatabase.lifecycleState), targetStates.includes(models.AutonomousContainerDatabase.LifecycleState.Terminated));
        });
    }
    /**
     * Waits forAutonomousContainerDatabaseDataguardAssociation till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetAutonomousContainerDatabaseDataguardAssociationResponse | null (null in case of 404 response)
     */
    forAutonomousContainerDatabaseDataguardAssociation(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getAutonomousContainerDatabaseDataguardAssociation(request), response => targetStates.includes(response.autonomousContainerDatabaseDataguardAssociation.lifecycleState), targetStates.includes(models.AutonomousContainerDatabaseDataguardAssociation.LifecycleState.Terminated));
        });
    }
    /**
     * Waits forAutonomousDatabase till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetAutonomousDatabaseResponse | null (null in case of 404 response)
     */
    forAutonomousDatabase(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getAutonomousDatabase(request), response => targetStates.includes(response.autonomousDatabase.lifecycleState), targetStates.includes(models.AutonomousDatabase.LifecycleState.Terminated));
        });
    }
    /**
     * Waits forAutonomousDatabaseBackup till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetAutonomousDatabaseBackupResponse | null (null in case of 404 response)
     */
    forAutonomousDatabaseBackup(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getAutonomousDatabaseBackup(request), response => targetStates.includes(response.autonomousDatabaseBackup.lifecycleState), targetStates.includes(models.AutonomousDatabaseBackup.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forAutonomousDatabaseDataguardAssociation till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetAutonomousDatabaseDataguardAssociationResponse | null (null in case of 404 response)
     */
    forAutonomousDatabaseDataguardAssociation(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getAutonomousDatabaseDataguardAssociation(request), response => targetStates.includes(response.autonomousDatabaseDataguardAssociation.lifecycleState), targetStates.includes(models.AutonomousDatabaseDataguardAssociation.LifecycleState.Terminated));
        });
    }
    /**
     * Waits forAutonomousDatabaseWallet till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetAutonomousDatabaseWalletResponse
     */
    forAutonomousDatabaseWallet(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericWaiter(this.config, () => this.client.getAutonomousDatabaseWallet(request), response => targetStates.includes(response.autonomousDatabaseWallet.lifecycleState));
        });
    }
    /**
     * Waits forAutonomousExadataInfrastructure till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetAutonomousExadataInfrastructureResponse | null (null in case of 404 response)
     */
    forAutonomousExadataInfrastructure(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getAutonomousExadataInfrastructure(request), response => targetStates.includes(response.autonomousExadataInfrastructure.lifecycleState), targetStates.includes(models.AutonomousExadataInfrastructure.LifecycleState.Terminated));
        });
    }
    /**
     * Waits forAutonomousPatch till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetAutonomousPatchResponse
     */
    forAutonomousPatch(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericWaiter(this.config, () => this.client.getAutonomousPatch(request), response => targetStates.includes(response.autonomousPatch.lifecycleState));
        });
    }
    /**
     * Waits forAutonomousVmCluster till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetAutonomousVmClusterResponse | null (null in case of 404 response)
     */
    forAutonomousVmCluster(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getAutonomousVmCluster(request), response => targetStates.includes(response.autonomousVmCluster.lifecycleState), targetStates.includes(models.AutonomousVmCluster.LifecycleState.Terminated));
        });
    }
    /**
     * Waits forBackup till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetBackupResponse | null (null in case of 404 response)
     */
    forBackup(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getBackup(request), response => targetStates.includes(response.backup.lifecycleState), targetStates.includes(models.Backup.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forBackupDestination till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetBackupDestinationResponse | null (null in case of 404 response)
     */
    forBackupDestination(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getBackupDestination(request), response => targetStates.includes(response.backupDestination.lifecycleState), targetStates.includes(models.BackupDestination.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forCloudExadataInfrastructure till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetCloudExadataInfrastructureResponse | null (null in case of 404 response)
     */
    forCloudExadataInfrastructure(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getCloudExadataInfrastructure(request), response => targetStates.includes(response.cloudExadataInfrastructure.lifecycleState), targetStates.includes(models.CloudExadataInfrastructure.LifecycleState.Terminated));
        });
    }
    /**
     * Waits forCloudVmCluster till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetCloudVmClusterResponse | null (null in case of 404 response)
     */
    forCloudVmCluster(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getCloudVmCluster(request), response => targetStates.includes(response.cloudVmCluster.lifecycleState), targetStates.includes(models.CloudVmCluster.LifecycleState.Terminated));
        });
    }
    /**
     * Waits forConsoleConnection till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetConsoleConnectionResponse | null (null in case of 404 response)
     */
    forConsoleConnection(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getConsoleConnection(request), response => targetStates.includes(response.consoleConnection.lifecycleState), targetStates.includes(models.ConsoleConnection.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forDataGuardAssociation till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetDataGuardAssociationResponse | null (null in case of 404 response)
     */
    forDataGuardAssociation(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getDataGuardAssociation(request), response => targetStates.includes(response.dataGuardAssociation.lifecycleState), targetStates.includes(models.DataGuardAssociation.LifecycleState.Terminated));
        });
    }
    /**
     * Waits forDatabase till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetDatabaseResponse | null (null in case of 404 response)
     */
    forDatabase(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getDatabase(request), response => targetStates.includes(response.database.lifecycleState), targetStates.includes(models.Database.LifecycleState.Terminated));
        });
    }
    /**
     * Waits forDatabaseSoftwareImage till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetDatabaseSoftwareImageResponse | null (null in case of 404 response)
     */
    forDatabaseSoftwareImage(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getDatabaseSoftwareImage(request), response => targetStates.includes(response.databaseSoftwareImage.lifecycleState), targetStates.includes(models.DatabaseSoftwareImage.LifecycleState.Terminated));
        });
    }
    /**
     * Waits forDatabaseUpgradeHistoryEntry till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetDatabaseUpgradeHistoryEntryResponse
     */
    forDatabaseUpgradeHistoryEntry(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericWaiter(this.config, () => this.client.getDatabaseUpgradeHistoryEntry(request), response => targetStates.includes(response.databaseUpgradeHistoryEntry.lifecycleState));
        });
    }
    /**
     * Waits forDbHome till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetDbHomeResponse | null (null in case of 404 response)
     */
    forDbHome(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getDbHome(request), response => targetStates.includes(response.dbHome.lifecycleState), targetStates.includes(models.DbHome.LifecycleState.Terminated));
        });
    }
    /**
     * Waits forDbNode till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetDbNodeResponse | null (null in case of 404 response)
     */
    forDbNode(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getDbNode(request), response => targetStates.includes(response.dbNode.lifecycleState), targetStates.includes(models.DbNode.LifecycleState.Terminated));
        });
    }
    /**
     * Waits forDbSystem till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetDbSystemResponse | null (null in case of 404 response)
     */
    forDbSystem(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getDbSystem(request), response => targetStates.includes(response.dbSystem.lifecycleState), targetStates.includes(models.DbSystem.LifecycleState.Terminated));
        });
    }
    /**
     * Waits forExadataInfrastructure till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetExadataInfrastructureResponse | null (null in case of 404 response)
     */
    forExadataInfrastructure(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getExadataInfrastructure(request), response => targetStates.includes(response.exadataInfrastructure.lifecycleState), targetStates.includes(models.ExadataInfrastructure.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forExadataIormConfig till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetExadataIormConfigResponse
     */
    forExadataIormConfig(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericWaiter(this.config, () => this.client.getExadataIormConfig(request), response => targetStates.includes(response.exadataIormConfig.lifecycleState));
        });
    }
    /**
     * Waits forExternalContainerDatabase till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetExternalContainerDatabaseResponse | null (null in case of 404 response)
     */
    forExternalContainerDatabase(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getExternalContainerDatabase(request), response => targetStates.includes(response.externalContainerDatabase.lifecycleState), targetStates.includes(models.ExternalContainerDatabase.LifecycleState.Terminated));
        });
    }
    /**
     * Waits forExternalDatabaseConnector till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetExternalDatabaseConnectorResponse | null (null in case of 404 response)
     */
    forExternalDatabaseConnector(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getExternalDatabaseConnector(request), response => targetStates.includes(response.externalDatabaseConnector.lifecycleState), targetStates.includes(models.ExternalDatabaseConnector.LifecycleState.Terminated));
        });
    }
    /**
     * Waits forExternalNonContainerDatabase till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetExternalNonContainerDatabaseResponse | null (null in case of 404 response)
     */
    forExternalNonContainerDatabase(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getExternalNonContainerDatabase(request), response => targetStates.includes(response.externalNonContainerDatabase.lifecycleState), targetStates.includes(models.ExternalNonContainerDatabase.LifecycleState.Terminated));
        });
    }
    /**
     * Waits forExternalPluggableDatabase till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetExternalPluggableDatabaseResponse | null (null in case of 404 response)
     */
    forExternalPluggableDatabase(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getExternalPluggableDatabase(request), response => targetStates.includes(response.externalPluggableDatabase.lifecycleState), targetStates.includes(models.ExternalPluggableDatabase.LifecycleState.Terminated));
        });
    }
    /**
     * Waits forKeyStore till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetKeyStoreResponse | null (null in case of 404 response)
     */
    forKeyStore(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getKeyStore(request), response => targetStates.includes(response.keyStore.lifecycleState), targetStates.includes(models.KeyStore.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forMaintenanceRun till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetMaintenanceRunResponse | null (null in case of 404 response)
     */
    forMaintenanceRun(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getMaintenanceRun(request), response => targetStates.includes(response.maintenanceRun.lifecycleState), targetStates.includes(models.MaintenanceRun.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forVmCluster till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetVmClusterResponse | null (null in case of 404 response)
     */
    forVmCluster(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getVmCluster(request), response => targetStates.includes(response.vmCluster.lifecycleState), targetStates.includes(models.VmCluster.LifecycleState.Terminated));
        });
    }
    /**
     * Waits forVmClusterNetwork till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetVmClusterNetworkResponse | null (null in case of 404 response)
     */
    forVmClusterNetwork(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getVmClusterNetwork(request), response => targetStates.includes(response.vmClusterNetwork.lifecycleState), targetStates.includes(models.VmClusterNetwork.LifecycleState.Terminated));
        });
    }
    /**
     * Waits forLaunchAutonomousExadataInfrastructure
     *
     * @param request the request to send
     * @return response returns LaunchAutonomousExadataInfrastructureResponse, GetWorkRequestResponse tuple
     */
    forLaunchAutonomousExadataInfrastructure(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const launchAutonomousExadataInfrastructureResponse = yield this.client.launchAutonomousExadataInfrastructure(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, launchAutonomousExadataInfrastructureResponse.opcWorkRequestId);
            return {
                response: launchAutonomousExadataInfrastructureResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forLaunchDbSystem
     *
     * @param request the request to send
     * @return response returns LaunchDbSystemResponse, GetWorkRequestResponse tuple
     */
    forLaunchDbSystem(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const launchDbSystemResponse = yield this.client.launchDbSystem(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, launchDbSystemResponse.opcWorkRequestId);
            return { response: launchDbSystemResponse, workRequestResponse: getWorkRequestResponse };
        });
    }
    /**
     * Waits forMigrateExadataDbSystemResourceModel
     *
     * @param request the request to send
     * @return response returns MigrateExadataDbSystemResourceModelResponse, GetWorkRequestResponse tuple
     */
    forMigrateExadataDbSystemResourceModel(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const migrateExadataDbSystemResourceModelResponse = yield this.client.migrateExadataDbSystemResourceModel(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, migrateExadataDbSystemResourceModelResponse.opcWorkRequestId);
            return {
                response: migrateExadataDbSystemResourceModelResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forMigrateVaultKey
     *
     * @param request the request to send
     * @return response returns MigrateVaultKeyResponse, GetWorkRequestResponse tuple
     */
    forMigrateVaultKey(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const migrateVaultKeyResponse = yield this.client.migrateVaultKey(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, migrateVaultKeyResponse.opcWorkRequestId);
            return { response: migrateVaultKeyResponse, workRequestResponse: getWorkRequestResponse };
        });
    }
    /**
     * Waits forRegisterAutonomousDatabaseDataSafe
     *
     * @param request the request to send
     * @return response returns RegisterAutonomousDatabaseDataSafeResponse, GetWorkRequestResponse tuple
     */
    forRegisterAutonomousDatabaseDataSafe(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const registerAutonomousDatabaseDataSafeResponse = yield this.client.registerAutonomousDatabaseDataSafe(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, registerAutonomousDatabaseDataSafeResponse.opcWorkRequestId);
            return {
                response: registerAutonomousDatabaseDataSafeResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forReinstateAutonomousContainerDatabaseDataguardAssociation
     *
     * @param request the request to send
     * @return response returns ReinstateAutonomousContainerDatabaseDataguardAssociationResponse, GetWorkRequestResponse tuple
     */
    forReinstateAutonomousContainerDatabaseDataguardAssociation(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const reinstateAutonomousContainerDatabaseDataguardAssociationResponse = yield this.client.reinstateAutonomousContainerDatabaseDataguardAssociation(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, reinstateAutonomousContainerDatabaseDataguardAssociationResponse.opcWorkRequestId);
            return {
                response: reinstateAutonomousContainerDatabaseDataguardAssociationResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forReinstateDataGuardAssociation
     *
     * @param request the request to send
     * @return response returns ReinstateDataGuardAssociationResponse, GetWorkRequestResponse tuple
     */
    forReinstateDataGuardAssociation(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const reinstateDataGuardAssociationResponse = yield this.client.reinstateDataGuardAssociation(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, reinstateDataGuardAssociationResponse.opcWorkRequestId);
            return {
                response: reinstateDataGuardAssociationResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forRestartAutonomousContainerDatabase
     *
     * @param request the request to send
     * @return response returns RestartAutonomousContainerDatabaseResponse, GetWorkRequestResponse tuple
     */
    forRestartAutonomousContainerDatabase(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const restartAutonomousContainerDatabaseResponse = yield this.client.restartAutonomousContainerDatabase(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, restartAutonomousContainerDatabaseResponse.opcWorkRequestId);
            return {
                response: restartAutonomousContainerDatabaseResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forRestartAutonomousDatabase
     *
     * @param request the request to send
     * @return response returns RestartAutonomousDatabaseResponse, GetWorkRequestResponse tuple
     */
    forRestartAutonomousDatabase(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const restartAutonomousDatabaseResponse = yield this.client.restartAutonomousDatabase(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, restartAutonomousDatabaseResponse.opcWorkRequestId);
            return {
                response: restartAutonomousDatabaseResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forRestoreAutonomousDatabase
     *
     * @param request the request to send
     * @return response returns RestoreAutonomousDatabaseResponse, GetWorkRequestResponse tuple
     */
    forRestoreAutonomousDatabase(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const restoreAutonomousDatabaseResponse = yield this.client.restoreAutonomousDatabase(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, restoreAutonomousDatabaseResponse.opcWorkRequestId);
            return {
                response: restoreAutonomousDatabaseResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forRestoreDatabase
     *
     * @param request the request to send
     * @return response returns RestoreDatabaseResponse, GetWorkRequestResponse tuple
     */
    forRestoreDatabase(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const restoreDatabaseResponse = yield this.client.restoreDatabase(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, restoreDatabaseResponse.opcWorkRequestId);
            return { response: restoreDatabaseResponse, workRequestResponse: getWorkRequestResponse };
        });
    }
    /**
     * Waits forRotateAutonomousContainerDatabaseEncryptionKey
     *
     * @param request the request to send
     * @return response returns RotateAutonomousContainerDatabaseEncryptionKeyResponse, GetWorkRequestResponse tuple
     */
    forRotateAutonomousContainerDatabaseEncryptionKey(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const rotateAutonomousContainerDatabaseEncryptionKeyResponse = yield this.client.rotateAutonomousContainerDatabaseEncryptionKey(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, rotateAutonomousContainerDatabaseEncryptionKeyResponse.opcWorkRequestId);
            return {
                response: rotateAutonomousContainerDatabaseEncryptionKeyResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forRotateAutonomousDatabaseEncryptionKey
     *
     * @param request the request to send
     * @return response returns RotateAutonomousDatabaseEncryptionKeyResponse, GetWorkRequestResponse tuple
     */
    forRotateAutonomousDatabaseEncryptionKey(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const rotateAutonomousDatabaseEncryptionKeyResponse = yield this.client.rotateAutonomousDatabaseEncryptionKey(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, rotateAutonomousDatabaseEncryptionKeyResponse.opcWorkRequestId);
            return {
                response: rotateAutonomousDatabaseEncryptionKeyResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forRotateOrdsCerts
     *
     * @param request the request to send
     * @return response returns RotateOrdsCertsResponse, GetWorkRequestResponse tuple
     */
    forRotateOrdsCerts(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const rotateOrdsCertsResponse = yield this.client.rotateOrdsCerts(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, rotateOrdsCertsResponse.opcWorkRequestId);
            return { response: rotateOrdsCertsResponse, workRequestResponse: getWorkRequestResponse };
        });
    }
    /**
     * Waits forRotateSslCerts
     *
     * @param request the request to send
     * @return response returns RotateSslCertsResponse, GetWorkRequestResponse tuple
     */
    forRotateSslCerts(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const rotateSslCertsResponse = yield this.client.rotateSslCerts(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, rotateSslCertsResponse.opcWorkRequestId);
            return { response: rotateSslCertsResponse, workRequestResponse: getWorkRequestResponse };
        });
    }
    /**
     * Waits forRotateVaultKey
     *
     * @param request the request to send
     * @return response returns RotateVaultKeyResponse, GetWorkRequestResponse tuple
     */
    forRotateVaultKey(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const rotateVaultKeyResponse = yield this.client.rotateVaultKey(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, rotateVaultKeyResponse.opcWorkRequestId);
            return { response: rotateVaultKeyResponse, workRequestResponse: getWorkRequestResponse };
        });
    }
    /**
     * Waits forScanExternalContainerDatabasePluggableDatabases
     *
     * @param request the request to send
     * @return response returns ScanExternalContainerDatabasePluggableDatabasesResponse, GetWorkRequestResponse tuple
     */
    forScanExternalContainerDatabasePluggableDatabases(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const scanExternalContainerDatabasePluggableDatabasesResponse = yield this.client.scanExternalContainerDatabasePluggableDatabases(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, scanExternalContainerDatabasePluggableDatabasesResponse.opcWorkRequestId);
            return {
                response: scanExternalContainerDatabasePluggableDatabasesResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forStartAutonomousDatabase
     *
     * @param request the request to send
     * @return response returns StartAutonomousDatabaseResponse, GetWorkRequestResponse tuple
     */
    forStartAutonomousDatabase(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const startAutonomousDatabaseResponse = yield this.client.startAutonomousDatabase(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, startAutonomousDatabaseResponse.opcWorkRequestId);
            return {
                response: startAutonomousDatabaseResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forStopAutonomousDatabase
     *
     * @param request the request to send
     * @return response returns StopAutonomousDatabaseResponse, GetWorkRequestResponse tuple
     */
    forStopAutonomousDatabase(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const stopAutonomousDatabaseResponse = yield this.client.stopAutonomousDatabase(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, stopAutonomousDatabaseResponse.opcWorkRequestId);
            return {
                response: stopAutonomousDatabaseResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forSwitchoverAutonomousContainerDatabaseDataguardAssociation
     *
     * @param request the request to send
     * @return response returns SwitchoverAutonomousContainerDatabaseDataguardAssociationResponse, GetWorkRequestResponse tuple
     */
    forSwitchoverAutonomousContainerDatabaseDataguardAssociation(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const switchoverAutonomousContainerDatabaseDataguardAssociationResponse = yield this.client.switchoverAutonomousContainerDatabaseDataguardAssociation(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, switchoverAutonomousContainerDatabaseDataguardAssociationResponse.opcWorkRequestId);
            return {
                response: switchoverAutonomousContainerDatabaseDataguardAssociationResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forSwitchoverAutonomousDatabase
     *
     * @param request the request to send
     * @return response returns SwitchoverAutonomousDatabaseResponse, GetWorkRequestResponse tuple
     */
    forSwitchoverAutonomousDatabase(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const switchoverAutonomousDatabaseResponse = yield this.client.switchoverAutonomousDatabase(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, switchoverAutonomousDatabaseResponse.opcWorkRequestId);
            return {
                response: switchoverAutonomousDatabaseResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forSwitchoverDataGuardAssociation
     *
     * @param request the request to send
     * @return response returns SwitchoverDataGuardAssociationResponse, GetWorkRequestResponse tuple
     */
    forSwitchoverDataGuardAssociation(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const switchoverDataGuardAssociationResponse = yield this.client.switchoverDataGuardAssociation(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, switchoverDataGuardAssociationResponse.opcWorkRequestId);
            return {
                response: switchoverDataGuardAssociationResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forTerminateAutonomousContainerDatabase
     *
     * @param request the request to send
     * @return response returns TerminateAutonomousContainerDatabaseResponse, GetWorkRequestResponse tuple
     */
    forTerminateAutonomousContainerDatabase(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const terminateAutonomousContainerDatabaseResponse = yield this.client.terminateAutonomousContainerDatabase(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, terminateAutonomousContainerDatabaseResponse.opcWorkRequestId);
            return {
                response: terminateAutonomousContainerDatabaseResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forTerminateAutonomousExadataInfrastructure
     *
     * @param request the request to send
     * @return response returns TerminateAutonomousExadataInfrastructureResponse, GetWorkRequestResponse tuple
     */
    forTerminateAutonomousExadataInfrastructure(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const terminateAutonomousExadataInfrastructureResponse = yield this.client.terminateAutonomousExadataInfrastructure(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, terminateAutonomousExadataInfrastructureResponse.opcWorkRequestId);
            return {
                response: terminateAutonomousExadataInfrastructureResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forTerminateDbSystem
     *
     * @param request the request to send
     * @return response returns TerminateDbSystemResponse, GetWorkRequestResponse tuple
     */
    forTerminateDbSystem(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const terminateDbSystemResponse = yield this.client.terminateDbSystem(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, terminateDbSystemResponse.opcWorkRequestId);
            return { response: terminateDbSystemResponse, workRequestResponse: getWorkRequestResponse };
        });
    }
    /**
     * Waits forUpdateAutonomousContainerDatabase
     *
     * @param request the request to send
     * @return response returns UpdateAutonomousContainerDatabaseResponse, GetWorkRequestResponse tuple
     */
    forUpdateAutonomousContainerDatabase(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const updateAutonomousContainerDatabaseResponse = yield this.client.updateAutonomousContainerDatabase(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, updateAutonomousContainerDatabaseResponse.opcWorkRequestId);
            return {
                response: updateAutonomousContainerDatabaseResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forUpdateAutonomousDatabase
     *
     * @param request the request to send
     * @return response returns UpdateAutonomousDatabaseResponse, GetWorkRequestResponse tuple
     */
    forUpdateAutonomousDatabase(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const updateAutonomousDatabaseResponse = yield this.client.updateAutonomousDatabase(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, updateAutonomousDatabaseResponse.opcWorkRequestId);
            return {
                response: updateAutonomousDatabaseResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forUpdateAutonomousDatabaseRegionalWallet
     *
     * @param request the request to send
     * @return response returns UpdateAutonomousDatabaseRegionalWalletResponse, GetWorkRequestResponse tuple
     */
    forUpdateAutonomousDatabaseRegionalWallet(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const updateAutonomousDatabaseRegionalWalletResponse = yield this.client.updateAutonomousDatabaseRegionalWallet(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, updateAutonomousDatabaseRegionalWalletResponse.opcWorkRequestId);
            return {
                response: updateAutonomousDatabaseRegionalWalletResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forUpdateAutonomousDatabaseWallet
     *
     * @param request the request to send
     * @return response returns UpdateAutonomousDatabaseWalletResponse, GetWorkRequestResponse tuple
     */
    forUpdateAutonomousDatabaseWallet(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const updateAutonomousDatabaseWalletResponse = yield this.client.updateAutonomousDatabaseWallet(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, updateAutonomousDatabaseWalletResponse.opcWorkRequestId);
            return {
                response: updateAutonomousDatabaseWalletResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forUpdateAutonomousExadataInfrastructure
     *
     * @param request the request to send
     * @return response returns UpdateAutonomousExadataInfrastructureResponse, GetWorkRequestResponse tuple
     */
    forUpdateAutonomousExadataInfrastructure(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const updateAutonomousExadataInfrastructureResponse = yield this.client.updateAutonomousExadataInfrastructure(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, updateAutonomousExadataInfrastructureResponse.opcWorkRequestId);
            return {
                response: updateAutonomousExadataInfrastructureResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forUpdateAutonomousVmCluster
     *
     * @param request the request to send
     * @return response returns UpdateAutonomousVmClusterResponse, GetWorkRequestResponse tuple
     */
    forUpdateAutonomousVmCluster(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const updateAutonomousVmClusterResponse = yield this.client.updateAutonomousVmCluster(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, updateAutonomousVmClusterResponse.opcWorkRequestId);
            return {
                response: updateAutonomousVmClusterResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forUpdateCloudExadataInfrastructure
     *
     * @param request the request to send
     * @return response returns UpdateCloudExadataInfrastructureResponse, GetWorkRequestResponse tuple
     */
    forUpdateCloudExadataInfrastructure(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const updateCloudExadataInfrastructureResponse = yield this.client.updateCloudExadataInfrastructure(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, updateCloudExadataInfrastructureResponse.opcWorkRequestId);
            return {
                response: updateCloudExadataInfrastructureResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forUpdateCloudVmCluster
     *
     * @param request the request to send
     * @return response returns UpdateCloudVmClusterResponse, GetWorkRequestResponse tuple
     */
    forUpdateCloudVmCluster(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const updateCloudVmClusterResponse = yield this.client.updateCloudVmCluster(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, updateCloudVmClusterResponse.opcWorkRequestId);
            return { response: updateCloudVmClusterResponse, workRequestResponse: getWorkRequestResponse };
        });
    }
    /**
     * Waits forUpdateCloudVmClusterIormConfig
     *
     * @param request the request to send
     * @return response returns UpdateCloudVmClusterIormConfigResponse, GetWorkRequestResponse tuple
     */
    forUpdateCloudVmClusterIormConfig(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const updateCloudVmClusterIormConfigResponse = yield this.client.updateCloudVmClusterIormConfig(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, updateCloudVmClusterIormConfigResponse.opcWorkRequestId);
            return {
                response: updateCloudVmClusterIormConfigResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forUpdateDatabase
     *
     * @param request the request to send
     * @return response returns UpdateDatabaseResponse, GetWorkRequestResponse tuple
     */
    forUpdateDatabase(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const updateDatabaseResponse = yield this.client.updateDatabase(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, updateDatabaseResponse.opcWorkRequestId);
            return { response: updateDatabaseResponse, workRequestResponse: getWorkRequestResponse };
        });
    }
    /**
     * Waits forUpdateDbHome
     *
     * @param request the request to send
     * @return response returns UpdateDbHomeResponse, GetWorkRequestResponse tuple
     */
    forUpdateDbHome(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const updateDbHomeResponse = yield this.client.updateDbHome(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, updateDbHomeResponse.opcWorkRequestId);
            return { response: updateDbHomeResponse, workRequestResponse: getWorkRequestResponse };
        });
    }
    /**
     * Waits forUpdateDbSystem
     *
     * @param request the request to send
     * @return response returns UpdateDbSystemResponse, GetWorkRequestResponse tuple
     */
    forUpdateDbSystem(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const updateDbSystemResponse = yield this.client.updateDbSystem(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, updateDbSystemResponse.opcWorkRequestId);
            return { response: updateDbSystemResponse, workRequestResponse: getWorkRequestResponse };
        });
    }
    /**
     * Waits forUpdateExadataInfrastructure
     *
     * @param request the request to send
     * @return response returns UpdateExadataInfrastructureResponse, GetWorkRequestResponse tuple
     */
    forUpdateExadataInfrastructure(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const updateExadataInfrastructureResponse = yield this.client.updateExadataInfrastructure(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, updateExadataInfrastructureResponse.opcWorkRequestId);
            return {
                response: updateExadataInfrastructureResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forUpdateExadataIormConfig
     *
     * @param request the request to send
     * @return response returns UpdateExadataIormConfigResponse, GetWorkRequestResponse tuple
     */
    forUpdateExadataIormConfig(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const updateExadataIormConfigResponse = yield this.client.updateExadataIormConfig(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, updateExadataIormConfigResponse.opcWorkRequestId);
            return {
                response: updateExadataIormConfigResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forUpdateExternalContainerDatabase
     *
     * @param request the request to send
     * @return response returns UpdateExternalContainerDatabaseResponse, GetWorkRequestResponse tuple
     */
    forUpdateExternalContainerDatabase(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const updateExternalContainerDatabaseResponse = yield this.client.updateExternalContainerDatabase(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, updateExternalContainerDatabaseResponse.opcWorkRequestId);
            return {
                response: updateExternalContainerDatabaseResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forUpdateExternalDatabaseConnector
     *
     * @param request the request to send
     * @return response returns UpdateExternalDatabaseConnectorResponse, GetWorkRequestResponse tuple
     */
    forUpdateExternalDatabaseConnector(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const updateExternalDatabaseConnectorResponse = yield this.client.updateExternalDatabaseConnector(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, updateExternalDatabaseConnectorResponse.opcWorkRequestId);
            return {
                response: updateExternalDatabaseConnectorResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forUpdateExternalNonContainerDatabase
     *
     * @param request the request to send
     * @return response returns UpdateExternalNonContainerDatabaseResponse, GetWorkRequestResponse tuple
     */
    forUpdateExternalNonContainerDatabase(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const updateExternalNonContainerDatabaseResponse = yield this.client.updateExternalNonContainerDatabase(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, updateExternalNonContainerDatabaseResponse.opcWorkRequestId);
            return {
                response: updateExternalNonContainerDatabaseResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forUpdateExternalPluggableDatabase
     *
     * @param request the request to send
     * @return response returns UpdateExternalPluggableDatabaseResponse, GetWorkRequestResponse tuple
     */
    forUpdateExternalPluggableDatabase(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const updateExternalPluggableDatabaseResponse = yield this.client.updateExternalPluggableDatabase(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, updateExternalPluggableDatabaseResponse.opcWorkRequestId);
            return {
                response: updateExternalPluggableDatabaseResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forUpdateVmCluster
     *
     * @param request the request to send
     * @return response returns UpdateVmClusterResponse, GetWorkRequestResponse tuple
     */
    forUpdateVmCluster(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const updateVmClusterResponse = yield this.client.updateVmCluster(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, updateVmClusterResponse.opcWorkRequestId);
            return { response: updateVmClusterResponse, workRequestResponse: getWorkRequestResponse };
        });
    }
    /**
     * Waits forUpdateVmClusterNetwork
     *
     * @param request the request to send
     * @return response returns UpdateVmClusterNetworkResponse, GetWorkRequestResponse tuple
     */
    forUpdateVmClusterNetwork(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const updateVmClusterNetworkResponse = yield this.client.updateVmClusterNetwork(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, updateVmClusterNetworkResponse.opcWorkRequestId);
            return {
                response: updateVmClusterNetworkResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forUpgradeDatabase
     *
     * @param request the request to send
     * @return response returns UpgradeDatabaseResponse, GetWorkRequestResponse tuple
     */
    forUpgradeDatabase(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const upgradeDatabaseResponse = yield this.client.upgradeDatabase(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, upgradeDatabaseResponse.opcWorkRequestId);
            return { response: upgradeDatabaseResponse, workRequestResponse: getWorkRequestResponse };
        });
    }
    /**
     * Waits forValidateVmClusterNetwork
     *
     * @param request the request to send
     * @return response returns ValidateVmClusterNetworkResponse, GetWorkRequestResponse tuple
     */
    forValidateVmClusterNetwork(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const validateVmClusterNetworkResponse = yield this.client.validateVmClusterNetwork(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, validateVmClusterNetworkResponse.opcWorkRequestId);
            return {
                response: validateVmClusterNetworkResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
}
exports.DatabaseWaiter = DatabaseWaiter;
