// oci-datacatalog/lib/model/index.js
"use strict";
/**
 * Data Catalog API
 * Use the Data Catalog APIs to collect, organize, find, access, understand, enrich, and activate technical, business, and operational metadata.
 * OpenAPI spec version: 20190325
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataAssetSummary = exports.DataAssetPermissionsSummary = exports.DataAssetCollection = exports.DataAsset = exports.CustomPropertyTypeUsage = exports.CustomPropertySummary = exports.CustomPropertySetUsage = exports.CustomPropertyGetUsage = exports.CustomPropertyDataType = exports.CustomPropertyCollection = exports.CustomProperty = exports.CreateTermRelationshipDetails = exports.CreateTermDetails = exports.CreateTagDetails = exports.CreatePatternDetails = exports.CreateNamespaceDetails = exports.CreateJobExecutionDetails = exports.CreateJobDetails = exports.CreateJobDefinitionDetails = exports.CreateGlossaryDetails = exports.CreateFolderDetails = exports.CreateEntityDetails = exports.CreateDataAssetDetails = exports.CreateCustomPropertyDetails = exports.CreateConnectionDetails = exports.CreateCatalogPrivateEndpointDetails = exports.CreateCatalogDetails = exports.CreateAttributeDetails = exports.ConnectionSummary = exports.ConnectionResult = exports.ConnectionCollection = exports.ConnectionAliasSummary = exports.Connection = exports.ChangeCatalogPrivateEndpointCompartmentDetails = exports.ChangeCatalogCompartmentDetails = exports.CatalogSummary = exports.CatalogPrivateEndpointSummary = exports.CatalogPrivateEndpoint = exports.CatalogPermissionsSummary = exports.Catalog = exports.BaseTagSummary = exports.BaseTag = exports.BasePermissionsSummary = exports.AttributeTagSummary = exports.AttributeTagCollection = exports.AttributeTag = exports.AttributeSummary = exports.AttributeCollection = exports.Attribute = exports.AttachCatalogPrivateEndpointDetails = void 0;
exports.JobMetricSummary = exports.JobMetricCollection = exports.JobMetric = exports.JobLogSummary = exports.JobLogCollection = exports.JobLog = exports.JobLifecycleState = exports.JobExecutionSummary = exports.JobExecutionState = exports.JobExecutionCollection = exports.JobExecution = exports.JobDefinitionSummary = exports.JobDefinitionScope = exports.JobDefinitionPermissionsSummary = exports.JobDefinitionCollection = exports.JobDefinition = exports.JobCollection = exports.Job = exports.ImportGlossaryDetails = exports.ImportConnectionDetails = exports.HarvestStatus = exports.GlossaryTreeElement = exports.GlossarySummary = exports.GlossaryPermissionsSummary = exports.GlossaryCollection = exports.Glossary = exports.FolderTagSummary = exports.FolderTagCollection = exports.FolderTag = exports.FolderSummary = exports.FolderCollection = exports.Folder = exports.FacetedSearchStringFilterRequest = exports.FacetedSearchSortRequest = exports.FacetedSearchFilterRequest = exports.FacetedSearchDateFilterRequest = exports.FacetedSearchCustomProperty = exports.FacetedSearchAggregation = exports.EntityTagSummary = exports.EntityTagCollection = exports.EntityTag = exports.EntitySummary = exports.EntityCollection = exports.Entity = exports.DetachCatalogPrivateEndpointDetails = exports.DerivedLogicalEntities = exports.DataSelectorPatternDetails = exports.DataAssetTagSummary = exports.DataAssetTagCollection = exports.DataAssetTag = void 0;
exports.UpdateFolderDetails = exports.UpdateEntityDetails = exports.UpdateDataAssetDetails = exports.UpdateCustomPropertyDetails = exports.UpdateConnectionDetails = exports.UpdateCatalogPrivateEndpointDetails = exports.UpdateCatalogDetails = exports.UpdateAttributeDetails = exports.TypeSummary = exports.TypeCustomPropertyDetails = exports.TypeCollection = exports.Type = exports.TermWorkflowStatus = exports.TermSummary = exports.TermRelationshipSummary = exports.TermRelationshipCollection = exports.TermRelationship = exports.TermCollection = exports.TermAssociatedObject = exports.Term = exports.SuggestResults = exports.SuggestListItem = exports.SearchTermSummary = exports.SearchTagSummary = exports.SearchResultCollection = exports.SearchResult = exports.SearchCriteria = exports.RuleType = exports.RuleSummary = exports.RuleOriginType = exports.RuleCollection = exports.RuleAttribute = exports.RecommendationType = exports.RecommendationStatus = exports.RecommendationResourceType = exports.RecommendationDetails = exports.RecommendationCollection = exports.PropertyDefinition = exports.ProcessRecommendationDetails = exports.PatternSummary = exports.PatternCollection = exports.Pattern = exports.ParseConnectionDetails = exports.NamespaceSummary = exports.NamespaceCollection = exports.Namespace = exports.LifecycleState = exports.JobType = exports.JobSummary = exports.JobScheduleType = void 0;
exports.WorkRequestResource = exports.WorkRequestLog = exports.WorkRequestError = exports.WorkRequest = exports.ValidatePatternResult = exports.ValidatePatternDetails = exports.ValidateConnectionResult = exports.ValidateConnectionDetails = exports.UploadCredentialsDetails = exports.UpdateTermRelationshipDetails = exports.UpdateTermDetails = exports.UpdatePatternDetails = exports.UpdateNamespaceDetails = exports.UpdateJobDetails = exports.UpdateJobDefinitionDetails = exports.UpdateGlossaryDetails = void 0;
const AttachCatalogPrivateEndpointDetails = __importStar(require("./attach-catalog-private-endpoint-details"));
exports.AttachCatalogPrivateEndpointDetails = AttachCatalogPrivateEndpointDetails.AttachCatalogPrivateEndpointDetails;
const Attribute = __importStar(require("./attribute"));
exports.Attribute = Attribute.Attribute;
const AttributeCollection = __importStar(require("./attribute-collection"));
exports.AttributeCollection = AttributeCollection.AttributeCollection;
const AttributeSummary = __importStar(require("./attribute-summary"));
exports.AttributeSummary = AttributeSummary.AttributeSummary;
const AttributeTag = __importStar(require("./attribute-tag"));
exports.AttributeTag = AttributeTag.AttributeTag;
const AttributeTagCollection = __importStar(require("./attribute-tag-collection"));
exports.AttributeTagCollection = AttributeTagCollection.AttributeTagCollection;
const AttributeTagSummary = __importStar(require("./attribute-tag-summary"));
exports.AttributeTagSummary = AttributeTagSummary.AttributeTagSummary;
const BasePermissionsSummary = __importStar(require("./base-permissions-summary"));
exports.BasePermissionsSummary = BasePermissionsSummary.BasePermissionsSummary;
const BaseTag = __importStar(require("./base-tag"));
exports.BaseTag = BaseTag.BaseTag;
const BaseTagSummary = __importStar(require("./base-tag-summary"));
exports.BaseTagSummary = BaseTagSummary.BaseTagSummary;
const Catalog = __importStar(require("./catalog"));
exports.Catalog = Catalog.Catalog;
const CatalogPermissionsSummary = __importStar(require("./catalog-permissions-summary"));
exports.CatalogPermissionsSummary = CatalogPermissionsSummary.CatalogPermissionsSummary;
const CatalogPrivateEndpoint = __importStar(require("./catalog-private-endpoint"));
exports.CatalogPrivateEndpoint = CatalogPrivateEndpoint.CatalogPrivateEndpoint;
const CatalogPrivateEndpointSummary = __importStar(require("./catalog-private-endpoint-summary"));
exports.CatalogPrivateEndpointSummary = CatalogPrivateEndpointSummary.CatalogPrivateEndpointSummary;
const CatalogSummary = __importStar(require("./catalog-summary"));
exports.CatalogSummary = CatalogSummary.CatalogSummary;
const ChangeCatalogCompartmentDetails = __importStar(require("./change-catalog-compartment-details"));
exports.ChangeCatalogCompartmentDetails = ChangeCatalogCompartmentDetails.ChangeCatalogCompartmentDetails;
const ChangeCatalogPrivateEndpointCompartmentDetails = __importStar(require("./change-catalog-private-endpoint-compartment-details"));
exports.ChangeCatalogPrivateEndpointCompartmentDetails = ChangeCatalogPrivateEndpointCompartmentDetails.ChangeCatalogPrivateEndpointCompartmentDetails;
const Connection = __importStar(require("./connection"));
exports.Connection = Connection.Connection;
const ConnectionAliasSummary = __importStar(require("./connection-alias-summary"));
exports.ConnectionAliasSummary = ConnectionAliasSummary.ConnectionAliasSummary;
const ConnectionCollection = __importStar(require("./connection-collection"));
exports.ConnectionCollection = ConnectionCollection.ConnectionCollection;
const ConnectionResult = __importStar(require("./connection-result"));
exports.ConnectionResult = ConnectionResult.ConnectionResult;
const ConnectionSummary = __importStar(require("./connection-summary"));
exports.ConnectionSummary = ConnectionSummary.ConnectionSummary;
const CreateAttributeDetails = __importStar(require("./create-attribute-details"));
exports.CreateAttributeDetails = CreateAttributeDetails.CreateAttributeDetails;
const CreateCatalogDetails = __importStar(require("./create-catalog-details"));
exports.CreateCatalogDetails = CreateCatalogDetails.CreateCatalogDetails;
const CreateCatalogPrivateEndpointDetails = __importStar(require("./create-catalog-private-endpoint-details"));
exports.CreateCatalogPrivateEndpointDetails = CreateCatalogPrivateEndpointDetails.CreateCatalogPrivateEndpointDetails;
const CreateConnectionDetails = __importStar(require("./create-connection-details"));
exports.CreateConnectionDetails = CreateConnectionDetails.CreateConnectionDetails;
const CreateCustomPropertyDetails = __importStar(require("./create-custom-property-details"));
exports.CreateCustomPropertyDetails = CreateCustomPropertyDetails.CreateCustomPropertyDetails;
const CreateDataAssetDetails = __importStar(require("./create-data-asset-details"));
exports.CreateDataAssetDetails = CreateDataAssetDetails.CreateDataAssetDetails;
const CreateEntityDetails = __importStar(require("./create-entity-details"));
exports.CreateEntityDetails = CreateEntityDetails.CreateEntityDetails;
const CreateFolderDetails = __importStar(require("./create-folder-details"));
exports.CreateFolderDetails = CreateFolderDetails.CreateFolderDetails;
const CreateGlossaryDetails = __importStar(require("./create-glossary-details"));
exports.CreateGlossaryDetails = CreateGlossaryDetails.CreateGlossaryDetails;
const CreateJobDefinitionDetails = __importStar(require("./create-job-definition-details"));
exports.CreateJobDefinitionDetails = CreateJobDefinitionDetails.CreateJobDefinitionDetails;
const CreateJobDetails = __importStar(require("./create-job-details"));
exports.CreateJobDetails = CreateJobDetails.CreateJobDetails;
const CreateJobExecutionDetails = __importStar(require("./create-job-execution-details"));
exports.CreateJobExecutionDetails = CreateJobExecutionDetails.CreateJobExecutionDetails;
const CreateNamespaceDetails = __importStar(require("./create-namespace-details"));
exports.CreateNamespaceDetails = CreateNamespaceDetails.CreateNamespaceDetails;
const CreatePatternDetails = __importStar(require("./create-pattern-details"));
exports.CreatePatternDetails = CreatePatternDetails.CreatePatternDetails;
const CreateTagDetails = __importStar(require("./create-tag-details"));
exports.CreateTagDetails = CreateTagDetails.CreateTagDetails;
const CreateTermDetails = __importStar(require("./create-term-details"));
exports.CreateTermDetails = CreateTermDetails.CreateTermDetails;
const CreateTermRelationshipDetails = __importStar(require("./create-term-relationship-details"));
exports.CreateTermRelationshipDetails = CreateTermRelationshipDetails.CreateTermRelationshipDetails;
const CustomProperty = __importStar(require("./custom-property"));
exports.CustomProperty = CustomProperty.CustomProperty;
const CustomPropertyCollection = __importStar(require("./custom-property-collection"));
exports.CustomPropertyCollection = CustomPropertyCollection.CustomPropertyCollection;
const CustomPropertyDataType = __importStar(require("./custom-property-data-type"));
exports.CustomPropertyDataType = CustomPropertyDataType.CustomPropertyDataType;
const CustomPropertyGetUsage = __importStar(require("./custom-property-get-usage"));
exports.CustomPropertyGetUsage = CustomPropertyGetUsage.CustomPropertyGetUsage;
const CustomPropertySetUsage = __importStar(require("./custom-property-set-usage"));
exports.CustomPropertySetUsage = CustomPropertySetUsage.CustomPropertySetUsage;
const CustomPropertySummary = __importStar(require("./custom-property-summary"));
exports.CustomPropertySummary = CustomPropertySummary.CustomPropertySummary;
const CustomPropertyTypeUsage = __importStar(require("./custom-property-type-usage"));
exports.CustomPropertyTypeUsage = CustomPropertyTypeUsage.CustomPropertyTypeUsage;
const DataAsset = __importStar(require("./data-asset"));
exports.DataAsset = DataAsset.DataAsset;
const DataAssetCollection = __importStar(require("./data-asset-collection"));
exports.DataAssetCollection = DataAssetCollection.DataAssetCollection;
const DataAssetPermissionsSummary = __importStar(require("./data-asset-permissions-summary"));
exports.DataAssetPermissionsSummary = DataAssetPermissionsSummary.DataAssetPermissionsSummary;
const DataAssetSummary = __importStar(require("./data-asset-summary"));
exports.DataAssetSummary = DataAssetSummary.DataAssetSummary;
const DataAssetTag = __importStar(require("./data-asset-tag"));
exports.DataAssetTag = DataAssetTag.DataAssetTag;
const DataAssetTagCollection = __importStar(require("./data-asset-tag-collection"));
exports.DataAssetTagCollection = DataAssetTagCollection.DataAssetTagCollection;
const DataAssetTagSummary = __importStar(require("./data-asset-tag-summary"));
exports.DataAssetTagSummary = DataAssetTagSummary.DataAssetTagSummary;
const DataSelectorPatternDetails = __importStar(require("./data-selector-pattern-details"));
exports.DataSelectorPatternDetails = DataSelectorPatternDetails.DataSelectorPatternDetails;
const DerivedLogicalEntities = __importStar(require("./derived-logical-entities"));
exports.DerivedLogicalEntities = DerivedLogicalEntities.DerivedLogicalEntities;
const DetachCatalogPrivateEndpointDetails = __importStar(require("./detach-catalog-private-endpoint-details"));
exports.DetachCatalogPrivateEndpointDetails = DetachCatalogPrivateEndpointDetails.DetachCatalogPrivateEndpointDetails;
const Entity = __importStar(require("./entity"));
exports.Entity = Entity.Entity;
const EntityCollection = __importStar(require("./entity-collection"));
exports.EntityCollection = EntityCollection.EntityCollection;
const EntitySummary = __importStar(require("./entity-summary"));
exports.EntitySummary = EntitySummary.EntitySummary;
const EntityTag = __importStar(require("./entity-tag"));
exports.EntityTag = EntityTag.EntityTag;
const EntityTagCollection = __importStar(require("./entity-tag-collection"));
exports.EntityTagCollection = EntityTagCollection.EntityTagCollection;
const EntityTagSummary = __importStar(require("./entity-tag-summary"));
exports.EntityTagSummary = EntityTagSummary.EntityTagSummary;
const FacetedSearchAggregation = __importStar(require("./faceted-search-aggregation"));
exports.FacetedSearchAggregation = FacetedSearchAggregation.FacetedSearchAggregation;
const FacetedSearchCustomProperty = __importStar(require("./faceted-search-custom-property"));
exports.FacetedSearchCustomProperty = FacetedSearchCustomProperty.FacetedSearchCustomProperty;
const FacetedSearchDateFilterRequest = __importStar(require("./faceted-search-date-filter-request"));
exports.FacetedSearchDateFilterRequest = FacetedSearchDateFilterRequest.FacetedSearchDateFilterRequest;
const FacetedSearchFilterRequest = __importStar(require("./faceted-search-filter-request"));
exports.FacetedSearchFilterRequest = FacetedSearchFilterRequest.FacetedSearchFilterRequest;
const FacetedSearchSortRequest = __importStar(require("./faceted-search-sort-request"));
exports.FacetedSearchSortRequest = FacetedSearchSortRequest.FacetedSearchSortRequest;
const FacetedSearchStringFilterRequest = __importStar(require("./faceted-search-string-filter-request"));
exports.FacetedSearchStringFilterRequest = FacetedSearchStringFilterRequest.FacetedSearchStringFilterRequest;
const Folder = __importStar(require("./folder"));
exports.Folder = Folder.Folder;
const FolderCollection = __importStar(require("./folder-collection"));
exports.FolderCollection = FolderCollection.FolderCollection;
const FolderSummary = __importStar(require("./folder-summary"));
exports.FolderSummary = FolderSummary.FolderSummary;
const FolderTag = __importStar(require("./folder-tag"));
exports.FolderTag = FolderTag.FolderTag;
const FolderTagCollection = __importStar(require("./folder-tag-collection"));
exports.FolderTagCollection = FolderTagCollection.FolderTagCollection;
const FolderTagSummary = __importStar(require("./folder-tag-summary"));
exports.FolderTagSummary = FolderTagSummary.FolderTagSummary;
const Glossary = __importStar(require("./glossary"));
exports.Glossary = Glossary.Glossary;
const GlossaryCollection = __importStar(require("./glossary-collection"));
exports.GlossaryCollection = GlossaryCollection.GlossaryCollection;
const GlossaryPermissionsSummary = __importStar(require("./glossary-permissions-summary"));
exports.GlossaryPermissionsSummary = GlossaryPermissionsSummary.GlossaryPermissionsSummary;
const GlossarySummary = __importStar(require("./glossary-summary"));
exports.GlossarySummary = GlossarySummary.GlossarySummary;
const GlossaryTreeElement = __importStar(require("./glossary-tree-element"));
exports.GlossaryTreeElement = GlossaryTreeElement.GlossaryTreeElement;
const HarvestStatus = __importStar(require("./harvest-status"));
exports.HarvestStatus = HarvestStatus.HarvestStatus;
const ImportConnectionDetails = __importStar(require("./import-connection-details"));
exports.ImportConnectionDetails = ImportConnectionDetails.ImportConnectionDetails;
const ImportGlossaryDetails = __importStar(require("./import-glossary-details"));
exports.ImportGlossaryDetails = ImportGlossaryDetails.ImportGlossaryDetails;
const Job = __importStar(require("./job"));
exports.Job = Job.Job;
const JobCollection = __importStar(require("./job-collection"));
exports.JobCollection = JobCollection.JobCollection;
const JobDefinition = __importStar(require("./job-definition"));
exports.JobDefinition = JobDefinition.JobDefinition;
const JobDefinitionCollection = __importStar(require("./job-definition-collection"));
exports.JobDefinitionCollection = JobDefinitionCollection.JobDefinitionCollection;
const JobDefinitionPermissionsSummary = __importStar(require("./job-definition-permissions-summary"));
exports.JobDefinitionPermissionsSummary = JobDefinitionPermissionsSummary.JobDefinitionPermissionsSummary;
const JobDefinitionScope = __importStar(require("./job-definition-scope"));
exports.JobDefinitionScope = JobDefinitionScope.JobDefinitionScope;
const JobDefinitionSummary = __importStar(require("./job-definition-summary"));
exports.JobDefinitionSummary = JobDefinitionSummary.JobDefinitionSummary;
const JobExecution = __importStar(require("./job-execution"));
exports.JobExecution = JobExecution.JobExecution;
const JobExecutionCollection = __importStar(require("./job-execution-collection"));
exports.JobExecutionCollection = JobExecutionCollection.JobExecutionCollection;
const JobExecutionState = __importStar(require("./job-execution-state"));
exports.JobExecutionState = JobExecutionState.JobExecutionState;
const JobExecutionSummary = __importStar(require("./job-execution-summary"));
exports.JobExecutionSummary = JobExecutionSummary.JobExecutionSummary;
const JobLifecycleState = __importStar(require("./job-lifecycle-state"));
exports.JobLifecycleState = JobLifecycleState.JobLifecycleState;
const JobLog = __importStar(require("./job-log"));
exports.JobLog = JobLog.JobLog;
const JobLogCollection = __importStar(require("./job-log-collection"));
exports.JobLogCollection = JobLogCollection.JobLogCollection;
const JobLogSummary = __importStar(require("./job-log-summary"));
exports.JobLogSummary = JobLogSummary.JobLogSummary;
const JobMetric = __importStar(require("./job-metric"));
exports.JobMetric = JobMetric.JobMetric;
const JobMetricCollection = __importStar(require("./job-metric-collection"));
exports.JobMetricCollection = JobMetricCollection.JobMetricCollection;
const JobMetricSummary = __importStar(require("./job-metric-summary"));
exports.JobMetricSummary = JobMetricSummary.JobMetricSummary;
const JobScheduleType = __importStar(require("./job-schedule-type"));
exports.JobScheduleType = JobScheduleType.JobScheduleType;
const JobSummary = __importStar(require("./job-summary"));
exports.JobSummary = JobSummary.JobSummary;
const JobType = __importStar(require("./job-type"));
exports.JobType = JobType.JobType;
const LifecycleState = __importStar(require("./lifecycle-state"));
exports.LifecycleState = LifecycleState.LifecycleState;
const Namespace = __importStar(require("./namespace"));
exports.Namespace = Namespace.Namespace;
const NamespaceCollection = __importStar(require("./namespace-collection"));
exports.NamespaceCollection = NamespaceCollection.NamespaceCollection;
const NamespaceSummary = __importStar(require("./namespace-summary"));
exports.NamespaceSummary = NamespaceSummary.NamespaceSummary;
const ParseConnectionDetails = __importStar(require("./parse-connection-details"));
exports.ParseConnectionDetails = ParseConnectionDetails.ParseConnectionDetails;
const Pattern = __importStar(require("./pattern"));
exports.Pattern = Pattern.Pattern;
const PatternCollection = __importStar(require("./pattern-collection"));
exports.PatternCollection = PatternCollection.PatternCollection;
const PatternSummary = __importStar(require("./pattern-summary"));
exports.PatternSummary = PatternSummary.PatternSummary;
const ProcessRecommendationDetails = __importStar(require("./process-recommendation-details"));
exports.ProcessRecommendationDetails = ProcessRecommendationDetails.ProcessRecommendationDetails;
const PropertyDefinition = __importStar(require("./property-definition"));
exports.PropertyDefinition = PropertyDefinition.PropertyDefinition;
const RecommendationCollection = __importStar(require("./recommendation-collection"));
exports.RecommendationCollection = RecommendationCollection.RecommendationCollection;
const RecommendationDetails = __importStar(require("./recommendation-details"));
exports.RecommendationDetails = RecommendationDetails.RecommendationDetails;
const RecommendationResourceType = __importStar(require("./recommendation-resource-type"));
exports.RecommendationResourceType = RecommendationResourceType.RecommendationResourceType;
const RecommendationStatus = __importStar(require("./recommendation-status"));
exports.RecommendationStatus = RecommendationStatus.RecommendationStatus;
const RecommendationType = __importStar(require("./recommendation-type"));
exports.RecommendationType = RecommendationType.RecommendationType;
const RuleAttribute = __importStar(require("./rule-attribute"));
exports.RuleAttribute = RuleAttribute.RuleAttribute;
const RuleCollection = __importStar(require("./rule-collection"));
exports.RuleCollection = RuleCollection.RuleCollection;
const RuleOriginType = __importStar(require("./rule-origin-type"));
exports.RuleOriginType = RuleOriginType.RuleOriginType;
const RuleSummary = __importStar(require("./rule-summary"));
exports.RuleSummary = RuleSummary.RuleSummary;
const RuleType = __importStar(require("./rule-type"));
exports.RuleType = RuleType.RuleType;
const SearchCriteria = __importStar(require("./search-criteria"));
exports.SearchCriteria = SearchCriteria.SearchCriteria;
const SearchResult = __importStar(require("./search-result"));
exports.SearchResult = SearchResult.SearchResult;
const SearchResultCollection = __importStar(require("./search-result-collection"));
exports.SearchResultCollection = SearchResultCollection.SearchResultCollection;
const SearchTagSummary = __importStar(require("./search-tag-summary"));
exports.SearchTagSummary = SearchTagSummary.SearchTagSummary;
const SearchTermSummary = __importStar(require("./search-term-summary"));
exports.SearchTermSummary = SearchTermSummary.SearchTermSummary;
const SuggestListItem = __importStar(require("./suggest-list-item"));
exports.SuggestListItem = SuggestListItem.SuggestListItem;
const SuggestResults = __importStar(require("./suggest-results"));
exports.SuggestResults = SuggestResults.SuggestResults;
const Term = __importStar(require("./term"));
exports.Term = Term.Term;
const TermAssociatedObject = __importStar(require("./term-associated-object"));
exports.TermAssociatedObject = TermAssociatedObject.TermAssociatedObject;
const TermCollection = __importStar(require("./term-collection"));
exports.TermCollection = TermCollection.TermCollection;
const TermRelationship = __importStar(require("./term-relationship"));
exports.TermRelationship = TermRelationship.TermRelationship;
const TermRelationshipCollection = __importStar(require("./term-relationship-collection"));
exports.TermRelationshipCollection = TermRelationshipCollection.TermRelationshipCollection;
const TermRelationshipSummary = __importStar(require("./term-relationship-summary"));
exports.TermRelationshipSummary = TermRelationshipSummary.TermRelationshipSummary;
const TermSummary = __importStar(require("./term-summary"));
exports.TermSummary = TermSummary.TermSummary;
const TermWorkflowStatus = __importStar(require("./term-workflow-status"));
exports.TermWorkflowStatus = TermWorkflowStatus.TermWorkflowStatus;
const Type = __importStar(require("./type"));
exports.Type = Type.Type;
const TypeCollection = __importStar(require("./type-collection"));
exports.TypeCollection = TypeCollection.TypeCollection;
const TypeCustomPropertyDetails = __importStar(require("./type-custom-property-details"));
exports.TypeCustomPropertyDetails = TypeCustomPropertyDetails.TypeCustomPropertyDetails;
const TypeSummary = __importStar(require("./type-summary"));
exports.TypeSummary = TypeSummary.TypeSummary;
const UpdateAttributeDetails = __importStar(require("./update-attribute-details"));
exports.UpdateAttributeDetails = UpdateAttributeDetails.UpdateAttributeDetails;
const UpdateCatalogDetails = __importStar(require("./update-catalog-details"));
exports.UpdateCatalogDetails = UpdateCatalogDetails.UpdateCatalogDetails;
const UpdateCatalogPrivateEndpointDetails = __importStar(require("./update-catalog-private-endpoint-details"));
exports.UpdateCatalogPrivateEndpointDetails = UpdateCatalogPrivateEndpointDetails.UpdateCatalogPrivateEndpointDetails;
const UpdateConnectionDetails = __importStar(require("./update-connection-details"));
exports.UpdateConnectionDetails = UpdateConnectionDetails.UpdateConnectionDetails;
const UpdateCustomPropertyDetails = __importStar(require("./update-custom-property-details"));
exports.UpdateCustomPropertyDetails = UpdateCustomPropertyDetails.UpdateCustomPropertyDetails;
const UpdateDataAssetDetails = __importStar(require("./update-data-asset-details"));
exports.UpdateDataAssetDetails = UpdateDataAssetDetails.UpdateDataAssetDetails;
const UpdateEntityDetails = __importStar(require("./update-entity-details"));
exports.UpdateEntityDetails = UpdateEntityDetails.UpdateEntityDetails;
const UpdateFolderDetails = __importStar(require("./update-folder-details"));
exports.UpdateFolderDetails = UpdateFolderDetails.UpdateFolderDetails;
const UpdateGlossaryDetails = __importStar(require("./update-glossary-details"));
exports.UpdateGlossaryDetails = UpdateGlossaryDetails.UpdateGlossaryDetails;
const UpdateJobDefinitionDetails = __importStar(require("./update-job-definition-details"));
exports.UpdateJobDefinitionDetails = UpdateJobDefinitionDetails.UpdateJobDefinitionDetails;
const UpdateJobDetails = __importStar(require("./update-job-details"));
exports.UpdateJobDetails = UpdateJobDetails.UpdateJobDetails;
const UpdateNamespaceDetails = __importStar(require("./update-namespace-details"));
exports.UpdateNamespaceDetails = UpdateNamespaceDetails.UpdateNamespaceDetails;
const UpdatePatternDetails = __importStar(require("./update-pattern-details"));
exports.UpdatePatternDetails = UpdatePatternDetails.UpdatePatternDetails;
const UpdateTermDetails = __importStar(require("./update-term-details"));
exports.UpdateTermDetails = UpdateTermDetails.UpdateTermDetails;
const UpdateTermRelationshipDetails = __importStar(require("./update-term-relationship-details"));
exports.UpdateTermRelationshipDetails = UpdateTermRelationshipDetails.UpdateTermRelationshipDetails;
const UploadCredentialsDetails = __importStar(require("./upload-credentials-details"));
exports.UploadCredentialsDetails = UploadCredentialsDetails.UploadCredentialsDetails;
const ValidateConnectionDetails = __importStar(require("./validate-connection-details"));
exports.ValidateConnectionDetails = ValidateConnectionDetails.ValidateConnectionDetails;
const ValidateConnectionResult = __importStar(require("./validate-connection-result"));
exports.ValidateConnectionResult = ValidateConnectionResult.ValidateConnectionResult;
const ValidatePatternDetails = __importStar(require("./validate-pattern-details"));
exports.ValidatePatternDetails = ValidatePatternDetails.ValidatePatternDetails;
const ValidatePatternResult = __importStar(require("./validate-pattern-result"));
exports.ValidatePatternResult = ValidatePatternResult.ValidatePatternResult;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestLog = __importStar(require("./work-request-log"));
exports.WorkRequestLog = WorkRequestLog.WorkRequestLog;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
