// oci-datacatalog/lib/model/job-schedule-type.js
"use strict";
/**
 * Data Catalog API
 * Use the Data Catalog APIs to collect, organize, find, access, understand, enrich, and activate technical, business, and operational metadata.
 * OpenAPI spec version: 20190325
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.JobScheduleType = void 0;
/**
 * Type of job schedule.
 * SCHEDULED - The job will execute at a later time based on a schedule.
 * IMMEDIATE - The job will execute as soon as possible.
 *
 **/
var JobScheduleType;
(function (JobScheduleType) {
    JobScheduleType["Scheduled"] = "SCHEDULED";
    JobScheduleType["Immediate"] = "IMMEDIATE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    JobScheduleType["UnknownValue"] = "UNKNOWN_VALUE";
})(JobScheduleType = exports.JobScheduleType || (exports.JobScheduleType = {}));
(function (JobScheduleType) {
    function getJsonObj(obj) {
        return { obj };
    }
    JobScheduleType.getJsonObj = getJsonObj;
})(JobScheduleType = exports.JobScheduleType || (exports.JobScheduleType = {}));
