// oci-datacatalog/lib/model/recommendation-resource-type.js
"use strict";
/**
 * Data Catalog API
 * Use the Data Catalog APIs to collect, organize, find, access, understand, enrich, and activate technical, business, and operational metadata.
 * OpenAPI spec version: 20190325
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecommendationResourceType = void 0;
/**
 * Data Catalog resource type which can be used in recommendations.
 * - DATA_ENTITY
 * - ATTRIBUTE
 * - TERM
 * - CATEGORY
 *
 **/
var RecommendationResourceType;
(function (RecommendationResourceType) {
    RecommendationResourceType["DataEntity"] = "DATA_ENTITY";
    RecommendationResourceType["Attribute"] = "ATTRIBUTE";
    RecommendationResourceType["Term"] = "TERM";
    RecommendationResourceType["Category"] = "CATEGORY";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    RecommendationResourceType["UnknownValue"] = "UNKNOWN_VALUE";
})(RecommendationResourceType = exports.RecommendationResourceType || (exports.RecommendationResourceType = {}));
(function (RecommendationResourceType) {
    function getJsonObj(obj) {
        return { obj };
    }
    RecommendationResourceType.getJsonObj = getJsonObj;
})(RecommendationResourceType = exports.RecommendationResourceType || (exports.RecommendationResourceType = {}));
