// oci-datacatalog/lib/model/rule-origin-type.js
"use strict";
/**
 * Data Catalog API
 * Use the Data Catalog APIs to collect, organize, find, access, understand, enrich, and activate technical, business, and operational metadata.
 * OpenAPI spec version: 20190325
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RuleOriginType = void 0;
/**
 * Origin type for the given rule.
 * SOURCE - Represents rule is harvested and is as defined in source system.
 * USER - Represents user defined rules at data catalog system.
 * PROFILING - Represents rules which are inferred via profiling.
 *
 **/
var RuleOriginType;
(function (RuleOriginType) {
    RuleOriginType["Source"] = "SOURCE";
    RuleOriginType["User"] = "USER";
    RuleOriginType["Profiling"] = "PROFILING";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    RuleOriginType["UnknownValue"] = "UNKNOWN_VALUE";
})(RuleOriginType = exports.RuleOriginType || (exports.RuleOriginType = {}));
(function (RuleOriginType) {
    function getJsonObj(obj) {
        return { obj };
    }
    RuleOriginType.getJsonObj = getJsonObj;
})(RuleOriginType = exports.RuleOriginType || (exports.RuleOriginType = {}));
