// oci-apigateway/lib/request/index.js
"use strict";
/**
 * API Gateway API
 * API for the API Gateway service. Use this API to manage gateways, deployments, and related items.
For more information, see
[Overview of API Gateway](/iaas/Content/APIGateway/Concepts/apigatewayoverview.htm).

 * OpenAPI spec version: 20190501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWorkRequestsRequest = exports.ListWorkRequestLogsRequest = exports.ListWorkRequestErrorsRequest = exports.ListGatewaysRequest = exports.ListDeploymentsRequest = exports.ListSdksRequest = exports.ListSdkLanguageTypesRequest = exports.ListCertificatesRequest = exports.ListApisRequest = void 0;
const ListApisRequest = __importStar(require("./list-apis-request"));
exports.ListApisRequest = ListApisRequest.ListApisRequest;
const ListCertificatesRequest = __importStar(require("./list-certificates-request"));
exports.ListCertificatesRequest = ListCertificatesRequest.ListCertificatesRequest;
const ListSdkLanguageTypesRequest = __importStar(require("./list-sdk-language-types-request"));
exports.ListSdkLanguageTypesRequest = ListSdkLanguageTypesRequest.ListSdkLanguageTypesRequest;
const ListSdksRequest = __importStar(require("./list-sdks-request"));
exports.ListSdksRequest = ListSdksRequest.ListSdksRequest;
const ListDeploymentsRequest = __importStar(require("./list-deployments-request"));
exports.ListDeploymentsRequest = ListDeploymentsRequest.ListDeploymentsRequest;
const ListGatewaysRequest = __importStar(require("./list-gateways-request"));
exports.ListGatewaysRequest = ListGatewaysRequest.ListGatewaysRequest;
const ListWorkRequestErrorsRequest = __importStar(require("./list-work-request-errors-request"));
exports.ListWorkRequestErrorsRequest = ListWorkRequestErrorsRequest.ListWorkRequestErrorsRequest;
const ListWorkRequestLogsRequest = __importStar(require("./list-work-request-logs-request"));
exports.ListWorkRequestLogsRequest = ListWorkRequestLogsRequest.ListWorkRequestLogsRequest;
const ListWorkRequestsRequest = __importStar(require("./list-work-requests-request"));
exports.ListWorkRequestsRequest = ListWorkRequestsRequest.ListWorkRequestsRequest;
