// oci-apigateway/lib/request/list-deployments-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListDeploymentsRequest = void 0;
var ListDeploymentsRequest;
(function (ListDeploymentsRequest) {
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListDeploymentsRequest.SortOrder || (ListDeploymentsRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["DisplayName"] = "displayName";
    })(SortBy = ListDeploymentsRequest.SortBy || (ListDeploymentsRequest.SortBy = {}));
})(ListDeploymentsRequest = exports.ListDeploymentsRequest || (exports.ListDeploymentsRequest = {}));
