// oci-computeinstanceagent/lib/request/list-instance-agent-plugins-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20180530
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListInstanceAgentPluginsRequest = void 0;
var ListInstanceAgentPluginsRequest;
(function (ListInstanceAgentPluginsRequest) {
    let Status;
    (function (Status) {
        Status["Running"] = "RUNNING";
        Status["Stopped"] = "STOPPED";
        Status["NotSupported"] = "NOT_SUPPORTED";
        Status["Invalid"] = "INVALID";
    })(Status = ListInstanceAgentPluginsRequest.Status || (ListInstanceAgentPluginsRequest.Status = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["Timecreated"] = "TIMECREATED";
        SortBy["Displayname"] = "DISPLAYNAME";
    })(SortBy = ListInstanceAgentPluginsRequest.SortBy || (ListInstanceAgentPluginsRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListInstanceAgentPluginsRequest.SortOrder || (ListInstanceAgentPluginsRequest.SortOrder = {}));
})(ListInstanceAgentPluginsRequest = exports.ListInstanceAgentPluginsRequest || (exports.ListInstanceAgentPluginsRequest = {}));
