// oci-mysql/lib/model/analytics-cluster-memory-estimate-status.js
"use strict";
/**
 * MySQL Database Service API
 * The API for the MySQL Database Service
 * OpenAPI spec version: 20190415
 * Contact: mysql-cloud-dev_ww_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AnalyticsClusterMemoryEstimateStatus = void 0;
/**
 * DEPRECATED -- please use HeatWave API instead.
 * Possible operation status.
 *
 **/
var AnalyticsClusterMemoryEstimateStatus;
(function (AnalyticsClusterMemoryEstimateStatus) {
    AnalyticsClusterMemoryEstimateStatus["Accepted"] = "ACCEPTED";
    AnalyticsClusterMemoryEstimateStatus["InProgress"] = "IN_PROGRESS";
    AnalyticsClusterMemoryEstimateStatus["Failed"] = "FAILED";
    AnalyticsClusterMemoryEstimateStatus["Succeeded"] = "SUCCEEDED";
    AnalyticsClusterMemoryEstimateStatus["Canceling"] = "CANCELING";
    AnalyticsClusterMemoryEstimateStatus["Canceled"] = "CANCELED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AnalyticsClusterMemoryEstimateStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(AnalyticsClusterMemoryEstimateStatus = exports.AnalyticsClusterMemoryEstimateStatus || (exports.AnalyticsClusterMemoryEstimateStatus = {}));
(function (AnalyticsClusterMemoryEstimateStatus) {
    function getJsonObj(obj) {
        return { obj };
    }
    AnalyticsClusterMemoryEstimateStatus.getJsonObj = getJsonObj;
})(AnalyticsClusterMemoryEstimateStatus = exports.AnalyticsClusterMemoryEstimateStatus || (exports.AnalyticsClusterMemoryEstimateStatus = {}));
