// oci-mysql/lib/model/db-system.js
"use strict";
/**
 * MySQL Database Service API
 * The API for the MySQL Database Service
 * OpenAPI spec version: 20190415
 * Contact: mysql-cloud-dev_ww_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DbSystem = void 0;
const model = __importStar(require("../model"));
var DbSystem;
(function (DbSystem) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Inactive"] = "INACTIVE";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = DbSystem.LifecycleState || (DbSystem.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "currentPlacement": obj.currentPlacement
                ? model.DbSystemPlacement.getJsonObj(obj.currentPlacement)
                : undefined,
            "analyticsCluster": obj.analyticsCluster
                ? model.AnalyticsClusterSummary.getJsonObj(obj.analyticsCluster)
                : undefined,
            "heatWaveCluster": obj.heatWaveCluster
                ? model.HeatWaveClusterSummary.getJsonObj(obj.heatWaveCluster)
                : undefined,
            "backupPolicy": obj.backupPolicy
                ? model.BackupPolicy.getJsonObj(obj.backupPolicy)
                : undefined,
            "source": obj.source ? model.DbSystemSource.getJsonObj(obj.source) : undefined,
            "endpoints": obj.endpoints
                ? obj.endpoints.map(item => {
                    return model.DbSystemEndpoint.getJsonObj(item);
                })
                : undefined,
            "channels": obj.channels
                ? obj.channels.map(item => {
                    return model.ChannelSummary.getJsonObj(item);
                })
                : undefined,
            "maintenance": obj.maintenance
                ? model.MaintenanceDetails.getJsonObj(obj.maintenance)
                : undefined
        });
        return jsonObj;
    }
    DbSystem.getJsonObj = getJsonObj;
})(DbSystem = exports.DbSystem || (exports.DbSystem = {}));
