// oci-mysql/lib/model/index.js
"use strict";
/**
 * MySQL Database Service API
 * The API for the MySQL Database Service
 * OpenAPI spec version: 20190415
 * Contact: mysql-cloud-dev_ww_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateBackupPolicyDetails = exports.UpdateBackupDetails = exports.UpdateAnalyticsClusterDetails = exports.StopDbSystemDetails = exports.ShapeSummary = exports.RestartDbSystemDetails = exports.MaintenanceDetails = exports.InnoDbShutdownMode = exports.HeatWaveNode = exports.HeatWaveClusterTableMemoryEstimate = exports.HeatWaveClusterSummary = exports.HeatWaveClusterSchemaMemoryEstimate = exports.HeatWaveClusterMemoryEstimateStatus = exports.HeatWaveClusterMemoryEstimate = exports.HeatWaveCluster = exports.DbSystemSummary = exports.DbSystemSource = exports.DbSystemSnapshot = exports.DbSystemPlacement = exports.DbSystemEndpoint = exports.DbSystem = exports.CreateMaintenanceDetails = exports.CreateDbSystemSourceDetails = exports.CreateDbSystemDetails = exports.CreateConfigurationDetails = exports.CreateChannelTargetDetails = exports.CreateChannelSourceDetails = exports.CreateChannelDetails = exports.CreateBackupPolicyDetails = exports.CreateBackupDetails = exports.ConfigurationVariables = exports.ConfigurationSummary = exports.Configuration = exports.ChannelTarget = exports.ChannelSummary = exports.ChannelSource = exports.Channel = exports.CaCertificate = exports.BackupSummary = exports.BackupPolicy = exports.Backup = exports.AnalyticsClusterTableMemoryEstimate = exports.AnalyticsClusterSummary = exports.AnalyticsClusterSchemaMemoryEstimate = exports.AnalyticsClusterNode = exports.AnalyticsClusterMemoryEstimateStatus = exports.AnalyticsClusterMemoryEstimate = exports.AnalyticsCluster = exports.AddHeatWaveClusterDetails = exports.AddAnalyticsClusterDetails = void 0;
exports.UpdateChannelTargetFromDbSystemDetails = exports.UpdateChannelSourceFromMysqlDetails = exports.PemCaCertificate = exports.DbSystemSourceImportFromUrl = exports.DbSystemSourceFromNone = exports.DbSystemSourceFromBackup = exports.CreateDbSystemSourceImportFromUrlDetails = exports.CreateDbSystemSourceFromNoneDetails = exports.CreateDbSystemSourceFromBackupDetails = exports.CreateChannelTargetFromDbSystemDetails = exports.CreateChannelSourceFromMysqlDetails = exports.ChannelTargetDbSystem = exports.ChannelSourceMysql = exports.WorkRequestSummary = exports.WorkRequestResourceActionType = exports.WorkRequestResource = exports.WorkRequestOperationType = exports.WorkRequestOperationStatus = exports.WorkRequestLogEntry = exports.WorkRequestError = exports.WorkRequest = exports.VersionSummary = exports.Version = exports.UpdateMaintenanceDetails = exports.UpdateHeatWaveClusterDetails = exports.UpdateDbSystemDetails = exports.UpdateConfigurationDetails = exports.UpdateChannelTargetDetails = exports.UpdateChannelSourceDetails = exports.UpdateChannelDetails = void 0;
const AddAnalyticsClusterDetails = __importStar(require("./add-analytics-cluster-details"));
exports.AddAnalyticsClusterDetails = AddAnalyticsClusterDetails.AddAnalyticsClusterDetails;
const AddHeatWaveClusterDetails = __importStar(require("./add-heat-wave-cluster-details"));
exports.AddHeatWaveClusterDetails = AddHeatWaveClusterDetails.AddHeatWaveClusterDetails;
const AnalyticsCluster = __importStar(require("./analytics-cluster"));
exports.AnalyticsCluster = AnalyticsCluster.AnalyticsCluster;
const AnalyticsClusterMemoryEstimate = __importStar(require("./analytics-cluster-memory-estimate"));
exports.AnalyticsClusterMemoryEstimate = AnalyticsClusterMemoryEstimate.AnalyticsClusterMemoryEstimate;
const AnalyticsClusterMemoryEstimateStatus = __importStar(require("./analytics-cluster-memory-estimate-status"));
exports.AnalyticsClusterMemoryEstimateStatus = AnalyticsClusterMemoryEstimateStatus.AnalyticsClusterMemoryEstimateStatus;
const AnalyticsClusterNode = __importStar(require("./analytics-cluster-node"));
exports.AnalyticsClusterNode = AnalyticsClusterNode.AnalyticsClusterNode;
const AnalyticsClusterSchemaMemoryEstimate = __importStar(require("./analytics-cluster-schema-memory-estimate"));
exports.AnalyticsClusterSchemaMemoryEstimate = AnalyticsClusterSchemaMemoryEstimate.AnalyticsClusterSchemaMemoryEstimate;
const AnalyticsClusterSummary = __importStar(require("./analytics-cluster-summary"));
exports.AnalyticsClusterSummary = AnalyticsClusterSummary.AnalyticsClusterSummary;
const AnalyticsClusterTableMemoryEstimate = __importStar(require("./analytics-cluster-table-memory-estimate"));
exports.AnalyticsClusterTableMemoryEstimate = AnalyticsClusterTableMemoryEstimate.AnalyticsClusterTableMemoryEstimate;
const Backup = __importStar(require("./backup"));
exports.Backup = Backup.Backup;
const BackupPolicy = __importStar(require("./backup-policy"));
exports.BackupPolicy = BackupPolicy.BackupPolicy;
const BackupSummary = __importStar(require("./backup-summary"));
exports.BackupSummary = BackupSummary.BackupSummary;
const CaCertificate = __importStar(require("./ca-certificate"));
exports.CaCertificate = CaCertificate.CaCertificate;
const Channel = __importStar(require("./channel"));
exports.Channel = Channel.Channel;
const ChannelSource = __importStar(require("./channel-source"));
exports.ChannelSource = ChannelSource.ChannelSource;
const ChannelSummary = __importStar(require("./channel-summary"));
exports.ChannelSummary = ChannelSummary.ChannelSummary;
const ChannelTarget = __importStar(require("./channel-target"));
exports.ChannelTarget = ChannelTarget.ChannelTarget;
const Configuration = __importStar(require("./configuration"));
exports.Configuration = Configuration.Configuration;
const ConfigurationSummary = __importStar(require("./configuration-summary"));
exports.ConfigurationSummary = ConfigurationSummary.ConfigurationSummary;
const ConfigurationVariables = __importStar(require("./configuration-variables"));
exports.ConfigurationVariables = ConfigurationVariables.ConfigurationVariables;
const CreateBackupDetails = __importStar(require("./create-backup-details"));
exports.CreateBackupDetails = CreateBackupDetails.CreateBackupDetails;
const CreateBackupPolicyDetails = __importStar(require("./create-backup-policy-details"));
exports.CreateBackupPolicyDetails = CreateBackupPolicyDetails.CreateBackupPolicyDetails;
const CreateChannelDetails = __importStar(require("./create-channel-details"));
exports.CreateChannelDetails = CreateChannelDetails.CreateChannelDetails;
const CreateChannelSourceDetails = __importStar(require("./create-channel-source-details"));
exports.CreateChannelSourceDetails = CreateChannelSourceDetails.CreateChannelSourceDetails;
const CreateChannelTargetDetails = __importStar(require("./create-channel-target-details"));
exports.CreateChannelTargetDetails = CreateChannelTargetDetails.CreateChannelTargetDetails;
const CreateConfigurationDetails = __importStar(require("./create-configuration-details"));
exports.CreateConfigurationDetails = CreateConfigurationDetails.CreateConfigurationDetails;
const CreateDbSystemDetails = __importStar(require("./create-db-system-details"));
exports.CreateDbSystemDetails = CreateDbSystemDetails.CreateDbSystemDetails;
const CreateDbSystemSourceDetails = __importStar(require("./create-db-system-source-details"));
exports.CreateDbSystemSourceDetails = CreateDbSystemSourceDetails.CreateDbSystemSourceDetails;
const CreateMaintenanceDetails = __importStar(require("./create-maintenance-details"));
exports.CreateMaintenanceDetails = CreateMaintenanceDetails.CreateMaintenanceDetails;
const DbSystem = __importStar(require("./db-system"));
exports.DbSystem = DbSystem.DbSystem;
const DbSystemEndpoint = __importStar(require("./db-system-endpoint"));
exports.DbSystemEndpoint = DbSystemEndpoint.DbSystemEndpoint;
const DbSystemPlacement = __importStar(require("./db-system-placement"));
exports.DbSystemPlacement = DbSystemPlacement.DbSystemPlacement;
const DbSystemSnapshot = __importStar(require("./db-system-snapshot"));
exports.DbSystemSnapshot = DbSystemSnapshot.DbSystemSnapshot;
const DbSystemSource = __importStar(require("./db-system-source"));
exports.DbSystemSource = DbSystemSource.DbSystemSource;
const DbSystemSummary = __importStar(require("./db-system-summary"));
exports.DbSystemSummary = DbSystemSummary.DbSystemSummary;
const HeatWaveCluster = __importStar(require("./heat-wave-cluster"));
exports.HeatWaveCluster = HeatWaveCluster.HeatWaveCluster;
const HeatWaveClusterMemoryEstimate = __importStar(require("./heat-wave-cluster-memory-estimate"));
exports.HeatWaveClusterMemoryEstimate = HeatWaveClusterMemoryEstimate.HeatWaveClusterMemoryEstimate;
const HeatWaveClusterMemoryEstimateStatus = __importStar(require("./heat-wave-cluster-memory-estimate-status"));
exports.HeatWaveClusterMemoryEstimateStatus = HeatWaveClusterMemoryEstimateStatus.HeatWaveClusterMemoryEstimateStatus;
const HeatWaveClusterSchemaMemoryEstimate = __importStar(require("./heat-wave-cluster-schema-memory-estimate"));
exports.HeatWaveClusterSchemaMemoryEstimate = HeatWaveClusterSchemaMemoryEstimate.HeatWaveClusterSchemaMemoryEstimate;
const HeatWaveClusterSummary = __importStar(require("./heat-wave-cluster-summary"));
exports.HeatWaveClusterSummary = HeatWaveClusterSummary.HeatWaveClusterSummary;
const HeatWaveClusterTableMemoryEstimate = __importStar(require("./heat-wave-cluster-table-memory-estimate"));
exports.HeatWaveClusterTableMemoryEstimate = HeatWaveClusterTableMemoryEstimate.HeatWaveClusterTableMemoryEstimate;
const HeatWaveNode = __importStar(require("./heat-wave-node"));
exports.HeatWaveNode = HeatWaveNode.HeatWaveNode;
const InnoDbShutdownMode = __importStar(require("./inno-db-shutdown-mode"));
exports.InnoDbShutdownMode = InnoDbShutdownMode.InnoDbShutdownMode;
const MaintenanceDetails = __importStar(require("./maintenance-details"));
exports.MaintenanceDetails = MaintenanceDetails.MaintenanceDetails;
const RestartDbSystemDetails = __importStar(require("./restart-db-system-details"));
exports.RestartDbSystemDetails = RestartDbSystemDetails.RestartDbSystemDetails;
const ShapeSummary = __importStar(require("./shape-summary"));
exports.ShapeSummary = ShapeSummary.ShapeSummary;
const StopDbSystemDetails = __importStar(require("./stop-db-system-details"));
exports.StopDbSystemDetails = StopDbSystemDetails.StopDbSystemDetails;
const UpdateAnalyticsClusterDetails = __importStar(require("./update-analytics-cluster-details"));
exports.UpdateAnalyticsClusterDetails = UpdateAnalyticsClusterDetails.UpdateAnalyticsClusterDetails;
const UpdateBackupDetails = __importStar(require("./update-backup-details"));
exports.UpdateBackupDetails = UpdateBackupDetails.UpdateBackupDetails;
const UpdateBackupPolicyDetails = __importStar(require("./update-backup-policy-details"));
exports.UpdateBackupPolicyDetails = UpdateBackupPolicyDetails.UpdateBackupPolicyDetails;
const UpdateChannelDetails = __importStar(require("./update-channel-details"));
exports.UpdateChannelDetails = UpdateChannelDetails.UpdateChannelDetails;
const UpdateChannelSourceDetails = __importStar(require("./update-channel-source-details"));
exports.UpdateChannelSourceDetails = UpdateChannelSourceDetails.UpdateChannelSourceDetails;
const UpdateChannelTargetDetails = __importStar(require("./update-channel-target-details"));
exports.UpdateChannelTargetDetails = UpdateChannelTargetDetails.UpdateChannelTargetDetails;
const UpdateConfigurationDetails = __importStar(require("./update-configuration-details"));
exports.UpdateConfigurationDetails = UpdateConfigurationDetails.UpdateConfigurationDetails;
const UpdateDbSystemDetails = __importStar(require("./update-db-system-details"));
exports.UpdateDbSystemDetails = UpdateDbSystemDetails.UpdateDbSystemDetails;
const UpdateHeatWaveClusterDetails = __importStar(require("./update-heat-wave-cluster-details"));
exports.UpdateHeatWaveClusterDetails = UpdateHeatWaveClusterDetails.UpdateHeatWaveClusterDetails;
const UpdateMaintenanceDetails = __importStar(require("./update-maintenance-details"));
exports.UpdateMaintenanceDetails = UpdateMaintenanceDetails.UpdateMaintenanceDetails;
const Version = __importStar(require("./version"));
exports.Version = Version.Version;
const VersionSummary = __importStar(require("./version-summary"));
exports.VersionSummary = VersionSummary.VersionSummary;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestOperationStatus = __importStar(require("./work-request-operation-status"));
exports.WorkRequestOperationStatus = WorkRequestOperationStatus.WorkRequestOperationStatus;
const WorkRequestOperationType = __importStar(require("./work-request-operation-type"));
exports.WorkRequestOperationType = WorkRequestOperationType.WorkRequestOperationType;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestResourceActionType = __importStar(require("./work-request-resource-action-type"));
exports.WorkRequestResourceActionType = WorkRequestResourceActionType.WorkRequestResourceActionType;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const ChannelSourceMysql = __importStar(require("./channel-source-mysql"));
exports.ChannelSourceMysql = ChannelSourceMysql.ChannelSourceMysql;
const ChannelTargetDbSystem = __importStar(require("./channel-target-db-system"));
exports.ChannelTargetDbSystem = ChannelTargetDbSystem.ChannelTargetDbSystem;
const CreateChannelSourceFromMysqlDetails = __importStar(require("./create-channel-source-from-mysql-details"));
exports.CreateChannelSourceFromMysqlDetails = CreateChannelSourceFromMysqlDetails.CreateChannelSourceFromMysqlDetails;
const CreateChannelTargetFromDbSystemDetails = __importStar(require("./create-channel-target-from-db-system-details"));
exports.CreateChannelTargetFromDbSystemDetails = CreateChannelTargetFromDbSystemDetails.CreateChannelTargetFromDbSystemDetails;
const CreateDbSystemSourceFromBackupDetails = __importStar(require("./create-db-system-source-from-backup-details"));
exports.CreateDbSystemSourceFromBackupDetails = CreateDbSystemSourceFromBackupDetails.CreateDbSystemSourceFromBackupDetails;
const CreateDbSystemSourceFromNoneDetails = __importStar(require("./create-db-system-source-from-none-details"));
exports.CreateDbSystemSourceFromNoneDetails = CreateDbSystemSourceFromNoneDetails.CreateDbSystemSourceFromNoneDetails;
const CreateDbSystemSourceImportFromUrlDetails = __importStar(require("./create-db-system-source-import-from-url-details"));
exports.CreateDbSystemSourceImportFromUrlDetails = CreateDbSystemSourceImportFromUrlDetails.CreateDbSystemSourceImportFromUrlDetails;
const DbSystemSourceFromBackup = __importStar(require("./db-system-source-from-backup"));
exports.DbSystemSourceFromBackup = DbSystemSourceFromBackup.DbSystemSourceFromBackup;
const DbSystemSourceFromNone = __importStar(require("./db-system-source-from-none"));
exports.DbSystemSourceFromNone = DbSystemSourceFromNone.DbSystemSourceFromNone;
const DbSystemSourceImportFromUrl = __importStar(require("./db-system-source-import-from-url"));
exports.DbSystemSourceImportFromUrl = DbSystemSourceImportFromUrl.DbSystemSourceImportFromUrl;
const PemCaCertificate = __importStar(require("./pem-ca-certificate"));
exports.PemCaCertificate = PemCaCertificate.PemCaCertificate;
const UpdateChannelSourceFromMysqlDetails = __importStar(require("./update-channel-source-from-mysql-details"));
exports.UpdateChannelSourceFromMysqlDetails = UpdateChannelSourceFromMysqlDetails.UpdateChannelSourceFromMysqlDetails;
const UpdateChannelTargetFromDbSystemDetails = __importStar(require("./update-channel-target-from-db-system-details"));
exports.UpdateChannelTargetFromDbSystemDetails = UpdateChannelTargetFromDbSystemDetails.UpdateChannelTargetFromDbSystemDetails;
