// oci-mysql/lib/model/shape-summary.js
"use strict";
/**
 * MySQL Database Service API
 * The API for the MySQL Database Service
 * OpenAPI spec version: 20190415
 * Contact: mysql-cloud-dev_ww_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShapeSummary = void 0;
var ShapeSummary;
(function (ShapeSummary) {
    let IsSupportedFor;
    (function (IsSupportedFor) {
        IsSupportedFor["Dbsystem"] = "DBSYSTEM";
        IsSupportedFor["Analyticscluster"] = "ANALYTICSCLUSTER";
        IsSupportedFor["Heatwavecluster"] = "HEATWAVECLUSTER";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        IsSupportedFor["UnknownValue"] = "UNKNOWN_VALUE";
    })(IsSupportedFor = ShapeSummary.IsSupportedFor || (ShapeSummary.IsSupportedFor = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ShapeSummary.getJsonObj = getJsonObj;
})(ShapeSummary = exports.ShapeSummary || (exports.ShapeSummary = {}));
