// oci-mysql/lib/model/work-request-operation-type.js
"use strict";
/**
 * MySQL Database Service API
 * The API for the MySQL Database Service
 * OpenAPI spec version: 20190415
 * Contact: mysql-cloud-dev_ww_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestOperationType = void 0;
/**
 * Possible operation types.
 **/
var WorkRequestOperationType;
(function (WorkRequestOperationType) {
    WorkRequestOperationType["CreateDbsystem"] = "CREATE_DBSYSTEM";
    WorkRequestOperationType["UpdateDbsystem"] = "UPDATE_DBSYSTEM";
    WorkRequestOperationType["DeleteDbsystem"] = "DELETE_DBSYSTEM";
    WorkRequestOperationType["StartDbsystem"] = "START_DBSYSTEM";
    WorkRequestOperationType["StopDbsystem"] = "STOP_DBSYSTEM";
    WorkRequestOperationType["RestartDbsystem"] = "RESTART_DBSYSTEM";
    WorkRequestOperationType["AddAnalyticsCluster"] = "ADD_ANALYTICS_CLUSTER";
    WorkRequestOperationType["UpdateAnalyticsCluster"] = "UPDATE_ANALYTICS_CLUSTER";
    WorkRequestOperationType["DeleteAnalyticsCluster"] = "DELETE_ANALYTICS_CLUSTER";
    WorkRequestOperationType["StartAnalyticsCluster"] = "START_ANALYTICS_CLUSTER";
    WorkRequestOperationType["StopAnalyticsCluster"] = "STOP_ANALYTICS_CLUSTER";
    WorkRequestOperationType["RestartAnalyticsCluster"] = "RESTART_ANALYTICS_CLUSTER";
    WorkRequestOperationType["GenerateAnalyticsClusterMemoryEstimate"] = "GENERATE_ANALYTICS_CLUSTER_MEMORY_ESTIMATE";
    WorkRequestOperationType["AddHeatwaveCluster"] = "ADD_HEATWAVE_CLUSTER";
    WorkRequestOperationType["UpdateHeatwaveCluster"] = "UPDATE_HEATWAVE_CLUSTER";
    WorkRequestOperationType["DeleteHeatwaveCluster"] = "DELETE_HEATWAVE_CLUSTER";
    WorkRequestOperationType["StartHeatwaveCluster"] = "START_HEATWAVE_CLUSTER";
    WorkRequestOperationType["StopHeatwaveCluster"] = "STOP_HEATWAVE_CLUSTER";
    WorkRequestOperationType["RestartHeatwaveCluster"] = "RESTART_HEATWAVE_CLUSTER";
    WorkRequestOperationType["GenerateHeatwaveClusterMemoryEstimate"] = "GENERATE_HEATWAVE_CLUSTER_MEMORY_ESTIMATE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    WorkRequestOperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(WorkRequestOperationType = exports.WorkRequestOperationType || (exports.WorkRequestOperationType = {}));
(function (WorkRequestOperationType) {
    function getJsonObj(obj) {
        return { obj };
    }
    WorkRequestOperationType.getJsonObj = getJsonObj;
})(WorkRequestOperationType = exports.WorkRequestOperationType || (exports.WorkRequestOperationType = {}));
