// oci-cims/lib/model/contact.js
"use strict";
/**
 * Support Management API
 * Use the Support Management API to manage support requests. For more information, see [Getting Help and Contacting Support](/iaas/Content/GSG/Tasks/contactingsupport.htm). **Note**: Before you can create service requests with this API, you need to have an Oracle Single Sign On (SSO) account, and you need to register your Customer Support Identifier (CSI) with My Oracle Support.
 * OpenAPI spec version: 20181231
 * Contact: oci_ops_cims_dev_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Contact = void 0;
var Contact;
(function (Contact) {
    let ContactType;
    (function (ContactType) {
        ContactType["Primary"] = "PRIMARY";
        ContactType["Alternate"] = "ALTERNATE";
        ContactType["Secondary"] = "SECONDARY";
        ContactType["Admin"] = "ADMIN";
        ContactType["Manager"] = "MANAGER";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ContactType["UnknownValue"] = "UNKNOWN_VALUE";
    })(ContactType = Contact.ContactType || (Contact.ContactType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    Contact.getJsonObj = getJsonObj;
})(Contact = exports.Contact || (exports.Contact = {}));
