// oci-cims/lib/model/index.js
"use strict";
/**
 * Support Management API
 * Use the Support Management API to manage support requests. For more information, see [Getting Help and Contacting Support](/iaas/Content/GSG/Tasks/contactingsupport.htm). **Note**: Before you can create service requests with this API, you need to have an Oracle Single Sign On (SSO) account, and you need to register your Customer Support Identifier (CSI) with My Oracle Support.
 * OpenAPI spec version: 20181231
 * Contact: oci_ops_cims_dev_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateActivityItemDetails = exports.TechSupportItem = exports.LimitItem = exports.CreateTechSupportItemDetails = exports.CreateLimitItemDetails = exports.ActivityItem = exports.ValidationResponse = exports.User = exports.UpdateTicketDetails = exports.UpdateResourceDetails = exports.UpdateItemDetails = exports.UpdateIncident = exports.Unit = exports.TimeZone = exports.Ticket = exports.TenancyInformation = exports.SubCategory = exports.Status = exports.SortOrder = exports.SortBy = exports.ServiceCategory = exports.Scope = exports.Resource = exports.Region = exports.ProblemType = exports.LifecycleState = exports.LifecycleDetails = exports.Item = exports.IssueType = exports.IncidentType = exports.IncidentSummary = exports.IncidentResourceType = exports.Incident = exports.ErrorCode = exports.CreateUserDetails = exports.CreateTicketDetails = exports.CreateSubCategoryDetails = exports.CreateResourceDetails = exports.CreateItemDetails = exports.CreateIssueTypeDetails = exports.CreateIncident = exports.CreateCategoryDetails = exports.ContextualData = exports.ContactList = exports.Contact = exports.Classifier = exports.Category = exports.AvailabilityDomain = void 0;
const AvailabilityDomain = __importStar(require("./availability-domain"));
exports.AvailabilityDomain = AvailabilityDomain.AvailabilityDomain;
const Category = __importStar(require("./category"));
exports.Category = Category.Category;
const Classifier = __importStar(require("./classifier"));
exports.Classifier = Classifier.Classifier;
const Contact = __importStar(require("./contact"));
exports.Contact = Contact.Contact;
const ContactList = __importStar(require("./contact-list"));
exports.ContactList = ContactList.ContactList;
const ContextualData = __importStar(require("./contextual-data"));
exports.ContextualData = ContextualData.ContextualData;
const CreateCategoryDetails = __importStar(require("./create-category-details"));
exports.CreateCategoryDetails = CreateCategoryDetails.CreateCategoryDetails;
const CreateIncident = __importStar(require("./create-incident"));
exports.CreateIncident = CreateIncident.CreateIncident;
const CreateIssueTypeDetails = __importStar(require("./create-issue-type-details"));
exports.CreateIssueTypeDetails = CreateIssueTypeDetails.CreateIssueTypeDetails;
const CreateItemDetails = __importStar(require("./create-item-details"));
exports.CreateItemDetails = CreateItemDetails.CreateItemDetails;
const CreateResourceDetails = __importStar(require("./create-resource-details"));
exports.CreateResourceDetails = CreateResourceDetails.CreateResourceDetails;
const CreateSubCategoryDetails = __importStar(require("./create-sub-category-details"));
exports.CreateSubCategoryDetails = CreateSubCategoryDetails.CreateSubCategoryDetails;
const CreateTicketDetails = __importStar(require("./create-ticket-details"));
exports.CreateTicketDetails = CreateTicketDetails.CreateTicketDetails;
const CreateUserDetails = __importStar(require("./create-user-details"));
exports.CreateUserDetails = CreateUserDetails.CreateUserDetails;
const ErrorCode = __importStar(require("./error-code"));
exports.ErrorCode = ErrorCode.ErrorCode;
const Incident = __importStar(require("./incident"));
exports.Incident = Incident.Incident;
const IncidentResourceType = __importStar(require("./incident-resource-type"));
exports.IncidentResourceType = IncidentResourceType.IncidentResourceType;
const IncidentSummary = __importStar(require("./incident-summary"));
exports.IncidentSummary = IncidentSummary.IncidentSummary;
const IncidentType = __importStar(require("./incident-type"));
exports.IncidentType = IncidentType.IncidentType;
const IssueType = __importStar(require("./issue-type"));
exports.IssueType = IssueType.IssueType;
const Item = __importStar(require("./item"));
exports.Item = Item.Item;
const LifecycleDetails = __importStar(require("./lifecycle-details"));
exports.LifecycleDetails = LifecycleDetails.LifecycleDetails;
const LifecycleState = __importStar(require("./lifecycle-state"));
exports.LifecycleState = LifecycleState.LifecycleState;
const ProblemType = __importStar(require("./problem-type"));
exports.ProblemType = ProblemType.ProblemType;
const Region = __importStar(require("./region"));
exports.Region = Region.Region;
const Resource = __importStar(require("./resource"));
exports.Resource = Resource.Resource;
const Scope = __importStar(require("./scope"));
exports.Scope = Scope.Scope;
const ServiceCategory = __importStar(require("./service-category"));
exports.ServiceCategory = ServiceCategory.ServiceCategory;
const SortBy = __importStar(require("./sort-by"));
exports.SortBy = SortBy.SortBy;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const Status = __importStar(require("./status"));
exports.Status = Status.Status;
const SubCategory = __importStar(require("./sub-category"));
exports.SubCategory = SubCategory.SubCategory;
const TenancyInformation = __importStar(require("./tenancy-information"));
exports.TenancyInformation = TenancyInformation.TenancyInformation;
const Ticket = __importStar(require("./ticket"));
exports.Ticket = Ticket.Ticket;
const TimeZone = __importStar(require("./time-zone"));
exports.TimeZone = TimeZone.TimeZone;
const Unit = __importStar(require("./unit"));
exports.Unit = Unit.Unit;
const UpdateIncident = __importStar(require("./update-incident"));
exports.UpdateIncident = UpdateIncident.UpdateIncident;
const UpdateItemDetails = __importStar(require("./update-item-details"));
exports.UpdateItemDetails = UpdateItemDetails.UpdateItemDetails;
const UpdateResourceDetails = __importStar(require("./update-resource-details"));
exports.UpdateResourceDetails = UpdateResourceDetails.UpdateResourceDetails;
const UpdateTicketDetails = __importStar(require("./update-ticket-details"));
exports.UpdateTicketDetails = UpdateTicketDetails.UpdateTicketDetails;
const User = __importStar(require("./user"));
exports.User = User.User;
const ValidationResponse = __importStar(require("./validation-response"));
exports.ValidationResponse = ValidationResponse.ValidationResponse;
const ActivityItem = __importStar(require("./activity-item"));
exports.ActivityItem = ActivityItem.ActivityItem;
const CreateLimitItemDetails = __importStar(require("./create-limit-item-details"));
exports.CreateLimitItemDetails = CreateLimitItemDetails.CreateLimitItemDetails;
const CreateTechSupportItemDetails = __importStar(require("./create-tech-support-item-details"));
exports.CreateTechSupportItemDetails = CreateTechSupportItemDetails.CreateTechSupportItemDetails;
const LimitItem = __importStar(require("./limit-item"));
exports.LimitItem = LimitItem.LimitItem;
const TechSupportItem = __importStar(require("./tech-support-item"));
exports.TechSupportItem = TechSupportItem.TechSupportItem;
const UpdateActivityItemDetails = __importStar(require("./update-activity-item-details"));
exports.UpdateActivityItemDetails = UpdateActivityItemDetails.UpdateActivityItemDetails;
