// oci-resourcesearch/lib/model/index.js
"use strict";
/**
 * Search Service API
 * Search for resources in your cloud network.
 * OpenAPI spec version: 20180409
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StructuredSearchDetails = exports.FreeTextSearchDetails = exports.SearchDetails = exports.SearchContext = exports.ResourceType = exports.ResourceSummaryCollection = exports.ResourceSummary = exports.QueryableFieldDescription = void 0;
const QueryableFieldDescription = __importStar(require("./queryable-field-description"));
exports.QueryableFieldDescription = QueryableFieldDescription.QueryableFieldDescription;
const ResourceSummary = __importStar(require("./resource-summary"));
exports.ResourceSummary = ResourceSummary.ResourceSummary;
const ResourceSummaryCollection = __importStar(require("./resource-summary-collection"));
exports.ResourceSummaryCollection = ResourceSummaryCollection.ResourceSummaryCollection;
const ResourceType = __importStar(require("./resource-type"));
exports.ResourceType = ResourceType.ResourceType;
const SearchContext = __importStar(require("./search-context"));
exports.SearchContext = SearchContext.SearchContext;
const SearchDetails = __importStar(require("./search-details"));
exports.SearchDetails = SearchDetails.SearchDetails;
const FreeTextSearchDetails = __importStar(require("./free-text-search-details"));
exports.FreeTextSearchDetails = FreeTextSearchDetails.FreeTextSearchDetails;
const StructuredSearchDetails = __importStar(require("./structured-search-details"));
exports.StructuredSearchDetails = StructuredSearchDetails.StructuredSearchDetails;
