// oci-usageapi/lib/model/index.js
"use strict";
/**
 * Usage API
 * Use the Usage API to view your Oracle Cloud usage and costs. The API allows you to request data that meets the specified filter criteria, and to group that data by the dimension of your choosing. The Usage API is used by the Cost Analysis tool in the Console. Also see [Using the Usage API](/Content/Billing/Concepts/costanalysisoverview.htm#cost_analysis_using_the_api) for more information.
 * OpenAPI spec version: 20200107
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsageSummary = exports.UsageAggregation = exports.UpdateScheduleReportDetails = exports.UpdateQueryDetails = exports.UpdateCustomTableDetails = exports.Tag = exports.ScheduleReportSummary = exports.ScheduleReportCollection = exports.ScheduleReport = exports.SavedScheduleReport = exports.SavedCustomTable = exports.RequestSummarizedUsagesDetails = exports.ReportQuery = exports.QuerySummary = exports.QueryDefinition = exports.QueryCollection = exports.Query = exports.Forecast = exports.Filter = exports.Dimension = exports.CustomTableSummary = exports.CustomTableCollection = exports.CustomTable = exports.CreateScheduleReportDetails = exports.CreateQueryDetails = exports.CreateCustomTableDetails = exports.CostAnalysisUI = exports.ConfigurationAggregation = exports.Configuration = void 0;
const Configuration = __importStar(require("./configuration"));
exports.Configuration = Configuration.Configuration;
const ConfigurationAggregation = __importStar(require("./configuration-aggregation"));
exports.ConfigurationAggregation = ConfigurationAggregation.ConfigurationAggregation;
const CostAnalysisUI = __importStar(require("./cost-analysis-ui"));
exports.CostAnalysisUI = CostAnalysisUI.CostAnalysisUI;
const CreateCustomTableDetails = __importStar(require("./create-custom-table-details"));
exports.CreateCustomTableDetails = CreateCustomTableDetails.CreateCustomTableDetails;
const CreateQueryDetails = __importStar(require("./create-query-details"));
exports.CreateQueryDetails = CreateQueryDetails.CreateQueryDetails;
const CreateScheduleReportDetails = __importStar(require("./create-schedule-report-details"));
exports.CreateScheduleReportDetails = CreateScheduleReportDetails.CreateScheduleReportDetails;
const CustomTable = __importStar(require("./custom-table"));
exports.CustomTable = CustomTable.CustomTable;
const CustomTableCollection = __importStar(require("./custom-table-collection"));
exports.CustomTableCollection = CustomTableCollection.CustomTableCollection;
const CustomTableSummary = __importStar(require("./custom-table-summary"));
exports.CustomTableSummary = CustomTableSummary.CustomTableSummary;
const Dimension = __importStar(require("./dimension"));
exports.Dimension = Dimension.Dimension;
const Filter = __importStar(require("./filter"));
exports.Filter = Filter.Filter;
const Forecast = __importStar(require("./forecast"));
exports.Forecast = Forecast.Forecast;
const Query = __importStar(require("./query"));
exports.Query = Query.Query;
const QueryCollection = __importStar(require("./query-collection"));
exports.QueryCollection = QueryCollection.QueryCollection;
const QueryDefinition = __importStar(require("./query-definition"));
exports.QueryDefinition = QueryDefinition.QueryDefinition;
const QuerySummary = __importStar(require("./query-summary"));
exports.QuerySummary = QuerySummary.QuerySummary;
const ReportQuery = __importStar(require("./report-query"));
exports.ReportQuery = ReportQuery.ReportQuery;
const RequestSummarizedUsagesDetails = __importStar(require("./request-summarized-usages-details"));
exports.RequestSummarizedUsagesDetails = RequestSummarizedUsagesDetails.RequestSummarizedUsagesDetails;
const SavedCustomTable = __importStar(require("./saved-custom-table"));
exports.SavedCustomTable = SavedCustomTable.SavedCustomTable;
const SavedScheduleReport = __importStar(require("./saved-schedule-report"));
exports.SavedScheduleReport = SavedScheduleReport.SavedScheduleReport;
const ScheduleReport = __importStar(require("./schedule-report"));
exports.ScheduleReport = ScheduleReport.ScheduleReport;
const ScheduleReportCollection = __importStar(require("./schedule-report-collection"));
exports.ScheduleReportCollection = ScheduleReportCollection.ScheduleReportCollection;
const ScheduleReportSummary = __importStar(require("./schedule-report-summary"));
exports.ScheduleReportSummary = ScheduleReportSummary.ScheduleReportSummary;
const Tag = __importStar(require("./tag"));
exports.Tag = Tag.Tag;
const UpdateCustomTableDetails = __importStar(require("./update-custom-table-details"));
exports.UpdateCustomTableDetails = UpdateCustomTableDetails.UpdateCustomTableDetails;
const UpdateQueryDetails = __importStar(require("./update-query-details"));
exports.UpdateQueryDetails = UpdateQueryDetails.UpdateQueryDetails;
const UpdateScheduleReportDetails = __importStar(require("./update-schedule-report-details"));
exports.UpdateScheduleReportDetails = UpdateScheduleReportDetails.UpdateScheduleReportDetails;
const UsageAggregation = __importStar(require("./usage-aggregation"));
exports.UsageAggregation = UsageAggregation.UsageAggregation;
const UsageSummary = __importStar(require("./usage-summary"));
exports.UsageSummary = UsageSummary.UsageSummary;
