// oci-marketplace/lib/model/agreement-summary.js
"use strict";
/**
 * Marketplace Service API
 * Manage applications in Oracle Cloud Infrastructure Marketplace.
 * OpenAPI spec version: 20181001
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AgreementSummary = void 0;
var AgreementSummary;
(function (AgreementSummary) {
    let Author;
    (function (Author) {
        Author["Oracle"] = "ORACLE";
        Author["Partner"] = "PARTNER";
        Author["Pii"] = "PII";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Author["UnknownValue"] = "UNKNOWN_VALUE";
    })(Author = AgreementSummary.Author || (AgreementSummary.Author = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AgreementSummary.getJsonObj = getJsonObj;
})(AgreementSummary = exports.AgreementSummary || (exports.AgreementSummary = {}));
