// oci-marketplace/lib/model/index.js
"use strict";
/**
 * Marketplace Service API
 * Manage applications in Oracle Cloud Infrastructure Marketplace.
 * OpenAPI spec version: 20181001
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImagePublicationPackage = exports.ImageListingPackage = exports.CreateImagePublicationPackage = exports.UploadData = exports.UpdatePublicationDetails = exports.UpdateAcceptedAgreementDetails = exports.TaxSummary = exports.SupportContact = exports.Screenshot = exports.ReportTypeSummary = exports.ReportTypeCollection = exports.ReportSummary = exports.ReportCollection = exports.Region = exports.PublisherSummary = exports.Publisher = exports.PublicationSummary = exports.PublicationPackageSummary = exports.PublicationPackage = exports.Publication = exports.PricingTypeEnum = exports.PricingStrategyEnum = exports.PricingModel = exports.PricingCurrencyEnum = exports.PackageTypeEnum = exports.OrchestrationVariableTypeEnum = exports.OrchestrationVariable = exports.OperatingSystem = exports.NamedLink = exports.ListingType = exports.ListingSummary = exports.ListingPackageSummary = exports.ListingPackage = exports.Listing = exports.LinkEnum = exports.Link = exports.Item = exports.EulaTypeEnum = exports.Eula = exports.ErrorEntity = exports.DocumentationLink = exports.CreatePublicationPackage = exports.CreatePublicationDetails = exports.CreateAcceptedAgreementDetails = exports.ChangePublicationCompartmentDetails = exports.CategorySummary = exports.AgreementSummary = exports.Agreement = exports.AcceptedAgreementSummary = exports.AcceptedAgreement = void 0;
exports.TextBasedEula = exports.OrchestrationPublicationPackage = exports.OrchestrationListingPackage = void 0;
const AcceptedAgreement = __importStar(require("./accepted-agreement"));
exports.AcceptedAgreement = AcceptedAgreement.AcceptedAgreement;
const AcceptedAgreementSummary = __importStar(require("./accepted-agreement-summary"));
exports.AcceptedAgreementSummary = AcceptedAgreementSummary.AcceptedAgreementSummary;
const Agreement = __importStar(require("./agreement"));
exports.Agreement = Agreement.Agreement;
const AgreementSummary = __importStar(require("./agreement-summary"));
exports.AgreementSummary = AgreementSummary.AgreementSummary;
const CategorySummary = __importStar(require("./category-summary"));
exports.CategorySummary = CategorySummary.CategorySummary;
const ChangePublicationCompartmentDetails = __importStar(require("./change-publication-compartment-details"));
exports.ChangePublicationCompartmentDetails = ChangePublicationCompartmentDetails.ChangePublicationCompartmentDetails;
const CreateAcceptedAgreementDetails = __importStar(require("./create-accepted-agreement-details"));
exports.CreateAcceptedAgreementDetails = CreateAcceptedAgreementDetails.CreateAcceptedAgreementDetails;
const CreatePublicationDetails = __importStar(require("./create-publication-details"));
exports.CreatePublicationDetails = CreatePublicationDetails.CreatePublicationDetails;
const CreatePublicationPackage = __importStar(require("./create-publication-package"));
exports.CreatePublicationPackage = CreatePublicationPackage.CreatePublicationPackage;
const DocumentationLink = __importStar(require("./documentation-link"));
exports.DocumentationLink = DocumentationLink.DocumentationLink;
const ErrorEntity = __importStar(require("./error-entity"));
exports.ErrorEntity = ErrorEntity.ErrorEntity;
const Eula = __importStar(require("./eula"));
exports.Eula = Eula.Eula;
const EulaTypeEnum = __importStar(require("./eula-type-enum"));
exports.EulaTypeEnum = EulaTypeEnum.EulaTypeEnum;
const Item = __importStar(require("./item"));
exports.Item = Item.Item;
const Link = __importStar(require("./link"));
exports.Link = Link.Link;
const LinkEnum = __importStar(require("./link-enum"));
exports.LinkEnum = LinkEnum.LinkEnum;
const Listing = __importStar(require("./listing"));
exports.Listing = Listing.Listing;
const ListingPackage = __importStar(require("./listing-package"));
exports.ListingPackage = ListingPackage.ListingPackage;
const ListingPackageSummary = __importStar(require("./listing-package-summary"));
exports.ListingPackageSummary = ListingPackageSummary.ListingPackageSummary;
const ListingSummary = __importStar(require("./listing-summary"));
exports.ListingSummary = ListingSummary.ListingSummary;
const ListingType = __importStar(require("./listing-type"));
exports.ListingType = ListingType.ListingType;
const NamedLink = __importStar(require("./named-link"));
exports.NamedLink = NamedLink.NamedLink;
const OperatingSystem = __importStar(require("./operating-system"));
exports.OperatingSystem = OperatingSystem.OperatingSystem;
const OrchestrationVariable = __importStar(require("./orchestration-variable"));
exports.OrchestrationVariable = OrchestrationVariable.OrchestrationVariable;
const OrchestrationVariableTypeEnum = __importStar(require("./orchestration-variable-type-enum"));
exports.OrchestrationVariableTypeEnum = OrchestrationVariableTypeEnum.OrchestrationVariableTypeEnum;
const PackageTypeEnum = __importStar(require("./package-type-enum"));
exports.PackageTypeEnum = PackageTypeEnum.PackageTypeEnum;
const PricingCurrencyEnum = __importStar(require("./pricing-currency-enum"));
exports.PricingCurrencyEnum = PricingCurrencyEnum.PricingCurrencyEnum;
const PricingModel = __importStar(require("./pricing-model"));
exports.PricingModel = PricingModel.PricingModel;
const PricingStrategyEnum = __importStar(require("./pricing-strategy-enum"));
exports.PricingStrategyEnum = PricingStrategyEnum.PricingStrategyEnum;
const PricingTypeEnum = __importStar(require("./pricing-type-enum"));
exports.PricingTypeEnum = PricingTypeEnum.PricingTypeEnum;
const Publication = __importStar(require("./publication"));
exports.Publication = Publication.Publication;
const PublicationPackage = __importStar(require("./publication-package"));
exports.PublicationPackage = PublicationPackage.PublicationPackage;
const PublicationPackageSummary = __importStar(require("./publication-package-summary"));
exports.PublicationPackageSummary = PublicationPackageSummary.PublicationPackageSummary;
const PublicationSummary = __importStar(require("./publication-summary"));
exports.PublicationSummary = PublicationSummary.PublicationSummary;
const Publisher = __importStar(require("./publisher"));
exports.Publisher = Publisher.Publisher;
const PublisherSummary = __importStar(require("./publisher-summary"));
exports.PublisherSummary = PublisherSummary.PublisherSummary;
const Region = __importStar(require("./region"));
exports.Region = Region.Region;
const ReportCollection = __importStar(require("./report-collection"));
exports.ReportCollection = ReportCollection.ReportCollection;
const ReportSummary = __importStar(require("./report-summary"));
exports.ReportSummary = ReportSummary.ReportSummary;
const ReportTypeCollection = __importStar(require("./report-type-collection"));
exports.ReportTypeCollection = ReportTypeCollection.ReportTypeCollection;
const ReportTypeSummary = __importStar(require("./report-type-summary"));
exports.ReportTypeSummary = ReportTypeSummary.ReportTypeSummary;
const Screenshot = __importStar(require("./screenshot"));
exports.Screenshot = Screenshot.Screenshot;
const SupportContact = __importStar(require("./support-contact"));
exports.SupportContact = SupportContact.SupportContact;
const TaxSummary = __importStar(require("./tax-summary"));
exports.TaxSummary = TaxSummary.TaxSummary;
const UpdateAcceptedAgreementDetails = __importStar(require("./update-accepted-agreement-details"));
exports.UpdateAcceptedAgreementDetails = UpdateAcceptedAgreementDetails.UpdateAcceptedAgreementDetails;
const UpdatePublicationDetails = __importStar(require("./update-publication-details"));
exports.UpdatePublicationDetails = UpdatePublicationDetails.UpdatePublicationDetails;
const UploadData = __importStar(require("./upload-data"));
exports.UploadData = UploadData.UploadData;
const CreateImagePublicationPackage = __importStar(require("./create-image-publication-package"));
exports.CreateImagePublicationPackage = CreateImagePublicationPackage.CreateImagePublicationPackage;
const ImageListingPackage = __importStar(require("./image-listing-package"));
exports.ImageListingPackage = ImageListingPackage.ImageListingPackage;
const ImagePublicationPackage = __importStar(require("./image-publication-package"));
exports.ImagePublicationPackage = ImagePublicationPackage.ImagePublicationPackage;
const OrchestrationListingPackage = __importStar(require("./orchestration-listing-package"));
exports.OrchestrationListingPackage = OrchestrationListingPackage.OrchestrationListingPackage;
const OrchestrationPublicationPackage = __importStar(require("./orchestration-publication-package"));
exports.OrchestrationPublicationPackage = OrchestrationPublicationPackage.OrchestrationPublicationPackage;
const TextBasedEula = __importStar(require("./text-based-eula"));
exports.TextBasedEula = TextBasedEula.TextBasedEula;
