// oci-marketplace/lib/model/listing-summary.js
"use strict";
/**
 * Marketplace Service API
 * Manage applications in Oracle Cloud Infrastructure Marketplace.
 * OpenAPI spec version: 20181001
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListingSummary = void 0;
const model = __importStar(require("../model"));
var ListingSummary;
(function (ListingSummary) {
    let PricingTypes;
    (function (PricingTypes) {
        PricingTypes["Free"] = "FREE";
        PricingTypes["Byol"] = "BYOL";
        PricingTypes["Paygo"] = "PAYGO";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        PricingTypes["UnknownValue"] = "UNKNOWN_VALUE";
    })(PricingTypes = ListingSummary.PricingTypes || (ListingSummary.PricingTypes = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "icon": obj.icon ? model.UploadData.getJsonObj(obj.icon) : undefined,
            "regions": obj.regions
                ? obj.regions.map(item => {
                    return model.Region.getJsonObj(item);
                })
                : undefined,
            "publisher": obj.publisher ? model.PublisherSummary.getJsonObj(obj.publisher) : undefined,
            "supportedOperatingSystems": obj.supportedOperatingSystems
                ? obj.supportedOperatingSystems.map(item => {
                    return model.OperatingSystem.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    ListingSummary.getJsonObj = getJsonObj;
})(ListingSummary = exports.ListingSummary || (exports.ListingSummary = {}));
