// oci-marketplace/lib/model/pricing-strategy-enum.js
"use strict";
/**
 * Marketplace Service API
 * Manage applications in Oracle Cloud Infrastructure Marketplace.
 * OpenAPI spec version: 20181001
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PricingStrategyEnum = void 0;
/**
 * The possible values for a listing package's pricing strategy.
 **/
var PricingStrategyEnum;
(function (PricingStrategyEnum) {
    PricingStrategyEnum["PerOcpuLinear"] = "PER_OCPU_LINEAR";
    PricingStrategyEnum["PerOcpuMinBilling"] = "PER_OCPU_MIN_BILLING";
    PricingStrategyEnum["PerInstance"] = "PER_INSTANCE";
    PricingStrategyEnum["PerInstanceMonthlyInclusive"] = "PER_INSTANCE_MONTHLY_INCLUSIVE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    PricingStrategyEnum["UnknownValue"] = "UNKNOWN_VALUE";
})(PricingStrategyEnum = exports.PricingStrategyEnum || (exports.PricingStrategyEnum = {}));
(function (PricingStrategyEnum) {
    function getJsonObj(obj) {
        return { obj };
    }
    PricingStrategyEnum.getJsonObj = getJsonObj;
})(PricingStrategyEnum = exports.PricingStrategyEnum || (exports.PricingStrategyEnum = {}));
