// oci-marketplace/lib/model/pricing-currency-enum.js
"use strict";
/**
 * Marketplace Service API
 * Manage applications in Oracle Cloud Infrastructure Marketplace.
 * OpenAPI spec version: 20181001
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PricingCurrencyEnum = void 0;
/**
 * Possible values for the pricing currency code.
 **/
var PricingCurrencyEnum;
(function (PricingCurrencyEnum) {
    PricingCurrencyEnum["Usd"] = "USD";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    PricingCurrencyEnum["UnknownValue"] = "UNKNOWN_VALUE";
})(PricingCurrencyEnum = exports.PricingCurrencyEnum || (exports.PricingCurrencyEnum = {}));
(function (PricingCurrencyEnum) {
    function getJsonObj(obj) {
        return { obj };
    }
    PricingCurrencyEnum.getJsonObj = getJsonObj;
})(PricingCurrencyEnum = exports.PricingCurrencyEnum || (exports.PricingCurrencyEnum = {}));
