// oci-marketplace/lib/model/pricing-type-enum.js
"use strict";
/**
 * Marketplace Service API
 * Manage applications in Oracle Cloud Infrastructure Marketplace.
 * OpenAPI spec version: 20181001
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PricingTypeEnum = void 0;
/**
 * Possible values for the listing package's pricing model.
 **/
var PricingTypeEnum;
(function (PricingTypeEnum) {
    PricingTypeEnum["Free"] = "FREE";
    PricingTypeEnum["Byol"] = "BYOL";
    PricingTypeEnum["Paygo"] = "PAYGO";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    PricingTypeEnum["UnknownValue"] = "UNKNOWN_VALUE";
})(PricingTypeEnum = exports.PricingTypeEnum || (exports.PricingTypeEnum = {}));
(function (PricingTypeEnum) {
    function getJsonObj(obj) {
        return { obj };
    }
    PricingTypeEnum.getJsonObj = getJsonObj;
})(PricingTypeEnum = exports.PricingTypeEnum || (exports.PricingTypeEnum = {}));
