// oci-announcementsservice/lib/request/list-announcements-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListAnnouncementsRequest = void 0;
var ListAnnouncementsRequest;
(function (ListAnnouncementsRequest) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Inactive"] = "INACTIVE";
    })(LifecycleState = ListAnnouncementsRequest.LifecycleState || (ListAnnouncementsRequest.LifecycleState = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeOneValue"] = "timeOneValue";
        SortBy["TimeTwoValue"] = "timeTwoValue";
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["ReferenceTicketNumber"] = "referenceTicketNumber";
        SortBy["Summary"] = "summary";
        SortBy["AnnouncementType"] = "announcementType";
    })(SortBy = ListAnnouncementsRequest.SortBy || (ListAnnouncementsRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListAnnouncementsRequest.SortOrder || (ListAnnouncementsRequest.SortOrder = {}));
})(ListAnnouncementsRequest = exports.ListAnnouncementsRequest || (exports.ListAnnouncementsRequest = {}));
