// oci-artifacts/lib/model/container-repository-readme.js
"use strict";
/**
 * Container Images API
 * API covering the [Registry](/iaas/Content/Registry/Concepts/registryoverview.htm) services.
Use this API to manage resources such as container images and repositories.

 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContainerRepositoryReadme = void 0;
var ContainerRepositoryReadme;
(function (ContainerRepositoryReadme) {
    let Format;
    (function (Format) {
        Format["TextMarkdown"] = "TEXT_MARKDOWN";
        Format["TextPlain"] = "TEXT_PLAIN";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Format["UnknownValue"] = "UNKNOWN_VALUE";
    })(Format = ContainerRepositoryReadme.Format || (ContainerRepositoryReadme.Format = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ContainerRepositoryReadme.getJsonObj = getJsonObj;
})(ContainerRepositoryReadme = exports.ContainerRepositoryReadme || (exports.ContainerRepositoryReadme = {}));
