// oci-artifacts/lib/model/index.js
"use strict";
/**
 * Container Images API
 * API covering the [Registry](/iaas/Content/Registry/Concepts/registryoverview.htm) services.
Use this API to manage resources such as container images and repositories.

 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateContainerRepositoryDetails = exports.UpdateContainerConfigurationDetails = exports.RestoreContainerImageDetails = exports.RemoveContainerVersionDetails = exports.CreateContainerRepositoryDetails = exports.CreateContainerImageSignatureDetails = exports.ContainerVersion = exports.ContainerRepositorySummary = exports.ContainerRepositoryReadme = exports.ContainerRepositoryCollection = exports.ContainerRepository = exports.ContainerImageSummary = exports.ContainerImageSignatureSummary = exports.ContainerImageSignatureCollection = exports.ContainerImageSignature = exports.ContainerImageLayer = exports.ContainerImageCollection = exports.ContainerImage = exports.ContainerConfiguration = exports.ChangeContainerRepositoryCompartmentDetails = void 0;
const ChangeContainerRepositoryCompartmentDetails = __importStar(require("./change-container-repository-compartment-details"));
exports.ChangeContainerRepositoryCompartmentDetails = ChangeContainerRepositoryCompartmentDetails.ChangeContainerRepositoryCompartmentDetails;
const ContainerConfiguration = __importStar(require("./container-configuration"));
exports.ContainerConfiguration = ContainerConfiguration.ContainerConfiguration;
const ContainerImage = __importStar(require("./container-image"));
exports.ContainerImage = ContainerImage.ContainerImage;
const ContainerImageCollection = __importStar(require("./container-image-collection"));
exports.ContainerImageCollection = ContainerImageCollection.ContainerImageCollection;
const ContainerImageLayer = __importStar(require("./container-image-layer"));
exports.ContainerImageLayer = ContainerImageLayer.ContainerImageLayer;
const ContainerImageSignature = __importStar(require("./container-image-signature"));
exports.ContainerImageSignature = ContainerImageSignature.ContainerImageSignature;
const ContainerImageSignatureCollection = __importStar(require("./container-image-signature-collection"));
exports.ContainerImageSignatureCollection = ContainerImageSignatureCollection.ContainerImageSignatureCollection;
const ContainerImageSignatureSummary = __importStar(require("./container-image-signature-summary"));
exports.ContainerImageSignatureSummary = ContainerImageSignatureSummary.ContainerImageSignatureSummary;
const ContainerImageSummary = __importStar(require("./container-image-summary"));
exports.ContainerImageSummary = ContainerImageSummary.ContainerImageSummary;
const ContainerRepository = __importStar(require("./container-repository"));
exports.ContainerRepository = ContainerRepository.ContainerRepository;
const ContainerRepositoryCollection = __importStar(require("./container-repository-collection"));
exports.ContainerRepositoryCollection = ContainerRepositoryCollection.ContainerRepositoryCollection;
const ContainerRepositoryReadme = __importStar(require("./container-repository-readme"));
exports.ContainerRepositoryReadme = ContainerRepositoryReadme.ContainerRepositoryReadme;
const ContainerRepositorySummary = __importStar(require("./container-repository-summary"));
exports.ContainerRepositorySummary = ContainerRepositorySummary.ContainerRepositorySummary;
const ContainerVersion = __importStar(require("./container-version"));
exports.ContainerVersion = ContainerVersion.ContainerVersion;
const CreateContainerImageSignatureDetails = __importStar(require("./create-container-image-signature-details"));
exports.CreateContainerImageSignatureDetails = CreateContainerImageSignatureDetails.CreateContainerImageSignatureDetails;
const CreateContainerRepositoryDetails = __importStar(require("./create-container-repository-details"));
exports.CreateContainerRepositoryDetails = CreateContainerRepositoryDetails.CreateContainerRepositoryDetails;
const RemoveContainerVersionDetails = __importStar(require("./remove-container-version-details"));
exports.RemoveContainerVersionDetails = RemoveContainerVersionDetails.RemoveContainerVersionDetails;
const RestoreContainerImageDetails = __importStar(require("./restore-container-image-details"));
exports.RestoreContainerImageDetails = RestoreContainerImageDetails.RestoreContainerImageDetails;
const UpdateContainerConfigurationDetails = __importStar(require("./update-container-configuration-details"));
exports.UpdateContainerConfigurationDetails = UpdateContainerConfigurationDetails.UpdateContainerConfigurationDetails;
const UpdateContainerRepositoryDetails = __importStar(require("./update-container-repository-details"));
exports.UpdateContainerRepositoryDetails = UpdateContainerRepositoryDetails.UpdateContainerRepositoryDetails;
