// oci-streaming/lib/request/list-connect-harnesses-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20180418
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListConnectHarnessesRequest = void 0;
var ListConnectHarnessesRequest;
(function (ListConnectHarnessesRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["Name"] = "NAME";
        SortBy["Timecreated"] = "TIMECREATED";
    })(SortBy = ListConnectHarnessesRequest.SortBy || (ListConnectHarnessesRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListConnectHarnessesRequest.SortOrder || (ListConnectHarnessesRequest.SortOrder = {}));
})(ListConnectHarnessesRequest = exports.ListConnectHarnessesRequest || (exports.ListConnectHarnessesRequest = {}));
