// oci-resourcemanager/lib/model/destroy-job-operation-details.js
"use strict";
/**
 * Resource Manager API
 * API for the Resource Manager service.
Use this API to install, configure, and manage resources via the "infrastructure-as-code" model.
For more information, see
[Overview of Resource Manager](/iaas/Content/ResourceManager/Concepts/resourcemanager.htm).

 * OpenAPI spec version: 20180917
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DestroyJobOperationDetails = void 0;
const model = __importStar(require("../model"));
var DestroyJobOperationDetails;
(function (DestroyJobOperationDetails) {
    let ExecutionPlanStrategy;
    (function (ExecutionPlanStrategy) {
        ExecutionPlanStrategy["AutoApproved"] = "AUTO_APPROVED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ExecutionPlanStrategy["UnknownValue"] = "UNKNOWN_VALUE";
    })(ExecutionPlanStrategy = DestroyJobOperationDetails.ExecutionPlanStrategy || (DestroyJobOperationDetails.ExecutionPlanStrategy = {}));
    function getJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.JobOperationDetails.getJsonObj(obj))), {});
        return jsonObj;
    }
    DestroyJobOperationDetails.getJsonObj = getJsonObj;
    DestroyJobOperationDetails.operation = "DESTROY";
})(DestroyJobOperationDetails = exports.DestroyJobOperationDetails || (exports.DestroyJobOperationDetails = {}));
