// oci-resourcemanager/lib/model/failure-details.js
"use strict";
/**
 * Resource Manager API
 * API for the Resource Manager service.
Use this API to install, configure, and manage resources via the "infrastructure-as-code" model.
For more information, see
[Overview of Resource Manager](/iaas/Content/ResourceManager/Concepts/resourcemanager.htm).

 * OpenAPI spec version: 20180917
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FailureDetails = void 0;
var FailureDetails;
(function (FailureDetails) {
    let Code;
    (function (Code) {
        Code["InternalServiceError"] = "INTERNAL_SERVICE_ERROR";
        Code["TerraformExecutionError"] = "TERRAFORM_EXECUTION_ERROR";
        Code["TerraformConfigUnzipFailed"] = "TERRAFORM_CONFIG_UNZIP_FAILED";
        Code["InvalidWorkingDirectory"] = "INVALID_WORKING_DIRECTORY";
        Code["JobTimeout"] = "JOB_TIMEOUT";
        Code["TerraformConfigVirusFound"] = "TERRAFORM_CONFIG_VIRUS_FOUND";
        Code["TerraformGitCloneFailure"] = "TERRAFORM_GIT_CLONE_FAILURE";
        Code["TerraformGitCheckoutFailure"] = "TERRAFORM_GIT_CHECKOUT_FAILURE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Code["UnknownValue"] = "UNKNOWN_VALUE";
    })(Code = FailureDetails.Code || (FailureDetails.Code = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    FailureDetails.getJsonObj = getJsonObj;
})(FailureDetails = exports.FailureDetails || (exports.FailureDetails = {}));
