// oci-database/lib/model/maintenance-run-summary.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MaintenanceRunSummary = void 0;
var MaintenanceRunSummary;
(function (MaintenanceRunSummary) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Scheduled"] = "SCHEDULED";
        LifecycleState["InProgress"] = "IN_PROGRESS";
        LifecycleState["Succeeded"] = "SUCCEEDED";
        LifecycleState["Skipped"] = "SKIPPED";
        LifecycleState["Failed"] = "FAILED";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Canceled"] = "CANCELED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = MaintenanceRunSummary.LifecycleState || (MaintenanceRunSummary.LifecycleState = {}));
    let TargetResourceType;
    (function (TargetResourceType) {
        TargetResourceType["AutonomousExadataInfrastructure"] = "AUTONOMOUS_EXADATA_INFRASTRUCTURE";
        TargetResourceType["AutonomousContainerDatabase"] = "AUTONOMOUS_CONTAINER_DATABASE";
        TargetResourceType["ExadataDbSystem"] = "EXADATA_DB_SYSTEM";
        TargetResourceType["CloudExadataInfrastructure"] = "CLOUD_EXADATA_INFRASTRUCTURE";
        TargetResourceType["ExaccInfrastructure"] = "EXACC_INFRASTRUCTURE";
        TargetResourceType["AutonomousDatabase"] = "AUTONOMOUS_DATABASE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        TargetResourceType["UnknownValue"] = "UNKNOWN_VALUE";
    })(TargetResourceType = MaintenanceRunSummary.TargetResourceType || (MaintenanceRunSummary.TargetResourceType = {}));
    let MaintenanceType;
    (function (MaintenanceType) {
        MaintenanceType["Planned"] = "PLANNED";
        MaintenanceType["Unplanned"] = "UNPLANNED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        MaintenanceType["UnknownValue"] = "UNKNOWN_VALUE";
    })(MaintenanceType = MaintenanceRunSummary.MaintenanceType || (MaintenanceRunSummary.MaintenanceType = {}));
    let MaintenanceSubtype;
    (function (MaintenanceSubtype) {
        MaintenanceSubtype["Quarterly"] = "QUARTERLY";
        MaintenanceSubtype["Hardware"] = "HARDWARE";
        MaintenanceSubtype["Critical"] = "CRITICAL";
        MaintenanceSubtype["Infrastructure"] = "INFRASTRUCTURE";
        MaintenanceSubtype["Database"] = "DATABASE";
        MaintenanceSubtype["Oneoff"] = "ONEOFF";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        MaintenanceSubtype["UnknownValue"] = "UNKNOWN_VALUE";
    })(MaintenanceSubtype = MaintenanceRunSummary.MaintenanceSubtype || (MaintenanceRunSummary.MaintenanceSubtype = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    MaintenanceRunSummary.getJsonObj = getJsonObj;
})(MaintenanceRunSummary = exports.MaintenanceRunSummary || (exports.MaintenanceRunSummary = {}));
