// oci-dataintegration/lib/request/list-patch-changes-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200430
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListPatchChangesRequest = void 0;
var ListPatchChangesRequest;
(function (ListPatchChangesRequest) {
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListPatchChangesRequest.SortOrder || (ListPatchChangesRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "TIME_CREATED";
        SortBy["DisplayName"] = "DISPLAY_NAME";
    })(SortBy = ListPatchChangesRequest.SortBy || (ListPatchChangesRequest.SortBy = {}));
})(ListPatchChangesRequest = exports.ListPatchChangesRequest || (exports.ListPatchChangesRequest = {}));
