// oci-events/lib/model/index.js
"use strict";
/**
 * Events API
 * API for the Events Service. Use this API to manage rules and actions that create automation
in your tenancy. For more information, see [Overview of Events](/iaas/Content/Events/Concepts/eventsoverview.htm).

 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StreamingServiceAction = exports.NotificationServiceAction = exports.FaaSAction = exports.CreateStreamingServiceActionDetails = exports.CreateNotificationServiceActionDetails = exports.CreateFaaSActionDetails = exports.UpdateRuleDetails = exports.RuleSummary = exports.Rule = exports.CreateRuleDetails = exports.ChangeRuleCompartmentDetails = exports.ActionList = exports.ActionDetailsList = exports.ActionDetails = exports.Action = void 0;
const Action = __importStar(require("./action"));
exports.Action = Action.Action;
const ActionDetails = __importStar(require("./action-details"));
exports.ActionDetails = ActionDetails.ActionDetails;
const ActionDetailsList = __importStar(require("./action-details-list"));
exports.ActionDetailsList = ActionDetailsList.ActionDetailsList;
const ActionList = __importStar(require("./action-list"));
exports.ActionList = ActionList.ActionList;
const ChangeRuleCompartmentDetails = __importStar(require("./change-rule-compartment-details"));
exports.ChangeRuleCompartmentDetails = ChangeRuleCompartmentDetails.ChangeRuleCompartmentDetails;
const CreateRuleDetails = __importStar(require("./create-rule-details"));
exports.CreateRuleDetails = CreateRuleDetails.CreateRuleDetails;
const Rule = __importStar(require("./rule"));
exports.Rule = Rule.Rule;
const RuleSummary = __importStar(require("./rule-summary"));
exports.RuleSummary = RuleSummary.RuleSummary;
const UpdateRuleDetails = __importStar(require("./update-rule-details"));
exports.UpdateRuleDetails = UpdateRuleDetails.UpdateRuleDetails;
const CreateFaaSActionDetails = __importStar(require("./create-faa-saction-details"));
exports.CreateFaaSActionDetails = CreateFaaSActionDetails.CreateFaaSActionDetails;
const CreateNotificationServiceActionDetails = __importStar(require("./create-notification-service-action-details"));
exports.CreateNotificationServiceActionDetails = CreateNotificationServiceActionDetails.CreateNotificationServiceActionDetails;
const CreateStreamingServiceActionDetails = __importStar(require("./create-streaming-service-action-details"));
exports.CreateStreamingServiceActionDetails = CreateStreamingServiceActionDetails.CreateStreamingServiceActionDetails;
const FaaSAction = __importStar(require("./faa-saction"));
exports.FaaSAction = FaaSAction.FaaSAction;
const NotificationServiceAction = __importStar(require("./notification-service-action"));
exports.NotificationServiceAction = NotificationServiceAction.NotificationServiceAction;
const StreamingServiceAction = __importStar(require("./streaming-service-action"));
exports.StreamingServiceAction = StreamingServiceAction.StreamingServiceAction;
