// oci-dts/lib/model/index.js
"use strict";
/**
 * Data Transfer Service API
 * Data Transfer Service API Specification
 * OpenAPI spec version: 1.0.015
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateTransferPackageDetails = exports.UpdateTransferJobDetails = exports.UpdateTransferDeviceDetails = exports.UpdateTransferApplianceDetails = exports.UpdateApplianceExportJobDetails = exports.TransferPackageSummary = exports.TransferPackage = exports.TransferJobSummary = exports.TransferJob = exports.TransferDeviceSummary = exports.TransferDevice = exports.TransferApplianceSummary = exports.TransferAppliancePublicKey = exports.TransferApplianceEntitlementSummary = exports.TransferApplianceEntitlement = exports.TransferApplianceEncryptionPassphrase = exports.TransferApplianceCertificate = exports.TransferAppliance = exports.ShippingVendors = exports.ShippingAddress = exports.NewTransferDevice = exports.MultipleTransferPackages = exports.MultipleTransferDevices = exports.MultipleTransferAppliances = exports.DetachDevicesDetails = exports.CreateTransferPackageDetails = exports.CreateTransferJobDetails = exports.CreateTransferDeviceDetails = exports.CreateTransferApplianceEntitlementDetails = exports.CreateTransferApplianceDetails = exports.CreateApplianceExportJobDetails = exports.ChangeTransferJobCompartmentDetails = exports.ChangeApplianceExportJobCompartmentDetails = exports.AttachDevicesDetails = exports.ApplianceExportJobSummary = exports.ApplianceExportJob = void 0;
const ApplianceExportJob = __importStar(require("./appliance-export-job"));
exports.ApplianceExportJob = ApplianceExportJob.ApplianceExportJob;
const ApplianceExportJobSummary = __importStar(require("./appliance-export-job-summary"));
exports.ApplianceExportJobSummary = ApplianceExportJobSummary.ApplianceExportJobSummary;
const AttachDevicesDetails = __importStar(require("./attach-devices-details"));
exports.AttachDevicesDetails = AttachDevicesDetails.AttachDevicesDetails;
const ChangeApplianceExportJobCompartmentDetails = __importStar(require("./change-appliance-export-job-compartment-details"));
exports.ChangeApplianceExportJobCompartmentDetails = ChangeApplianceExportJobCompartmentDetails.ChangeApplianceExportJobCompartmentDetails;
const ChangeTransferJobCompartmentDetails = __importStar(require("./change-transfer-job-compartment-details"));
exports.ChangeTransferJobCompartmentDetails = ChangeTransferJobCompartmentDetails.ChangeTransferJobCompartmentDetails;
const CreateApplianceExportJobDetails = __importStar(require("./create-appliance-export-job-details"));
exports.CreateApplianceExportJobDetails = CreateApplianceExportJobDetails.CreateApplianceExportJobDetails;
const CreateTransferApplianceDetails = __importStar(require("./create-transfer-appliance-details"));
exports.CreateTransferApplianceDetails = CreateTransferApplianceDetails.CreateTransferApplianceDetails;
const CreateTransferApplianceEntitlementDetails = __importStar(require("./create-transfer-appliance-entitlement-details"));
exports.CreateTransferApplianceEntitlementDetails = CreateTransferApplianceEntitlementDetails.CreateTransferApplianceEntitlementDetails;
const CreateTransferDeviceDetails = __importStar(require("./create-transfer-device-details"));
exports.CreateTransferDeviceDetails = CreateTransferDeviceDetails.CreateTransferDeviceDetails;
const CreateTransferJobDetails = __importStar(require("./create-transfer-job-details"));
exports.CreateTransferJobDetails = CreateTransferJobDetails.CreateTransferJobDetails;
const CreateTransferPackageDetails = __importStar(require("./create-transfer-package-details"));
exports.CreateTransferPackageDetails = CreateTransferPackageDetails.CreateTransferPackageDetails;
const DetachDevicesDetails = __importStar(require("./detach-devices-details"));
exports.DetachDevicesDetails = DetachDevicesDetails.DetachDevicesDetails;
const MultipleTransferAppliances = __importStar(require("./multiple-transfer-appliances"));
exports.MultipleTransferAppliances = MultipleTransferAppliances.MultipleTransferAppliances;
const MultipleTransferDevices = __importStar(require("./multiple-transfer-devices"));
exports.MultipleTransferDevices = MultipleTransferDevices.MultipleTransferDevices;
const MultipleTransferPackages = __importStar(require("./multiple-transfer-packages"));
exports.MultipleTransferPackages = MultipleTransferPackages.MultipleTransferPackages;
const NewTransferDevice = __importStar(require("./new-transfer-device"));
exports.NewTransferDevice = NewTransferDevice.NewTransferDevice;
const ShippingAddress = __importStar(require("./shipping-address"));
exports.ShippingAddress = ShippingAddress.ShippingAddress;
const ShippingVendors = __importStar(require("./shipping-vendors"));
exports.ShippingVendors = ShippingVendors.ShippingVendors;
const TransferAppliance = __importStar(require("./transfer-appliance"));
exports.TransferAppliance = TransferAppliance.TransferAppliance;
const TransferApplianceCertificate = __importStar(require("./transfer-appliance-certificate"));
exports.TransferApplianceCertificate = TransferApplianceCertificate.TransferApplianceCertificate;
const TransferApplianceEncryptionPassphrase = __importStar(require("./transfer-appliance-encryption-passphrase"));
exports.TransferApplianceEncryptionPassphrase = TransferApplianceEncryptionPassphrase.TransferApplianceEncryptionPassphrase;
const TransferApplianceEntitlement = __importStar(require("./transfer-appliance-entitlement"));
exports.TransferApplianceEntitlement = TransferApplianceEntitlement.TransferApplianceEntitlement;
const TransferApplianceEntitlementSummary = __importStar(require("./transfer-appliance-entitlement-summary"));
exports.TransferApplianceEntitlementSummary = TransferApplianceEntitlementSummary.TransferApplianceEntitlementSummary;
const TransferAppliancePublicKey = __importStar(require("./transfer-appliance-public-key"));
exports.TransferAppliancePublicKey = TransferAppliancePublicKey.TransferAppliancePublicKey;
const TransferApplianceSummary = __importStar(require("./transfer-appliance-summary"));
exports.TransferApplianceSummary = TransferApplianceSummary.TransferApplianceSummary;
const TransferDevice = __importStar(require("./transfer-device"));
exports.TransferDevice = TransferDevice.TransferDevice;
const TransferDeviceSummary = __importStar(require("./transfer-device-summary"));
exports.TransferDeviceSummary = TransferDeviceSummary.TransferDeviceSummary;
const TransferJob = __importStar(require("./transfer-job"));
exports.TransferJob = TransferJob.TransferJob;
const TransferJobSummary = __importStar(require("./transfer-job-summary"));
exports.TransferJobSummary = TransferJobSummary.TransferJobSummary;
const TransferPackage = __importStar(require("./transfer-package"));
exports.TransferPackage = TransferPackage.TransferPackage;
const TransferPackageSummary = __importStar(require("./transfer-package-summary"));
exports.TransferPackageSummary = TransferPackageSummary.TransferPackageSummary;
const UpdateApplianceExportJobDetails = __importStar(require("./update-appliance-export-job-details"));
exports.UpdateApplianceExportJobDetails = UpdateApplianceExportJobDetails.UpdateApplianceExportJobDetails;
const UpdateTransferApplianceDetails = __importStar(require("./update-transfer-appliance-details"));
exports.UpdateTransferApplianceDetails = UpdateTransferApplianceDetails.UpdateTransferApplianceDetails;
const UpdateTransferDeviceDetails = __importStar(require("./update-transfer-device-details"));
exports.UpdateTransferDeviceDetails = UpdateTransferDeviceDetails.UpdateTransferDeviceDetails;
const UpdateTransferJobDetails = __importStar(require("./update-transfer-job-details"));
exports.UpdateTransferJobDetails = UpdateTransferJobDetails.UpdateTransferJobDetails;
const UpdateTransferPackageDetails = __importStar(require("./update-transfer-package-details"));
exports.UpdateTransferPackageDetails = UpdateTransferPackageDetails.UpdateTransferPackageDetails;
