// oci-dts/lib/model/transfer-device.js
"use strict";
/**
 * Data Transfer Service API
 * Data Transfer Service API Specification
 * OpenAPI spec version: 1.0.015
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransferDevice = void 0;
var TransferDevice;
(function (TransferDevice) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Preparing"] = "PREPARING";
        LifecycleState["Ready"] = "READY";
        LifecycleState["Packaged"] = "PACKAGED";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Processing"] = "PROCESSING";
        LifecycleState["Complete"] = "COMPLETE";
        LifecycleState["Missing"] = "MISSING";
        LifecycleState["Error"] = "ERROR";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Cancelled"] = "CANCELLED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = TransferDevice.LifecycleState || (TransferDevice.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    TransferDevice.getJsonObj = getJsonObj;
})(TransferDevice = exports.TransferDevice || (exports.TransferDevice = {}));
