// oci-dts/lib/model/transfer-appliance.js
"use strict";
/**
 * Data Transfer Service API
 * Data Transfer Service API Specification
 * OpenAPI spec version: 1.0.015
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransferAppliance = void 0;
const model = __importStar(require("../model"));
var TransferAppliance;
(function (TransferAppliance) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Requested"] = "REQUESTED";
        LifecycleState["OraclePreparing"] = "ORACLE_PREPARING";
        LifecycleState["Shipping"] = "SHIPPING";
        LifecycleState["Delivered"] = "DELIVERED";
        LifecycleState["Preparing"] = "PREPARING";
        LifecycleState["Finalized"] = "FINALIZED";
        LifecycleState["ReturnDelayed"] = "RETURN_DELAYED";
        LifecycleState["ReturnShipped"] = "RETURN_SHIPPED";
        LifecycleState["ReturnShippedCancelled"] = "RETURN_SHIPPED_CANCELLED";
        LifecycleState["OracleReceived"] = "ORACLE_RECEIVED";
        LifecycleState["OracleReceivedCancelled"] = "ORACLE_RECEIVED_CANCELLED";
        LifecycleState["Processing"] = "PROCESSING";
        LifecycleState["Complete"] = "COMPLETE";
        LifecycleState["CustomerNeverReceived"] = "CUSTOMER_NEVER_RECEIVED";
        LifecycleState["OracleNeverReceived"] = "ORACLE_NEVER_RECEIVED";
        LifecycleState["CustomerLost"] = "CUSTOMER_LOST";
        LifecycleState["Cancelled"] = "CANCELLED";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Rejected"] = "REJECTED";
        LifecycleState["Error"] = "ERROR";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = TransferAppliance.LifecycleState || (TransferAppliance.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "customerShippingAddress": obj.customerShippingAddress
                ? model.ShippingAddress.getJsonObj(obj.customerShippingAddress)
                : undefined
        });
        return jsonObj;
    }
    TransferAppliance.getJsonObj = getJsonObj;
})(TransferAppliance = exports.TransferAppliance || (exports.TransferAppliance = {}));
