// oci-dts/lib/model/update-transfer-job-details.js
"use strict";
/**
 * Data Transfer Service API
 * Data Transfer Service API Specification
 * OpenAPI spec version: 1.0.015
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateTransferJobDetails = void 0;
var UpdateTransferJobDetails;
(function (UpdateTransferJobDetails) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Closed"] = "CLOSED";
    })(LifecycleState = UpdateTransferJobDetails.LifecycleState || (UpdateTransferJobDetails.LifecycleState = {}));
    let DeviceType;
    (function (DeviceType) {
        DeviceType["Disk"] = "DISK";
        DeviceType["Appliance"] = "APPLIANCE";
    })(DeviceType = UpdateTransferJobDetails.DeviceType || (UpdateTransferJobDetails.DeviceType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    UpdateTransferJobDetails.getJsonObj = getJsonObj;
})(UpdateTransferJobDetails = exports.UpdateTransferJobDetails || (exports.UpdateTransferJobDetails = {}));
