// oci-apmcontrolplane/lib/model/index.js
"use strict";
/**
 * Application Performance Monitoring (APM) Control Plane API
 * Provide a set of APIs for tenant to perform operations like create, update, delete and list APM domains, and also
work request APIs to monitor progress of these operations.

 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestResource = exports.WorkRequestLogEntry = exports.WorkRequestError = exports.WorkRequest = exports.UpdateApmDomainDetails = exports.SortOrders = exports.RemoveDataKeyDetails = exports.OperationTypes = exports.OperationStatus = exports.LifecycleStates = exports.GenerateDataKeyDetails = exports.DataKeyTypes = exports.DataKeySummary = exports.DataKey = exports.CreateApmDomainDetails = exports.ChangeApmDomainCompartmentDetails = exports.BaseKeyDetails = exports.BaseDomainDetails = exports.ApmDomainSummary = exports.ApmDomain = exports.ActionTypes = void 0;
const ActionTypes = __importStar(require("./action-types"));
exports.ActionTypes = ActionTypes.ActionTypes;
const ApmDomain = __importStar(require("./apm-domain"));
exports.ApmDomain = ApmDomain.ApmDomain;
const ApmDomainSummary = __importStar(require("./apm-domain-summary"));
exports.ApmDomainSummary = ApmDomainSummary.ApmDomainSummary;
const BaseDomainDetails = __importStar(require("./base-domain-details"));
exports.BaseDomainDetails = BaseDomainDetails.BaseDomainDetails;
const BaseKeyDetails = __importStar(require("./base-key-details"));
exports.BaseKeyDetails = BaseKeyDetails.BaseKeyDetails;
const ChangeApmDomainCompartmentDetails = __importStar(require("./change-apm-domain-compartment-details"));
exports.ChangeApmDomainCompartmentDetails = ChangeApmDomainCompartmentDetails.ChangeApmDomainCompartmentDetails;
const CreateApmDomainDetails = __importStar(require("./create-apm-domain-details"));
exports.CreateApmDomainDetails = CreateApmDomainDetails.CreateApmDomainDetails;
const DataKey = __importStar(require("./data-key"));
exports.DataKey = DataKey.DataKey;
const DataKeySummary = __importStar(require("./data-key-summary"));
exports.DataKeySummary = DataKeySummary.DataKeySummary;
const DataKeyTypes = __importStar(require("./data-key-types"));
exports.DataKeyTypes = DataKeyTypes.DataKeyTypes;
const GenerateDataKeyDetails = __importStar(require("./generate-data-key-details"));
exports.GenerateDataKeyDetails = GenerateDataKeyDetails.GenerateDataKeyDetails;
const LifecycleStates = __importStar(require("./lifecycle-states"));
exports.LifecycleStates = LifecycleStates.LifecycleStates;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationTypes = __importStar(require("./operation-types"));
exports.OperationTypes = OperationTypes.OperationTypes;
const RemoveDataKeyDetails = __importStar(require("./remove-data-key-details"));
exports.RemoveDataKeyDetails = RemoveDataKeyDetails.RemoveDataKeyDetails;
const SortOrders = __importStar(require("./sort-orders"));
exports.SortOrders = SortOrders.SortOrders;
const UpdateApmDomainDetails = __importStar(require("./update-apm-domain-details"));
exports.UpdateApmDomainDetails = UpdateApmDomainDetails.UpdateApmDomainDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
