// oci-apmcontrolplane/lib/model/operation-status.js
"use strict";
/**
 * Application Performance Monitoring (APM) Control Plane API
 * Provide a set of APIs for tenant to perform operations like create, update, delete and list APM domains, and also
work request APIs to monitor progress of these operations.

 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationStatus = void 0;
/**
 * Possible statuses for a work request.
 **/
var OperationStatus;
(function (OperationStatus) {
    OperationStatus["Accepted"] = "ACCEPTED";
    OperationStatus["InProgress"] = "IN_PROGRESS";
    OperationStatus["Failed"] = "FAILED";
    OperationStatus["Succeeded"] = "SUCCEEDED";
    OperationStatus["Canceling"] = "CANCELING";
    OperationStatus["Canceled"] = "CANCELED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationStatus = exports.OperationStatus || (exports.OperationStatus = {}));
(function (OperationStatus) {
    function getJsonObj(obj) {
        return { obj };
    }
    OperationStatus.getJsonObj = getJsonObj;
})(OperationStatus = exports.OperationStatus || (exports.OperationStatus = {}));
