// oci-cloudguard/lib/model/actor-type.js
"use strict";
/**
 * Cloud Guard APIs
 * A description of the Cloud Guard APIs
 * OpenAPI spec version: 20200131
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ActorType = void 0;
/**
 * Actor type who performed the operation
 **/
var ActorType;
(function (ActorType) {
    ActorType["CloudGuardService"] = "CLOUD_GUARD_SERVICE";
    ActorType["Correlation"] = "CORRELATION";
    ActorType["Responder"] = "RESPONDER";
    ActorType["User"] = "USER";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ActorType["UnknownValue"] = "UNKNOWN_VALUE";
})(ActorType = exports.ActorType || (exports.ActorType = {}));
(function (ActorType) {
    function getJsonObj(obj) {
        return { obj };
    }
    ActorType.getJsonObj = getJsonObj;
})(ActorType = exports.ActorType || (exports.ActorType = {}));
