// oci-cloudguard/lib/model/condition-filter-type.js
"use strict";
/**
 * Cloud Guard APIs
 * A description of the Cloud Guard APIs
 * OpenAPI spec version: 20200131
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConditionFilterType = void 0;
/**
 * Filter type can be config filter or condition filter
 **/
var ConditionFilterType;
(function (ConditionFilterType) {
    ConditionFilterType["Condition"] = "CONDITION";
    ConditionFilterType["Config"] = "CONFIG";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ConditionFilterType["UnknownValue"] = "UNKNOWN_VALUE";
})(ConditionFilterType = exports.ConditionFilterType || (exports.ConditionFilterType = {}));
(function (ConditionFilterType) {
    function getJsonObj(obj) {
        return { obj };
    }
    ConditionFilterType.getJsonObj = getJsonObj;
})(ConditionFilterType = exports.ConditionFilterType || (exports.ConditionFilterType = {}));
