// oci-cloudguard/lib/model/condition-operator-name.js
"use strict";
/**
 * Cloud Guard APIs
 * A description of the Cloud Guard APIs
 * OpenAPI spec version: 20200131
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConditionOperatorName = void 0;
/**
 * Operation names supported for Condition Operator
 **/
var ConditionOperatorName;
(function (ConditionOperatorName) {
    ConditionOperatorName["And"] = "AND";
    ConditionOperatorName["Or"] = "OR";
    ConditionOperatorName["In"] = "IN";
    ConditionOperatorName["NotIn"] = "NOT_IN";
    ConditionOperatorName["Equals"] = "EQUALS";
    ConditionOperatorName["NotEquals"] = "NOT_EQUALS";
    ConditionOperatorName["LessThan"] = "LESS_THAN";
    ConditionOperatorName["GreaterThan"] = "GREATER_THAN";
    ConditionOperatorName["Range"] = "RANGE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ConditionOperatorName["UnknownValue"] = "UNKNOWN_VALUE";
})(ConditionOperatorName = exports.ConditionOperatorName || (exports.ConditionOperatorName = {}));
(function (ConditionOperatorName) {
    function getJsonObj(obj) {
        return { obj };
    }
    ConditionOperatorName.getJsonObj = getJsonObj;
})(ConditionOperatorName = exports.ConditionOperatorName || (exports.ConditionOperatorName = {}));
