// oci-cloudguard/lib/model/event-status.js
"use strict";
/**
 * Cloud Guard APIs
 * A description of the Cloud Guard APIs
 * OpenAPI spec version: 20200131
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventStatus = void 0;
/**
 * Possible event states.
 **/
var EventStatus;
(function (EventStatus) {
    EventStatus["Reopen"] = "REOPEN";
    EventStatus["Open"] = "OPEN";
    EventStatus["Update"] = "UPDATE";
    EventStatus["Resolve"] = "RESOLVE";
    EventStatus["Dismiss"] = "DISMISS";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    EventStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(EventStatus = exports.EventStatus || (exports.EventStatus = {}));
(function (EventStatus) {
    function getJsonObj(obj) {
        return { obj };
    }
    EventStatus.getJsonObj = getJsonObj;
})(EventStatus = exports.EventStatus || (exports.EventStatus = {}));
