// oci-cloudguard/lib/model/managed-list-type.js
"use strict";
/**
 * Cloud Guard APIs
 * A description of the Cloud Guard APIs
 * OpenAPI spec version: 20200131
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagedListType = void 0;
/**
 * Possible cloud guard list types
 **/
var ManagedListType;
(function (ManagedListType) {
    ManagedListType["CidrBlock"] = "CIDR_BLOCK";
    ManagedListType["Users"] = "USERS";
    ManagedListType["Groups"] = "GROUPS";
    ManagedListType["Ipv4Address"] = "IPV4ADDRESS";
    ManagedListType["Ipv6Address"] = "IPV6ADDRESS";
    ManagedListType["ResourceOcid"] = "RESOURCE_OCID";
    ManagedListType["Region"] = "REGION";
    ManagedListType["Country"] = "COUNTRY";
    ManagedListType["State"] = "STATE";
    ManagedListType["City"] = "CITY";
    ManagedListType["Tags"] = "TAGS";
    ManagedListType["Generic"] = "GENERIC";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ManagedListType["UnknownValue"] = "UNKNOWN_VALUE";
})(ManagedListType = exports.ManagedListType || (exports.ManagedListType = {}));
(function (ManagedListType) {
    function getJsonObj(obj) {
        return { obj };
    }
    ManagedListType.getJsonObj = getJsonObj;
})(ManagedListType = exports.ManagedListType || (exports.ManagedListType = {}));
