// oci-cloudguard/lib/model/problem-dimension.js
"use strict";
/**
 * Cloud Guard APIs
 * A description of the Cloud Guard APIs
 * OpenAPI spec version: 20200131
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProblemDimension = void 0;
/**
 * Possible dimensions for distinguishing Problems
 **/
var ProblemDimension;
(function (ProblemDimension) {
    ProblemDimension["ResourceType"] = "RESOURCE_TYPE";
    ProblemDimension["Region"] = "REGION";
    ProblemDimension["CompartmentId"] = "COMPARTMENT_ID";
    ProblemDimension["RiskLevel"] = "RISK_LEVEL";
})(ProblemDimension = exports.ProblemDimension || (exports.ProblemDimension = {}));
(function (ProblemDimension) {
    function getJsonObj(obj) {
        return { obj };
    }
    ProblemDimension.getJsonObj = getJsonObj;
})(ProblemDimension = exports.ProblemDimension || (exports.ProblemDimension = {}));
