// oci-cloudguard/lib/model/problem-lifecycle-detail.js
"use strict";
/**
 * Cloud Guard APIs
 * A description of the Cloud Guard APIs
 * OpenAPI spec version: 20200131
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProblemLifecycleDetail = void 0;
/**
 * Possible lifecycle states.
 **/
var ProblemLifecycleDetail;
(function (ProblemLifecycleDetail) {
    ProblemLifecycleDetail["Open"] = "OPEN";
    ProblemLifecycleDetail["Resolved"] = "RESOLVED";
    ProblemLifecycleDetail["Dismissed"] = "DISMISSED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ProblemLifecycleDetail["UnknownValue"] = "UNKNOWN_VALUE";
})(ProblemLifecycleDetail = exports.ProblemLifecycleDetail || (exports.ProblemLifecycleDetail = {}));
(function (ProblemLifecycleDetail) {
    function getJsonObj(obj) {
        return { obj };
    }
    ProblemLifecycleDetail.getJsonObj = getJsonObj;
})(ProblemLifecycleDetail = exports.ProblemLifecycleDetail || (exports.ProblemLifecycleDetail = {}));
