// oci-cloudguard/lib/model/recommendation-type.js
"use strict";
/**
 * Cloud Guard APIs
 * A description of the Cloud Guard APIs
 * OpenAPI spec version: 20200131
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecommendationType = void 0;
/**
 * Recommendation types
 **/
var RecommendationType;
(function (RecommendationType) {
    RecommendationType["DetectorProblems"] = "DETECTOR_PROBLEMS";
    RecommendationType["ResolvedProblems"] = "RESOLVED_PROBLEMS";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    RecommendationType["UnknownValue"] = "UNKNOWN_VALUE";
})(RecommendationType = exports.RecommendationType || (exports.RecommendationType = {}));
(function (RecommendationType) {
    function getJsonObj(obj) {
        return { obj };
    }
    RecommendationType.getJsonObj = getJsonObj;
})(RecommendationType = exports.RecommendationType || (exports.RecommendationType = {}));
