// oci-cloudguard/lib/model/responder-execution-status.js
"use strict";
/**
 * Cloud Guard APIs
 * A description of the Cloud Guard APIs
 * OpenAPI spec version: 20200131
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResponderExecutionStatus = void 0;
/**
 * Possible responder execution status.
 **/
var ResponderExecutionStatus;
(function (ResponderExecutionStatus) {
    ResponderExecutionStatus["Started"] = "STARTED";
    ResponderExecutionStatus["AwaitingConfirmation"] = "AWAITING_CONFIRMATION";
    ResponderExecutionStatus["Succeeded"] = "SUCCEEDED";
    ResponderExecutionStatus["Failed"] = "FAILED";
    ResponderExecutionStatus["Skipped"] = "SKIPPED";
})(ResponderExecutionStatus = exports.ResponderExecutionStatus || (exports.ResponderExecutionStatus = {}));
(function (ResponderExecutionStatus) {
    function getJsonObj(obj) {
        return { obj };
    }
    ResponderExecutionStatus.getJsonObj = getJsonObj;
})(ResponderExecutionStatus = exports.ResponderExecutionStatus || (exports.ResponderExecutionStatus = {}));
