// oci-cloudguard/lib/model/responder-execution-modes.js
"use strict";
/**
 * Cloud Guard APIs
 * A description of the Cloud Guard APIs
 * OpenAPI spec version: 20200131
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResponderExecutionModes = void 0;
/**
 * Possible responder execution modes.
 **/
var ResponderExecutionModes;
(function (ResponderExecutionModes) {
    ResponderExecutionModes["Manual"] = "MANUAL";
    ResponderExecutionModes["Automated"] = "AUTOMATED";
    ResponderExecutionModes["All"] = "ALL";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ResponderExecutionModes["UnknownValue"] = "UNKNOWN_VALUE";
})(ResponderExecutionModes = exports.ResponderExecutionModes || (exports.ResponderExecutionModes = {}));
(function (ResponderExecutionModes) {
    function getJsonObj(obj) {
        return { obj };
    }
    ResponderExecutionModes.getJsonObj = getJsonObj;
})(ResponderExecutionModes = exports.ResponderExecutionModes || (exports.ResponderExecutionModes = {}));
