// oci-cloudguard/lib/model/responder-dimension.js
"use strict";
/**
 * Cloud Guard APIs
 * A description of the Cloud Guard APIs
 * OpenAPI spec version: 20200131
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResponderDimension = void 0;
/**
 * Possible Dimensions for distinguishing Responder Executions
 **/
var ResponderDimension;
(function (ResponderDimension) {
    ResponderDimension["ResponderRuleType"] = "RESPONDER_RULE_TYPE";
    ResponderDimension["ResponderExecutionStatus"] = "RESPONDER_EXECUTION_STATUS";
})(ResponderDimension = exports.ResponderDimension || (exports.ResponderDimension = {}));
(function (ResponderDimension) {
    function getJsonObj(obj) {
        return { obj };
    }
    ResponderDimension.getJsonObj = getJsonObj;
})(ResponderDimension = exports.ResponderDimension || (exports.ResponderDimension = {}));
