// oci-dts/lib/request/index.js
"use strict";
/**
 * Data Transfer Service API
 * Data Transfer Service API Specification
 * OpenAPI spec version: 1.0.015
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListTransferPackagesRequest = exports.ListTransferJobsRequest = exports.ListTransferDevicesRequest = exports.ListTransferAppliancesRequest = exports.ListApplianceExportJobsRequest = void 0;
const ListApplianceExportJobsRequest = __importStar(require("./list-appliance-export-jobs-request"));
exports.ListApplianceExportJobsRequest = ListApplianceExportJobsRequest.ListApplianceExportJobsRequest;
const ListTransferAppliancesRequest = __importStar(require("./list-transfer-appliances-request"));
exports.ListTransferAppliancesRequest = ListTransferAppliancesRequest.ListTransferAppliancesRequest;
const ListTransferDevicesRequest = __importStar(require("./list-transfer-devices-request"));
exports.ListTransferDevicesRequest = ListTransferDevicesRequest.ListTransferDevicesRequest;
const ListTransferJobsRequest = __importStar(require("./list-transfer-jobs-request"));
exports.ListTransferJobsRequest = ListTransferJobsRequest.ListTransferJobsRequest;
const ListTransferPackagesRequest = __importStar(require("./list-transfer-packages-request"));
exports.ListTransferPackagesRequest = ListTransferPackagesRequest.ListTransferPackagesRequest;
