// oci-dts/lib/request/list-transfer-packages-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 1.0.015
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListTransferPackagesRequest = void 0;
var ListTransferPackagesRequest;
(function (ListTransferPackagesRequest) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Preparing"] = "PREPARING";
        LifecycleState["Shipping"] = "SHIPPING";
        LifecycleState["Received"] = "RECEIVED";
        LifecycleState["Processing"] = "PROCESSING";
        LifecycleState["Processed"] = "PROCESSED";
        LifecycleState["Returned"] = "RETURNED";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Cancelled"] = "CANCELLED";
        LifecycleState["CancelledReturned"] = "CANCELLED_RETURNED";
    })(LifecycleState = ListTransferPackagesRequest.LifecycleState || (ListTransferPackagesRequest.LifecycleState = {}));
})(ListTransferPackagesRequest = exports.ListTransferPackagesRequest || (exports.ListTransferPackagesRequest = {}));
