// oci-dts/lib/request/list-transfer-appliances-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 1.0.015
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListTransferAppliancesRequest = void 0;
var ListTransferAppliancesRequest;
(function (ListTransferAppliancesRequest) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Requested"] = "REQUESTED";
        LifecycleState["OraclePreparing"] = "ORACLE_PREPARING";
        LifecycleState["Shipping"] = "SHIPPING";
        LifecycleState["Delivered"] = "DELIVERED";
        LifecycleState["Preparing"] = "PREPARING";
        LifecycleState["Finalized"] = "FINALIZED";
        LifecycleState["ReturnDelayed"] = "RETURN_DELAYED";
        LifecycleState["ReturnShipped"] = "RETURN_SHIPPED";
        LifecycleState["ReturnShippedCancelled"] = "RETURN_SHIPPED_CANCELLED";
        LifecycleState["OracleReceived"] = "ORACLE_RECEIVED";
        LifecycleState["OracleReceivedCancelled"] = "ORACLE_RECEIVED_CANCELLED";
        LifecycleState["Processing"] = "PROCESSING";
        LifecycleState["Complete"] = "COMPLETE";
        LifecycleState["CustomerNeverReceived"] = "CUSTOMER_NEVER_RECEIVED";
        LifecycleState["OracleNeverReceived"] = "ORACLE_NEVER_RECEIVED";
        LifecycleState["CustomerLost"] = "CUSTOMER_LOST";
        LifecycleState["Cancelled"] = "CANCELLED";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Rejected"] = "REJECTED";
        LifecycleState["Error"] = "ERROR";
    })(LifecycleState = ListTransferAppliancesRequest.LifecycleState || (ListTransferAppliancesRequest.LifecycleState = {}));
})(ListTransferAppliancesRequest = exports.ListTransferAppliancesRequest || (exports.ListTransferAppliancesRequest = {}));
