// oci-dts/lib/request/list-transfer-devices-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 1.0.015
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListTransferDevicesRequest = void 0;
var ListTransferDevicesRequest;
(function (ListTransferDevicesRequest) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Preparing"] = "PREPARING";
        LifecycleState["Ready"] = "READY";
        LifecycleState["Packaged"] = "PACKAGED";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Processing"] = "PROCESSING";
        LifecycleState["Complete"] = "COMPLETE";
        LifecycleState["Missing"] = "MISSING";
        LifecycleState["Error"] = "ERROR";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Cancelled"] = "CANCELLED";
    })(LifecycleState = ListTransferDevicesRequest.LifecycleState || (ListTransferDevicesRequest.LifecycleState = {}));
})(ListTransferDevicesRequest = exports.ListTransferDevicesRequest || (exports.ListTransferDevicesRequest = {}));
