// oci-dataintegration/lib/model/compression.js
"use strict";
/**
 * Data Integration API
 * Use the Data Integration Service APIs to perform common extract, load, and transform (ETL) tasks.
 * OpenAPI spec version: 20200430
 * Contact: di_dis_ww_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Compression = void 0;
var Compression;
(function (Compression) {
    let Codec;
    (function (Codec) {
        Codec["None"] = "NONE";
        Codec["Auto"] = "AUTO";
        Codec["Gzip"] = "GZIP";
        Codec["Bzip2"] = "BZIP2";
        Codec["Deflate"] = "DEFLATE";
        Codec["Lz4"] = "LZ4";
        Codec["Snappy"] = "SNAPPY";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Codec["UnknownValue"] = "UNKNOWN_VALUE";
    })(Codec = Compression.Codec || (Compression.Codec = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    Compression.getJsonObj = getJsonObj;
})(Compression = exports.Compression || (exports.Compression = {}));
