// oci-dataintegration/lib/model/count-statistic-summary.js
"use strict";
/**
 * Data Integration API
 * Use the Data Integration Service APIs to perform common extract, load, and transform (ETL) tasks.
 * OpenAPI spec version: 20200430
 * Contact: di_dis_ww_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CountStatisticSummary = void 0;
var CountStatisticSummary;
(function (CountStatisticSummary) {
    let ObjectType;
    (function (ObjectType) {
        ObjectType["Project"] = "PROJECT";
        ObjectType["Folder"] = "FOLDER";
        ObjectType["DataFlow"] = "DATA_FLOW";
        ObjectType["DataAsset"] = "DATA_ASSET";
        ObjectType["Connection"] = "CONNECTION";
        ObjectType["Task"] = "TASK";
        ObjectType["Application"] = "APPLICATION";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ObjectType["UnknownValue"] = "UNKNOWN_VALUE";
    })(ObjectType = CountStatisticSummary.ObjectType || (CountStatisticSummary.ObjectType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CountStatisticSummary.getJsonObj = getJsonObj;
})(CountStatisticSummary = exports.CountStatisticSummary || (exports.CountStatisticSummary = {}));
