// oci-dataintegration/lib/model/entity-shape-from-file.js
"use strict";
/**
 * Data Integration API
 * Use the Data Integration Service APIs to perform common extract, load, and transform (ETL) tasks.
 * OpenAPI spec version: 20200430
 * Contact: di_dis_ww_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.EntityShapeFromFile = void 0;
const model = __importStar(require("../model"));
var EntityShapeFromFile;
(function (EntityShapeFromFile) {
    let EntityType;
    (function (EntityType) {
        EntityType["Table"] = "TABLE";
        EntityType["View"] = "VIEW";
        EntityType["File"] = "FILE";
        EntityType["Queue"] = "QUEUE";
        EntityType["Stream"] = "STREAM";
        EntityType["Other"] = "OTHER";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        EntityType["UnknownValue"] = "UNKNOWN_VALUE";
    })(EntityType = EntityShapeFromFile.EntityType || (EntityShapeFromFile.EntityType = {}));
    function getJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj ? obj : model.EntityShape.getJsonObj(obj))), {
            "parentRef": obj.parentRef ? model.ParentReference.getJsonObj(obj.parentRef) : undefined,
            "shape": obj.shape ? model.Shape.getJsonObj(obj.shape) : undefined,
            "types": obj.types ? model.TypeLibrary.getJsonObj(obj.types) : undefined,
            "uniqueKeys": obj.uniqueKeys
                ? obj.uniqueKeys.map(item => {
                    return model.UniqueKey.getJsonObj(item);
                })
                : undefined,
            "foreignKeys": obj.foreignKeys
                ? obj.foreignKeys.map(item => {
                    return model.ForeignKey.getJsonObj(item);
                })
                : undefined,
            "dataFormat": obj.dataFormat ? model.DataFormat.getJsonObj(obj.dataFormat) : undefined
        });
        return jsonObj;
    }
    EntityShapeFromFile.getJsonObj = getJsonObj;
    EntityShapeFromFile.modelType = "FILE_ENTITY";
})(EntityShapeFromFile = exports.EntityShapeFromFile || (exports.EntityShapeFromFile = {}));
