// oci-dataintegration/lib/model/index.js
"use strict";
/**
 * Data Integration API
 * Use the Data Integration Service APIs to perform common extract, load, and transform (ETL) tasks.
 * OpenAPI spec version: 20200430
 * Contact: di_dis_ww_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateTaskValidationDetails = exports.CreateTaskRunDetails = exports.CreateTaskDetails = exports.CreateSourceApplicationInfo = exports.CreateProjectDetails = exports.CreatePipelineValidationDetails = exports.CreatePipelineDetails = exports.CreatePatchDetails = exports.CreateFolderDetails = exports.CreateExternalPublicationValidationDetails = exports.CreateExternalPublicationDetails = exports.CreateEntityShapeDetails = exports.CreateDataFlowValidationDetails = exports.CreateDataFlowDetails = exports.CreateDataAssetDetails = exports.CreateConnectionValidationDetails = exports.CreateConnectionDetails = exports.CreateConfigProvider = exports.CreateApplicationDetails = exports.CountStatisticSummary = exports.CountStatistic = exports.ConnectionValidationSummaryCollection = exports.ConnectionValidationSummary = exports.ConnectionValidation = exports.ConnectionSummaryCollection = exports.ConnectionSummary = exports.ConnectionProperty = exports.ConnectionDetails = exports.Connection = exports.ConfigurationDetails = exports.ConfigValues = exports.ConfigProvider = exports.ConfigParameterValue = exports.ConfigParameterDefinition = exports.ConfigDefinition = exports.Compression = exports.ChildReferenceDetail = exports.ChildReference = exports.ChangeCompartmentDetails = exports.BaseType = exports.ApplicationSummaryCollection = exports.ApplicationSummary = exports.ApplicationDetails = exports.Application = exports.AggregatorSummary = exports.AbstractWriteAttribute = exports.AbstractReadAttribute = exports.AbstractFrequencyDetails = exports.AbstractFormatAttribute = exports.AbstractDataOperationConfig = void 0;
exports.ParameterValue = exports.OracleWriteAttributes = exports.OracleReadAttributes = exports.OracleAtpWriteAttributes = exports.OracleAdwcWriteAttributes = exports.Operator = exports.ObjectMetadata = exports.NativeShapeField = exports.Message = exports.LastRunDetails = exports.KeyRange = exports.KeyAttribute = exports.Key = exports.FolderSummaryCollection = exports.FolderSummary = exports.FolderDetails = exports.Folder = exports.FlowPortLink = exports.FlowNode = exports.FieldMap = exports.ExternalPublicationValidationSummaryCollection = exports.ExternalPublicationValidationSummary = exports.ExternalPublicationValidation = exports.ExternalPublicationSummaryCollection = exports.ExternalPublicationSummary = exports.ExternalPublication = exports.Expression = exports.ErrorDetails = exports.EntityShape = exports.EnrichedEntity = exports.DynamicTypeHandler = exports.DependentObjectSummaryCollection = exports.DependentObjectSummary = exports.DependentObject = exports.DataFormat = exports.DataFlowValidationSummaryCollection = exports.DataFlowValidationSummary = exports.DataFlowValidation = exports.DataFlowSummaryCollection = exports.DataFlowSummary = exports.DataFlowDetails = exports.DataFlow = exports.DataEntitySummaryCollection = exports.DataEntitySummary = exports.DataEntityDetails = exports.DataEntity = exports.DataAssetSummaryCollection = exports.DataAssetSummary = exports.DataAsset = exports.CreateWorkspaceDetails = void 0;
exports.TaskSummary = exports.TaskSchedule = exports.TaskRunSummaryCollection = exports.TaskRunSummary = exports.TaskRunLogSummary = exports.TaskRunDetails = exports.TaskRun = exports.Task = exports.StructuredType = exports.SourceApplicationInfo = exports.SortKeyRule = exports.SortKey = exports.SortClause = exports.SensitiveAttribute = exports.SecretConfig = exports.SchemaSummaryCollection = exports.SchemaSummary = exports.SchemaDriftConfig = exports.Schema = exports.Schedule = exports.RootObject = exports.ResourceConfiguration = exports.RegistryMetadata = exports.ReferenceUsedBy = exports.ReferenceSummaryCollection = exports.ReferenceSummary = exports.Reference = exports.PushDownOperation = exports.PublishedObjectSummaryCollection = exports.PublishedObjectSummary = exports.PublishedObject = exports.ProjectionRule = exports.ProjectSummaryCollection = exports.ProjectSummary = exports.ProjectDetails = exports.Project = exports.PipelineValidationSummaryCollection = exports.PipelineValidationSummary = exports.PipelineValidation = exports.PipelineSummaryCollection = exports.PipelineSummary = exports.Pipeline = exports.PatchSummaryCollection = exports.PatchSummary = exports.PatchObjectMetadata = exports.PatchChangeSummaryCollection = exports.PatchChangeSummary = exports.Patch = exports.PartitionConfig = exports.ParentReference = void 0;
exports.ConnectionFromOracleDetails = exports.ConnectionFromOracle = exports.ConnectionFromObjectStorageDetails = exports.ConnectionFromObjectStorage = exports.ConnectionFromMySQLDetails = exports.ConnectionFromMySQL = exports.ConnectionFromJdbcDetails = exports.ConnectionFromJdbc = exports.ConnectionFromAtpDetails = exports.ConnectionFromAtp = exports.ConnectionFromAdwcDetails = exports.ConnectionFromAdwc = exports.ConfiguredType = exports.ConditionalInputLink = exports.CompositeType = exports.CompositeFieldMap = exports.AvroFormatAttribute = exports.Aggregator = exports.AbstractField = exports.WorkspaceSummary = exports.Workspace = exports.WorkRequestSummary = exports.WorkRequestResource = exports.WorkRequestLogEntry = exports.WorkRequestError = exports.WorkRequest = exports.Variable = exports.ValidationMessage = exports.UpdateWorkspaceDetails = exports.UpdateTaskRunDetails = exports.UpdateTaskDetails = exports.UpdateReferenceDetails = exports.UpdateProjectDetails = exports.UpdatePipelineDetails = exports.UpdateFolderDetails = exports.UpdateExternalPublicationDetails = exports.UpdateDataFlowDetails = exports.UpdateDataAssetDetails = exports.UpdateConnectionDetails = exports.UpdateApplicationDetails = exports.UniqueKey = exports.UIProperties = exports.TypedObject = exports.TypeSystem = exports.TypeLibrary = exports.Time = exports.TaskValidationSummaryCollection = exports.TaskValidationSummary = exports.TaskValidation = exports.TaskSummaryCollection = void 0;
exports.DerivedField = exports.DataType = exports.DataEntitySummaryFromView = exports.DataEntitySummaryFromTable = exports.DataEntitySummaryFromFile = exports.DataEntityFromViewEntityDetails = exports.DataEntityFromView = exports.DataEntityFromTableEntityDetails = exports.DataEntityFromTable = exports.DataEntityFromFileEntityDetails = exports.DataEntityFromFile = exports.DataAssetSummaryFromOracle = exports.DataAssetSummaryFromObjectStorage = exports.DataAssetSummaryFromMySQL = exports.DataAssetSummaryFromJdbc = exports.DataAssetSummaryFromAtp = exports.DataAssetSummaryFromAdwc = exports.DataAssetFromOracleDetails = exports.DataAssetFromObjectStorageDetails = exports.DataAssetFromMySQL = exports.DataAssetFromJdbc = exports.DataAssetFromAtpDetails = exports.DataAssetFromAdwcDetails = exports.DailyFrequencyDetails = exports.CsvFormatAttribute = exports.CreateTaskValidationFromPipelineTask = exports.CreateTaskValidationFromIntegrationTask = exports.CreateTaskValidationFromDataLoaderTask = exports.CreateTaskFromPipelineTask = exports.CreateTaskFromIntegrationTask = exports.CreateTaskFromDataLoaderTask = exports.CreateEntityShapeFromFile = exports.CreateDataAssetFromOracle = exports.CreateDataAssetFromObjectStorage = exports.CreateDataAssetFromMySQL = exports.CreateDataAssetFromJdbc = exports.CreateDataAssetFromAtp = exports.CreateDataAssetFromAdwc = exports.CreateConnectionFromOracle = exports.CreateConnectionFromObjectStorage = exports.CreateConnectionFromMySQL = exports.CreateConnectionFromJdbc = exports.CreateConnectionFromAtp = exports.CreateConnectionFromAdwc = exports.ConnectionSummaryFromOracle = exports.ConnectionSummaryFromObjectStorage = exports.ConnectionSummaryFromMySQL = exports.ConnectionSummaryFromJdbc = exports.ConnectionSummaryFromAtp = exports.ConnectionSummaryFromAdwc = void 0;
exports.RenameRule = exports.ReadOperationConfig = exports.Query = exports.PublishedObjectSummaryFromIntegrationTask = exports.PublishedObjectSummaryFromDataLoaderTask = exports.PublishedObjectFromPipelineTaskSummary = exports.PublishedObjectFromPipelineTask = exports.PublishedObjectFromIntegrationTask = exports.PublishedObjectFromDataLoaderTask = exports.ProxyField = exports.Projection = exports.PrimaryKey = exports.Parameter = exports.OutputPort = exports.OutputLink = exports.OutputField = exports.OracleWriteAttribute = exports.OracleReadAttribute = exports.OracleAtpWriteAttribute = exports.OracleAdwcWriteAttribute = exports.OciVaultSecretConfig = exports.NamePatternRule = exports.NameListRule = exports.MonthlyFrequencyDetails = exports.Minus = exports.MergeOperator = exports.MacroField = exports.KeyRangePartitionConfig = exports.JsonFormatAttribute = exports.Joiner = exports.Join = exports.JavaType = exports.Intersect = exports.InputPort = exports.InputLink = exports.InputField = exports.HourlyFrequencyDetails = exports.ForeignKey = exports.FlowPort = exports.FilterPush = exports.Filter = exports.EntityShapeFromFile = exports.EndOperator = exports.DynamicType = exports.DynamicProxyField = exports.DynamicInputField = exports.Distinct = exports.DirectNamedFieldMap = exports.DirectFieldMap = exports.DerivedType = void 0;
exports.WriteOperationConfig = exports.UpdateTaskFromPipelineTask = exports.UpdateTaskFromIntegrationTask = exports.UpdateTaskFromDataLoaderTask = exports.UpdateDataAssetFromOracle = exports.UpdateDataAssetFromObjectStorage = exports.UpdateDataAssetFromMySQL = exports.UpdateDataAssetFromJdbc = exports.UpdateDataAssetFromAtp = exports.UpdateDataAssetFromAdwc = exports.UpdateConnectionFromOracle = exports.UpdateConnectionFromObjectStorage = exports.UpdateConnectionFromMySQL = exports.UpdateConnectionFromJdbc = exports.UpdateConnectionFromAtp = exports.UpdateConnectionFromAdwc = exports.Union = exports.TypedNamePatternRule = exports.TypeListRule = exports.TaskSummaryFromPipelineTask = exports.TaskSummaryFromIntegrationTask = exports.TaskSummaryFromDataLoaderTask = exports.TaskOperator = exports.TaskFromPipelineTaskDetails = exports.TaskFromIntegrationTaskDetails = exports.TaskFromDataLoaderTaskDetails = exports.Target = exports.StartOperator = exports.Source = exports.SortOper = exports.Sort = exports.ShapeField = exports.Shape = exports.Select = exports.RuleTypeConfig = exports.RuleBasedFieldMap = void 0;
const AbstractDataOperationConfig = __importStar(require("./abstract-data-operation-config"));
exports.AbstractDataOperationConfig = AbstractDataOperationConfig.AbstractDataOperationConfig;
const AbstractFormatAttribute = __importStar(require("./abstract-format-attribute"));
exports.AbstractFormatAttribute = AbstractFormatAttribute.AbstractFormatAttribute;
const AbstractFrequencyDetails = __importStar(require("./abstract-frequency-details"));
exports.AbstractFrequencyDetails = AbstractFrequencyDetails.AbstractFrequencyDetails;
const AbstractReadAttribute = __importStar(require("./abstract-read-attribute"));
exports.AbstractReadAttribute = AbstractReadAttribute.AbstractReadAttribute;
const AbstractWriteAttribute = __importStar(require("./abstract-write-attribute"));
exports.AbstractWriteAttribute = AbstractWriteAttribute.AbstractWriteAttribute;
const AggregatorSummary = __importStar(require("./aggregator-summary"));
exports.AggregatorSummary = AggregatorSummary.AggregatorSummary;
const Application = __importStar(require("./application"));
exports.Application = Application.Application;
const ApplicationDetails = __importStar(require("./application-details"));
exports.ApplicationDetails = ApplicationDetails.ApplicationDetails;
const ApplicationSummary = __importStar(require("./application-summary"));
exports.ApplicationSummary = ApplicationSummary.ApplicationSummary;
const ApplicationSummaryCollection = __importStar(require("./application-summary-collection"));
exports.ApplicationSummaryCollection = ApplicationSummaryCollection.ApplicationSummaryCollection;
const BaseType = __importStar(require("./base-type"));
exports.BaseType = BaseType.BaseType;
const ChangeCompartmentDetails = __importStar(require("./change-compartment-details"));
exports.ChangeCompartmentDetails = ChangeCompartmentDetails.ChangeCompartmentDetails;
const ChildReference = __importStar(require("./child-reference"));
exports.ChildReference = ChildReference.ChildReference;
const ChildReferenceDetail = __importStar(require("./child-reference-detail"));
exports.ChildReferenceDetail = ChildReferenceDetail.ChildReferenceDetail;
const Compression = __importStar(require("./compression"));
exports.Compression = Compression.Compression;
const ConfigDefinition = __importStar(require("./config-definition"));
exports.ConfigDefinition = ConfigDefinition.ConfigDefinition;
const ConfigParameterDefinition = __importStar(require("./config-parameter-definition"));
exports.ConfigParameterDefinition = ConfigParameterDefinition.ConfigParameterDefinition;
const ConfigParameterValue = __importStar(require("./config-parameter-value"));
exports.ConfigParameterValue = ConfigParameterValue.ConfigParameterValue;
const ConfigProvider = __importStar(require("./config-provider"));
exports.ConfigProvider = ConfigProvider.ConfigProvider;
const ConfigValues = __importStar(require("./config-values"));
exports.ConfigValues = ConfigValues.ConfigValues;
const ConfigurationDetails = __importStar(require("./configuration-details"));
exports.ConfigurationDetails = ConfigurationDetails.ConfigurationDetails;
const Connection = __importStar(require("./connection"));
exports.Connection = Connection.Connection;
const ConnectionDetails = __importStar(require("./connection-details"));
exports.ConnectionDetails = ConnectionDetails.ConnectionDetails;
const ConnectionProperty = __importStar(require("./connection-property"));
exports.ConnectionProperty = ConnectionProperty.ConnectionProperty;
const ConnectionSummary = __importStar(require("./connection-summary"));
exports.ConnectionSummary = ConnectionSummary.ConnectionSummary;
const ConnectionSummaryCollection = __importStar(require("./connection-summary-collection"));
exports.ConnectionSummaryCollection = ConnectionSummaryCollection.ConnectionSummaryCollection;
const ConnectionValidation = __importStar(require("./connection-validation"));
exports.ConnectionValidation = ConnectionValidation.ConnectionValidation;
const ConnectionValidationSummary = __importStar(require("./connection-validation-summary"));
exports.ConnectionValidationSummary = ConnectionValidationSummary.ConnectionValidationSummary;
const ConnectionValidationSummaryCollection = __importStar(require("./connection-validation-summary-collection"));
exports.ConnectionValidationSummaryCollection = ConnectionValidationSummaryCollection.ConnectionValidationSummaryCollection;
const CountStatistic = __importStar(require("./count-statistic"));
exports.CountStatistic = CountStatistic.CountStatistic;
const CountStatisticSummary = __importStar(require("./count-statistic-summary"));
exports.CountStatisticSummary = CountStatisticSummary.CountStatisticSummary;
const CreateApplicationDetails = __importStar(require("./create-application-details"));
exports.CreateApplicationDetails = CreateApplicationDetails.CreateApplicationDetails;
const CreateConfigProvider = __importStar(require("./create-config-provider"));
exports.CreateConfigProvider = CreateConfigProvider.CreateConfigProvider;
const CreateConnectionDetails = __importStar(require("./create-connection-details"));
exports.CreateConnectionDetails = CreateConnectionDetails.CreateConnectionDetails;
const CreateConnectionValidationDetails = __importStar(require("./create-connection-validation-details"));
exports.CreateConnectionValidationDetails = CreateConnectionValidationDetails.CreateConnectionValidationDetails;
const CreateDataAssetDetails = __importStar(require("./create-data-asset-details"));
exports.CreateDataAssetDetails = CreateDataAssetDetails.CreateDataAssetDetails;
const CreateDataFlowDetails = __importStar(require("./create-data-flow-details"));
exports.CreateDataFlowDetails = CreateDataFlowDetails.CreateDataFlowDetails;
const CreateDataFlowValidationDetails = __importStar(require("./create-data-flow-validation-details"));
exports.CreateDataFlowValidationDetails = CreateDataFlowValidationDetails.CreateDataFlowValidationDetails;
const CreateEntityShapeDetails = __importStar(require("./create-entity-shape-details"));
exports.CreateEntityShapeDetails = CreateEntityShapeDetails.CreateEntityShapeDetails;
const CreateExternalPublicationDetails = __importStar(require("./create-external-publication-details"));
exports.CreateExternalPublicationDetails = CreateExternalPublicationDetails.CreateExternalPublicationDetails;
const CreateExternalPublicationValidationDetails = __importStar(require("./create-external-publication-validation-details"));
exports.CreateExternalPublicationValidationDetails = CreateExternalPublicationValidationDetails.CreateExternalPublicationValidationDetails;
const CreateFolderDetails = __importStar(require("./create-folder-details"));
exports.CreateFolderDetails = CreateFolderDetails.CreateFolderDetails;
const CreatePatchDetails = __importStar(require("./create-patch-details"));
exports.CreatePatchDetails = CreatePatchDetails.CreatePatchDetails;
const CreatePipelineDetails = __importStar(require("./create-pipeline-details"));
exports.CreatePipelineDetails = CreatePipelineDetails.CreatePipelineDetails;
const CreatePipelineValidationDetails = __importStar(require("./create-pipeline-validation-details"));
exports.CreatePipelineValidationDetails = CreatePipelineValidationDetails.CreatePipelineValidationDetails;
const CreateProjectDetails = __importStar(require("./create-project-details"));
exports.CreateProjectDetails = CreateProjectDetails.CreateProjectDetails;
const CreateSourceApplicationInfo = __importStar(require("./create-source-application-info"));
exports.CreateSourceApplicationInfo = CreateSourceApplicationInfo.CreateSourceApplicationInfo;
const CreateTaskDetails = __importStar(require("./create-task-details"));
exports.CreateTaskDetails = CreateTaskDetails.CreateTaskDetails;
const CreateTaskRunDetails = __importStar(require("./create-task-run-details"));
exports.CreateTaskRunDetails = CreateTaskRunDetails.CreateTaskRunDetails;
const CreateTaskValidationDetails = __importStar(require("./create-task-validation-details"));
exports.CreateTaskValidationDetails = CreateTaskValidationDetails.CreateTaskValidationDetails;
const CreateWorkspaceDetails = __importStar(require("./create-workspace-details"));
exports.CreateWorkspaceDetails = CreateWorkspaceDetails.CreateWorkspaceDetails;
const DataAsset = __importStar(require("./data-asset"));
exports.DataAsset = DataAsset.DataAsset;
const DataAssetSummary = __importStar(require("./data-asset-summary"));
exports.DataAssetSummary = DataAssetSummary.DataAssetSummary;
const DataAssetSummaryCollection = __importStar(require("./data-asset-summary-collection"));
exports.DataAssetSummaryCollection = DataAssetSummaryCollection.DataAssetSummaryCollection;
const DataEntity = __importStar(require("./data-entity"));
exports.DataEntity = DataEntity.DataEntity;
const DataEntityDetails = __importStar(require("./data-entity-details"));
exports.DataEntityDetails = DataEntityDetails.DataEntityDetails;
const DataEntitySummary = __importStar(require("./data-entity-summary"));
exports.DataEntitySummary = DataEntitySummary.DataEntitySummary;
const DataEntitySummaryCollection = __importStar(require("./data-entity-summary-collection"));
exports.DataEntitySummaryCollection = DataEntitySummaryCollection.DataEntitySummaryCollection;
const DataFlow = __importStar(require("./data-flow"));
exports.DataFlow = DataFlow.DataFlow;
const DataFlowDetails = __importStar(require("./data-flow-details"));
exports.DataFlowDetails = DataFlowDetails.DataFlowDetails;
const DataFlowSummary = __importStar(require("./data-flow-summary"));
exports.DataFlowSummary = DataFlowSummary.DataFlowSummary;
const DataFlowSummaryCollection = __importStar(require("./data-flow-summary-collection"));
exports.DataFlowSummaryCollection = DataFlowSummaryCollection.DataFlowSummaryCollection;
const DataFlowValidation = __importStar(require("./data-flow-validation"));
exports.DataFlowValidation = DataFlowValidation.DataFlowValidation;
const DataFlowValidationSummary = __importStar(require("./data-flow-validation-summary"));
exports.DataFlowValidationSummary = DataFlowValidationSummary.DataFlowValidationSummary;
const DataFlowValidationSummaryCollection = __importStar(require("./data-flow-validation-summary-collection"));
exports.DataFlowValidationSummaryCollection = DataFlowValidationSummaryCollection.DataFlowValidationSummaryCollection;
const DataFormat = __importStar(require("./data-format"));
exports.DataFormat = DataFormat.DataFormat;
const DependentObject = __importStar(require("./dependent-object"));
exports.DependentObject = DependentObject.DependentObject;
const DependentObjectSummary = __importStar(require("./dependent-object-summary"));
exports.DependentObjectSummary = DependentObjectSummary.DependentObjectSummary;
const DependentObjectSummaryCollection = __importStar(require("./dependent-object-summary-collection"));
exports.DependentObjectSummaryCollection = DependentObjectSummaryCollection.DependentObjectSummaryCollection;
const DynamicTypeHandler = __importStar(require("./dynamic-type-handler"));
exports.DynamicTypeHandler = DynamicTypeHandler.DynamicTypeHandler;
const EnrichedEntity = __importStar(require("./enriched-entity"));
exports.EnrichedEntity = EnrichedEntity.EnrichedEntity;
const EntityShape = __importStar(require("./entity-shape"));
exports.EntityShape = EntityShape.EntityShape;
const ErrorDetails = __importStar(require("./error-details"));
exports.ErrorDetails = ErrorDetails.ErrorDetails;
const Expression = __importStar(require("./expression"));
exports.Expression = Expression.Expression;
const ExternalPublication = __importStar(require("./external-publication"));
exports.ExternalPublication = ExternalPublication.ExternalPublication;
const ExternalPublicationSummary = __importStar(require("./external-publication-summary"));
exports.ExternalPublicationSummary = ExternalPublicationSummary.ExternalPublicationSummary;
const ExternalPublicationSummaryCollection = __importStar(require("./external-publication-summary-collection"));
exports.ExternalPublicationSummaryCollection = ExternalPublicationSummaryCollection.ExternalPublicationSummaryCollection;
const ExternalPublicationValidation = __importStar(require("./external-publication-validation"));
exports.ExternalPublicationValidation = ExternalPublicationValidation.ExternalPublicationValidation;
const ExternalPublicationValidationSummary = __importStar(require("./external-publication-validation-summary"));
exports.ExternalPublicationValidationSummary = ExternalPublicationValidationSummary.ExternalPublicationValidationSummary;
const ExternalPublicationValidationSummaryCollection = __importStar(require("./external-publication-validation-summary-collection"));
exports.ExternalPublicationValidationSummaryCollection = ExternalPublicationValidationSummaryCollection.ExternalPublicationValidationSummaryCollection;
const FieldMap = __importStar(require("./field-map"));
exports.FieldMap = FieldMap.FieldMap;
const FlowNode = __importStar(require("./flow-node"));
exports.FlowNode = FlowNode.FlowNode;
const FlowPortLink = __importStar(require("./flow-port-link"));
exports.FlowPortLink = FlowPortLink.FlowPortLink;
const Folder = __importStar(require("./folder"));
exports.Folder = Folder.Folder;
const FolderDetails = __importStar(require("./folder-details"));
exports.FolderDetails = FolderDetails.FolderDetails;
const FolderSummary = __importStar(require("./folder-summary"));
exports.FolderSummary = FolderSummary.FolderSummary;
const FolderSummaryCollection = __importStar(require("./folder-summary-collection"));
exports.FolderSummaryCollection = FolderSummaryCollection.FolderSummaryCollection;
const Key = __importStar(require("./key"));
exports.Key = Key.Key;
const KeyAttribute = __importStar(require("./key-attribute"));
exports.KeyAttribute = KeyAttribute.KeyAttribute;
const KeyRange = __importStar(require("./key-range"));
exports.KeyRange = KeyRange.KeyRange;
const LastRunDetails = __importStar(require("./last-run-details"));
exports.LastRunDetails = LastRunDetails.LastRunDetails;
const Message = __importStar(require("./message"));
exports.Message = Message.Message;
const NativeShapeField = __importStar(require("./native-shape-field"));
exports.NativeShapeField = NativeShapeField.NativeShapeField;
const ObjectMetadata = __importStar(require("./object-metadata"));
exports.ObjectMetadata = ObjectMetadata.ObjectMetadata;
const Operator = __importStar(require("./operator"));
exports.Operator = Operator.Operator;
const OracleAdwcWriteAttributes = __importStar(require("./oracle-adwc-write-attributes"));
exports.OracleAdwcWriteAttributes = OracleAdwcWriteAttributes.OracleAdwcWriteAttributes;
const OracleAtpWriteAttributes = __importStar(require("./oracle-atp-write-attributes"));
exports.OracleAtpWriteAttributes = OracleAtpWriteAttributes.OracleAtpWriteAttributes;
const OracleReadAttributes = __importStar(require("./oracle-read-attributes"));
exports.OracleReadAttributes = OracleReadAttributes.OracleReadAttributes;
const OracleWriteAttributes = __importStar(require("./oracle-write-attributes"));
exports.OracleWriteAttributes = OracleWriteAttributes.OracleWriteAttributes;
const ParameterValue = __importStar(require("./parameter-value"));
exports.ParameterValue = ParameterValue.ParameterValue;
const ParentReference = __importStar(require("./parent-reference"));
exports.ParentReference = ParentReference.ParentReference;
const PartitionConfig = __importStar(require("./partition-config"));
exports.PartitionConfig = PartitionConfig.PartitionConfig;
const Patch = __importStar(require("./patch"));
exports.Patch = Patch.Patch;
const PatchChangeSummary = __importStar(require("./patch-change-summary"));
exports.PatchChangeSummary = PatchChangeSummary.PatchChangeSummary;
const PatchChangeSummaryCollection = __importStar(require("./patch-change-summary-collection"));
exports.PatchChangeSummaryCollection = PatchChangeSummaryCollection.PatchChangeSummaryCollection;
const PatchObjectMetadata = __importStar(require("./patch-object-metadata"));
exports.PatchObjectMetadata = PatchObjectMetadata.PatchObjectMetadata;
const PatchSummary = __importStar(require("./patch-summary"));
exports.PatchSummary = PatchSummary.PatchSummary;
const PatchSummaryCollection = __importStar(require("./patch-summary-collection"));
exports.PatchSummaryCollection = PatchSummaryCollection.PatchSummaryCollection;
const Pipeline = __importStar(require("./pipeline"));
exports.Pipeline = Pipeline.Pipeline;
const PipelineSummary = __importStar(require("./pipeline-summary"));
exports.PipelineSummary = PipelineSummary.PipelineSummary;
const PipelineSummaryCollection = __importStar(require("./pipeline-summary-collection"));
exports.PipelineSummaryCollection = PipelineSummaryCollection.PipelineSummaryCollection;
const PipelineValidation = __importStar(require("./pipeline-validation"));
exports.PipelineValidation = PipelineValidation.PipelineValidation;
const PipelineValidationSummary = __importStar(require("./pipeline-validation-summary"));
exports.PipelineValidationSummary = PipelineValidationSummary.PipelineValidationSummary;
const PipelineValidationSummaryCollection = __importStar(require("./pipeline-validation-summary-collection"));
exports.PipelineValidationSummaryCollection = PipelineValidationSummaryCollection.PipelineValidationSummaryCollection;
const Project = __importStar(require("./project"));
exports.Project = Project.Project;
const ProjectDetails = __importStar(require("./project-details"));
exports.ProjectDetails = ProjectDetails.ProjectDetails;
const ProjectSummary = __importStar(require("./project-summary"));
exports.ProjectSummary = ProjectSummary.ProjectSummary;
const ProjectSummaryCollection = __importStar(require("./project-summary-collection"));
exports.ProjectSummaryCollection = ProjectSummaryCollection.ProjectSummaryCollection;
const ProjectionRule = __importStar(require("./projection-rule"));
exports.ProjectionRule = ProjectionRule.ProjectionRule;
const PublishedObject = __importStar(require("./published-object"));
exports.PublishedObject = PublishedObject.PublishedObject;
const PublishedObjectSummary = __importStar(require("./published-object-summary"));
exports.PublishedObjectSummary = PublishedObjectSummary.PublishedObjectSummary;
const PublishedObjectSummaryCollection = __importStar(require("./published-object-summary-collection"));
exports.PublishedObjectSummaryCollection = PublishedObjectSummaryCollection.PublishedObjectSummaryCollection;
const PushDownOperation = __importStar(require("./push-down-operation"));
exports.PushDownOperation = PushDownOperation.PushDownOperation;
const Reference = __importStar(require("./reference"));
exports.Reference = Reference.Reference;
const ReferenceSummary = __importStar(require("./reference-summary"));
exports.ReferenceSummary = ReferenceSummary.ReferenceSummary;
const ReferenceSummaryCollection = __importStar(require("./reference-summary-collection"));
exports.ReferenceSummaryCollection = ReferenceSummaryCollection.ReferenceSummaryCollection;
const ReferenceUsedBy = __importStar(require("./reference-used-by"));
exports.ReferenceUsedBy = ReferenceUsedBy.ReferenceUsedBy;
const RegistryMetadata = __importStar(require("./registry-metadata"));
exports.RegistryMetadata = RegistryMetadata.RegistryMetadata;
const ResourceConfiguration = __importStar(require("./resource-configuration"));
exports.ResourceConfiguration = ResourceConfiguration.ResourceConfiguration;
const RootObject = __importStar(require("./root-object"));
exports.RootObject = RootObject.RootObject;
const Schedule = __importStar(require("./schedule"));
exports.Schedule = Schedule.Schedule;
const Schema = __importStar(require("./schema"));
exports.Schema = Schema.Schema;
const SchemaDriftConfig = __importStar(require("./schema-drift-config"));
exports.SchemaDriftConfig = SchemaDriftConfig.SchemaDriftConfig;
const SchemaSummary = __importStar(require("./schema-summary"));
exports.SchemaSummary = SchemaSummary.SchemaSummary;
const SchemaSummaryCollection = __importStar(require("./schema-summary-collection"));
exports.SchemaSummaryCollection = SchemaSummaryCollection.SchemaSummaryCollection;
const SecretConfig = __importStar(require("./secret-config"));
exports.SecretConfig = SecretConfig.SecretConfig;
const SensitiveAttribute = __importStar(require("./sensitive-attribute"));
exports.SensitiveAttribute = SensitiveAttribute.SensitiveAttribute;
const SortClause = __importStar(require("./sort-clause"));
exports.SortClause = SortClause.SortClause;
const SortKey = __importStar(require("./sort-key"));
exports.SortKey = SortKey.SortKey;
const SortKeyRule = __importStar(require("./sort-key-rule"));
exports.SortKeyRule = SortKeyRule.SortKeyRule;
const SourceApplicationInfo = __importStar(require("./source-application-info"));
exports.SourceApplicationInfo = SourceApplicationInfo.SourceApplicationInfo;
const StructuredType = __importStar(require("./structured-type"));
exports.StructuredType = StructuredType.StructuredType;
const Task = __importStar(require("./task"));
exports.Task = Task.Task;
const TaskRun = __importStar(require("./task-run"));
exports.TaskRun = TaskRun.TaskRun;
const TaskRunDetails = __importStar(require("./task-run-details"));
exports.TaskRunDetails = TaskRunDetails.TaskRunDetails;
const TaskRunLogSummary = __importStar(require("./task-run-log-summary"));
exports.TaskRunLogSummary = TaskRunLogSummary.TaskRunLogSummary;
const TaskRunSummary = __importStar(require("./task-run-summary"));
exports.TaskRunSummary = TaskRunSummary.TaskRunSummary;
const TaskRunSummaryCollection = __importStar(require("./task-run-summary-collection"));
exports.TaskRunSummaryCollection = TaskRunSummaryCollection.TaskRunSummaryCollection;
const TaskSchedule = __importStar(require("./task-schedule"));
exports.TaskSchedule = TaskSchedule.TaskSchedule;
const TaskSummary = __importStar(require("./task-summary"));
exports.TaskSummary = TaskSummary.TaskSummary;
const TaskSummaryCollection = __importStar(require("./task-summary-collection"));
exports.TaskSummaryCollection = TaskSummaryCollection.TaskSummaryCollection;
const TaskValidation = __importStar(require("./task-validation"));
exports.TaskValidation = TaskValidation.TaskValidation;
const TaskValidationSummary = __importStar(require("./task-validation-summary"));
exports.TaskValidationSummary = TaskValidationSummary.TaskValidationSummary;
const TaskValidationSummaryCollection = __importStar(require("./task-validation-summary-collection"));
exports.TaskValidationSummaryCollection = TaskValidationSummaryCollection.TaskValidationSummaryCollection;
const Time = __importStar(require("./time"));
exports.Time = Time.Time;
const TypeLibrary = __importStar(require("./type-library"));
exports.TypeLibrary = TypeLibrary.TypeLibrary;
const TypeSystem = __importStar(require("./type-system"));
exports.TypeSystem = TypeSystem.TypeSystem;
const TypedObject = __importStar(require("./typed-object"));
exports.TypedObject = TypedObject.TypedObject;
const UIProperties = __importStar(require("./u-iproperties"));
exports.UIProperties = UIProperties.UIProperties;
const UniqueKey = __importStar(require("./unique-key"));
exports.UniqueKey = UniqueKey.UniqueKey;
const UpdateApplicationDetails = __importStar(require("./update-application-details"));
exports.UpdateApplicationDetails = UpdateApplicationDetails.UpdateApplicationDetails;
const UpdateConnectionDetails = __importStar(require("./update-connection-details"));
exports.UpdateConnectionDetails = UpdateConnectionDetails.UpdateConnectionDetails;
const UpdateDataAssetDetails = __importStar(require("./update-data-asset-details"));
exports.UpdateDataAssetDetails = UpdateDataAssetDetails.UpdateDataAssetDetails;
const UpdateDataFlowDetails = __importStar(require("./update-data-flow-details"));
exports.UpdateDataFlowDetails = UpdateDataFlowDetails.UpdateDataFlowDetails;
const UpdateExternalPublicationDetails = __importStar(require("./update-external-publication-details"));
exports.UpdateExternalPublicationDetails = UpdateExternalPublicationDetails.UpdateExternalPublicationDetails;
const UpdateFolderDetails = __importStar(require("./update-folder-details"));
exports.UpdateFolderDetails = UpdateFolderDetails.UpdateFolderDetails;
const UpdatePipelineDetails = __importStar(require("./update-pipeline-details"));
exports.UpdatePipelineDetails = UpdatePipelineDetails.UpdatePipelineDetails;
const UpdateProjectDetails = __importStar(require("./update-project-details"));
exports.UpdateProjectDetails = UpdateProjectDetails.UpdateProjectDetails;
const UpdateReferenceDetails = __importStar(require("./update-reference-details"));
exports.UpdateReferenceDetails = UpdateReferenceDetails.UpdateReferenceDetails;
const UpdateTaskDetails = __importStar(require("./update-task-details"));
exports.UpdateTaskDetails = UpdateTaskDetails.UpdateTaskDetails;
const UpdateTaskRunDetails = __importStar(require("./update-task-run-details"));
exports.UpdateTaskRunDetails = UpdateTaskRunDetails.UpdateTaskRunDetails;
const UpdateWorkspaceDetails = __importStar(require("./update-workspace-details"));
exports.UpdateWorkspaceDetails = UpdateWorkspaceDetails.UpdateWorkspaceDetails;
const ValidationMessage = __importStar(require("./validation-message"));
exports.ValidationMessage = ValidationMessage.ValidationMessage;
const Variable = __importStar(require("./variable"));
exports.Variable = Variable.Variable;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const Workspace = __importStar(require("./workspace"));
exports.Workspace = Workspace.Workspace;
const WorkspaceSummary = __importStar(require("./workspace-summary"));
exports.WorkspaceSummary = WorkspaceSummary.WorkspaceSummary;
const AbstractField = __importStar(require("./abstract-field"));
exports.AbstractField = AbstractField.AbstractField;
const Aggregator = __importStar(require("./aggregator"));
exports.Aggregator = Aggregator.Aggregator;
const AvroFormatAttribute = __importStar(require("./avro-format-attribute"));
exports.AvroFormatAttribute = AvroFormatAttribute.AvroFormatAttribute;
const CompositeFieldMap = __importStar(require("./composite-field-map"));
exports.CompositeFieldMap = CompositeFieldMap.CompositeFieldMap;
const CompositeType = __importStar(require("./composite-type"));
exports.CompositeType = CompositeType.CompositeType;
const ConditionalInputLink = __importStar(require("./conditional-input-link"));
exports.ConditionalInputLink = ConditionalInputLink.ConditionalInputLink;
const ConfiguredType = __importStar(require("./configured-type"));
exports.ConfiguredType = ConfiguredType.ConfiguredType;
const ConnectionFromAdwc = __importStar(require("./connection-from-adwc"));
exports.ConnectionFromAdwc = ConnectionFromAdwc.ConnectionFromAdwc;
const ConnectionFromAdwcDetails = __importStar(require("./connection-from-adwc-details"));
exports.ConnectionFromAdwcDetails = ConnectionFromAdwcDetails.ConnectionFromAdwcDetails;
const ConnectionFromAtp = __importStar(require("./connection-from-atp"));
exports.ConnectionFromAtp = ConnectionFromAtp.ConnectionFromAtp;
const ConnectionFromAtpDetails = __importStar(require("./connection-from-atp-details"));
exports.ConnectionFromAtpDetails = ConnectionFromAtpDetails.ConnectionFromAtpDetails;
const ConnectionFromJdbc = __importStar(require("./connection-from-jdbc"));
exports.ConnectionFromJdbc = ConnectionFromJdbc.ConnectionFromJdbc;
const ConnectionFromJdbcDetails = __importStar(require("./connection-from-jdbc-details"));
exports.ConnectionFromJdbcDetails = ConnectionFromJdbcDetails.ConnectionFromJdbcDetails;
const ConnectionFromMySQL = __importStar(require("./connection-from-my-sq-l"));
exports.ConnectionFromMySQL = ConnectionFromMySQL.ConnectionFromMySQL;
const ConnectionFromMySQLDetails = __importStar(require("./connection-from-my-sq-ldetails"));
exports.ConnectionFromMySQLDetails = ConnectionFromMySQLDetails.ConnectionFromMySQLDetails;
const ConnectionFromObjectStorage = __importStar(require("./connection-from-object-storage"));
exports.ConnectionFromObjectStorage = ConnectionFromObjectStorage.ConnectionFromObjectStorage;
const ConnectionFromObjectStorageDetails = __importStar(require("./connection-from-object-storage-details"));
exports.ConnectionFromObjectStorageDetails = ConnectionFromObjectStorageDetails.ConnectionFromObjectStorageDetails;
const ConnectionFromOracle = __importStar(require("./connection-from-oracle"));
exports.ConnectionFromOracle = ConnectionFromOracle.ConnectionFromOracle;
const ConnectionFromOracleDetails = __importStar(require("./connection-from-oracle-details"));
exports.ConnectionFromOracleDetails = ConnectionFromOracleDetails.ConnectionFromOracleDetails;
const ConnectionSummaryFromAdwc = __importStar(require("./connection-summary-from-adwc"));
exports.ConnectionSummaryFromAdwc = ConnectionSummaryFromAdwc.ConnectionSummaryFromAdwc;
const ConnectionSummaryFromAtp = __importStar(require("./connection-summary-from-atp"));
exports.ConnectionSummaryFromAtp = ConnectionSummaryFromAtp.ConnectionSummaryFromAtp;
const ConnectionSummaryFromJdbc = __importStar(require("./connection-summary-from-jdbc"));
exports.ConnectionSummaryFromJdbc = ConnectionSummaryFromJdbc.ConnectionSummaryFromJdbc;
const ConnectionSummaryFromMySQL = __importStar(require("./connection-summary-from-my-sq-l"));
exports.ConnectionSummaryFromMySQL = ConnectionSummaryFromMySQL.ConnectionSummaryFromMySQL;
const ConnectionSummaryFromObjectStorage = __importStar(require("./connection-summary-from-object-storage"));
exports.ConnectionSummaryFromObjectStorage = ConnectionSummaryFromObjectStorage.ConnectionSummaryFromObjectStorage;
const ConnectionSummaryFromOracle = __importStar(require("./connection-summary-from-oracle"));
exports.ConnectionSummaryFromOracle = ConnectionSummaryFromOracle.ConnectionSummaryFromOracle;
const CreateConnectionFromAdwc = __importStar(require("./create-connection-from-adwc"));
exports.CreateConnectionFromAdwc = CreateConnectionFromAdwc.CreateConnectionFromAdwc;
const CreateConnectionFromAtp = __importStar(require("./create-connection-from-atp"));
exports.CreateConnectionFromAtp = CreateConnectionFromAtp.CreateConnectionFromAtp;
const CreateConnectionFromJdbc = __importStar(require("./create-connection-from-jdbc"));
exports.CreateConnectionFromJdbc = CreateConnectionFromJdbc.CreateConnectionFromJdbc;
const CreateConnectionFromMySQL = __importStar(require("./create-connection-from-my-sq-l"));
exports.CreateConnectionFromMySQL = CreateConnectionFromMySQL.CreateConnectionFromMySQL;
const CreateConnectionFromObjectStorage = __importStar(require("./create-connection-from-object-storage"));
exports.CreateConnectionFromObjectStorage = CreateConnectionFromObjectStorage.CreateConnectionFromObjectStorage;
const CreateConnectionFromOracle = __importStar(require("./create-connection-from-oracle"));
exports.CreateConnectionFromOracle = CreateConnectionFromOracle.CreateConnectionFromOracle;
const CreateDataAssetFromAdwc = __importStar(require("./create-data-asset-from-adwc"));
exports.CreateDataAssetFromAdwc = CreateDataAssetFromAdwc.CreateDataAssetFromAdwc;
const CreateDataAssetFromAtp = __importStar(require("./create-data-asset-from-atp"));
exports.CreateDataAssetFromAtp = CreateDataAssetFromAtp.CreateDataAssetFromAtp;
const CreateDataAssetFromJdbc = __importStar(require("./create-data-asset-from-jdbc"));
exports.CreateDataAssetFromJdbc = CreateDataAssetFromJdbc.CreateDataAssetFromJdbc;
const CreateDataAssetFromMySQL = __importStar(require("./create-data-asset-from-my-sq-l"));
exports.CreateDataAssetFromMySQL = CreateDataAssetFromMySQL.CreateDataAssetFromMySQL;
const CreateDataAssetFromObjectStorage = __importStar(require("./create-data-asset-from-object-storage"));
exports.CreateDataAssetFromObjectStorage = CreateDataAssetFromObjectStorage.CreateDataAssetFromObjectStorage;
const CreateDataAssetFromOracle = __importStar(require("./create-data-asset-from-oracle"));
exports.CreateDataAssetFromOracle = CreateDataAssetFromOracle.CreateDataAssetFromOracle;
const CreateEntityShapeFromFile = __importStar(require("./create-entity-shape-from-file"));
exports.CreateEntityShapeFromFile = CreateEntityShapeFromFile.CreateEntityShapeFromFile;
const CreateTaskFromDataLoaderTask = __importStar(require("./create-task-from-data-loader-task"));
exports.CreateTaskFromDataLoaderTask = CreateTaskFromDataLoaderTask.CreateTaskFromDataLoaderTask;
const CreateTaskFromIntegrationTask = __importStar(require("./create-task-from-integration-task"));
exports.CreateTaskFromIntegrationTask = CreateTaskFromIntegrationTask.CreateTaskFromIntegrationTask;
const CreateTaskFromPipelineTask = __importStar(require("./create-task-from-pipeline-task"));
exports.CreateTaskFromPipelineTask = CreateTaskFromPipelineTask.CreateTaskFromPipelineTask;
const CreateTaskValidationFromDataLoaderTask = __importStar(require("./create-task-validation-from-data-loader-task"));
exports.CreateTaskValidationFromDataLoaderTask = CreateTaskValidationFromDataLoaderTask.CreateTaskValidationFromDataLoaderTask;
const CreateTaskValidationFromIntegrationTask = __importStar(require("./create-task-validation-from-integration-task"));
exports.CreateTaskValidationFromIntegrationTask = CreateTaskValidationFromIntegrationTask.CreateTaskValidationFromIntegrationTask;
const CreateTaskValidationFromPipelineTask = __importStar(require("./create-task-validation-from-pipeline-task"));
exports.CreateTaskValidationFromPipelineTask = CreateTaskValidationFromPipelineTask.CreateTaskValidationFromPipelineTask;
const CsvFormatAttribute = __importStar(require("./csv-format-attribute"));
exports.CsvFormatAttribute = CsvFormatAttribute.CsvFormatAttribute;
const DailyFrequencyDetails = __importStar(require("./daily-frequency-details"));
exports.DailyFrequencyDetails = DailyFrequencyDetails.DailyFrequencyDetails;
const DataAssetFromAdwcDetails = __importStar(require("./data-asset-from-adwc-details"));
exports.DataAssetFromAdwcDetails = DataAssetFromAdwcDetails.DataAssetFromAdwcDetails;
const DataAssetFromAtpDetails = __importStar(require("./data-asset-from-atp-details"));
exports.DataAssetFromAtpDetails = DataAssetFromAtpDetails.DataAssetFromAtpDetails;
const DataAssetFromJdbc = __importStar(require("./data-asset-from-jdbc"));
exports.DataAssetFromJdbc = DataAssetFromJdbc.DataAssetFromJdbc;
const DataAssetFromMySQL = __importStar(require("./data-asset-from-my-sq-l"));
exports.DataAssetFromMySQL = DataAssetFromMySQL.DataAssetFromMySQL;
const DataAssetFromObjectStorageDetails = __importStar(require("./data-asset-from-object-storage-details"));
exports.DataAssetFromObjectStorageDetails = DataAssetFromObjectStorageDetails.DataAssetFromObjectStorageDetails;
const DataAssetFromOracleDetails = __importStar(require("./data-asset-from-oracle-details"));
exports.DataAssetFromOracleDetails = DataAssetFromOracleDetails.DataAssetFromOracleDetails;
const DataAssetSummaryFromAdwc = __importStar(require("./data-asset-summary-from-adwc"));
exports.DataAssetSummaryFromAdwc = DataAssetSummaryFromAdwc.DataAssetSummaryFromAdwc;
const DataAssetSummaryFromAtp = __importStar(require("./data-asset-summary-from-atp"));
exports.DataAssetSummaryFromAtp = DataAssetSummaryFromAtp.DataAssetSummaryFromAtp;
const DataAssetSummaryFromJdbc = __importStar(require("./data-asset-summary-from-jdbc"));
exports.DataAssetSummaryFromJdbc = DataAssetSummaryFromJdbc.DataAssetSummaryFromJdbc;
const DataAssetSummaryFromMySQL = __importStar(require("./data-asset-summary-from-my-sq-l"));
exports.DataAssetSummaryFromMySQL = DataAssetSummaryFromMySQL.DataAssetSummaryFromMySQL;
const DataAssetSummaryFromObjectStorage = __importStar(require("./data-asset-summary-from-object-storage"));
exports.DataAssetSummaryFromObjectStorage = DataAssetSummaryFromObjectStorage.DataAssetSummaryFromObjectStorage;
const DataAssetSummaryFromOracle = __importStar(require("./data-asset-summary-from-oracle"));
exports.DataAssetSummaryFromOracle = DataAssetSummaryFromOracle.DataAssetSummaryFromOracle;
const DataEntityFromFile = __importStar(require("./data-entity-from-file"));
exports.DataEntityFromFile = DataEntityFromFile.DataEntityFromFile;
const DataEntityFromFileEntityDetails = __importStar(require("./data-entity-from-file-entity-details"));
exports.DataEntityFromFileEntityDetails = DataEntityFromFileEntityDetails.DataEntityFromFileEntityDetails;
const DataEntityFromTable = __importStar(require("./data-entity-from-table"));
exports.DataEntityFromTable = DataEntityFromTable.DataEntityFromTable;
const DataEntityFromTableEntityDetails = __importStar(require("./data-entity-from-table-entity-details"));
exports.DataEntityFromTableEntityDetails = DataEntityFromTableEntityDetails.DataEntityFromTableEntityDetails;
const DataEntityFromView = __importStar(require("./data-entity-from-view"));
exports.DataEntityFromView = DataEntityFromView.DataEntityFromView;
const DataEntityFromViewEntityDetails = __importStar(require("./data-entity-from-view-entity-details"));
exports.DataEntityFromViewEntityDetails = DataEntityFromViewEntityDetails.DataEntityFromViewEntityDetails;
const DataEntitySummaryFromFile = __importStar(require("./data-entity-summary-from-file"));
exports.DataEntitySummaryFromFile = DataEntitySummaryFromFile.DataEntitySummaryFromFile;
const DataEntitySummaryFromTable = __importStar(require("./data-entity-summary-from-table"));
exports.DataEntitySummaryFromTable = DataEntitySummaryFromTable.DataEntitySummaryFromTable;
const DataEntitySummaryFromView = __importStar(require("./data-entity-summary-from-view"));
exports.DataEntitySummaryFromView = DataEntitySummaryFromView.DataEntitySummaryFromView;
const DataType = __importStar(require("./data-type"));
exports.DataType = DataType.DataType;
const DerivedField = __importStar(require("./derived-field"));
exports.DerivedField = DerivedField.DerivedField;
const DerivedType = __importStar(require("./derived-type"));
exports.DerivedType = DerivedType.DerivedType;
const DirectFieldMap = __importStar(require("./direct-field-map"));
exports.DirectFieldMap = DirectFieldMap.DirectFieldMap;
const DirectNamedFieldMap = __importStar(require("./direct-named-field-map"));
exports.DirectNamedFieldMap = DirectNamedFieldMap.DirectNamedFieldMap;
const Distinct = __importStar(require("./distinct"));
exports.Distinct = Distinct.Distinct;
const DynamicInputField = __importStar(require("./dynamic-input-field"));
exports.DynamicInputField = DynamicInputField.DynamicInputField;
const DynamicProxyField = __importStar(require("./dynamic-proxy-field"));
exports.DynamicProxyField = DynamicProxyField.DynamicProxyField;
const DynamicType = __importStar(require("./dynamic-type"));
exports.DynamicType = DynamicType.DynamicType;
const EndOperator = __importStar(require("./end-operator"));
exports.EndOperator = EndOperator.EndOperator;
const EntityShapeFromFile = __importStar(require("./entity-shape-from-file"));
exports.EntityShapeFromFile = EntityShapeFromFile.EntityShapeFromFile;
const Filter = __importStar(require("./filter"));
exports.Filter = Filter.Filter;
const FilterPush = __importStar(require("./filter-push"));
exports.FilterPush = FilterPush.FilterPush;
const FlowPort = __importStar(require("./flow-port"));
exports.FlowPort = FlowPort.FlowPort;
const ForeignKey = __importStar(require("./foreign-key"));
exports.ForeignKey = ForeignKey.ForeignKey;
const HourlyFrequencyDetails = __importStar(require("./hourly-frequency-details"));
exports.HourlyFrequencyDetails = HourlyFrequencyDetails.HourlyFrequencyDetails;
const InputField = __importStar(require("./input-field"));
exports.InputField = InputField.InputField;
const InputLink = __importStar(require("./input-link"));
exports.InputLink = InputLink.InputLink;
const InputPort = __importStar(require("./input-port"));
exports.InputPort = InputPort.InputPort;
const Intersect = __importStar(require("./intersect"));
exports.Intersect = Intersect.Intersect;
const JavaType = __importStar(require("./java-type"));
exports.JavaType = JavaType.JavaType;
const Join = __importStar(require("./join"));
exports.Join = Join.Join;
const Joiner = __importStar(require("./joiner"));
exports.Joiner = Joiner.Joiner;
const JsonFormatAttribute = __importStar(require("./json-format-attribute"));
exports.JsonFormatAttribute = JsonFormatAttribute.JsonFormatAttribute;
const KeyRangePartitionConfig = __importStar(require("./key-range-partition-config"));
exports.KeyRangePartitionConfig = KeyRangePartitionConfig.KeyRangePartitionConfig;
const MacroField = __importStar(require("./macro-field"));
exports.MacroField = MacroField.MacroField;
const MergeOperator = __importStar(require("./merge-operator"));
exports.MergeOperator = MergeOperator.MergeOperator;
const Minus = __importStar(require("./minus"));
exports.Minus = Minus.Minus;
const MonthlyFrequencyDetails = __importStar(require("./monthly-frequency-details"));
exports.MonthlyFrequencyDetails = MonthlyFrequencyDetails.MonthlyFrequencyDetails;
const NameListRule = __importStar(require("./name-list-rule"));
exports.NameListRule = NameListRule.NameListRule;
const NamePatternRule = __importStar(require("./name-pattern-rule"));
exports.NamePatternRule = NamePatternRule.NamePatternRule;
const OciVaultSecretConfig = __importStar(require("./oci-vault-secret-config"));
exports.OciVaultSecretConfig = OciVaultSecretConfig.OciVaultSecretConfig;
const OracleAdwcWriteAttribute = __importStar(require("./oracle-adwc-write-attribute"));
exports.OracleAdwcWriteAttribute = OracleAdwcWriteAttribute.OracleAdwcWriteAttribute;
const OracleAtpWriteAttribute = __importStar(require("./oracle-atp-write-attribute"));
exports.OracleAtpWriteAttribute = OracleAtpWriteAttribute.OracleAtpWriteAttribute;
const OracleReadAttribute = __importStar(require("./oracle-read-attribute"));
exports.OracleReadAttribute = OracleReadAttribute.OracleReadAttribute;
const OracleWriteAttribute = __importStar(require("./oracle-write-attribute"));
exports.OracleWriteAttribute = OracleWriteAttribute.OracleWriteAttribute;
const OutputField = __importStar(require("./output-field"));
exports.OutputField = OutputField.OutputField;
const OutputLink = __importStar(require("./output-link"));
exports.OutputLink = OutputLink.OutputLink;
const OutputPort = __importStar(require("./output-port"));
exports.OutputPort = OutputPort.OutputPort;
const Parameter = __importStar(require("./parameter"));
exports.Parameter = Parameter.Parameter;
const PrimaryKey = __importStar(require("./primary-key"));
exports.PrimaryKey = PrimaryKey.PrimaryKey;
const Projection = __importStar(require("./projection"));
exports.Projection = Projection.Projection;
const ProxyField = __importStar(require("./proxy-field"));
exports.ProxyField = ProxyField.ProxyField;
const PublishedObjectFromDataLoaderTask = __importStar(require("./published-object-from-data-loader-task"));
exports.PublishedObjectFromDataLoaderTask = PublishedObjectFromDataLoaderTask.PublishedObjectFromDataLoaderTask;
const PublishedObjectFromIntegrationTask = __importStar(require("./published-object-from-integration-task"));
exports.PublishedObjectFromIntegrationTask = PublishedObjectFromIntegrationTask.PublishedObjectFromIntegrationTask;
const PublishedObjectFromPipelineTask = __importStar(require("./published-object-from-pipeline-task"));
exports.PublishedObjectFromPipelineTask = PublishedObjectFromPipelineTask.PublishedObjectFromPipelineTask;
const PublishedObjectFromPipelineTaskSummary = __importStar(require("./published-object-from-pipeline-task-summary"));
exports.PublishedObjectFromPipelineTaskSummary = PublishedObjectFromPipelineTaskSummary.PublishedObjectFromPipelineTaskSummary;
const PublishedObjectSummaryFromDataLoaderTask = __importStar(require("./published-object-summary-from-data-loader-task"));
exports.PublishedObjectSummaryFromDataLoaderTask = PublishedObjectSummaryFromDataLoaderTask.PublishedObjectSummaryFromDataLoaderTask;
const PublishedObjectSummaryFromIntegrationTask = __importStar(require("./published-object-summary-from-integration-task"));
exports.PublishedObjectSummaryFromIntegrationTask = PublishedObjectSummaryFromIntegrationTask.PublishedObjectSummaryFromIntegrationTask;
const Query = __importStar(require("./query"));
exports.Query = Query.Query;
const ReadOperationConfig = __importStar(require("./read-operation-config"));
exports.ReadOperationConfig = ReadOperationConfig.ReadOperationConfig;
const RenameRule = __importStar(require("./rename-rule"));
exports.RenameRule = RenameRule.RenameRule;
const RuleBasedFieldMap = __importStar(require("./rule-based-field-map"));
exports.RuleBasedFieldMap = RuleBasedFieldMap.RuleBasedFieldMap;
const RuleTypeConfig = __importStar(require("./rule-type-config"));
exports.RuleTypeConfig = RuleTypeConfig.RuleTypeConfig;
const Select = __importStar(require("./select"));
exports.Select = Select.Select;
const Shape = __importStar(require("./shape"));
exports.Shape = Shape.Shape;
const ShapeField = __importStar(require("./shape-field"));
exports.ShapeField = ShapeField.ShapeField;
const Sort = __importStar(require("./sort"));
exports.Sort = Sort.Sort;
const SortOper = __importStar(require("./sort-oper"));
exports.SortOper = SortOper.SortOper;
const Source = __importStar(require("./source"));
exports.Source = Source.Source;
const StartOperator = __importStar(require("./start-operator"));
exports.StartOperator = StartOperator.StartOperator;
const Target = __importStar(require("./target"));
exports.Target = Target.Target;
const TaskFromDataLoaderTaskDetails = __importStar(require("./task-from-data-loader-task-details"));
exports.TaskFromDataLoaderTaskDetails = TaskFromDataLoaderTaskDetails.TaskFromDataLoaderTaskDetails;
const TaskFromIntegrationTaskDetails = __importStar(require("./task-from-integration-task-details"));
exports.TaskFromIntegrationTaskDetails = TaskFromIntegrationTaskDetails.TaskFromIntegrationTaskDetails;
const TaskFromPipelineTaskDetails = __importStar(require("./task-from-pipeline-task-details"));
exports.TaskFromPipelineTaskDetails = TaskFromPipelineTaskDetails.TaskFromPipelineTaskDetails;
const TaskOperator = __importStar(require("./task-operator"));
exports.TaskOperator = TaskOperator.TaskOperator;
const TaskSummaryFromDataLoaderTask = __importStar(require("./task-summary-from-data-loader-task"));
exports.TaskSummaryFromDataLoaderTask = TaskSummaryFromDataLoaderTask.TaskSummaryFromDataLoaderTask;
const TaskSummaryFromIntegrationTask = __importStar(require("./task-summary-from-integration-task"));
exports.TaskSummaryFromIntegrationTask = TaskSummaryFromIntegrationTask.TaskSummaryFromIntegrationTask;
const TaskSummaryFromPipelineTask = __importStar(require("./task-summary-from-pipeline-task"));
exports.TaskSummaryFromPipelineTask = TaskSummaryFromPipelineTask.TaskSummaryFromPipelineTask;
const TypeListRule = __importStar(require("./type-list-rule"));
exports.TypeListRule = TypeListRule.TypeListRule;
const TypedNamePatternRule = __importStar(require("./typed-name-pattern-rule"));
exports.TypedNamePatternRule = TypedNamePatternRule.TypedNamePatternRule;
const Union = __importStar(require("./union"));
exports.Union = Union.Union;
const UpdateConnectionFromAdwc = __importStar(require("./update-connection-from-adwc"));
exports.UpdateConnectionFromAdwc = UpdateConnectionFromAdwc.UpdateConnectionFromAdwc;
const UpdateConnectionFromAtp = __importStar(require("./update-connection-from-atp"));
exports.UpdateConnectionFromAtp = UpdateConnectionFromAtp.UpdateConnectionFromAtp;
const UpdateConnectionFromJdbc = __importStar(require("./update-connection-from-jdbc"));
exports.UpdateConnectionFromJdbc = UpdateConnectionFromJdbc.UpdateConnectionFromJdbc;
const UpdateConnectionFromMySQL = __importStar(require("./update-connection-from-my-sq-l"));
exports.UpdateConnectionFromMySQL = UpdateConnectionFromMySQL.UpdateConnectionFromMySQL;
const UpdateConnectionFromObjectStorage = __importStar(require("./update-connection-from-object-storage"));
exports.UpdateConnectionFromObjectStorage = UpdateConnectionFromObjectStorage.UpdateConnectionFromObjectStorage;
const UpdateConnectionFromOracle = __importStar(require("./update-connection-from-oracle"));
exports.UpdateConnectionFromOracle = UpdateConnectionFromOracle.UpdateConnectionFromOracle;
const UpdateDataAssetFromAdwc = __importStar(require("./update-data-asset-from-adwc"));
exports.UpdateDataAssetFromAdwc = UpdateDataAssetFromAdwc.UpdateDataAssetFromAdwc;
const UpdateDataAssetFromAtp = __importStar(require("./update-data-asset-from-atp"));
exports.UpdateDataAssetFromAtp = UpdateDataAssetFromAtp.UpdateDataAssetFromAtp;
const UpdateDataAssetFromJdbc = __importStar(require("./update-data-asset-from-jdbc"));
exports.UpdateDataAssetFromJdbc = UpdateDataAssetFromJdbc.UpdateDataAssetFromJdbc;
const UpdateDataAssetFromMySQL = __importStar(require("./update-data-asset-from-my-sq-l"));
exports.UpdateDataAssetFromMySQL = UpdateDataAssetFromMySQL.UpdateDataAssetFromMySQL;
const UpdateDataAssetFromObjectStorage = __importStar(require("./update-data-asset-from-object-storage"));
exports.UpdateDataAssetFromObjectStorage = UpdateDataAssetFromObjectStorage.UpdateDataAssetFromObjectStorage;
const UpdateDataAssetFromOracle = __importStar(require("./update-data-asset-from-oracle"));
exports.UpdateDataAssetFromOracle = UpdateDataAssetFromOracle.UpdateDataAssetFromOracle;
const UpdateTaskFromDataLoaderTask = __importStar(require("./update-task-from-data-loader-task"));
exports.UpdateTaskFromDataLoaderTask = UpdateTaskFromDataLoaderTask.UpdateTaskFromDataLoaderTask;
const UpdateTaskFromIntegrationTask = __importStar(require("./update-task-from-integration-task"));
exports.UpdateTaskFromIntegrationTask = UpdateTaskFromIntegrationTask.UpdateTaskFromIntegrationTask;
const UpdateTaskFromPipelineTask = __importStar(require("./update-task-from-pipeline-task"));
exports.UpdateTaskFromPipelineTask = UpdateTaskFromPipelineTask.UpdateTaskFromPipelineTask;
const WriteOperationConfig = __importStar(require("./write-operation-config"));
exports.WriteOperationConfig = WriteOperationConfig.WriteOperationConfig;
