// oci-dataintegration/lib/model/patch-object-metadata.js
"use strict";
/**
 * Data Integration API
 * Use the Data Integration Service APIs to perform common extract, load, and transform (ETL) tasks.
 * OpenAPI spec version: 20200430
 * Contact: di_dis_ww_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchObjectMetadata = void 0;
var PatchObjectMetadata;
(function (PatchObjectMetadata) {
    let Type;
    (function (Type) {
        Type["IntegrationTask"] = "INTEGRATION_TASK";
        Type["DataLoaderTask"] = "DATA_LOADER_TASK";
        Type["PipelineTask"] = "PIPELINE_TASK";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Type["UnknownValue"] = "UNKNOWN_VALUE";
    })(Type = PatchObjectMetadata.Type || (PatchObjectMetadata.Type = {}));
    let Action;
    (function (Action) {
        Action["Created"] = "CREATED";
        Action["Deleted"] = "DELETED";
        Action["Updated"] = "UPDATED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Action["UnknownValue"] = "UNKNOWN_VALUE";
    })(Action = PatchObjectMetadata.Action || (PatchObjectMetadata.Action = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    PatchObjectMetadata.getJsonObj = getJsonObj;
})(PatchObjectMetadata = exports.PatchObjectMetadata || (exports.PatchObjectMetadata = {}));
