// oci-dataintegration/lib/model/reference.js
"use strict";
/**
 * Data Integration API
 * Use the Data Integration Service APIs to perform common extract, load, and transform (ETL) tasks.
 * OpenAPI spec version: 20200430
 * Contact: di_dis_ww_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Reference = void 0;
const model = __importStar(require("../model"));
var Reference;
(function (Reference) {
    let Type;
    (function (Type) {
        Type["OracleDataAsset"] = "ORACLE_DATA_ASSET";
        Type["OracleObjectStorageDataAsset"] = "ORACLE_OBJECT_STORAGE_DATA_ASSET";
        Type["OracleAtpDataAsset"] = "ORACLE_ATP_DATA_ASSET";
        Type["OracleAdwcDataAsset"] = "ORACLE_ADWC_DATA_ASSET";
        Type["MysqlDataAsset"] = "MYSQL_DATA_ASSET";
        Type["GenericJdbcDataAsset"] = "GENERIC_JDBC_DATA_ASSET";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Type["UnknownValue"] = "UNKNOWN_VALUE";
    })(Type = Reference.Type || (Reference.Type = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "usedBy": obj.usedBy
                ? obj.usedBy.map(item => {
                    return model.ReferenceUsedBy.getJsonObj(item);
                })
                : undefined,
            "childReferences": obj.childReferences
                ? obj.childReferences.map(item => {
                    return model.ChildReference.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    Reference.getJsonObj = getJsonObj;
})(Reference = exports.Reference || (exports.Reference = {}));
