// oci-integration/lib/request/list-integration-instances-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190131
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListIntegrationInstancesRequest = void 0;
var ListIntegrationInstancesRequest;
(function (ListIntegrationInstancesRequest) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Inactive"] = "INACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
    })(LifecycleState = ListIntegrationInstancesRequest.LifecycleState || (ListIntegrationInstancesRequest.LifecycleState = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListIntegrationInstancesRequest.SortOrder || (ListIntegrationInstancesRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["Timecreated"] = "TIMECREATED";
        SortBy["Displayname"] = "DISPLAYNAME";
    })(SortBy = ListIntegrationInstancesRequest.SortBy || (ListIntegrationInstancesRequest.SortBy = {}));
})(ListIntegrationInstancesRequest = exports.ListIntegrationInstancesRequest || (exports.ListIntegrationInstancesRequest = {}));
