// oci-goldengate/lib/model/deployment-type.js
"use strict";
/**
 * GoldenGate API
 * Use the Oracle Cloud Infrastructure GoldenGate APIs to perform data replication operations.

 * OpenAPI spec version: 20200407
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeploymentType = void 0;
/**
 * The deployment type.
 *
 **/
var DeploymentType;
(function (DeploymentType) {
    DeploymentType["Ogg"] = "OGG";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DeploymentType["UnknownValue"] = "UNKNOWN_VALUE";
})(DeploymentType = exports.DeploymentType || (exports.DeploymentType = {}));
(function (DeploymentType) {
    function getJsonObj(obj) {
        return { obj };
    }
    DeploymentType.getJsonObj = getJsonObj;
})(DeploymentType = exports.DeploymentType || (exports.DeploymentType = {}));
