// oci-goldengate/lib/model/index.js
"use strict";
/**
 * GoldenGate API
 * Use the Oracle Cloud Infrastructure GoldenGate APIs to perform data replication operations.

 * OpenAPI spec version: 20200407
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpgradeDeploymentCurrentReleaseDetails = exports.DefaultStopDeploymentDetails = exports.DefaultStartDeploymentDetails = exports.DefaultRestoreDeploymentDetails = exports.WorkRequestResource = exports.WorkRequestLogEntry = exports.WorkRequestError = exports.WorkRequest = exports.UpgradeDeploymentType = exports.UpgradeDeploymentDetails = exports.UpdateOggDeploymentDetails = exports.UpdateDeploymentDetails = exports.UpdateDeploymentBackupDetails = exports.UpdateDatabaseRegistrationDetails = exports.StopDeploymentType = exports.StopDeploymentDetails = exports.StartDeploymentType = exports.StartDeploymentDetails = exports.SortOrder = exports.RestoreDeploymentType = exports.RestoreDeploymentDetails = exports.OperationType = exports.OperationStatus = exports.OggDeployment = exports.LifecycleState = exports.LicenseModel = exports.DeploymentType = exports.DeploymentSummary = exports.DeploymentCollection = exports.DeploymentBackupType = exports.DeploymentBackupSummary = exports.DeploymentBackupCollection = exports.DeploymentBackup = exports.Deployment = exports.DatabaseRegistrationSummary = exports.DatabaseRegistrationCollection = exports.DatabaseRegistration = exports.CreateOggDeploymentDetails = exports.CreateDeploymentDetails = exports.CreateDeploymentBackupDetails = exports.CreateDatabaseRegistrationDetails = exports.ChangeDeploymentCompartmentDetails = exports.ChangeDeploymentBackupCompartmentDetails = exports.ChangeDatabaseRegistrationCompartmentDetails = exports.ActionType = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const ChangeDatabaseRegistrationCompartmentDetails = __importStar(require("./change-database-registration-compartment-details"));
exports.ChangeDatabaseRegistrationCompartmentDetails = ChangeDatabaseRegistrationCompartmentDetails.ChangeDatabaseRegistrationCompartmentDetails;
const ChangeDeploymentBackupCompartmentDetails = __importStar(require("./change-deployment-backup-compartment-details"));
exports.ChangeDeploymentBackupCompartmentDetails = ChangeDeploymentBackupCompartmentDetails.ChangeDeploymentBackupCompartmentDetails;
const ChangeDeploymentCompartmentDetails = __importStar(require("./change-deployment-compartment-details"));
exports.ChangeDeploymentCompartmentDetails = ChangeDeploymentCompartmentDetails.ChangeDeploymentCompartmentDetails;
const CreateDatabaseRegistrationDetails = __importStar(require("./create-database-registration-details"));
exports.CreateDatabaseRegistrationDetails = CreateDatabaseRegistrationDetails.CreateDatabaseRegistrationDetails;
const CreateDeploymentBackupDetails = __importStar(require("./create-deployment-backup-details"));
exports.CreateDeploymentBackupDetails = CreateDeploymentBackupDetails.CreateDeploymentBackupDetails;
const CreateDeploymentDetails = __importStar(require("./create-deployment-details"));
exports.CreateDeploymentDetails = CreateDeploymentDetails.CreateDeploymentDetails;
const CreateOggDeploymentDetails = __importStar(require("./create-ogg-deployment-details"));
exports.CreateOggDeploymentDetails = CreateOggDeploymentDetails.CreateOggDeploymentDetails;
const DatabaseRegistration = __importStar(require("./database-registration"));
exports.DatabaseRegistration = DatabaseRegistration.DatabaseRegistration;
const DatabaseRegistrationCollection = __importStar(require("./database-registration-collection"));
exports.DatabaseRegistrationCollection = DatabaseRegistrationCollection.DatabaseRegistrationCollection;
const DatabaseRegistrationSummary = __importStar(require("./database-registration-summary"));
exports.DatabaseRegistrationSummary = DatabaseRegistrationSummary.DatabaseRegistrationSummary;
const Deployment = __importStar(require("./deployment"));
exports.Deployment = Deployment.Deployment;
const DeploymentBackup = __importStar(require("./deployment-backup"));
exports.DeploymentBackup = DeploymentBackup.DeploymentBackup;
const DeploymentBackupCollection = __importStar(require("./deployment-backup-collection"));
exports.DeploymentBackupCollection = DeploymentBackupCollection.DeploymentBackupCollection;
const DeploymentBackupSummary = __importStar(require("./deployment-backup-summary"));
exports.DeploymentBackupSummary = DeploymentBackupSummary.DeploymentBackupSummary;
const DeploymentBackupType = __importStar(require("./deployment-backup-type"));
exports.DeploymentBackupType = DeploymentBackupType.DeploymentBackupType;
const DeploymentCollection = __importStar(require("./deployment-collection"));
exports.DeploymentCollection = DeploymentCollection.DeploymentCollection;
const DeploymentSummary = __importStar(require("./deployment-summary"));
exports.DeploymentSummary = DeploymentSummary.DeploymentSummary;
const DeploymentType = __importStar(require("./deployment-type"));
exports.DeploymentType = DeploymentType.DeploymentType;
const LicenseModel = __importStar(require("./license-model"));
exports.LicenseModel = LicenseModel.LicenseModel;
const LifecycleState = __importStar(require("./lifecycle-state"));
exports.LifecycleState = LifecycleState.LifecycleState;
const OggDeployment = __importStar(require("./ogg-deployment"));
exports.OggDeployment = OggDeployment.OggDeployment;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const RestoreDeploymentDetails = __importStar(require("./restore-deployment-details"));
exports.RestoreDeploymentDetails = RestoreDeploymentDetails.RestoreDeploymentDetails;
const RestoreDeploymentType = __importStar(require("./restore-deployment-type"));
exports.RestoreDeploymentType = RestoreDeploymentType.RestoreDeploymentType;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const StartDeploymentDetails = __importStar(require("./start-deployment-details"));
exports.StartDeploymentDetails = StartDeploymentDetails.StartDeploymentDetails;
const StartDeploymentType = __importStar(require("./start-deployment-type"));
exports.StartDeploymentType = StartDeploymentType.StartDeploymentType;
const StopDeploymentDetails = __importStar(require("./stop-deployment-details"));
exports.StopDeploymentDetails = StopDeploymentDetails.StopDeploymentDetails;
const StopDeploymentType = __importStar(require("./stop-deployment-type"));
exports.StopDeploymentType = StopDeploymentType.StopDeploymentType;
const UpdateDatabaseRegistrationDetails = __importStar(require("./update-database-registration-details"));
exports.UpdateDatabaseRegistrationDetails = UpdateDatabaseRegistrationDetails.UpdateDatabaseRegistrationDetails;
const UpdateDeploymentBackupDetails = __importStar(require("./update-deployment-backup-details"));
exports.UpdateDeploymentBackupDetails = UpdateDeploymentBackupDetails.UpdateDeploymentBackupDetails;
const UpdateDeploymentDetails = __importStar(require("./update-deployment-details"));
exports.UpdateDeploymentDetails = UpdateDeploymentDetails.UpdateDeploymentDetails;
const UpdateOggDeploymentDetails = __importStar(require("./update-ogg-deployment-details"));
exports.UpdateOggDeploymentDetails = UpdateOggDeploymentDetails.UpdateOggDeploymentDetails;
const UpgradeDeploymentDetails = __importStar(require("./upgrade-deployment-details"));
exports.UpgradeDeploymentDetails = UpgradeDeploymentDetails.UpgradeDeploymentDetails;
const UpgradeDeploymentType = __importStar(require("./upgrade-deployment-type"));
exports.UpgradeDeploymentType = UpgradeDeploymentType.UpgradeDeploymentType;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const DefaultRestoreDeploymentDetails = __importStar(require("./default-restore-deployment-details"));
exports.DefaultRestoreDeploymentDetails = DefaultRestoreDeploymentDetails.DefaultRestoreDeploymentDetails;
const DefaultStartDeploymentDetails = __importStar(require("./default-start-deployment-details"));
exports.DefaultStartDeploymentDetails = DefaultStartDeploymentDetails.DefaultStartDeploymentDetails;
const DefaultStopDeploymentDetails = __importStar(require("./default-stop-deployment-details"));
exports.DefaultStopDeploymentDetails = DefaultStopDeploymentDetails.DefaultStopDeploymentDetails;
const UpgradeDeploymentCurrentReleaseDetails = __importStar(require("./upgrade-deployment-current-release-details"));
exports.UpgradeDeploymentCurrentReleaseDetails = UpgradeDeploymentCurrentReleaseDetails.UpgradeDeploymentCurrentReleaseDetails;
