// oci-goldengate/lib/model/license-model.js
"use strict";
/**
 * GoldenGate API
 * Use the Oracle Cloud Infrastructure GoldenGate APIs to perform data replication operations.

 * OpenAPI spec version: 20200407
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LicenseModel = void 0;
/**
 * The Oracle license model that applies to a Deployment.
 *
 **/
var LicenseModel;
(function (LicenseModel) {
    LicenseModel["LicenseIncluded"] = "LICENSE_INCLUDED";
    LicenseModel["BringYourOwnLicense"] = "BRING_YOUR_OWN_LICENSE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    LicenseModel["UnknownValue"] = "UNKNOWN_VALUE";
})(LicenseModel = exports.LicenseModel || (exports.LicenseModel = {}));
(function (LicenseModel) {
    function getJsonObj(obj) {
        return { obj };
    }
    LicenseModel.getJsonObj = getJsonObj;
})(LicenseModel = exports.LicenseModel || (exports.LicenseModel = {}));
