// oci-computeinstanceagent/lib/model/instance-agent-plugin-summary.js
"use strict";
/**
 * Oracle Cloud Agent API
 * API for the Oracle Cloud Agent software running on compute instances. Oracle Cloud Agent
is a lightweight process that monitors and manages compute instances.

 * OpenAPI spec version: 20180530
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InstanceAgentPluginSummary = void 0;
var InstanceAgentPluginSummary;
(function (InstanceAgentPluginSummary) {
    let Status;
    (function (Status) {
        Status["Running"] = "RUNNING";
        Status["Stopped"] = "STOPPED";
        Status["NotSupported"] = "NOT_SUPPORTED";
        Status["Invalid"] = "INVALID";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Status["UnknownValue"] = "UNKNOWN_VALUE";
    })(Status = InstanceAgentPluginSummary.Status || (InstanceAgentPluginSummary.Status = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    InstanceAgentPluginSummary.getJsonObj = getJsonObj;
})(InstanceAgentPluginSummary = exports.InstanceAgentPluginSummary || (exports.InstanceAgentPluginSummary = {}));
