// oci-secrets/lib/model/secret-bundle-version-summary.js
"use strict";
/**
 * Secrets
 * API for retrieving secrets from vaults.
 * OpenAPI spec version: 20190301
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecretBundleVersionSummary = void 0;
var SecretBundleVersionSummary;
(function (SecretBundleVersionSummary) {
    let Stages;
    (function (Stages) {
        Stages["Current"] = "CURRENT";
        Stages["Pending"] = "PENDING";
        Stages["Latest"] = "LATEST";
        Stages["Previous"] = "PREVIOUS";
        Stages["Deprecated"] = "DEPRECATED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Stages["UnknownValue"] = "UNKNOWN_VALUE";
    })(Stages = SecretBundleVersionSummary.Stages || (SecretBundleVersionSummary.Stages = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SecretBundleVersionSummary.getJsonObj = getJsonObj;
})(SecretBundleVersionSummary = exports.SecretBundleVersionSummary || (exports.SecretBundleVersionSummary = {}));
