// oci-osmanagement/lib/model/arch-types.js
"use strict";
/**
 * OS Management API
 * API for the OS Management service. Use these API operations for working
with Managed instances and Managed instance groups.

 * OpenAPI spec version: 20190801
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ArchTypes = void 0;
/**
 * Possible CPU architecture types
 **/
var ArchTypes;
(function (ArchTypes) {
    ArchTypes["Ia32"] = "IA_32";
    ArchTypes["X8664"] = "X86_64";
    ArchTypes["Aarch64"] = "AARCH64";
    ArchTypes["Sparc"] = "SPARC";
    ArchTypes["Amd64Debian"] = "AMD64_DEBIAN";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ArchTypes["UnknownValue"] = "UNKNOWN_VALUE";
})(ArchTypes = exports.ArchTypes || (exports.ArchTypes = {}));
(function (ArchTypes) {
    function getJsonObj(obj) {
        return { obj };
    }
    ArchTypes.getJsonObj = getJsonObj;
})(ArchTypes = exports.ArchTypes || (exports.ArchTypes = {}));
