// oci-osmanagement/lib/model/checksum-types.js
"use strict";
/**
 * OS Management API
 * API for the OS Management service. Use these API operations for working
with Managed instances and Managed instance groups.

 * OpenAPI spec version: 20190801
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChecksumTypes = void 0;
/**
 * Possible checksum types
 **/
var ChecksumTypes;
(function (ChecksumTypes) {
    ChecksumTypes["Sha1"] = "SHA1";
    ChecksumTypes["Sha256"] = "SHA256";
    ChecksumTypes["Sha384"] = "SHA384";
    ChecksumTypes["Sha512"] = "SHA512";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ChecksumTypes["UnknownValue"] = "UNKNOWN_VALUE";
})(ChecksumTypes = exports.ChecksumTypes || (exports.ChecksumTypes = {}));
(function (ChecksumTypes) {
    function getJsonObj(obj) {
        return { obj };
    }
    ChecksumTypes.getJsonObj = getJsonObj;
})(ChecksumTypes = exports.ChecksumTypes || (exports.ChecksumTypes = {}));
