// oci-osmanagement/lib/model/index.js
"use strict";
/**
 * OS Management API
 * API for the OS Management service. Use these API operations for working
with Managed instances and Managed instance groups.

 * OpenAPI spec version: 20190801
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateScheduledJobDetails = exports.UpdateManagedInstanceGroupDetails = exports.SoftwareSourceSummary = exports.SoftwareSourceId = exports.SoftwareSource = exports.SoftwarePackageSummary = exports.SoftwarePackageSearchSummary = exports.SoftwarePackageFile = exports.SoftwarePackageDependency = exports.SoftwarePackage = exports.ScheduledJobSummary = exports.ScheduledJob = exports.ScheduleTypes = exports.RemovePackagesFromSoftwareSourceDetails = exports.Recurrence = exports.PackageUpdateTypes = exports.PackageName = exports.OsFamilies = exports.OperationTypes = exports.OperationStatus = exports.ManagedInstanceSummary = exports.ManagedInstanceGroupSummary = exports.ManagedInstanceGroup = exports.ManagedInstance = exports.LifecycleStates = exports.IsEligibleForInstallation = exports.IntervalTypes = exports.InstalledWindowsUpdateSummary = exports.InstalledPackageSummary = exports.InstallablePackageSummary = exports.Id = exports.ErratumSummary = exports.Erratum = exports.DetachParentSoftwareSourceFromManagedInstanceDetails = exports.DetachChildSoftwareSourceFromManagedInstanceDetails = exports.CreateSoftwareSourceDetails = exports.CreateScheduledJobDetails = exports.CreateManagedInstanceGroupDetails = exports.ChecksumTypes = exports.ChangeSoftwareSourceCompartmentDetails = exports.ChangeScheduledJobCompartmentDetails = exports.ChangeManagedInstanceGroupCompartmentDetails = exports.AvailableWindowsUpdateSummary = exports.AvailableUpdateSummary = exports.AvailableSoftwareSourceSummary = exports.AttachParentSoftwareSourceToManagedInstanceDetails = exports.AttachChildSoftwareSourceToManagedInstanceDetails = exports.ArchTypes = exports.ApiError = exports.AddPackagesToSoftwareSourceDetails = void 0;
exports.WorkRequestSummary = exports.WorkRequestResource = exports.WorkRequestLogEntry = exports.WorkRequestError = exports.WorkRequest = exports.WindowsUpdateSummary = exports.WindowsUpdate = exports.UpdateTypes = exports.UpdateSoftwareSourceDetails = void 0;
const AddPackagesToSoftwareSourceDetails = __importStar(require("./add-packages-to-software-source-details"));
exports.AddPackagesToSoftwareSourceDetails = AddPackagesToSoftwareSourceDetails.AddPackagesToSoftwareSourceDetails;
const ApiError = __importStar(require("./api-error"));
exports.ApiError = ApiError.ApiError;
const ArchTypes = __importStar(require("./arch-types"));
exports.ArchTypes = ArchTypes.ArchTypes;
const AttachChildSoftwareSourceToManagedInstanceDetails = __importStar(require("./attach-child-software-source-to-managed-instance-details"));
exports.AttachChildSoftwareSourceToManagedInstanceDetails = AttachChildSoftwareSourceToManagedInstanceDetails.AttachChildSoftwareSourceToManagedInstanceDetails;
const AttachParentSoftwareSourceToManagedInstanceDetails = __importStar(require("./attach-parent-software-source-to-managed-instance-details"));
exports.AttachParentSoftwareSourceToManagedInstanceDetails = AttachParentSoftwareSourceToManagedInstanceDetails.AttachParentSoftwareSourceToManagedInstanceDetails;
const AvailableSoftwareSourceSummary = __importStar(require("./available-software-source-summary"));
exports.AvailableSoftwareSourceSummary = AvailableSoftwareSourceSummary.AvailableSoftwareSourceSummary;
const AvailableUpdateSummary = __importStar(require("./available-update-summary"));
exports.AvailableUpdateSummary = AvailableUpdateSummary.AvailableUpdateSummary;
const AvailableWindowsUpdateSummary = __importStar(require("./available-windows-update-summary"));
exports.AvailableWindowsUpdateSummary = AvailableWindowsUpdateSummary.AvailableWindowsUpdateSummary;
const ChangeManagedInstanceGroupCompartmentDetails = __importStar(require("./change-managed-instance-group-compartment-details"));
exports.ChangeManagedInstanceGroupCompartmentDetails = ChangeManagedInstanceGroupCompartmentDetails.ChangeManagedInstanceGroupCompartmentDetails;
const ChangeScheduledJobCompartmentDetails = __importStar(require("./change-scheduled-job-compartment-details"));
exports.ChangeScheduledJobCompartmentDetails = ChangeScheduledJobCompartmentDetails.ChangeScheduledJobCompartmentDetails;
const ChangeSoftwareSourceCompartmentDetails = __importStar(require("./change-software-source-compartment-details"));
exports.ChangeSoftwareSourceCompartmentDetails = ChangeSoftwareSourceCompartmentDetails.ChangeSoftwareSourceCompartmentDetails;
const ChecksumTypes = __importStar(require("./checksum-types"));
exports.ChecksumTypes = ChecksumTypes.ChecksumTypes;
const CreateManagedInstanceGroupDetails = __importStar(require("./create-managed-instance-group-details"));
exports.CreateManagedInstanceGroupDetails = CreateManagedInstanceGroupDetails.CreateManagedInstanceGroupDetails;
const CreateScheduledJobDetails = __importStar(require("./create-scheduled-job-details"));
exports.CreateScheduledJobDetails = CreateScheduledJobDetails.CreateScheduledJobDetails;
const CreateSoftwareSourceDetails = __importStar(require("./create-software-source-details"));
exports.CreateSoftwareSourceDetails = CreateSoftwareSourceDetails.CreateSoftwareSourceDetails;
const DetachChildSoftwareSourceFromManagedInstanceDetails = __importStar(require("./detach-child-software-source-from-managed-instance-details"));
exports.DetachChildSoftwareSourceFromManagedInstanceDetails = DetachChildSoftwareSourceFromManagedInstanceDetails.DetachChildSoftwareSourceFromManagedInstanceDetails;
const DetachParentSoftwareSourceFromManagedInstanceDetails = __importStar(require("./detach-parent-software-source-from-managed-instance-details"));
exports.DetachParentSoftwareSourceFromManagedInstanceDetails = DetachParentSoftwareSourceFromManagedInstanceDetails.DetachParentSoftwareSourceFromManagedInstanceDetails;
const Erratum = __importStar(require("./erratum"));
exports.Erratum = Erratum.Erratum;
const ErratumSummary = __importStar(require("./erratum-summary"));
exports.ErratumSummary = ErratumSummary.ErratumSummary;
const Id = __importStar(require("./id"));
exports.Id = Id.Id;
const InstallablePackageSummary = __importStar(require("./installable-package-summary"));
exports.InstallablePackageSummary = InstallablePackageSummary.InstallablePackageSummary;
const InstalledPackageSummary = __importStar(require("./installed-package-summary"));
exports.InstalledPackageSummary = InstalledPackageSummary.InstalledPackageSummary;
const InstalledWindowsUpdateSummary = __importStar(require("./installed-windows-update-summary"));
exports.InstalledWindowsUpdateSummary = InstalledWindowsUpdateSummary.InstalledWindowsUpdateSummary;
const IntervalTypes = __importStar(require("./interval-types"));
exports.IntervalTypes = IntervalTypes.IntervalTypes;
const IsEligibleForInstallation = __importStar(require("./is-eligible-for-installation"));
exports.IsEligibleForInstallation = IsEligibleForInstallation.IsEligibleForInstallation;
const LifecycleStates = __importStar(require("./lifecycle-states"));
exports.LifecycleStates = LifecycleStates.LifecycleStates;
const ManagedInstance = __importStar(require("./managed-instance"));
exports.ManagedInstance = ManagedInstance.ManagedInstance;
const ManagedInstanceGroup = __importStar(require("./managed-instance-group"));
exports.ManagedInstanceGroup = ManagedInstanceGroup.ManagedInstanceGroup;
const ManagedInstanceGroupSummary = __importStar(require("./managed-instance-group-summary"));
exports.ManagedInstanceGroupSummary = ManagedInstanceGroupSummary.ManagedInstanceGroupSummary;
const ManagedInstanceSummary = __importStar(require("./managed-instance-summary"));
exports.ManagedInstanceSummary = ManagedInstanceSummary.ManagedInstanceSummary;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationTypes = __importStar(require("./operation-types"));
exports.OperationTypes = OperationTypes.OperationTypes;
const OsFamilies = __importStar(require("./os-families"));
exports.OsFamilies = OsFamilies.OsFamilies;
const PackageName = __importStar(require("./package-name"));
exports.PackageName = PackageName.PackageName;
const PackageUpdateTypes = __importStar(require("./package-update-types"));
exports.PackageUpdateTypes = PackageUpdateTypes.PackageUpdateTypes;
const Recurrence = __importStar(require("./recurrence"));
exports.Recurrence = Recurrence.Recurrence;
const RemovePackagesFromSoftwareSourceDetails = __importStar(require("./remove-packages-from-software-source-details"));
exports.RemovePackagesFromSoftwareSourceDetails = RemovePackagesFromSoftwareSourceDetails.RemovePackagesFromSoftwareSourceDetails;
const ScheduleTypes = __importStar(require("./schedule-types"));
exports.ScheduleTypes = ScheduleTypes.ScheduleTypes;
const ScheduledJob = __importStar(require("./scheduled-job"));
exports.ScheduledJob = ScheduledJob.ScheduledJob;
const ScheduledJobSummary = __importStar(require("./scheduled-job-summary"));
exports.ScheduledJobSummary = ScheduledJobSummary.ScheduledJobSummary;
const SoftwarePackage = __importStar(require("./software-package"));
exports.SoftwarePackage = SoftwarePackage.SoftwarePackage;
const SoftwarePackageDependency = __importStar(require("./software-package-dependency"));
exports.SoftwarePackageDependency = SoftwarePackageDependency.SoftwarePackageDependency;
const SoftwarePackageFile = __importStar(require("./software-package-file"));
exports.SoftwarePackageFile = SoftwarePackageFile.SoftwarePackageFile;
const SoftwarePackageSearchSummary = __importStar(require("./software-package-search-summary"));
exports.SoftwarePackageSearchSummary = SoftwarePackageSearchSummary.SoftwarePackageSearchSummary;
const SoftwarePackageSummary = __importStar(require("./software-package-summary"));
exports.SoftwarePackageSummary = SoftwarePackageSummary.SoftwarePackageSummary;
const SoftwareSource = __importStar(require("./software-source"));
exports.SoftwareSource = SoftwareSource.SoftwareSource;
const SoftwareSourceId = __importStar(require("./software-source-id"));
exports.SoftwareSourceId = SoftwareSourceId.SoftwareSourceId;
const SoftwareSourceSummary = __importStar(require("./software-source-summary"));
exports.SoftwareSourceSummary = SoftwareSourceSummary.SoftwareSourceSummary;
const UpdateManagedInstanceGroupDetails = __importStar(require("./update-managed-instance-group-details"));
exports.UpdateManagedInstanceGroupDetails = UpdateManagedInstanceGroupDetails.UpdateManagedInstanceGroupDetails;
const UpdateScheduledJobDetails = __importStar(require("./update-scheduled-job-details"));
exports.UpdateScheduledJobDetails = UpdateScheduledJobDetails.UpdateScheduledJobDetails;
const UpdateSoftwareSourceDetails = __importStar(require("./update-software-source-details"));
exports.UpdateSoftwareSourceDetails = UpdateSoftwareSourceDetails.UpdateSoftwareSourceDetails;
const UpdateTypes = __importStar(require("./update-types"));
exports.UpdateTypes = UpdateTypes.UpdateTypes;
const WindowsUpdate = __importStar(require("./windows-update"));
exports.WindowsUpdate = WindowsUpdate.WindowsUpdate;
const WindowsUpdateSummary = __importStar(require("./windows-update-summary"));
exports.WindowsUpdateSummary = WindowsUpdateSummary.WindowsUpdateSummary;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
