// oci-osmanagement/lib/model/operation-types.js
"use strict";
/**
 * OS Management API
 * API for the OS Management service. Use these API operations for working
with Managed instances and Managed instance groups.

 * OpenAPI spec version: 20190801
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationTypes = void 0;
/**
 * Possible package operation types
 **/
var OperationTypes;
(function (OperationTypes) {
    OperationTypes["Install"] = "INSTALL";
    OperationTypes["Update"] = "UPDATE";
    OperationTypes["Remove"] = "REMOVE";
    OperationTypes["Updateall"] = "UPDATEALL";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationTypes["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationTypes = exports.OperationTypes || (exports.OperationTypes = {}));
(function (OperationTypes) {
    function getJsonObj(obj) {
        return { obj };
    }
    OperationTypes.getJsonObj = getJsonObj;
})(OperationTypes = exports.OperationTypes || (exports.OperationTypes = {}));
