// oci-osmanagement/lib/model/os-families.js
"use strict";
/**
 * OS Management API
 * API for the OS Management service. Use these API operations for working
with Managed instances and Managed instance groups.

 * OpenAPI spec version: 20190801
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OsFamilies = void 0;
/**
 * Operating System types
 **/
var OsFamilies;
(function (OsFamilies) {
    OsFamilies["Linux"] = "LINUX";
    OsFamilies["Windows"] = "WINDOWS";
    OsFamilies["All"] = "ALL";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OsFamilies["UnknownValue"] = "UNKNOWN_VALUE";
})(OsFamilies = exports.OsFamilies || (exports.OsFamilies = {}));
(function (OsFamilies) {
    function getJsonObj(obj) {
        return { obj };
    }
    OsFamilies.getJsonObj = getJsonObj;
})(OsFamilies = exports.OsFamilies || (exports.OsFamilies = {}));
