// oci-osmanagement/lib/model/package-update-types.js
"use strict";
/**
 * OS Management API
 * API for the OS Management service. Use these API operations for working
with Managed instances and Managed instance groups.

 * OpenAPI spec version: 20190801
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PackageUpdateTypes = void 0;
/**
 * The purpose of an update.
 **/
var PackageUpdateTypes;
(function (PackageUpdateTypes) {
    PackageUpdateTypes["Security"] = "SECURITY";
    PackageUpdateTypes["Bugfix"] = "BUGFIX";
    PackageUpdateTypes["Enhancement"] = "ENHANCEMENT";
    PackageUpdateTypes["All"] = "ALL";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    PackageUpdateTypes["UnknownValue"] = "UNKNOWN_VALUE";
})(PackageUpdateTypes = exports.PackageUpdateTypes || (exports.PackageUpdateTypes = {}));
(function (PackageUpdateTypes) {
    function getJsonObj(obj) {
        return { obj };
    }
    PackageUpdateTypes.getJsonObj = getJsonObj;
})(PackageUpdateTypes = exports.PackageUpdateTypes || (exports.PackageUpdateTypes = {}));
