// oci-osmanagement/lib/model/recurrence.js
"use strict";
/**
 * OS Management API
 * API for the OS Management service. Use these API operations for working
with Managed instances and Managed instance groups.

 * OpenAPI spec version: 20190801
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Recurrence = void 0;
var Recurrence;
(function (Recurrence) {
    let IntervalType;
    (function (IntervalType) {
        IntervalType["Minutes"] = "MINUTES";
        IntervalType["Hours"] = "HOURS";
        IntervalType["Days"] = "DAYS";
        IntervalType["Weeks"] = "WEEKS";
    })(IntervalType = Recurrence.IntervalType || (Recurrence.IntervalType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    Recurrence.getJsonObj = getJsonObj;
})(Recurrence = exports.Recurrence || (exports.Recurrence = {}));
