// oci-osmanagement/lib/model/software-source-summary.js
"use strict";
/**
 * OS Management API
 * API for the OS Management service. Use these API operations for working
with Managed instances and Managed instance groups.

 * OpenAPI spec version: 20190801
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SoftwareSourceSummary = void 0;
var SoftwareSourceSummary;
(function (SoftwareSourceSummary) {
    let Status;
    (function (Status) {
        Status["Normal"] = "NORMAL";
        Status["Unreachable"] = "UNREACHABLE";
        Status["Error"] = "ERROR";
        Status["Warning"] = "WARNING";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Status["UnknownValue"] = "UNKNOWN_VALUE";
    })(Status = SoftwareSourceSummary.Status || (SoftwareSourceSummary.Status = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SoftwareSourceSummary.getJsonObj = getJsonObj;
})(SoftwareSourceSummary = exports.SoftwareSourceSummary || (exports.SoftwareSourceSummary = {}));
