// oci-osmanagement/lib/model/update-types.js
"use strict";
/**
 * OS Management API
 * API for the OS Management service. Use these API operations for working
with Managed instances and Managed instance groups.

 * OpenAPI spec version: 20190801
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateTypes = void 0;
/**
 * Possible update types
 **/
var UpdateTypes;
(function (UpdateTypes) {
    UpdateTypes["Security"] = "SECURITY";
    UpdateTypes["Bug"] = "BUG";
    UpdateTypes["Enhancement"] = "ENHANCEMENT";
    UpdateTypes["Other"] = "OTHER";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    UpdateTypes["UnknownValue"] = "UNKNOWN_VALUE";
})(UpdateTypes = exports.UpdateTypes || (exports.UpdateTypes = {}));
(function (UpdateTypes) {
    function getJsonObj(obj) {
        return { obj };
    }
    UpdateTypes.getJsonObj = getJsonObj;
})(UpdateTypes = exports.UpdateTypes || (exports.UpdateTypes = {}));
