// oci-budget/lib/model/reset-period.js
"use strict";
/**
 * Budgets API
 * Use the Budgets API to manage budgets and budget alerts.
 * OpenAPI spec version: 20190111
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResetPeriod = void 0;
/**
 * The reset period for the budget. Valid value is MONTHLY.
 *
 **/
var ResetPeriod;
(function (ResetPeriod) {
    ResetPeriod["Monthly"] = "MONTHLY";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ResetPeriod["UnknownValue"] = "UNKNOWN_VALUE";
})(ResetPeriod = exports.ResetPeriod || (exports.ResetPeriod = {}));
(function (ResetPeriod) {
    function getJsonObj(obj) {
        return { obj };
    }
    ResetPeriod.getJsonObj = getJsonObj;
})(ResetPeriod = exports.ResetPeriod || (exports.ResetPeriod = {}));
