// oci-budget/lib/model/sort-by.js
"use strict";
/**
 * Budgets API
 * Use the Budgets API to manage budgets and budget alerts.
 * OpenAPI spec version: 20190111
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SortBy = void 0;
/**
 * The field to sort by. If not specified, the default is timeCreated.
 * The default sort order for timeCreated is DESC.
 * The default sort order for displayName is ASC in alphanumeric order.
 *
 **/
var SortBy;
(function (SortBy) {
    SortBy["TimeCreated"] = "timeCreated";
    SortBy["DisplayName"] = "displayName";
})(SortBy = exports.SortBy || (exports.SortBy = {}));
(function (SortBy) {
    function getJsonObj(obj) {
        return { obj };
    }
    SortBy.getJsonObj = getJsonObj;
})(SortBy = exports.SortBy || (exports.SortBy = {}));
