// oci-budget/lib/model/threshold-type.js
"use strict";
/**
 * Budgets API
 * Use the Budgets API to manage budgets and budget alerts.
 * OpenAPI spec version: 20190111
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ThresholdType = void 0;
/**
 * The type of threshold. Valid values are PERCENTAGE or ABSOLUTE.
 **/
var ThresholdType;
(function (ThresholdType) {
    ThresholdType["Percentage"] = "PERCENTAGE";
    ThresholdType["Absolute"] = "ABSOLUTE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ThresholdType["UnknownValue"] = "UNKNOWN_VALUE";
})(ThresholdType = exports.ThresholdType || (exports.ThresholdType = {}));
(function (ThresholdType) {
    function getJsonObj(obj) {
        return { obj };
    }
    ThresholdType.getJsonObj = getJsonObj;
})(ThresholdType = exports.ThresholdType || (exports.ThresholdType = {}));
