// oci-filestorage/lib/request/list-snapshots-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20171215
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListSnapshotsRequest = void 0;
var ListSnapshotsRequest;
(function (ListSnapshotsRequest) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
    })(LifecycleState = ListSnapshotsRequest.LifecycleState || (ListSnapshotsRequest.LifecycleState = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListSnapshotsRequest.SortOrder || (ListSnapshotsRequest.SortOrder = {}));
})(ListSnapshotsRequest = exports.ListSnapshotsRequest || (exports.ListSnapshotsRequest = {}));
