// oci-ocvp/lib/model/action-types.js
"use strict";
/**
 * Oracle Cloud VMware Solution API
 * Use this API to manage your [Oracle Cloud VMware Solution](/iaas/Content/VMware/Concepts/ocvsoverview.htm).

 * OpenAPI spec version: 20200501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ActionTypes = void 0;
/**
 * Possible types of actions.
 **/
var ActionTypes;
(function (ActionTypes) {
    ActionTypes["Created"] = "CREATED";
    ActionTypes["Updated"] = "UPDATED";
    ActionTypes["Deleted"] = "DELETED";
    ActionTypes["InProgress"] = "IN_PROGRESS";
    ActionTypes["Related"] = "RELATED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ActionTypes["UnknownValue"] = "UNKNOWN_VALUE";
})(ActionTypes = exports.ActionTypes || (exports.ActionTypes = {}));
(function (ActionTypes) {
    function getJsonObj(obj) {
        return { obj };
    }
    ActionTypes.getJsonObj = getJsonObj;
})(ActionTypes = exports.ActionTypes || (exports.ActionTypes = {}));
