// oci-ocvp/lib/model/index.js
"use strict";
/**
 * Oracle Cloud VMware Solution API
 * Use this API to manage your [Oracle Cloud VMware Solution](/iaas/Content/VMware/Concepts/ocvsoverview.htm).

 * OpenAPI spec version: 20200501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequestCollection = exports.WorkRequest = exports.UpdateSddcDetails = exports.UpdateEsxiHostDetails = exports.SupportedVmwareSoftwareVersionSummary = exports.SupportedVmwareSoftwareVersionCollection = exports.SupportedSkuSummaryCollection = exports.SupportedSkuSummary = exports.SortOrders = exports.Sku = exports.SddcSummary = exports.SddcCollection = exports.Sddc = exports.OperationTypes = exports.OperationStatus = exports.LifecycleStates = exports.EsxiHostSummary = exports.EsxiHostCollection = exports.EsxiHost = exports.CreateSddcDetails = exports.CreateEsxiHostDetails = exports.ChangeSddcCompartmentDetails = exports.ActionTypes = void 0;
const ActionTypes = __importStar(require("./action-types"));
exports.ActionTypes = ActionTypes.ActionTypes;
const ChangeSddcCompartmentDetails = __importStar(require("./change-sddc-compartment-details"));
exports.ChangeSddcCompartmentDetails = ChangeSddcCompartmentDetails.ChangeSddcCompartmentDetails;
const CreateEsxiHostDetails = __importStar(require("./create-esxi-host-details"));
exports.CreateEsxiHostDetails = CreateEsxiHostDetails.CreateEsxiHostDetails;
const CreateSddcDetails = __importStar(require("./create-sddc-details"));
exports.CreateSddcDetails = CreateSddcDetails.CreateSddcDetails;
const EsxiHost = __importStar(require("./esxi-host"));
exports.EsxiHost = EsxiHost.EsxiHost;
const EsxiHostCollection = __importStar(require("./esxi-host-collection"));
exports.EsxiHostCollection = EsxiHostCollection.EsxiHostCollection;
const EsxiHostSummary = __importStar(require("./esxi-host-summary"));
exports.EsxiHostSummary = EsxiHostSummary.EsxiHostSummary;
const LifecycleStates = __importStar(require("./lifecycle-states"));
exports.LifecycleStates = LifecycleStates.LifecycleStates;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationTypes = __importStar(require("./operation-types"));
exports.OperationTypes = OperationTypes.OperationTypes;
const Sddc = __importStar(require("./sddc"));
exports.Sddc = Sddc.Sddc;
const SddcCollection = __importStar(require("./sddc-collection"));
exports.SddcCollection = SddcCollection.SddcCollection;
const SddcSummary = __importStar(require("./sddc-summary"));
exports.SddcSummary = SddcSummary.SddcSummary;
const Sku = __importStar(require("./sku"));
exports.Sku = Sku.Sku;
const SortOrders = __importStar(require("./sort-orders"));
exports.SortOrders = SortOrders.SortOrders;
const SupportedSkuSummary = __importStar(require("./supported-sku-summary"));
exports.SupportedSkuSummary = SupportedSkuSummary.SupportedSkuSummary;
const SupportedSkuSummaryCollection = __importStar(require("./supported-sku-summary-collection"));
exports.SupportedSkuSummaryCollection = SupportedSkuSummaryCollection.SupportedSkuSummaryCollection;
const SupportedVmwareSoftwareVersionCollection = __importStar(require("./supported-vmware-software-version-collection"));
exports.SupportedVmwareSoftwareVersionCollection = SupportedVmwareSoftwareVersionCollection.SupportedVmwareSoftwareVersionCollection;
const SupportedVmwareSoftwareVersionSummary = __importStar(require("./supported-vmware-software-version-summary"));
exports.SupportedVmwareSoftwareVersionSummary = SupportedVmwareSoftwareVersionSummary.SupportedVmwareSoftwareVersionSummary;
const UpdateEsxiHostDetails = __importStar(require("./update-esxi-host-details"));
exports.UpdateEsxiHostDetails = UpdateEsxiHostDetails.UpdateEsxiHostDetails;
const UpdateSddcDetails = __importStar(require("./update-sddc-details"));
exports.UpdateSddcDetails = UpdateSddcDetails.UpdateSddcDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestCollection = __importStar(require("./work-request-collection"));
exports.WorkRequestCollection = WorkRequestCollection.WorkRequestCollection;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
