// oci-ocvp/lib/model/operation-types.js
"use strict";
/**
 * Oracle Cloud VMware Solution API
 * Use this API to manage your [Oracle Cloud VMware Solution](/iaas/Content/VMware/Concepts/ocvsoverview.htm).

 * OpenAPI spec version: 20200501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationTypes = void 0;
/**
 * Possible asynchronous operation types.
 **/
var OperationTypes;
(function (OperationTypes) {
    OperationTypes["CreateSddc"] = "CREATE_SDDC";
    OperationTypes["DeleteSddc"] = "DELETE_SDDC";
    OperationTypes["CreateEsxiHost"] = "CREATE_ESXI_HOST";
    OperationTypes["DeleteEsxiHost"] = "DELETE_ESXI_HOST";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationTypes["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationTypes = exports.OperationTypes || (exports.OperationTypes = {}));
(function (OperationTypes) {
    function getJsonObj(obj) {
        return { obj };
    }
    OperationTypes.getJsonObj = getJsonObj;
})(OperationTypes = exports.OperationTypes || (exports.OperationTypes = {}));
