// oci-ocvp/lib/model/sku.js
"use strict";
/**
 * Oracle Cloud VMware Solution API
 * Use this API to manage your [Oracle Cloud VMware Solution](/iaas/Content/VMware/Concepts/ocvsoverview.htm).

 * OpenAPI spec version: 20200501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Sku = void 0;
/**
 * SKU offerings
 **/
var Sku;
(function (Sku) {
    Sku["Hour"] = "HOUR";
    Sku["Month"] = "MONTH";
    Sku["OneYear"] = "ONE_YEAR";
    Sku["ThreeYears"] = "THREE_YEARS";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    Sku["UnknownValue"] = "UNKNOWN_VALUE";
})(Sku = exports.Sku || (exports.Sku = {}));
(function (Sku) {
    function getJsonObj(obj) {
        return { obj };
    }
    Sku.getJsonObj = getJsonObj;
})(Sku = exports.Sku || (exports.Sku = {}));
