// oci-managementagent/lib/model/availability-status.js
"use strict";
/**
 * Management Agent API
 * API for Management Agent Cloud Service
 * OpenAPI spec version: 20200202
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AvailabilityStatus = void 0;
/**
 * Possible availability status.
 **/
var AvailabilityStatus;
(function (AvailabilityStatus) {
    AvailabilityStatus["Active"] = "ACTIVE";
    AvailabilityStatus["Silent"] = "SILENT";
    AvailabilityStatus["NotAvailable"] = "NOT_AVAILABLE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AvailabilityStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(AvailabilityStatus = exports.AvailabilityStatus || (exports.AvailabilityStatus = {}));
(function (AvailabilityStatus) {
    function getJsonObj(obj) {
        return { obj };
    }
    AvailabilityStatus.getJsonObj = getJsonObj;
})(AvailabilityStatus = exports.AvailabilityStatus || (exports.AvailabilityStatus = {}));
