// oci-managementagent/lib/model/index.js
"use strict";
/**
 * Management Agent API
 * API for Management Agent Cloud Service
 * OpenAPI spec version: 20200202
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkSubmissionKey = exports.WorkRequestSummary = exports.WorkRequestResource = exports.WorkRequestLogEntry = exports.WorkRequestError = exports.WorkRequest = exports.WorkDeliveryStatus = exports.UpdateManagementAgentInstallKeyDetails = exports.UpdateManagementAgentDetails = exports.PlatformTypes = exports.OperationTypes = exports.OperationStatus = exports.ManagementAgentSummary = exports.ManagementAgentPluginSummary = exports.ManagementAgentPluginDetails = exports.ManagementAgentPlugin = exports.ManagementAgentInstallKeySummary = exports.ManagementAgentInstallKey = exports.ManagementAgentImageSummary = exports.ManagementAgentImage = exports.ManagementAgentError = exports.ManagementAgent = exports.LifecycleStates = exports.EditModes = exports.DeployPluginsDetails = exports.Datatypes = exports.CreateManagementAgentInstallKeyDetails = exports.AvailabilityStatus = exports.AvailabilityHistorySummary = exports.ActionTypes = void 0;
const ActionTypes = __importStar(require("./action-types"));
exports.ActionTypes = ActionTypes.ActionTypes;
const AvailabilityHistorySummary = __importStar(require("./availability-history-summary"));
exports.AvailabilityHistorySummary = AvailabilityHistorySummary.AvailabilityHistorySummary;
const AvailabilityStatus = __importStar(require("./availability-status"));
exports.AvailabilityStatus = AvailabilityStatus.AvailabilityStatus;
const CreateManagementAgentInstallKeyDetails = __importStar(require("./create-management-agent-install-key-details"));
exports.CreateManagementAgentInstallKeyDetails = CreateManagementAgentInstallKeyDetails.CreateManagementAgentInstallKeyDetails;
const Datatypes = __importStar(require("./datatypes"));
exports.Datatypes = Datatypes.Datatypes;
const DeployPluginsDetails = __importStar(require("./deploy-plugins-details"));
exports.DeployPluginsDetails = DeployPluginsDetails.DeployPluginsDetails;
const EditModes = __importStar(require("./edit-modes"));
exports.EditModes = EditModes.EditModes;
const LifecycleStates = __importStar(require("./lifecycle-states"));
exports.LifecycleStates = LifecycleStates.LifecycleStates;
const ManagementAgent = __importStar(require("./management-agent"));
exports.ManagementAgent = ManagementAgent.ManagementAgent;
const ManagementAgentError = __importStar(require("./management-agent-error"));
exports.ManagementAgentError = ManagementAgentError.ManagementAgentError;
const ManagementAgentImage = __importStar(require("./management-agent-image"));
exports.ManagementAgentImage = ManagementAgentImage.ManagementAgentImage;
const ManagementAgentImageSummary = __importStar(require("./management-agent-image-summary"));
exports.ManagementAgentImageSummary = ManagementAgentImageSummary.ManagementAgentImageSummary;
const ManagementAgentInstallKey = __importStar(require("./management-agent-install-key"));
exports.ManagementAgentInstallKey = ManagementAgentInstallKey.ManagementAgentInstallKey;
const ManagementAgentInstallKeySummary = __importStar(require("./management-agent-install-key-summary"));
exports.ManagementAgentInstallKeySummary = ManagementAgentInstallKeySummary.ManagementAgentInstallKeySummary;
const ManagementAgentPlugin = __importStar(require("./management-agent-plugin"));
exports.ManagementAgentPlugin = ManagementAgentPlugin.ManagementAgentPlugin;
const ManagementAgentPluginDetails = __importStar(require("./management-agent-plugin-details"));
exports.ManagementAgentPluginDetails = ManagementAgentPluginDetails.ManagementAgentPluginDetails;
const ManagementAgentPluginSummary = __importStar(require("./management-agent-plugin-summary"));
exports.ManagementAgentPluginSummary = ManagementAgentPluginSummary.ManagementAgentPluginSummary;
const ManagementAgentSummary = __importStar(require("./management-agent-summary"));
exports.ManagementAgentSummary = ManagementAgentSummary.ManagementAgentSummary;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationTypes = __importStar(require("./operation-types"));
exports.OperationTypes = OperationTypes.OperationTypes;
const PlatformTypes = __importStar(require("./platform-types"));
exports.PlatformTypes = PlatformTypes.PlatformTypes;
const UpdateManagementAgentDetails = __importStar(require("./update-management-agent-details"));
exports.UpdateManagementAgentDetails = UpdateManagementAgentDetails.UpdateManagementAgentDetails;
const UpdateManagementAgentInstallKeyDetails = __importStar(require("./update-management-agent-install-key-details"));
exports.UpdateManagementAgentInstallKeyDetails = UpdateManagementAgentInstallKeyDetails.UpdateManagementAgentInstallKeyDetails;
const WorkDeliveryStatus = __importStar(require("./work-delivery-status"));
exports.WorkDeliveryStatus = WorkDeliveryStatus.WorkDeliveryStatus;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkSubmissionKey = __importStar(require("./work-submission-key"));
exports.WorkSubmissionKey = WorkSubmissionKey.WorkSubmissionKey;
