// oci-managementagent/lib/model/work-delivery-status.js
"use strict";
/**
 * Management Agent API
 * API for Management Agent Cloud Service
 * OpenAPI spec version: 20200202
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkDeliveryStatus = void 0;
/**
 * Possible Work Delivery states.
 **/
var WorkDeliveryStatus;
(function (WorkDeliveryStatus) {
    WorkDeliveryStatus["Accepted"] = "ACCEPTED";
    WorkDeliveryStatus["InProgress"] = "IN_PROGRESS";
    WorkDeliveryStatus["Failed"] = "FAILED";
    WorkDeliveryStatus["Succeeded"] = "SUCCEEDED";
    WorkDeliveryStatus["Canceling"] = "CANCELING";
    WorkDeliveryStatus["Canceled"] = "CANCELED";
})(WorkDeliveryStatus = exports.WorkDeliveryStatus || (exports.WorkDeliveryStatus = {}));
(function (WorkDeliveryStatus) {
    function getJsonObj(obj) {
        return { obj };
    }
    WorkDeliveryStatus.getJsonObj = getJsonObj;
})(WorkDeliveryStatus = exports.WorkDeliveryStatus || (exports.WorkDeliveryStatus = {}));
