// oci-dataflow/lib/request/list-runs-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200129
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListRunsRequest = void 0;
var ListRunsRequest;
(function (ListRunsRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["DisplayName"] = "displayName";
        SortBy["Language"] = "language";
        SortBy["RunDurationInMilliseconds"] = "runDurationInMilliseconds";
        SortBy["LifecycleState"] = "lifecycleState";
        SortBy["TotalOCpu"] = "totalOCpu";
        SortBy["DataReadInBytes"] = "dataReadInBytes";
        SortBy["DataWrittenInBytes"] = "dataWrittenInBytes";
    })(SortBy = ListRunsRequest.SortBy || (ListRunsRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListRunsRequest.SortOrder || (ListRunsRequest.SortOrder = {}));
})(ListRunsRequest = exports.ListRunsRequest || (exports.ListRunsRequest = {}));
