// oci-apmsynthetics/lib/model/index.js
"use strict";
/**
 * Application Performance Monitoring Synthetic Monitoring API
 * Use the Application Performance Monitoring Synthetic Monitoring API to query synthetic scripts and monitors.
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScriptedRestMonitorConfiguration = exports.ScriptedBrowserMonitorConfiguration = exports.RestMonitorConfiguration = exports.BrowserMonitorConfiguration = exports.VerifyText = exports.VantagePointInfo = exports.UpdateScriptDetails = exports.UpdateMonitorDetails = exports.SortOrders = exports.ScriptSummary = exports.ScriptParameterInfo = exports.ScriptParameter = exports.ScriptCollection = exports.Script = exports.RequestQueryParam = exports.RequestMethods = exports.RequestAuthenticationSchemes = exports.RequestAuthenticationDetails = exports.PublicVantagePointSummary = exports.PublicVantagePointCollection = exports.OAuthSchemes = exports.MonitorTypes = exports.MonitorSummary = exports.MonitorStatusCountMap = exports.MonitorStatus = exports.MonitorScriptParameterInfo = exports.MonitorScriptParameter = exports.MonitorResultData = exports.MonitorResult = exports.MonitorConfiguration = exports.MonitorCollection = exports.Monitor = exports.Header = exports.GeoSummary = exports.CreateScriptDetails = exports.CreateMonitorDetails = exports.ContentTypes = void 0;
const ContentTypes = __importStar(require("./content-types"));
exports.ContentTypes = ContentTypes.ContentTypes;
const CreateMonitorDetails = __importStar(require("./create-monitor-details"));
exports.CreateMonitorDetails = CreateMonitorDetails.CreateMonitorDetails;
const CreateScriptDetails = __importStar(require("./create-script-details"));
exports.CreateScriptDetails = CreateScriptDetails.CreateScriptDetails;
const GeoSummary = __importStar(require("./geo-summary"));
exports.GeoSummary = GeoSummary.GeoSummary;
const Header = __importStar(require("./header"));
exports.Header = Header.Header;
const Monitor = __importStar(require("./monitor"));
exports.Monitor = Monitor.Monitor;
const MonitorCollection = __importStar(require("./monitor-collection"));
exports.MonitorCollection = MonitorCollection.MonitorCollection;
const MonitorConfiguration = __importStar(require("./monitor-configuration"));
exports.MonitorConfiguration = MonitorConfiguration.MonitorConfiguration;
const MonitorResult = __importStar(require("./monitor-result"));
exports.MonitorResult = MonitorResult.MonitorResult;
const MonitorResultData = __importStar(require("./monitor-result-data"));
exports.MonitorResultData = MonitorResultData.MonitorResultData;
const MonitorScriptParameter = __importStar(require("./monitor-script-parameter"));
exports.MonitorScriptParameter = MonitorScriptParameter.MonitorScriptParameter;
const MonitorScriptParameterInfo = __importStar(require("./monitor-script-parameter-info"));
exports.MonitorScriptParameterInfo = MonitorScriptParameterInfo.MonitorScriptParameterInfo;
const MonitorStatus = __importStar(require("./monitor-status"));
exports.MonitorStatus = MonitorStatus.MonitorStatus;
const MonitorStatusCountMap = __importStar(require("./monitor-status-count-map"));
exports.MonitorStatusCountMap = MonitorStatusCountMap.MonitorStatusCountMap;
const MonitorSummary = __importStar(require("./monitor-summary"));
exports.MonitorSummary = MonitorSummary.MonitorSummary;
const MonitorTypes = __importStar(require("./monitor-types"));
exports.MonitorTypes = MonitorTypes.MonitorTypes;
const OAuthSchemes = __importStar(require("./o-auth-schemes"));
exports.OAuthSchemes = OAuthSchemes.OAuthSchemes;
const PublicVantagePointCollection = __importStar(require("./public-vantage-point-collection"));
exports.PublicVantagePointCollection = PublicVantagePointCollection.PublicVantagePointCollection;
const PublicVantagePointSummary = __importStar(require("./public-vantage-point-summary"));
exports.PublicVantagePointSummary = PublicVantagePointSummary.PublicVantagePointSummary;
const RequestAuthenticationDetails = __importStar(require("./request-authentication-details"));
exports.RequestAuthenticationDetails = RequestAuthenticationDetails.RequestAuthenticationDetails;
const RequestAuthenticationSchemes = __importStar(require("./request-authentication-schemes"));
exports.RequestAuthenticationSchemes = RequestAuthenticationSchemes.RequestAuthenticationSchemes;
const RequestMethods = __importStar(require("./request-methods"));
exports.RequestMethods = RequestMethods.RequestMethods;
const RequestQueryParam = __importStar(require("./request-query-param"));
exports.RequestQueryParam = RequestQueryParam.RequestQueryParam;
const Script = __importStar(require("./script"));
exports.Script = Script.Script;
const ScriptCollection = __importStar(require("./script-collection"));
exports.ScriptCollection = ScriptCollection.ScriptCollection;
const ScriptParameter = __importStar(require("./script-parameter"));
exports.ScriptParameter = ScriptParameter.ScriptParameter;
const ScriptParameterInfo = __importStar(require("./script-parameter-info"));
exports.ScriptParameterInfo = ScriptParameterInfo.ScriptParameterInfo;
const ScriptSummary = __importStar(require("./script-summary"));
exports.ScriptSummary = ScriptSummary.ScriptSummary;
const SortOrders = __importStar(require("./sort-orders"));
exports.SortOrders = SortOrders.SortOrders;
const UpdateMonitorDetails = __importStar(require("./update-monitor-details"));
exports.UpdateMonitorDetails = UpdateMonitorDetails.UpdateMonitorDetails;
const UpdateScriptDetails = __importStar(require("./update-script-details"));
exports.UpdateScriptDetails = UpdateScriptDetails.UpdateScriptDetails;
const VantagePointInfo = __importStar(require("./vantage-point-info"));
exports.VantagePointInfo = VantagePointInfo.VantagePointInfo;
const VerifyText = __importStar(require("./verify-text"));
exports.VerifyText = VerifyText.VerifyText;
const BrowserMonitorConfiguration = __importStar(require("./browser-monitor-configuration"));
exports.BrowserMonitorConfiguration = BrowserMonitorConfiguration.BrowserMonitorConfiguration;
const RestMonitorConfiguration = __importStar(require("./rest-monitor-configuration"));
exports.RestMonitorConfiguration = RestMonitorConfiguration.RestMonitorConfiguration;
const ScriptedBrowserMonitorConfiguration = __importStar(require("./scripted-browser-monitor-configuration"));
exports.ScriptedBrowserMonitorConfiguration = ScriptedBrowserMonitorConfiguration.ScriptedBrowserMonitorConfiguration;
const ScriptedRestMonitorConfiguration = __importStar(require("./scripted-rest-monitor-configuration"));
exports.ScriptedRestMonitorConfiguration = ScriptedRestMonitorConfiguration.ScriptedRestMonitorConfiguration;
