// oci-apmsynthetics/lib/model/request-methods.js
"use strict";
/**
 * Application Performance Monitoring Synthetic Monitoring API
 * Use the Application Performance Monitoring Synthetic Monitoring API to query synthetic scripts and monitors.
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RequestMethods = void 0;
/**
 * Monitor request method.
 **/
var RequestMethods;
(function (RequestMethods) {
    RequestMethods["Get"] = "GET";
    RequestMethods["Post"] = "POST";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    RequestMethods["UnknownValue"] = "UNKNOWN_VALUE";
})(RequestMethods = exports.RequestMethods || (exports.RequestMethods = {}));
(function (RequestMethods) {
    function getJsonObj(obj) {
        return { obj };
    }
    RequestMethods.getJsonObj = getJsonObj;
})(RequestMethods = exports.RequestMethods || (exports.RequestMethods = {}));
