// oci-apmsynthetics/lib/model/request-authentication-schemes.js
"use strict";
/**
 * Application Performance Monitoring Synthetic Monitoring API
 * Use the Application Performance Monitoring Synthetic Monitoring API to query synthetic scripts and monitors.
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RequestAuthenticationSchemes = void 0;
/**
 * Monitor request http auth scheme.
 **/
var RequestAuthenticationSchemes;
(function (RequestAuthenticationSchemes) {
    RequestAuthenticationSchemes["Oauth"] = "OAUTH";
    RequestAuthenticationSchemes["None"] = "NONE";
    RequestAuthenticationSchemes["Basic"] = "BASIC";
    RequestAuthenticationSchemes["Bearer"] = "BEARER";
})(RequestAuthenticationSchemes = exports.RequestAuthenticationSchemes || (exports.RequestAuthenticationSchemes = {}));
(function (RequestAuthenticationSchemes) {
    function getJsonObj(obj) {
        return { obj };
    }
    RequestAuthenticationSchemes.getJsonObj = getJsonObj;
})(RequestAuthenticationSchemes = exports.RequestAuthenticationSchemes || (exports.RequestAuthenticationSchemes = {}));
