// oci-events/lib/request/list-rules-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListRulesRequest = void 0;
var ListRulesRequest;
(function (ListRulesRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "TIME_CREATED";
        SortBy["Id"] = "ID";
        SortBy["DisplayName"] = "DISPLAY_NAME";
    })(SortBy = ListRulesRequest.SortBy || (ListRulesRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListRulesRequest.SortOrder || (ListRulesRequest.SortOrder = {}));
})(ListRulesRequest = exports.ListRulesRequest || (exports.ListRulesRequest = {}));
