// oci-waas/lib/model/access-rule-criteria.js
"use strict";
/**
 * Web Application Acceleration and Security Services API
 * OCI Web Application Acceleration and Security Services
 * OpenAPI spec version: 20181116
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccessRuleCriteria = void 0;
var AccessRuleCriteria;
(function (AccessRuleCriteria) {
    let Condition;
    (function (Condition) {
        Condition["UrlIs"] = "URL_IS";
        Condition["UrlIsNot"] = "URL_IS_NOT";
        Condition["UrlStartsWith"] = "URL_STARTS_WITH";
        Condition["UrlPartEndsWith"] = "URL_PART_ENDS_WITH";
        Condition["UrlPartContains"] = "URL_PART_CONTAINS";
        Condition["UrlRegex"] = "URL_REGEX";
        Condition["UrlDoesNotMatchRegex"] = "URL_DOES_NOT_MATCH_REGEX";
        Condition["UrlDoesNotStartWith"] = "URL_DOES_NOT_START_WITH";
        Condition["UrlPartDoesNotContain"] = "URL_PART_DOES_NOT_CONTAIN";
        Condition["UrlPartDoesNotEndWith"] = "URL_PART_DOES_NOT_END_WITH";
        Condition["IpIs"] = "IP_IS";
        Condition["IpIsNot"] = "IP_IS_NOT";
        Condition["IpInList"] = "IP_IN_LIST";
        Condition["IpNotInList"] = "IP_NOT_IN_LIST";
        Condition["HttpHeaderContains"] = "HTTP_HEADER_CONTAINS";
        Condition["HttpMethodIs"] = "HTTP_METHOD_IS";
        Condition["HttpMethodIsNot"] = "HTTP_METHOD_IS_NOT";
        Condition["CountryIs"] = "COUNTRY_IS";
        Condition["CountryIsNot"] = "COUNTRY_IS_NOT";
        Condition["UserAgentIs"] = "USER_AGENT_IS";
        Condition["UserAgentIsNot"] = "USER_AGENT_IS_NOT";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Condition["UnknownValue"] = "UNKNOWN_VALUE";
    })(Condition = AccessRuleCriteria.Condition || (AccessRuleCriteria.Condition = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AccessRuleCriteria.getJsonObj = getJsonObj;
})(AccessRuleCriteria = exports.AccessRuleCriteria || (exports.AccessRuleCriteria = {}));
