// oci-waas/lib/model/block-challenge-settings.js
"use strict";
/**
 * Web Application Acceleration and Security Services API
 * OCI Web Application Acceleration and Security Services
 * OpenAPI spec version: 20181116
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BlockChallengeSettings = void 0;
var BlockChallengeSettings;
(function (BlockChallengeSettings) {
    let BlockAction;
    (function (BlockAction) {
        BlockAction["SetResponseCode"] = "SET_RESPONSE_CODE";
        BlockAction["ShowErrorPage"] = "SHOW_ERROR_PAGE";
        BlockAction["ShowCaptcha"] = "SHOW_CAPTCHA";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        BlockAction["UnknownValue"] = "UNKNOWN_VALUE";
    })(BlockAction = BlockChallengeSettings.BlockAction || (BlockChallengeSettings.BlockAction = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    BlockChallengeSettings.getJsonObj = getJsonObj;
})(BlockChallengeSettings = exports.BlockChallengeSettings || (exports.BlockChallengeSettings = {}));
