// oci-waas/lib/model/caching-rule-criteria.js
"use strict";
/**
 * Web Application Acceleration and Security Services API
 * OCI Web Application Acceleration and Security Services
 * OpenAPI spec version: 20181116
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CachingRuleCriteria = void 0;
var CachingRuleCriteria;
(function (CachingRuleCriteria) {
    let Condition;
    (function (Condition) {
        Condition["UrlIs"] = "URL_IS";
        Condition["UrlStartsWith"] = "URL_STARTS_WITH";
        Condition["UrlPartEndsWith"] = "URL_PART_ENDS_WITH";
        Condition["UrlPartContains"] = "URL_PART_CONTAINS";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Condition["UnknownValue"] = "UNKNOWN_VALUE";
    })(Condition = CachingRuleCriteria.Condition || (CachingRuleCriteria.Condition = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CachingRuleCriteria.getJsonObj = getJsonObj;
})(CachingRuleCriteria = exports.CachingRuleCriteria || (exports.CachingRuleCriteria = {}));
