// oci-waas/lib/model/health-check.js
"use strict";
/**
 * Web Application Acceleration and Security Services API
 * OCI Web Application Acceleration and Security Services
 * OpenAPI spec version: 20181116
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.HealthCheck = void 0;
var HealthCheck;
(function (HealthCheck) {
    let Method;
    (function (Method) {
        Method["Get"] = "GET";
        Method["Head"] = "HEAD";
        Method["Post"] = "POST";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Method["UnknownValue"] = "UNKNOWN_VALUE";
    })(Method = HealthCheck.Method || (HealthCheck.Method = {}));
    let ExpectedResponseCodeGroup;
    (function (ExpectedResponseCodeGroup) {
        ExpectedResponseCodeGroup["_2Xx"] = "2XX";
        ExpectedResponseCodeGroup["_3Xx"] = "3XX";
        ExpectedResponseCodeGroup["_4Xx"] = "4XX";
        ExpectedResponseCodeGroup["_5Xx"] = "5XX";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ExpectedResponseCodeGroup["UnknownValue"] = "UNKNOWN_VALUE";
    })(ExpectedResponseCodeGroup = HealthCheck.ExpectedResponseCodeGroup || (HealthCheck.ExpectedResponseCodeGroup = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    HealthCheck.getJsonObj = getJsonObj;
})(HealthCheck = exports.HealthCheck || (exports.HealthCheck = {}));
