// oci-waas/lib/model/http-redirect-target.js
"use strict";
/**
 * Web Application Acceleration and Security Services API
 * OCI Web Application Acceleration and Security Services
 * OpenAPI spec version: 20181116
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.HttpRedirectTarget = void 0;
var HttpRedirectTarget;
(function (HttpRedirectTarget) {
    let Protocol;
    (function (Protocol) {
        Protocol["Http"] = "HTTP";
        Protocol["Https"] = "HTTPS";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Protocol["UnknownValue"] = "UNKNOWN_VALUE";
    })(Protocol = HttpRedirectTarget.Protocol || (HttpRedirectTarget.Protocol = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    HttpRedirectTarget.getJsonObj = getJsonObj;
})(HttpRedirectTarget = exports.HttpRedirectTarget || (exports.HttpRedirectTarget = {}));
