// oci-waas/lib/model/protection-rule-exclusion.js
"use strict";
/**
 * Web Application Acceleration and Security Services API
 * OCI Web Application Acceleration and Security Services
 * OpenAPI spec version: 20181116
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProtectionRuleExclusion = void 0;
var ProtectionRuleExclusion;
(function (ProtectionRuleExclusion) {
    let Target;
    (function (Target) {
        Target["RequestCookies"] = "REQUEST_COOKIES";
        Target["RequestCookieNames"] = "REQUEST_COOKIE_NAMES";
        Target["Args"] = "ARGS";
        Target["ArgsNames"] = "ARGS_NAMES";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Target["UnknownValue"] = "UNKNOWN_VALUE";
    })(Target = ProtectionRuleExclusion.Target || (ProtectionRuleExclusion.Target = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ProtectionRuleExclusion.getJsonObj = getJsonObj;
})(ProtectionRuleExclusion = exports.ProtectionRuleExclusion || (exports.ProtectionRuleExclusion = {}));
