// oci-waas/lib/model/threat-feed-action.js
"use strict";
/**
 * Web Application Acceleration and Security Services API
 * OCI Web Application Acceleration and Security Services
 * OpenAPI spec version: 20181116
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ThreatFeedAction = void 0;
var ThreatFeedAction;
(function (ThreatFeedAction) {
    let Action;
    (function (Action) {
        Action["Off"] = "OFF";
        Action["Detect"] = "DETECT";
        Action["Block"] = "BLOCK";
    })(Action = ThreatFeedAction.Action || (ThreatFeedAction.Action = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ThreatFeedAction.getJsonObj = getJsonObj;
})(ThreatFeedAction = exports.ThreatFeedAction || (exports.ThreatFeedAction = {}));
