// oci-waas/lib/model/waf-blocked-request.js
"use strict";
/**
 * Web Application Acceleration and Security Services API
 * OCI Web Application Acceleration and Security Services
 * OpenAPI spec version: 20181116
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WafBlockedRequest = void 0;
var WafBlockedRequest;
(function (WafBlockedRequest) {
    let WafFeature;
    (function (WafFeature) {
        WafFeature["ProtectionRules"] = "PROTECTION_RULES";
        WafFeature["JsChallenge"] = "JS_CHALLENGE";
        WafFeature["AccessRules"] = "ACCESS_RULES";
        WafFeature["ThreatFeeds"] = "THREAT_FEEDS";
        WafFeature["HumanInteractionChallenge"] = "HUMAN_INTERACTION_CHALLENGE";
        WafFeature["DeviceFingerprintChallenge"] = "DEVICE_FINGERPRINT_CHALLENGE";
        WafFeature["Captcha"] = "CAPTCHA";
        WafFeature["AddressRateLimiting"] = "ADDRESS_RATE_LIMITING";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        WafFeature["UnknownValue"] = "UNKNOWN_VALUE";
    })(WafFeature = WafBlockedRequest.WafFeature || (WafBlockedRequest.WafFeature = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    WafBlockedRequest.getJsonObj = getJsonObj;
})(WafBlockedRequest = exports.WafBlockedRequest || (exports.WafBlockedRequest = {}));
