// oci-waas/lib/model/work-request-status-values.js
"use strict";
/**
 * Web Application Acceleration and Security Services API
 * OCI Web Application Acceleration and Security Services
 * OpenAPI spec version: 20181116
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestStatusValues = void 0;
/**
 * The possible status values for a work request.
 **/
var WorkRequestStatusValues;
(function (WorkRequestStatusValues) {
    WorkRequestStatusValues["Accepted"] = "ACCEPTED";
    WorkRequestStatusValues["InProgress"] = "IN_PROGRESS";
    WorkRequestStatusValues["Failed"] = "FAILED";
    WorkRequestStatusValues["Succeeded"] = "SUCCEEDED";
    WorkRequestStatusValues["Canceling"] = "CANCELING";
    WorkRequestStatusValues["Canceled"] = "CANCELED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    WorkRequestStatusValues["UnknownValue"] = "UNKNOWN_VALUE";
})(WorkRequestStatusValues = exports.WorkRequestStatusValues || (exports.WorkRequestStatusValues = {}));
(function (WorkRequestStatusValues) {
    function getJsonObj(obj) {
        return { obj };
    }
    WorkRequestStatusValues.getJsonObj = getJsonObj;
})(WorkRequestStatusValues = exports.WorkRequestStatusValues || (exports.WorkRequestStatusValues = {}));
