// oci-waas/lib/model/work-request-operation-types.js
"use strict";
/**
 * Web Application Acceleration and Security Services API
 * OCI Web Application Acceleration and Security Services
 * OpenAPI spec version: 20181116
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestOperationTypes = void 0;
/**
 * An enum value indicating the operation being performed by the work request.
 **/
var WorkRequestOperationTypes;
(function (WorkRequestOperationTypes) {
    WorkRequestOperationTypes["CreateWaasPolicy"] = "CREATE_WAAS_POLICY";
    WorkRequestOperationTypes["UpdateWaasPolicy"] = "UPDATE_WAAS_POLICY";
    WorkRequestOperationTypes["DeleteWaasPolicy"] = "DELETE_WAAS_POLICY";
    WorkRequestOperationTypes["CreateHttpRedirect"] = "CREATE_HTTP_REDIRECT";
    WorkRequestOperationTypes["UpdateHttpRedirect"] = "UPDATE_HTTP_REDIRECT";
    WorkRequestOperationTypes["DeleteHttpRedirect"] = "DELETE_HTTP_REDIRECT";
    WorkRequestOperationTypes["PurgeWaasPolicyCache"] = "PURGE_WAAS_POLICY_CACHE";
    WorkRequestOperationTypes["CreateCustomProtectionRule"] = "CREATE_CUSTOM_PROTECTION_RULE";
    WorkRequestOperationTypes["UpdateCustomProtectionRule"] = "UPDATE_CUSTOM_PROTECTION_RULE";
    WorkRequestOperationTypes["DeleteCustomProtectionRule"] = "DELETE_CUSTOM_PROTECTION_RULE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    WorkRequestOperationTypes["UnknownValue"] = "UNKNOWN_VALUE";
})(WorkRequestOperationTypes = exports.WorkRequestOperationTypes || (exports.WorkRequestOperationTypes = {}));
(function (WorkRequestOperationTypes) {
    function getJsonObj(obj) {
        return { obj };
    }
    WorkRequestOperationTypes.getJsonObj = getJsonObj;
})(WorkRequestOperationTypes = exports.WorkRequestOperationTypes || (exports.WorkRequestOperationTypes = {}));
