// oci-applicationmigration/lib/model/index.js
"use strict";
/**
 * Application Migration API
 * Application Migration simplifies the migration of applications from Oracle Cloud Infrastructure Classic to Oracle Cloud Infrastructure.
You can use Application Migration API to migrate applications, such as Oracle Java Cloud Service, SOA Cloud Service, and Integration Classic
instances, to Oracle Cloud Infrastructure. For more information, see
[Overview of Application Migration](/iaas/application-migration/appmigrationoverview.htm).

 * OpenAPI spec version: 20191031
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SoacsDiscoveryDetails = exports.PcsDiscoveryDetails = exports.OicDiscoveryDetails = exports.OcicSourceDetails = exports.OcicAuthorizationTokenDetails = exports.OcicAuthorizationDetails = exports.OccSourceDetails = exports.OccAuthorizationDetails = exports.OacDiscoveryDetails = exports.JcsDiscoveryDetails = exports.InternalSourceDetails = exports.InternalAuthorizationDetails = exports.IcsDiscoveryDetails = exports.WorkRequestSummary = exports.WorkRequestResource = exports.WorkRequestLogEntry = exports.WorkRequestError = exports.WorkRequest = exports.UpdateSourceDetails = exports.UpdateMigrationDetails = exports.TargetDatabaseTypes = exports.SourceTypes = exports.SourceSummary = exports.SourceLifecycleStates = exports.SourceDetails = exports.SourceApplicationSummary = exports.SourceApplication = exports.Source = exports.SortOrders = exports.ResourceField = exports.OperationTypes = exports.OperationStatus = exports.MigrationTypes = exports.MigrationSummary = exports.MigrationStates = exports.MigrationLifecycleStates = exports.Migration = exports.DiscoveryDetails = exports.CreateSourceDetails = exports.CreateMigrationDetails = exports.ConfigurationField = exports.ChangeCompartmentDetails = exports.AuthorizationDetails = void 0;
const AuthorizationDetails = __importStar(require("./authorization-details"));
exports.AuthorizationDetails = AuthorizationDetails.AuthorizationDetails;
const ChangeCompartmentDetails = __importStar(require("./change-compartment-details"));
exports.ChangeCompartmentDetails = ChangeCompartmentDetails.ChangeCompartmentDetails;
const ConfigurationField = __importStar(require("./configuration-field"));
exports.ConfigurationField = ConfigurationField.ConfigurationField;
const CreateMigrationDetails = __importStar(require("./create-migration-details"));
exports.CreateMigrationDetails = CreateMigrationDetails.CreateMigrationDetails;
const CreateSourceDetails = __importStar(require("./create-source-details"));
exports.CreateSourceDetails = CreateSourceDetails.CreateSourceDetails;
const DiscoveryDetails = __importStar(require("./discovery-details"));
exports.DiscoveryDetails = DiscoveryDetails.DiscoveryDetails;
const Migration = __importStar(require("./migration"));
exports.Migration = Migration.Migration;
const MigrationLifecycleStates = __importStar(require("./migration-lifecycle-states"));
exports.MigrationLifecycleStates = MigrationLifecycleStates.MigrationLifecycleStates;
const MigrationStates = __importStar(require("./migration-states"));
exports.MigrationStates = MigrationStates.MigrationStates;
const MigrationSummary = __importStar(require("./migration-summary"));
exports.MigrationSummary = MigrationSummary.MigrationSummary;
const MigrationTypes = __importStar(require("./migration-types"));
exports.MigrationTypes = MigrationTypes.MigrationTypes;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationTypes = __importStar(require("./operation-types"));
exports.OperationTypes = OperationTypes.OperationTypes;
const ResourceField = __importStar(require("./resource-field"));
exports.ResourceField = ResourceField.ResourceField;
const SortOrders = __importStar(require("./sort-orders"));
exports.SortOrders = SortOrders.SortOrders;
const Source = __importStar(require("./source"));
exports.Source = Source.Source;
const SourceApplication = __importStar(require("./source-application"));
exports.SourceApplication = SourceApplication.SourceApplication;
const SourceApplicationSummary = __importStar(require("./source-application-summary"));
exports.SourceApplicationSummary = SourceApplicationSummary.SourceApplicationSummary;
const SourceDetails = __importStar(require("./source-details"));
exports.SourceDetails = SourceDetails.SourceDetails;
const SourceLifecycleStates = __importStar(require("./source-lifecycle-states"));
exports.SourceLifecycleStates = SourceLifecycleStates.SourceLifecycleStates;
const SourceSummary = __importStar(require("./source-summary"));
exports.SourceSummary = SourceSummary.SourceSummary;
const SourceTypes = __importStar(require("./source-types"));
exports.SourceTypes = SourceTypes.SourceTypes;
const TargetDatabaseTypes = __importStar(require("./target-database-types"));
exports.TargetDatabaseTypes = TargetDatabaseTypes.TargetDatabaseTypes;
const UpdateMigrationDetails = __importStar(require("./update-migration-details"));
exports.UpdateMigrationDetails = UpdateMigrationDetails.UpdateMigrationDetails;
const UpdateSourceDetails = __importStar(require("./update-source-details"));
exports.UpdateSourceDetails = UpdateSourceDetails.UpdateSourceDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const IcsDiscoveryDetails = __importStar(require("./ics-discovery-details"));
exports.IcsDiscoveryDetails = IcsDiscoveryDetails.IcsDiscoveryDetails;
const InternalAuthorizationDetails = __importStar(require("./internal-authorization-details"));
exports.InternalAuthorizationDetails = InternalAuthorizationDetails.InternalAuthorizationDetails;
const InternalSourceDetails = __importStar(require("./internal-source-details"));
exports.InternalSourceDetails = InternalSourceDetails.InternalSourceDetails;
const JcsDiscoveryDetails = __importStar(require("./jcs-discovery-details"));
exports.JcsDiscoveryDetails = JcsDiscoveryDetails.JcsDiscoveryDetails;
const OacDiscoveryDetails = __importStar(require("./oac-discovery-details"));
exports.OacDiscoveryDetails = OacDiscoveryDetails.OacDiscoveryDetails;
const OccAuthorizationDetails = __importStar(require("./occ-authorization-details"));
exports.OccAuthorizationDetails = OccAuthorizationDetails.OccAuthorizationDetails;
const OccSourceDetails = __importStar(require("./occ-source-details"));
exports.OccSourceDetails = OccSourceDetails.OccSourceDetails;
const OcicAuthorizationDetails = __importStar(require("./ocic-authorization-details"));
exports.OcicAuthorizationDetails = OcicAuthorizationDetails.OcicAuthorizationDetails;
const OcicAuthorizationTokenDetails = __importStar(require("./ocic-authorization-token-details"));
exports.OcicAuthorizationTokenDetails = OcicAuthorizationTokenDetails.OcicAuthorizationTokenDetails;
const OcicSourceDetails = __importStar(require("./ocic-source-details"));
exports.OcicSourceDetails = OcicSourceDetails.OcicSourceDetails;
const OicDiscoveryDetails = __importStar(require("./oic-discovery-details"));
exports.OicDiscoveryDetails = OicDiscoveryDetails.OicDiscoveryDetails;
const PcsDiscoveryDetails = __importStar(require("./pcs-discovery-details"));
exports.PcsDiscoveryDetails = PcsDiscoveryDetails.PcsDiscoveryDetails;
const SoacsDiscoveryDetails = __importStar(require("./soacs-discovery-details"));
exports.SoacsDiscoveryDetails = SoacsDiscoveryDetails.SoacsDiscoveryDetails;
