// oci-applicationmigration/lib/model/operation-types.js
"use strict";
/**
 * Application Migration API
 * Application Migration simplifies the migration of applications from Oracle Cloud Infrastructure Classic to Oracle Cloud Infrastructure.
You can use Application Migration API to migrate applications, such as Oracle Java Cloud Service, SOA Cloud Service, and Integration Classic
instances, to Oracle Cloud Infrastructure. For more information, see
[Overview of Application Migration](/iaas/application-migration/appmigrationoverview.htm).

 * OpenAPI spec version: 20191031
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationTypes = void 0;
/**
 * Possible operation types.
 **/
var OperationTypes;
(function (OperationTypes) {
    OperationTypes["CreateSource"] = "CREATE_SOURCE";
    OperationTypes["UpdateSource"] = "UPDATE_SOURCE";
    OperationTypes["DeleteSource"] = "DELETE_SOURCE";
    OperationTypes["CreateMigration"] = "CREATE_MIGRATION";
    OperationTypes["UpdateMigration"] = "UPDATE_MIGRATION";
    OperationTypes["DeleteMigration"] = "DELETE_MIGRATION";
    OperationTypes["AuthorizeSource"] = "AUTHORIZE_SOURCE";
    OperationTypes["DiscoverApplication"] = "DISCOVER_APPLICATION";
    OperationTypes["MigrateApplication"] = "MIGRATE_APPLICATION";
    OperationTypes["ChangeSourceCompartment"] = "CHANGE_SOURCE_COMPARTMENT";
    OperationTypes["ChangeMigrationCompartment"] = "CHANGE_MIGRATION_COMPARTMENT";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationTypes["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationTypes = exports.OperationTypes || (exports.OperationTypes = {}));
(function (OperationTypes) {
    function getJsonObj(obj) {
        return { obj };
    }
    OperationTypes.getJsonObj = getJsonObj;
})(OperationTypes = exports.OperationTypes || (exports.OperationTypes = {}));
