// oci-analytics/lib/model/analytics-instance-lifecycle-state.js
"use strict";
/**
 * Analytics API
 * Analytics API.

 * OpenAPI spec version: 20190331
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AnalyticsInstanceLifecycleState = void 0;
/**
 * The current state of an instance.
 *
 **/
var AnalyticsInstanceLifecycleState;
(function (AnalyticsInstanceLifecycleState) {
    AnalyticsInstanceLifecycleState["Active"] = "ACTIVE";
    AnalyticsInstanceLifecycleState["Creating"] = "CREATING";
    AnalyticsInstanceLifecycleState["Deleted"] = "DELETED";
    AnalyticsInstanceLifecycleState["Deleting"] = "DELETING";
    AnalyticsInstanceLifecycleState["Failed"] = "FAILED";
    AnalyticsInstanceLifecycleState["Inactive"] = "INACTIVE";
    AnalyticsInstanceLifecycleState["Updating"] = "UPDATING";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AnalyticsInstanceLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(AnalyticsInstanceLifecycleState = exports.AnalyticsInstanceLifecycleState || (exports.AnalyticsInstanceLifecycleState = {}));
(function (AnalyticsInstanceLifecycleState) {
    function getJsonObj(obj) {
        return { obj };
    }
    AnalyticsInstanceLifecycleState.getJsonObj = getJsonObj;
})(AnalyticsInstanceLifecycleState = exports.AnalyticsInstanceLifecycleState || (exports.AnalyticsInstanceLifecycleState = {}));
