// oci-analytics/lib/model/capacity-type.js
"use strict";
/**
 * Analytics API
 * Analytics API.

 * OpenAPI spec version: 20190331
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CapacityType = void 0;
/**
 * Supported capacity types.
 *
 **/
var CapacityType;
(function (CapacityType) {
    CapacityType["OlpuCount"] = "OLPU_COUNT";
    CapacityType["UserCount"] = "USER_COUNT";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    CapacityType["UnknownValue"] = "UNKNOWN_VALUE";
})(CapacityType = exports.CapacityType || (exports.CapacityType = {}));
(function (CapacityType) {
    function getJsonObj(obj) {
        return { obj };
    }
    CapacityType.getJsonObj = getJsonObj;
})(CapacityType = exports.CapacityType || (exports.CapacityType = {}));
