// oci-analytics/lib/model/feature-set.js
"use strict";
/**
 * Analytics API
 * Analytics API.

 * OpenAPI spec version: 20190331
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeatureSet = void 0;
/**
 * Analytics feature set.
 *
 **/
var FeatureSet;
(function (FeatureSet) {
    FeatureSet["SelfServiceAnalytics"] = "SELF_SERVICE_ANALYTICS";
    FeatureSet["EnterpriseAnalytics"] = "ENTERPRISE_ANALYTICS";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    FeatureSet["UnknownValue"] = "UNKNOWN_VALUE";
})(FeatureSet = exports.FeatureSet || (exports.FeatureSet = {}));
(function (FeatureSet) {
    function getJsonObj(obj) {
        return { obj };
    }
    FeatureSet.getJsonObj = getJsonObj;
})(FeatureSet = exports.FeatureSet || (exports.FeatureSet = {}));
