// oci-analytics/lib/model/index.js
"use strict";
/**
 * Analytics API
 * Analytics API.

 * OpenAPI spec version: 20190331
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicEndpointDetails = exports.PrivateEndpointDetails = exports.WorkRequestSummary = exports.WorkRequestStatus = exports.WorkRequestSortBy = exports.WorkRequestResourceType = exports.WorkRequestResource = exports.WorkRequestOperationType = exports.WorkRequestLog = exports.WorkRequestError = exports.WorkRequestActionResult = exports.WorkRequest = exports.VirtualCloudNetwork = exports.VanityUrlDetails = exports.UpdateVanityUrlDetails = exports.UpdatePrivateAccessChannelDetails = exports.UpdateAnalyticsInstanceDetails = exports.SortOrder = exports.SortBy = exports.ScaleAnalyticsInstanceDetails = exports.PrivateSourceDnsZone = exports.PrivateAccessChannel = exports.NetworkEndpointType = exports.NetworkEndpointDetails = exports.LicenseType = exports.FeatureSet = exports.CreateVanityUrlDetails = exports.CreatePrivateAccessChannelDetails = exports.CreateAnalyticsInstanceDetails = exports.ChangeCompartmentDetails = exports.ChangeAnalyticsInstanceNetworkEndpointDetails = exports.CapacityType = exports.Capacity = exports.AnalyticsInstanceSummary = exports.AnalyticsInstanceLifecycleState = exports.AnalyticsInstance = void 0;
const AnalyticsInstance = __importStar(require("./analytics-instance"));
exports.AnalyticsInstance = AnalyticsInstance.AnalyticsInstance;
const AnalyticsInstanceLifecycleState = __importStar(require("./analytics-instance-lifecycle-state"));
exports.AnalyticsInstanceLifecycleState = AnalyticsInstanceLifecycleState.AnalyticsInstanceLifecycleState;
const AnalyticsInstanceSummary = __importStar(require("./analytics-instance-summary"));
exports.AnalyticsInstanceSummary = AnalyticsInstanceSummary.AnalyticsInstanceSummary;
const Capacity = __importStar(require("./capacity"));
exports.Capacity = Capacity.Capacity;
const CapacityType = __importStar(require("./capacity-type"));
exports.CapacityType = CapacityType.CapacityType;
const ChangeAnalyticsInstanceNetworkEndpointDetails = __importStar(require("./change-analytics-instance-network-endpoint-details"));
exports.ChangeAnalyticsInstanceNetworkEndpointDetails = ChangeAnalyticsInstanceNetworkEndpointDetails.ChangeAnalyticsInstanceNetworkEndpointDetails;
const ChangeCompartmentDetails = __importStar(require("./change-compartment-details"));
exports.ChangeCompartmentDetails = ChangeCompartmentDetails.ChangeCompartmentDetails;
const CreateAnalyticsInstanceDetails = __importStar(require("./create-analytics-instance-details"));
exports.CreateAnalyticsInstanceDetails = CreateAnalyticsInstanceDetails.CreateAnalyticsInstanceDetails;
const CreatePrivateAccessChannelDetails = __importStar(require("./create-private-access-channel-details"));
exports.CreatePrivateAccessChannelDetails = CreatePrivateAccessChannelDetails.CreatePrivateAccessChannelDetails;
const CreateVanityUrlDetails = __importStar(require("./create-vanity-url-details"));
exports.CreateVanityUrlDetails = CreateVanityUrlDetails.CreateVanityUrlDetails;
const FeatureSet = __importStar(require("./feature-set"));
exports.FeatureSet = FeatureSet.FeatureSet;
const LicenseType = __importStar(require("./license-type"));
exports.LicenseType = LicenseType.LicenseType;
const NetworkEndpointDetails = __importStar(require("./network-endpoint-details"));
exports.NetworkEndpointDetails = NetworkEndpointDetails.NetworkEndpointDetails;
const NetworkEndpointType = __importStar(require("./network-endpoint-type"));
exports.NetworkEndpointType = NetworkEndpointType.NetworkEndpointType;
const PrivateAccessChannel = __importStar(require("./private-access-channel"));
exports.PrivateAccessChannel = PrivateAccessChannel.PrivateAccessChannel;
const PrivateSourceDnsZone = __importStar(require("./private-source-dns-zone"));
exports.PrivateSourceDnsZone = PrivateSourceDnsZone.PrivateSourceDnsZone;
const ScaleAnalyticsInstanceDetails = __importStar(require("./scale-analytics-instance-details"));
exports.ScaleAnalyticsInstanceDetails = ScaleAnalyticsInstanceDetails.ScaleAnalyticsInstanceDetails;
const SortBy = __importStar(require("./sort-by"));
exports.SortBy = SortBy.SortBy;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const UpdateAnalyticsInstanceDetails = __importStar(require("./update-analytics-instance-details"));
exports.UpdateAnalyticsInstanceDetails = UpdateAnalyticsInstanceDetails.UpdateAnalyticsInstanceDetails;
const UpdatePrivateAccessChannelDetails = __importStar(require("./update-private-access-channel-details"));
exports.UpdatePrivateAccessChannelDetails = UpdatePrivateAccessChannelDetails.UpdatePrivateAccessChannelDetails;
const UpdateVanityUrlDetails = __importStar(require("./update-vanity-url-details"));
exports.UpdateVanityUrlDetails = UpdateVanityUrlDetails.UpdateVanityUrlDetails;
const VanityUrlDetails = __importStar(require("./vanity-url-details"));
exports.VanityUrlDetails = VanityUrlDetails.VanityUrlDetails;
const VirtualCloudNetwork = __importStar(require("./virtual-cloud-network"));
exports.VirtualCloudNetwork = VirtualCloudNetwork.VirtualCloudNetwork;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestActionResult = __importStar(require("./work-request-action-result"));
exports.WorkRequestActionResult = WorkRequestActionResult.WorkRequestActionResult;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestLog = __importStar(require("./work-request-log"));
exports.WorkRequestLog = WorkRequestLog.WorkRequestLog;
const WorkRequestOperationType = __importStar(require("./work-request-operation-type"));
exports.WorkRequestOperationType = WorkRequestOperationType.WorkRequestOperationType;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestResourceType = __importStar(require("./work-request-resource-type"));
exports.WorkRequestResourceType = WorkRequestResourceType.WorkRequestResourceType;
const WorkRequestSortBy = __importStar(require("./work-request-sort-by"));
exports.WorkRequestSortBy = WorkRequestSortBy.WorkRequestSortBy;
const WorkRequestStatus = __importStar(require("./work-request-status"));
exports.WorkRequestStatus = WorkRequestStatus.WorkRequestStatus;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const PrivateEndpointDetails = __importStar(require("./private-endpoint-details"));
exports.PrivateEndpointDetails = PrivateEndpointDetails.PrivateEndpointDetails;
const PublicEndpointDetails = __importStar(require("./public-endpoint-details"));
exports.PublicEndpointDetails = PublicEndpointDetails.PublicEndpointDetails;
