// oci-analytics/lib/model/sort-by.js
"use strict";
/**
 * Analytics API
 * Analytics API.

 * OpenAPI spec version: 20190331
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SortBy = void 0;
/**
 * The field to sort by (one column only). Default sort order is
 * ascending exception of `timeCreated` column (descending).
 *
 **/
var SortBy;
(function (SortBy) {
    SortBy["CapacityType"] = "capacityType";
    SortBy["CapacityValue"] = "capacityValue";
    SortBy["FeatureSet"] = "featureSet";
    SortBy["LifecycleState"] = "lifecycleState";
    SortBy["Name"] = "name";
    SortBy["TimeCreated"] = "timeCreated";
})(SortBy = exports.SortBy || (exports.SortBy = {}));
(function (SortBy) {
    function getJsonObj(obj) {
        return { obj };
    }
    SortBy.getJsonObj = getJsonObj;
})(SortBy = exports.SortBy || (exports.SortBy = {}));
