// oci-analytics/lib/model/work-request-operation-type.js
"use strict";
/**
 * Analytics API
 * Analytics API.

 * OpenAPI spec version: 20190331
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestOperationType = void 0;
/**
 * The types of work request operations.
 *
 **/
var WorkRequestOperationType;
(function (WorkRequestOperationType) {
    WorkRequestOperationType["CreateAnalyticsInstance"] = "CREATE_ANALYTICS_INSTANCE";
    WorkRequestOperationType["DeleteAnalyticsInstance"] = "DELETE_ANALYTICS_INSTANCE";
    WorkRequestOperationType["StartAnalyticsInstance"] = "START_ANALYTICS_INSTANCE";
    WorkRequestOperationType["StopAnalyticsInstance"] = "STOP_ANALYTICS_INSTANCE";
    WorkRequestOperationType["ScaleAnalyticsInstance"] = "SCALE_ANALYTICS_INSTANCE";
    WorkRequestOperationType["ChangeAnalyticsInstanceCompartment"] = "CHANGE_ANALYTICS_INSTANCE_COMPARTMENT";
    WorkRequestOperationType["ChangeAnalyticsInstanceNetworkEndpoint"] = "CHANGE_ANALYTICS_INSTANCE_NETWORK_ENDPOINT";
    WorkRequestOperationType["CreateVanityUrl"] = "CREATE_VANITY_URL";
    WorkRequestOperationType["UpdateVanityUrl"] = "UPDATE_VANITY_URL";
    WorkRequestOperationType["DeleteVanityUrl"] = "DELETE_VANITY_URL";
    WorkRequestOperationType["CreatePrivateAccessChannel"] = "CREATE_PRIVATE_ACCESS_CHANNEL";
    WorkRequestOperationType["UpdatePrivateAccessChannel"] = "UPDATE_PRIVATE_ACCESS_CHANNEL";
    WorkRequestOperationType["DeletePrivateAccessChannel"] = "DELETE_PRIVATE_ACCESS_CHANNEL";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    WorkRequestOperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(WorkRequestOperationType = exports.WorkRequestOperationType || (exports.WorkRequestOperationType = {}));
(function (WorkRequestOperationType) {
    function getJsonObj(obj) {
        return { obj };
    }
    WorkRequestOperationType.getJsonObj = getJsonObj;
})(WorkRequestOperationType = exports.WorkRequestOperationType || (exports.WorkRequestOperationType = {}));
