// oci-analytics/lib/model/work-request-action-result.js
"use strict";
/**
 * Analytics API
 * Analytics API.

 * OpenAPI spec version: 20190331
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestActionResult = void 0;
/**
 * The way in which a resource was affected by a work request.
 *
 **/
var WorkRequestActionResult;
(function (WorkRequestActionResult) {
    WorkRequestActionResult["CompartmentChanged"] = "COMPARTMENT_CHANGED";
    WorkRequestActionResult["Created"] = "CREATED";
    WorkRequestActionResult["Deleted"] = "DELETED";
    WorkRequestActionResult["Started"] = "STARTED";
    WorkRequestActionResult["Stopped"] = "STOPPED";
    WorkRequestActionResult["Scaled"] = "SCALED";
    WorkRequestActionResult["NetworkEndpointChanged"] = "NETWORK_ENDPOINT_CHANGED";
    WorkRequestActionResult["VanityUrlCreated"] = "VANITY_URL_CREATED";
    WorkRequestActionResult["VanityUrlUpdated"] = "VANITY_URL_UPDATED";
    WorkRequestActionResult["VanityUrlDeleted"] = "VANITY_URL_DELETED";
    WorkRequestActionResult["PrivateAccessChannelCreated"] = "PRIVATE_ACCESS_CHANNEL_CREATED";
    WorkRequestActionResult["PrivateAccessChannelUpdated"] = "PRIVATE_ACCESS_CHANNEL_UPDATED";
    WorkRequestActionResult["PrivateAccessChannelDeleted"] = "PRIVATE_ACCESS_CHANNEL_DELETED";
    WorkRequestActionResult["None"] = "NONE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    WorkRequestActionResult["UnknownValue"] = "UNKNOWN_VALUE";
})(WorkRequestActionResult = exports.WorkRequestActionResult || (exports.WorkRequestActionResult = {}));
(function (WorkRequestActionResult) {
    function getJsonObj(obj) {
        return { obj };
    }
    WorkRequestActionResult.getJsonObj = getJsonObj;
})(WorkRequestActionResult = exports.WorkRequestActionResult || (exports.WorkRequestActionResult = {}));
