// oci-analytics/lib/model/work-request-sort-by.js
"use strict";
/**
 * Analytics API
 * Analytics API.

 * OpenAPI spec version: 20190331
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestSortBy = void 0;
/**
 * The supported work request `sort by` columns.
 *
 **/
var WorkRequestSortBy;
(function (WorkRequestSortBy) {
    WorkRequestSortBy["Id"] = "id";
    WorkRequestSortBy["OperationType"] = "operationType";
    WorkRequestSortBy["Status"] = "status";
    WorkRequestSortBy["TimeAccepted"] = "timeAccepted";
    WorkRequestSortBy["TimeStarted"] = "timeStarted";
    WorkRequestSortBy["TimeFinished"] = "timeFinished";
})(WorkRequestSortBy = exports.WorkRequestSortBy || (exports.WorkRequestSortBy = {}));
(function (WorkRequestSortBy) {
    function getJsonObj(obj) {
        return { obj };
    }
    WorkRequestSortBy.getJsonObj = getJsonObj;
})(WorkRequestSortBy = exports.WorkRequestSortBy || (exports.WorkRequestSortBy = {}));
