// oci-containerengine/lib/request/list-work-requests-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20180222
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWorkRequestsRequest = void 0;
var ListWorkRequestsRequest;
(function (ListWorkRequestsRequest) {
    let ResourceType;
    (function (ResourceType) {
        ResourceType["Cluster"] = "CLUSTER";
        ResourceType["Nodepool"] = "NODEPOOL";
    })(ResourceType = ListWorkRequestsRequest.ResourceType || (ListWorkRequestsRequest.ResourceType = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["Id"] = "ID";
        SortBy["OperationType"] = "OPERATION_TYPE";
        SortBy["Status"] = "STATUS";
        SortBy["TimeAccepted"] = "TIME_ACCEPTED";
        SortBy["TimeStarted"] = "TIME_STARTED";
        SortBy["TimeFinished"] = "TIME_FINISHED";
    })(SortBy = ListWorkRequestsRequest.SortBy || (ListWorkRequestsRequest.SortBy = {}));
})(ListWorkRequestsRequest = exports.ListWorkRequestsRequest || (exports.ListWorkRequestsRequest = {}));
