// oci-monitoring/lib/model/list-metrics-details.js
"use strict";
/**
 * Monitoring API
 * Use the Monitoring API to manage metric queries and alarms for assessing the health, capacity, and performance of your cloud resources.
Endpoints vary by operation. For PostMetric, use the `telemetry-ingestion` endpoints; for all other operations, use the `telemetry` endpoints.
For information about monitoring, see [Monitoring Overview](/iaas/Content/Monitoring/Concepts/monitoringoverview.htm).

 * OpenAPI spec version: 20180401
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListMetricsDetails = void 0;
var ListMetricsDetails;
(function (ListMetricsDetails) {
    let SortBy;
    (function (SortBy) {
        SortBy["Namespace"] = "NAMESPACE";
        SortBy["Name"] = "NAME";
        SortBy["Resourcegroup"] = "RESOURCEGROUP";
    })(SortBy = ListMetricsDetails.SortBy || (ListMetricsDetails.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListMetricsDetails.SortOrder || (ListMetricsDetails.SortOrder = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ListMetricsDetails.getJsonObj = getJsonObj;
})(ListMetricsDetails = exports.ListMetricsDetails || (exports.ListMetricsDetails = {}));
