// oci-databasemigration/lib/model/data-pump-table-exists-action.js
"use strict";
/**
 * Database Migration API
 * Use the Oracle Cloud Infrastructure Database Migration APIs to perform database migration operations.
 * OpenAPI spec version: 20200720
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataPumpTableExistsAction = void 0;
/**
 * IMPORT: Specifies the action to be performed when data is loaded into a preexisting table.
 *
 **/
var DataPumpTableExistsAction;
(function (DataPumpTableExistsAction) {
    DataPumpTableExistsAction["Truncate"] = "TRUNCATE";
    DataPumpTableExistsAction["Replace"] = "REPLACE";
    DataPumpTableExistsAction["Append"] = "APPEND";
    DataPumpTableExistsAction["Skip"] = "SKIP";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DataPumpTableExistsAction["UnknownValue"] = "UNKNOWN_VALUE";
})(DataPumpTableExistsAction = exports.DataPumpTableExistsAction || (exports.DataPumpTableExistsAction = {}));
(function (DataPumpTableExistsAction) {
    function getJsonObj(obj) {
        return { obj };
    }
    DataPumpTableExistsAction.getJsonObj = getJsonObj;
})(DataPumpTableExistsAction = exports.DataPumpTableExistsAction || (exports.DataPumpTableExistsAction = {}));
