// oci-databasemigration/lib/model/database-connection-types.js
"use strict";
/**
 * Database Migration API
 * Use the Oracle Cloud Infrastructure Database Migration APIs to perform database migration operations.
 * OpenAPI spec version: 20200720
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatabaseConnectionTypes = void 0;
/**
 * Supported database connection types
 *
 **/
var DatabaseConnectionTypes;
(function (DatabaseConnectionTypes) {
    DatabaseConnectionTypes["Manual"] = "MANUAL";
    DatabaseConnectionTypes["Autonomous"] = "AUTONOMOUS";
    DatabaseConnectionTypes["UserManagedOci"] = "USER_MANAGED_OCI";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DatabaseConnectionTypes["UnknownValue"] = "UNKNOWN_VALUE";
})(DatabaseConnectionTypes = exports.DatabaseConnectionTypes || (exports.DatabaseConnectionTypes = {}));
(function (DatabaseConnectionTypes) {
    function getJsonObj(obj) {
        return { obj };
    }
    DatabaseConnectionTypes.getJsonObj = getJsonObj;
})(DatabaseConnectionTypes = exports.DatabaseConnectionTypes || (exports.DatabaseConnectionTypes = {}));
