// oci-databasemigration/lib/model/index.js
"use strict";
/**
 * Database Migration API
 * Use the Oracle Cloud Infrastructure Database Migration APIs to perform database migration operations.
 * OpenAPI spec version: 20200720
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GoldenGateSettings = exports.GoldenGateHub = exports.GoldenGateDetails = exports.GenerateToken = exports.ExtractPerformanceProfile = exports.Extract = exports.DirectoryObject = exports.DatabaseObject = exports.DatabaseLinkDetails = exports.DatabaseConnectionTypes = exports.DataTransferMediumDetails = exports.DataPumpTableExistsAction = exports.DataPumpSettings = exports.DataPumpParameters = exports.DataPumpJobMode = exports.DataPumpExcludeParameters = exports.DataPumpEstimate = exports.CreateVaultDetails = exports.CreateSshDetails = exports.CreateReplicat = exports.CreatePrivateEndpoint = exports.CreateObjectStoreBucket = exports.CreateMigrationDetails = exports.CreateGoldenGateSettings = exports.CreateGoldenGateHub = exports.CreateGoldenGateDetails = exports.CreateExtract = exports.CreateDirectoryObject = exports.CreateDatabaseLinkDetails = exports.CreateDataTransferMediumDetails = exports.CreateDataPumpSettings = exports.CreateDataPumpParameters = exports.CreateConnectionDetails = exports.CreateConnectDescriptor = exports.CreateAgentDetails = exports.CreateAdminCredentials = exports.ConnectionSummary = exports.ConnectionCollection = exports.Connection = exports.ConnectDescriptor = exports.CloneMigrationDetails = exports.ChangeMigrationCompartmentDetails = exports.ChangeConnectionCompartmentDetails = exports.ChangeAgentCompartmentDetails = exports.AgentSummary = exports.AgentImageSummary = exports.AgentImageCollection = exports.AgentCollection = exports.Agent = exports.AdminCredentials = void 0;
exports.UpdatePrivateEndpoint = exports.UpdateObjectStoreBucket = exports.UpdateMigrationDetails = exports.UpdateJobDetails = exports.UpdateGoldenGateSettings = exports.UpdateGoldenGateHub = exports.UpdateGoldenGateDetails = exports.UpdateExtract = exports.UpdateDirectoryObject = exports.UpdateDatabaseLinkDetails = exports.UpdateDataTransferMediumDetails = exports.UpdateDataPumpSettings = exports.UpdateDataPumpParameters = exports.UpdateConnectionDetails = exports.UpdateConnectDescriptor = exports.UpdateAgentDetails = exports.UpdateAdminCredentials = exports.UnsupportedDatabaseObject = exports.StartMigrationDetails = exports.SshDetails = exports.SortOrders = exports.ResumeJobDetails = exports.Replicat = exports.PrivateEndpointDetails = exports.PhaseStatus = exports.ParLink = exports.OperationTypes = exports.OperationStatus = exports.OdmsPhaseActions = exports.OdmsJobPhases = exports.ObjectStoreBucket = exports.MigrationTypes = exports.MigrationSummary = exports.MigrationStatus = exports.MigrationPhaseSummary = exports.MigrationPhaseCollection = exports.MigrationJobProgressSummary = exports.MigrationJobProgressResource = exports.MigrationCollection = exports.Migration = exports.MetadataRemap = exports.LifecycleStates = exports.JobTypes = exports.JobSummary = exports.JobPhaseStatus = exports.JobOutputSummaryCollection = exports.JobOutputSummary = exports.JobLifecycleStates = exports.JobCollection = exports.Job = void 0;
exports.WorkRequestSummary = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequestCollection = exports.WorkRequest = exports.VaultDetails = exports.UpdateVaultDetails = exports.UpdateSshDetails = exports.UpdateReplicat = void 0;
const AdminCredentials = __importStar(require("./admin-credentials"));
exports.AdminCredentials = AdminCredentials.AdminCredentials;
const Agent = __importStar(require("./agent"));
exports.Agent = Agent.Agent;
const AgentCollection = __importStar(require("./agent-collection"));
exports.AgentCollection = AgentCollection.AgentCollection;
const AgentImageCollection = __importStar(require("./agent-image-collection"));
exports.AgentImageCollection = AgentImageCollection.AgentImageCollection;
const AgentImageSummary = __importStar(require("./agent-image-summary"));
exports.AgentImageSummary = AgentImageSummary.AgentImageSummary;
const AgentSummary = __importStar(require("./agent-summary"));
exports.AgentSummary = AgentSummary.AgentSummary;
const ChangeAgentCompartmentDetails = __importStar(require("./change-agent-compartment-details"));
exports.ChangeAgentCompartmentDetails = ChangeAgentCompartmentDetails.ChangeAgentCompartmentDetails;
const ChangeConnectionCompartmentDetails = __importStar(require("./change-connection-compartment-details"));
exports.ChangeConnectionCompartmentDetails = ChangeConnectionCompartmentDetails.ChangeConnectionCompartmentDetails;
const ChangeMigrationCompartmentDetails = __importStar(require("./change-migration-compartment-details"));
exports.ChangeMigrationCompartmentDetails = ChangeMigrationCompartmentDetails.ChangeMigrationCompartmentDetails;
const CloneMigrationDetails = __importStar(require("./clone-migration-details"));
exports.CloneMigrationDetails = CloneMigrationDetails.CloneMigrationDetails;
const ConnectDescriptor = __importStar(require("./connect-descriptor"));
exports.ConnectDescriptor = ConnectDescriptor.ConnectDescriptor;
const Connection = __importStar(require("./connection"));
exports.Connection = Connection.Connection;
const ConnectionCollection = __importStar(require("./connection-collection"));
exports.ConnectionCollection = ConnectionCollection.ConnectionCollection;
const ConnectionSummary = __importStar(require("./connection-summary"));
exports.ConnectionSummary = ConnectionSummary.ConnectionSummary;
const CreateAdminCredentials = __importStar(require("./create-admin-credentials"));
exports.CreateAdminCredentials = CreateAdminCredentials.CreateAdminCredentials;
const CreateAgentDetails = __importStar(require("./create-agent-details"));
exports.CreateAgentDetails = CreateAgentDetails.CreateAgentDetails;
const CreateConnectDescriptor = __importStar(require("./create-connect-descriptor"));
exports.CreateConnectDescriptor = CreateConnectDescriptor.CreateConnectDescriptor;
const CreateConnectionDetails = __importStar(require("./create-connection-details"));
exports.CreateConnectionDetails = CreateConnectionDetails.CreateConnectionDetails;
const CreateDataPumpParameters = __importStar(require("./create-data-pump-parameters"));
exports.CreateDataPumpParameters = CreateDataPumpParameters.CreateDataPumpParameters;
const CreateDataPumpSettings = __importStar(require("./create-data-pump-settings"));
exports.CreateDataPumpSettings = CreateDataPumpSettings.CreateDataPumpSettings;
const CreateDataTransferMediumDetails = __importStar(require("./create-data-transfer-medium-details"));
exports.CreateDataTransferMediumDetails = CreateDataTransferMediumDetails.CreateDataTransferMediumDetails;
const CreateDatabaseLinkDetails = __importStar(require("./create-database-link-details"));
exports.CreateDatabaseLinkDetails = CreateDatabaseLinkDetails.CreateDatabaseLinkDetails;
const CreateDirectoryObject = __importStar(require("./create-directory-object"));
exports.CreateDirectoryObject = CreateDirectoryObject.CreateDirectoryObject;
const CreateExtract = __importStar(require("./create-extract"));
exports.CreateExtract = CreateExtract.CreateExtract;
const CreateGoldenGateDetails = __importStar(require("./create-golden-gate-details"));
exports.CreateGoldenGateDetails = CreateGoldenGateDetails.CreateGoldenGateDetails;
const CreateGoldenGateHub = __importStar(require("./create-golden-gate-hub"));
exports.CreateGoldenGateHub = CreateGoldenGateHub.CreateGoldenGateHub;
const CreateGoldenGateSettings = __importStar(require("./create-golden-gate-settings"));
exports.CreateGoldenGateSettings = CreateGoldenGateSettings.CreateGoldenGateSettings;
const CreateMigrationDetails = __importStar(require("./create-migration-details"));
exports.CreateMigrationDetails = CreateMigrationDetails.CreateMigrationDetails;
const CreateObjectStoreBucket = __importStar(require("./create-object-store-bucket"));
exports.CreateObjectStoreBucket = CreateObjectStoreBucket.CreateObjectStoreBucket;
const CreatePrivateEndpoint = __importStar(require("./create-private-endpoint"));
exports.CreatePrivateEndpoint = CreatePrivateEndpoint.CreatePrivateEndpoint;
const CreateReplicat = __importStar(require("./create-replicat"));
exports.CreateReplicat = CreateReplicat.CreateReplicat;
const CreateSshDetails = __importStar(require("./create-ssh-details"));
exports.CreateSshDetails = CreateSshDetails.CreateSshDetails;
const CreateVaultDetails = __importStar(require("./create-vault-details"));
exports.CreateVaultDetails = CreateVaultDetails.CreateVaultDetails;
const DataPumpEstimate = __importStar(require("./data-pump-estimate"));
exports.DataPumpEstimate = DataPumpEstimate.DataPumpEstimate;
const DataPumpExcludeParameters = __importStar(require("./data-pump-exclude-parameters"));
exports.DataPumpExcludeParameters = DataPumpExcludeParameters.DataPumpExcludeParameters;
const DataPumpJobMode = __importStar(require("./data-pump-job-mode"));
exports.DataPumpJobMode = DataPumpJobMode.DataPumpJobMode;
const DataPumpParameters = __importStar(require("./data-pump-parameters"));
exports.DataPumpParameters = DataPumpParameters.DataPumpParameters;
const DataPumpSettings = __importStar(require("./data-pump-settings"));
exports.DataPumpSettings = DataPumpSettings.DataPumpSettings;
const DataPumpTableExistsAction = __importStar(require("./data-pump-table-exists-action"));
exports.DataPumpTableExistsAction = DataPumpTableExistsAction.DataPumpTableExistsAction;
const DataTransferMediumDetails = __importStar(require("./data-transfer-medium-details"));
exports.DataTransferMediumDetails = DataTransferMediumDetails.DataTransferMediumDetails;
const DatabaseConnectionTypes = __importStar(require("./database-connection-types"));
exports.DatabaseConnectionTypes = DatabaseConnectionTypes.DatabaseConnectionTypes;
const DatabaseLinkDetails = __importStar(require("./database-link-details"));
exports.DatabaseLinkDetails = DatabaseLinkDetails.DatabaseLinkDetails;
const DatabaseObject = __importStar(require("./database-object"));
exports.DatabaseObject = DatabaseObject.DatabaseObject;
const DirectoryObject = __importStar(require("./directory-object"));
exports.DirectoryObject = DirectoryObject.DirectoryObject;
const Extract = __importStar(require("./extract"));
exports.Extract = Extract.Extract;
const ExtractPerformanceProfile = __importStar(require("./extract-performance-profile"));
exports.ExtractPerformanceProfile = ExtractPerformanceProfile.ExtractPerformanceProfile;
const GenerateToken = __importStar(require("./generate-token"));
exports.GenerateToken = GenerateToken.GenerateToken;
const GoldenGateDetails = __importStar(require("./golden-gate-details"));
exports.GoldenGateDetails = GoldenGateDetails.GoldenGateDetails;
const GoldenGateHub = __importStar(require("./golden-gate-hub"));
exports.GoldenGateHub = GoldenGateHub.GoldenGateHub;
const GoldenGateSettings = __importStar(require("./golden-gate-settings"));
exports.GoldenGateSettings = GoldenGateSettings.GoldenGateSettings;
const Job = __importStar(require("./job"));
exports.Job = Job.Job;
const JobCollection = __importStar(require("./job-collection"));
exports.JobCollection = JobCollection.JobCollection;
const JobLifecycleStates = __importStar(require("./job-lifecycle-states"));
exports.JobLifecycleStates = JobLifecycleStates.JobLifecycleStates;
const JobOutputSummary = __importStar(require("./job-output-summary"));
exports.JobOutputSummary = JobOutputSummary.JobOutputSummary;
const JobOutputSummaryCollection = __importStar(require("./job-output-summary-collection"));
exports.JobOutputSummaryCollection = JobOutputSummaryCollection.JobOutputSummaryCollection;
const JobPhaseStatus = __importStar(require("./job-phase-status"));
exports.JobPhaseStatus = JobPhaseStatus.JobPhaseStatus;
const JobSummary = __importStar(require("./job-summary"));
exports.JobSummary = JobSummary.JobSummary;
const JobTypes = __importStar(require("./job-types"));
exports.JobTypes = JobTypes.JobTypes;
const LifecycleStates = __importStar(require("./lifecycle-states"));
exports.LifecycleStates = LifecycleStates.LifecycleStates;
const MetadataRemap = __importStar(require("./metadata-remap"));
exports.MetadataRemap = MetadataRemap.MetadataRemap;
const Migration = __importStar(require("./migration"));
exports.Migration = Migration.Migration;
const MigrationCollection = __importStar(require("./migration-collection"));
exports.MigrationCollection = MigrationCollection.MigrationCollection;
const MigrationJobProgressResource = __importStar(require("./migration-job-progress-resource"));
exports.MigrationJobProgressResource = MigrationJobProgressResource.MigrationJobProgressResource;
const MigrationJobProgressSummary = __importStar(require("./migration-job-progress-summary"));
exports.MigrationJobProgressSummary = MigrationJobProgressSummary.MigrationJobProgressSummary;
const MigrationPhaseCollection = __importStar(require("./migration-phase-collection"));
exports.MigrationPhaseCollection = MigrationPhaseCollection.MigrationPhaseCollection;
const MigrationPhaseSummary = __importStar(require("./migration-phase-summary"));
exports.MigrationPhaseSummary = MigrationPhaseSummary.MigrationPhaseSummary;
const MigrationStatus = __importStar(require("./migration-status"));
exports.MigrationStatus = MigrationStatus.MigrationStatus;
const MigrationSummary = __importStar(require("./migration-summary"));
exports.MigrationSummary = MigrationSummary.MigrationSummary;
const MigrationTypes = __importStar(require("./migration-types"));
exports.MigrationTypes = MigrationTypes.MigrationTypes;
const ObjectStoreBucket = __importStar(require("./object-store-bucket"));
exports.ObjectStoreBucket = ObjectStoreBucket.ObjectStoreBucket;
const OdmsJobPhases = __importStar(require("./odms-job-phases"));
exports.OdmsJobPhases = OdmsJobPhases.OdmsJobPhases;
const OdmsPhaseActions = __importStar(require("./odms-phase-actions"));
exports.OdmsPhaseActions = OdmsPhaseActions.OdmsPhaseActions;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationTypes = __importStar(require("./operation-types"));
exports.OperationTypes = OperationTypes.OperationTypes;
const ParLink = __importStar(require("./par-link"));
exports.ParLink = ParLink.ParLink;
const PhaseStatus = __importStar(require("./phase-status"));
exports.PhaseStatus = PhaseStatus.PhaseStatus;
const PrivateEndpointDetails = __importStar(require("./private-endpoint-details"));
exports.PrivateEndpointDetails = PrivateEndpointDetails.PrivateEndpointDetails;
const Replicat = __importStar(require("./replicat"));
exports.Replicat = Replicat.Replicat;
const ResumeJobDetails = __importStar(require("./resume-job-details"));
exports.ResumeJobDetails = ResumeJobDetails.ResumeJobDetails;
const SortOrders = __importStar(require("./sort-orders"));
exports.SortOrders = SortOrders.SortOrders;
const SshDetails = __importStar(require("./ssh-details"));
exports.SshDetails = SshDetails.SshDetails;
const StartMigrationDetails = __importStar(require("./start-migration-details"));
exports.StartMigrationDetails = StartMigrationDetails.StartMigrationDetails;
const UnsupportedDatabaseObject = __importStar(require("./unsupported-database-object"));
exports.UnsupportedDatabaseObject = UnsupportedDatabaseObject.UnsupportedDatabaseObject;
const UpdateAdminCredentials = __importStar(require("./update-admin-credentials"));
exports.UpdateAdminCredentials = UpdateAdminCredentials.UpdateAdminCredentials;
const UpdateAgentDetails = __importStar(require("./update-agent-details"));
exports.UpdateAgentDetails = UpdateAgentDetails.UpdateAgentDetails;
const UpdateConnectDescriptor = __importStar(require("./update-connect-descriptor"));
exports.UpdateConnectDescriptor = UpdateConnectDescriptor.UpdateConnectDescriptor;
const UpdateConnectionDetails = __importStar(require("./update-connection-details"));
exports.UpdateConnectionDetails = UpdateConnectionDetails.UpdateConnectionDetails;
const UpdateDataPumpParameters = __importStar(require("./update-data-pump-parameters"));
exports.UpdateDataPumpParameters = UpdateDataPumpParameters.UpdateDataPumpParameters;
const UpdateDataPumpSettings = __importStar(require("./update-data-pump-settings"));
exports.UpdateDataPumpSettings = UpdateDataPumpSettings.UpdateDataPumpSettings;
const UpdateDataTransferMediumDetails = __importStar(require("./update-data-transfer-medium-details"));
exports.UpdateDataTransferMediumDetails = UpdateDataTransferMediumDetails.UpdateDataTransferMediumDetails;
const UpdateDatabaseLinkDetails = __importStar(require("./update-database-link-details"));
exports.UpdateDatabaseLinkDetails = UpdateDatabaseLinkDetails.UpdateDatabaseLinkDetails;
const UpdateDirectoryObject = __importStar(require("./update-directory-object"));
exports.UpdateDirectoryObject = UpdateDirectoryObject.UpdateDirectoryObject;
const UpdateExtract = __importStar(require("./update-extract"));
exports.UpdateExtract = UpdateExtract.UpdateExtract;
const UpdateGoldenGateDetails = __importStar(require("./update-golden-gate-details"));
exports.UpdateGoldenGateDetails = UpdateGoldenGateDetails.UpdateGoldenGateDetails;
const UpdateGoldenGateHub = __importStar(require("./update-golden-gate-hub"));
exports.UpdateGoldenGateHub = UpdateGoldenGateHub.UpdateGoldenGateHub;
const UpdateGoldenGateSettings = __importStar(require("./update-golden-gate-settings"));
exports.UpdateGoldenGateSettings = UpdateGoldenGateSettings.UpdateGoldenGateSettings;
const UpdateJobDetails = __importStar(require("./update-job-details"));
exports.UpdateJobDetails = UpdateJobDetails.UpdateJobDetails;
const UpdateMigrationDetails = __importStar(require("./update-migration-details"));
exports.UpdateMigrationDetails = UpdateMigrationDetails.UpdateMigrationDetails;
const UpdateObjectStoreBucket = __importStar(require("./update-object-store-bucket"));
exports.UpdateObjectStoreBucket = UpdateObjectStoreBucket.UpdateObjectStoreBucket;
const UpdatePrivateEndpoint = __importStar(require("./update-private-endpoint"));
exports.UpdatePrivateEndpoint = UpdatePrivateEndpoint.UpdatePrivateEndpoint;
const UpdateReplicat = __importStar(require("./update-replicat"));
exports.UpdateReplicat = UpdateReplicat.UpdateReplicat;
const UpdateSshDetails = __importStar(require("./update-ssh-details"));
exports.UpdateSshDetails = UpdateSshDetails.UpdateSshDetails;
const UpdateVaultDetails = __importStar(require("./update-vault-details"));
exports.UpdateVaultDetails = UpdateVaultDetails.UpdateVaultDetails;
const VaultDetails = __importStar(require("./vault-details"));
exports.VaultDetails = VaultDetails.VaultDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestCollection = __importStar(require("./work-request-collection"));
exports.WorkRequestCollection = WorkRequestCollection.WorkRequestCollection;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
