// oci-databasemigration/lib/model/job-phase-status.js
"use strict";
/**
 * Database Migration API
 * Use the Oracle Cloud Infrastructure Database Migration APIs to perform database migration operations.
 * OpenAPI spec version: 20200720
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.JobPhaseStatus = void 0;
/**
 * Job Phase status.
 *
 **/
var JobPhaseStatus;
(function (JobPhaseStatus) {
    JobPhaseStatus["Pending"] = "PENDING";
    JobPhaseStatus["Started"] = "STARTED";
    JobPhaseStatus["Completed"] = "COMPLETED";
    JobPhaseStatus["Failed"] = "FAILED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    JobPhaseStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(JobPhaseStatus = exports.JobPhaseStatus || (exports.JobPhaseStatus = {}));
(function (JobPhaseStatus) {
    function getJsonObj(obj) {
        return { obj };
    }
    JobPhaseStatus.getJsonObj = getJsonObj;
})(JobPhaseStatus = exports.JobPhaseStatus || (exports.JobPhaseStatus = {}));
