// oci-databasemigration/lib/model/lifecycle-states.js
"use strict";
/**
 * Database Migration API
 * Use the Oracle Cloud Infrastructure Database Migration APIs to perform database migration operations.
 * OpenAPI spec version: 20200720
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LifecycleStates = void 0;
/**
 * Possible lifecycle states.
 *
 **/
var LifecycleStates;
(function (LifecycleStates) {
    LifecycleStates["Creating"] = "CREATING";
    LifecycleStates["Updating"] = "UPDATING";
    LifecycleStates["Active"] = "ACTIVE";
    LifecycleStates["Inactive"] = "INACTIVE";
    LifecycleStates["Deleting"] = "DELETING";
    LifecycleStates["Deleted"] = "DELETED";
    LifecycleStates["Failed"] = "FAILED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    LifecycleStates["UnknownValue"] = "UNKNOWN_VALUE";
})(LifecycleStates = exports.LifecycleStates || (exports.LifecycleStates = {}));
(function (LifecycleStates) {
    function getJsonObj(obj) {
        return { obj };
    }
    LifecycleStates.getJsonObj = getJsonObj;
})(LifecycleStates = exports.LifecycleStates || (exports.LifecycleStates = {}));
