// oci-databasemigration/lib/model/migration-status.js
"use strict";
/**
 * Database Migration API
 * Use the Oracle Cloud Infrastructure Database Migration APIs to perform database migration operations.
 * OpenAPI spec version: 20200720
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MigrationStatus = void 0;
/**
 * Possible Migration status.
 *
 **/
var MigrationStatus;
(function (MigrationStatus) {
    MigrationStatus["Ready"] = "READY";
    MigrationStatus["Aborting"] = "ABORTING";
    MigrationStatus["Validating"] = "VALIDATING";
    MigrationStatus["Validated"] = "VALIDATED";
    MigrationStatus["Waiting"] = "WAITING";
    MigrationStatus["Migrating"] = "MIGRATING";
    MigrationStatus["Done"] = "DONE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    MigrationStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(MigrationStatus = exports.MigrationStatus || (exports.MigrationStatus = {}));
(function (MigrationStatus) {
    function getJsonObj(obj) {
        return { obj };
    }
    MigrationStatus.getJsonObj = getJsonObj;
})(MigrationStatus = exports.MigrationStatus || (exports.MigrationStatus = {}));
