// oci-databasemigration/lib/model/odms-job-phases.js
"use strict";
/**
 * Database Migration API
 * Use the Oracle Cloud Infrastructure Database Migration APIs to perform database migration operations.
 * OpenAPI spec version: 20200720
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OdmsJobPhases = void 0;
/**
 * Possible ODMS Job Phases.
 *
 **/
var OdmsJobPhases;
(function (OdmsJobPhases) {
    OdmsJobPhases["OdmsValidateTgt"] = "ODMS_VALIDATE_TGT";
    OdmsJobPhases["OdmsValidateSrc"] = "ODMS_VALIDATE_SRC";
    OdmsJobPhases["OdmsValidateGgHub"] = "ODMS_VALIDATE_GG_HUB";
    OdmsJobPhases["OdmsValidateDatapumpSettings"] = "ODMS_VALIDATE_DATAPUMP_SETTINGS";
    OdmsJobPhases["OdmsValidateDatapumpSettingsSrc"] = "ODMS_VALIDATE_DATAPUMP_SETTINGS_SRC";
    OdmsJobPhases["OdmsValidateDatapumpSettingsTgt"] = "ODMS_VALIDATE_DATAPUMP_SETTINGS_TGT";
    OdmsJobPhases["OdmsValidate"] = "ODMS_VALIDATE";
    OdmsJobPhases["OdmsPrepare"] = "ODMS_PREPARE";
    OdmsJobPhases["OdmsInitialLoadExport"] = "ODMS_INITIAL_LOAD_EXPORT";
    OdmsJobPhases["OdmsDataUpload"] = "ODMS_DATA_UPLOAD";
    OdmsJobPhases["OdmsInitialLoadImport"] = "ODMS_INITIAL_LOAD_IMPORT";
    OdmsJobPhases["OdmsPostInitialLoad"] = "ODMS_POST_INITIAL_LOAD";
    OdmsJobPhases["OdmsPrepareReplicationTarget"] = "ODMS_PREPARE_REPLICATION_TARGET";
    OdmsJobPhases["OdmsMonitorReplicationLag"] = "ODMS_MONITOR_REPLICATION_LAG";
    OdmsJobPhases["OdmsSwitchover"] = "ODMS_SWITCHOVER";
    OdmsJobPhases["OdmsCleanup"] = "ODMS_CLEANUP";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OdmsJobPhases["UnknownValue"] = "UNKNOWN_VALUE";
})(OdmsJobPhases = exports.OdmsJobPhases || (exports.OdmsJobPhases = {}));
(function (OdmsJobPhases) {
    function getJsonObj(obj) {
        return { obj };
    }
    OdmsJobPhases.getJsonObj = getJsonObj;
})(OdmsJobPhases = exports.OdmsJobPhases || (exports.OdmsJobPhases = {}));
