// oci-databasemigration/lib/model/operation-types.js
"use strict";
/**
 * Database Migration API
 * Use the Oracle Cloud Infrastructure Database Migration APIs to perform database migration operations.
 * OpenAPI spec version: 20200720
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationTypes = void 0;
/**
 * Possible operation types.
 *
 **/
var OperationTypes;
(function (OperationTypes) {
    OperationTypes["CreateAgent"] = "CREATE_AGENT";
    OperationTypes["DeleteAgent"] = "DELETE_AGENT";
    OperationTypes["CreateMigration"] = "CREATE_MIGRATION";
    OperationTypes["CloneMigration"] = "CLONE_MIGRATION";
    OperationTypes["DeleteMigration"] = "DELETE_MIGRATION";
    OperationTypes["UpdateMigration"] = "UPDATE_MIGRATION";
    OperationTypes["StartMigration"] = "START_MIGRATION";
    OperationTypes["ValidateMigration"] = "VALIDATE_MIGRATION";
    OperationTypes["CreateConnection"] = "CREATE_CONNECTION";
    OperationTypes["DeleteConnection"] = "DELETE_CONNECTION";
    OperationTypes["UpdateConnection"] = "UPDATE_CONNECTION";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationTypes["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationTypes = exports.OperationTypes || (exports.OperationTypes = {}));
(function (OperationTypes) {
    function getJsonObj(obj) {
        return { obj };
    }
    OperationTypes.getJsonObj = getJsonObj;
})(OperationTypes = exports.OperationTypes || (exports.OperationTypes = {}));
