// oci-waas/lib/request/index.js
"use strict";
/**
 * Web Application Acceleration and Security Services API
 * OCI Web Application Acceleration and Security Services
 * OpenAPI spec version: 20181116
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWorkRequestsRequest = exports.ListWafLogsRequest = exports.ListWafBlockedRequestsRequest = exports.ListWaasPolicyCustomProtectionRulesRequest = exports.ListWaasPoliciesRequest = exports.ListRecommendationsRequest = exports.ListProtectionRulesRequest = exports.ListEdgeSubnetsRequest = exports.ListCustomProtectionRulesRequest = exports.ListCertificatesRequest = exports.ListAddressListsRequest = exports.ListHttpRedirectsRequest = void 0;
const ListHttpRedirectsRequest = __importStar(require("./list-http-redirects-request"));
exports.ListHttpRedirectsRequest = ListHttpRedirectsRequest.ListHttpRedirectsRequest;
const ListAddressListsRequest = __importStar(require("./list-address-lists-request"));
exports.ListAddressListsRequest = ListAddressListsRequest.ListAddressListsRequest;
const ListCertificatesRequest = __importStar(require("./list-certificates-request"));
exports.ListCertificatesRequest = ListCertificatesRequest.ListCertificatesRequest;
const ListCustomProtectionRulesRequest = __importStar(require("./list-custom-protection-rules-request"));
exports.ListCustomProtectionRulesRequest = ListCustomProtectionRulesRequest.ListCustomProtectionRulesRequest;
const ListEdgeSubnetsRequest = __importStar(require("./list-edge-subnets-request"));
exports.ListEdgeSubnetsRequest = ListEdgeSubnetsRequest.ListEdgeSubnetsRequest;
const ListProtectionRulesRequest = __importStar(require("./list-protection-rules-request"));
exports.ListProtectionRulesRequest = ListProtectionRulesRequest.ListProtectionRulesRequest;
const ListRecommendationsRequest = __importStar(require("./list-recommendations-request"));
exports.ListRecommendationsRequest = ListRecommendationsRequest.ListRecommendationsRequest;
const ListWaasPoliciesRequest = __importStar(require("./list-waas-policies-request"));
exports.ListWaasPoliciesRequest = ListWaasPoliciesRequest.ListWaasPoliciesRequest;
const ListWaasPolicyCustomProtectionRulesRequest = __importStar(require("./list-waas-policy-custom-protection-rules-request"));
exports.ListWaasPolicyCustomProtectionRulesRequest = ListWaasPolicyCustomProtectionRulesRequest.ListWaasPolicyCustomProtectionRulesRequest;
const ListWafBlockedRequestsRequest = __importStar(require("./list-waf-blocked-requests-request"));
exports.ListWafBlockedRequestsRequest = ListWafBlockedRequestsRequest.ListWafBlockedRequestsRequest;
const ListWafLogsRequest = __importStar(require("./list-waf-logs-request"));
exports.ListWafLogsRequest = ListWafLogsRequest.ListWafLogsRequest;
const ListWorkRequestsRequest = __importStar(require("./list-work-requests-request"));
exports.ListWorkRequestsRequest = ListWorkRequestsRequest.ListWorkRequestsRequest;
