// oci-waas/lib/request/list-edge-subnets-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20181116
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListEdgeSubnetsRequest = void 0;
var ListEdgeSubnetsRequest;
(function (ListEdgeSubnetsRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["Cidr"] = "cidr";
        SortBy["Region"] = "region";
        SortBy["TimeModified"] = "timeModified";
    })(SortBy = ListEdgeSubnetsRequest.SortBy || (ListEdgeSubnetsRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListEdgeSubnetsRequest.SortOrder || (ListEdgeSubnetsRequest.SortOrder = {}));
})(ListEdgeSubnetsRequest = exports.ListEdgeSubnetsRequest || (exports.ListEdgeSubnetsRequest = {}));
