// oci-waas/lib/request/list-waf-logs-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20181116
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWafLogsRequest = void 0;
var ListWafLogsRequest;
(function (ListWafLogsRequest) {
    let Action;
    (function (Action) {
        Action["Block"] = "BLOCK";
        Action["Detect"] = "DETECT";
        Action["Bypass"] = "BYPASS";
        Action["Log"] = "LOG";
        Action["Redirected"] = "REDIRECTED";
    })(Action = ListWafLogsRequest.Action || (ListWafLogsRequest.Action = {}));
    let HttpMethod;
    (function (HttpMethod) {
        HttpMethod["Options"] = "OPTIONS";
        HttpMethod["Get"] = "GET";
        HttpMethod["Head"] = "HEAD";
        HttpMethod["Post"] = "POST";
        HttpMethod["Put"] = "PUT";
        HttpMethod["Delete"] = "DELETE";
        HttpMethod["Trace"] = "TRACE";
        HttpMethod["Connect"] = "CONNECT";
    })(HttpMethod = ListWafLogsRequest.HttpMethod || (ListWafLogsRequest.HttpMethod = {}));
    let LogType;
    (function (LogType) {
        LogType["Access"] = "ACCESS";
        LogType["ProtectionRules"] = "PROTECTION_RULES";
        LogType["JsChallenge"] = "JS_CHALLENGE";
        LogType["Captcha"] = "CAPTCHA";
        LogType["AccessRules"] = "ACCESS_RULES";
        LogType["ThreatFeeds"] = "THREAT_FEEDS";
        LogType["HumanInteractionChallenge"] = "HUMAN_INTERACTION_CHALLENGE";
        LogType["DeviceFingerprintChallenge"] = "DEVICE_FINGERPRINT_CHALLENGE";
        LogType["AddressRateLimiting"] = "ADDRESS_RATE_LIMITING";
    })(LogType = ListWafLogsRequest.LogType || (ListWafLogsRequest.LogType = {}));
})(ListWafLogsRequest = exports.ListWafLogsRequest || (exports.ListWafLogsRequest = {}));
