// oci-loadbalancer/lib/model/create-load-balancer-details.js
"use strict";
/**
 * Load Balancing API
 * API for the Load Balancing service. Use this API to manage load balancers, backend sets, and related items. For more
information, see [Overview of Load Balancing](/iaas/Content/Balance/Concepts/balanceoverview.htm).

 * OpenAPI spec version: 20170115
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateLoadBalancerDetails = void 0;
const model = __importStar(require("../model"));
const common = require("oci-common");
var CreateLoadBalancerDetails;
(function (CreateLoadBalancerDetails) {
    let IpMode;
    (function (IpMode) {
        IpMode["Ipv4"] = "IPV4";
        IpMode["Ipv6"] = "IPV6";
    })(IpMode = CreateLoadBalancerDetails.IpMode || (CreateLoadBalancerDetails.IpMode = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "shapeDetails": obj.shapeDetails
                ? model.ShapeDetails.getJsonObj(obj.shapeDetails)
                : undefined,
            "reservedIps": obj.reservedIps
                ? obj.reservedIps.map(item => {
                    return model.ReservedIP.getJsonObj(item);
                })
                : undefined,
            "listeners": obj.listeners
                ? common.mapContainer(obj.listeners, model.ListenerDetails.getJsonObj)
                : undefined,
            "hostnames": obj.hostnames
                ? common.mapContainer(obj.hostnames, model.HostnameDetails.getJsonObj)
                : undefined,
            "backendSets": obj.backendSets
                ? common.mapContainer(obj.backendSets, model.BackendSetDetails.getJsonObj)
                : undefined,
            "certificates": obj.certificates
                ? common.mapContainer(obj.certificates, model.CertificateDetails.getJsonObj)
                : undefined,
            "sslCipherSuites": obj.sslCipherSuites
                ? common.mapContainer(obj.sslCipherSuites, model.SSLCipherSuiteDetails.getJsonObj)
                : undefined,
            "pathRouteSets": obj.pathRouteSets
                ? common.mapContainer(obj.pathRouteSets, model.PathRouteSetDetails.getJsonObj)
                : undefined,
            "ruleSets": obj.ruleSets
                ? common.mapContainer(obj.ruleSets, model.RuleSetDetails.getJsonObj)
                : undefined
        });
        return jsonObj;
    }
    CreateLoadBalancerDetails.getJsonObj = getJsonObj;
})(CreateLoadBalancerDetails = exports.CreateLoadBalancerDetails || (exports.CreateLoadBalancerDetails = {}));
