// oci-loadbalancer/lib/model/index.js
"use strict";
/**
 * Load Balancing API
 * API for the Load Balancing service. Use this API to manage load balancers, backend sets, and related items. For more
information, see [Overview of Load Balancing](/iaas/Content/Balance/Concepts/balanceoverview.htm).

 * OpenAPI spec version: 20170115
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RuleSetDetails = exports.RuleSet = exports.RuleCondition = exports.Rule = exports.RoutingRule = exports.RoutingPolicyDetails = exports.RoutingPolicy = exports.ReservedIP = exports.RedirectUri = exports.PathRouteSetDetails = exports.PathRouteSet = exports.PathRoute = exports.PathMatchType = exports.LoadBalancerShape = exports.LoadBalancerProtocol = exports.LoadBalancerPolicy = exports.LoadBalancerHealthSummary = exports.LoadBalancerHealth = exports.LoadBalancer = exports.ListenerRuleSummary = exports.ListenerDetails = exports.Listener = exports.LBCookieSessionPersistenceConfigurationDetails = exports.IpAddress = exports.HostnameDetails = exports.Hostname = exports.HealthCheckerDetails = exports.HealthChecker = exports.HealthCheckResult = exports.CreateSSLCipherSuiteDetails = exports.CreateRuleSetDetails = exports.CreateRoutingPolicyDetails = exports.CreatePathRouteSetDetails = exports.CreateLoadBalancerDetails = exports.CreateListenerDetails = exports.CreateHostnameDetails = exports.CreateCertificateDetails = exports.CreateBackendSetDetails = exports.CreateBackendDetails = exports.ConnectionConfiguration = exports.ChangeLoadBalancerCompartmentDetails = exports.CertificateDetails = exports.Certificate = exports.BackendSetHealth = exports.BackendSetDetails = exports.BackendSet = exports.BackendHealth = exports.BackendDetails = exports.Backend = exports.Action = void 0;
exports.SourceVcnIpAddressCondition = exports.SourceVcnIdCondition = exports.SourceIpAddressCondition = exports.RemoveHttpResponseHeaderRule = exports.RemoveHttpRequestHeaderRule = exports.RedirectRule = exports.PathMatchCondition = exports.HttpHeaderRule = exports.ForwardToBackendSet = exports.ExtendHttpResponseHeaderValueRule = exports.ExtendHttpRequestHeaderValueRule = exports.ControlAccessUsingHttpMethodsRule = exports.AllowRule = exports.AddHttpResponseHeaderRule = exports.AddHttpRequestHeaderRule = exports.WorkRequestError = exports.WorkRequest = exports.UpdateSSLCipherSuiteDetails = exports.UpdateRuleSetDetails = exports.UpdateRoutingPolicyDetails = exports.UpdatePathRouteSetDetails = exports.UpdateNetworkSecurityGroupsDetails = exports.UpdateLoadBalancerShapeDetails = exports.UpdateLoadBalancerDetails = exports.UpdateListenerDetails = exports.UpdateHostnameDetails = exports.UpdateHealthCheckerDetails = exports.UpdateBackendSetDetails = exports.UpdateBackendDetails = exports.ShapeDetails = exports.SessionPersistenceConfigurationDetails = exports.SSLConfigurationDetails = exports.SSLConfiguration = exports.SSLCipherSuiteDetails = exports.SSLCipherSuite = void 0;
const Action = __importStar(require("./action"));
exports.Action = Action.Action;
const Backend = __importStar(require("./backend"));
exports.Backend = Backend.Backend;
const BackendDetails = __importStar(require("./backend-details"));
exports.BackendDetails = BackendDetails.BackendDetails;
const BackendHealth = __importStar(require("./backend-health"));
exports.BackendHealth = BackendHealth.BackendHealth;
const BackendSet = __importStar(require("./backend-set"));
exports.BackendSet = BackendSet.BackendSet;
const BackendSetDetails = __importStar(require("./backend-set-details"));
exports.BackendSetDetails = BackendSetDetails.BackendSetDetails;
const BackendSetHealth = __importStar(require("./backend-set-health"));
exports.BackendSetHealth = BackendSetHealth.BackendSetHealth;
const Certificate = __importStar(require("./certificate"));
exports.Certificate = Certificate.Certificate;
const CertificateDetails = __importStar(require("./certificate-details"));
exports.CertificateDetails = CertificateDetails.CertificateDetails;
const ChangeLoadBalancerCompartmentDetails = __importStar(require("./change-load-balancer-compartment-details"));
exports.ChangeLoadBalancerCompartmentDetails = ChangeLoadBalancerCompartmentDetails.ChangeLoadBalancerCompartmentDetails;
const ConnectionConfiguration = __importStar(require("./connection-configuration"));
exports.ConnectionConfiguration = ConnectionConfiguration.ConnectionConfiguration;
const CreateBackendDetails = __importStar(require("./create-backend-details"));
exports.CreateBackendDetails = CreateBackendDetails.CreateBackendDetails;
const CreateBackendSetDetails = __importStar(require("./create-backend-set-details"));
exports.CreateBackendSetDetails = CreateBackendSetDetails.CreateBackendSetDetails;
const CreateCertificateDetails = __importStar(require("./create-certificate-details"));
exports.CreateCertificateDetails = CreateCertificateDetails.CreateCertificateDetails;
const CreateHostnameDetails = __importStar(require("./create-hostname-details"));
exports.CreateHostnameDetails = CreateHostnameDetails.CreateHostnameDetails;
const CreateListenerDetails = __importStar(require("./create-listener-details"));
exports.CreateListenerDetails = CreateListenerDetails.CreateListenerDetails;
const CreateLoadBalancerDetails = __importStar(require("./create-load-balancer-details"));
exports.CreateLoadBalancerDetails = CreateLoadBalancerDetails.CreateLoadBalancerDetails;
const CreatePathRouteSetDetails = __importStar(require("./create-path-route-set-details"));
exports.CreatePathRouteSetDetails = CreatePathRouteSetDetails.CreatePathRouteSetDetails;
const CreateRoutingPolicyDetails = __importStar(require("./create-routing-policy-details"));
exports.CreateRoutingPolicyDetails = CreateRoutingPolicyDetails.CreateRoutingPolicyDetails;
const CreateRuleSetDetails = __importStar(require("./create-rule-set-details"));
exports.CreateRuleSetDetails = CreateRuleSetDetails.CreateRuleSetDetails;
const CreateSSLCipherSuiteDetails = __importStar(require("./create-ss-lcipher-suite-details"));
exports.CreateSSLCipherSuiteDetails = CreateSSLCipherSuiteDetails.CreateSSLCipherSuiteDetails;
const HealthCheckResult = __importStar(require("./health-check-result"));
exports.HealthCheckResult = HealthCheckResult.HealthCheckResult;
const HealthChecker = __importStar(require("./health-checker"));
exports.HealthChecker = HealthChecker.HealthChecker;
const HealthCheckerDetails = __importStar(require("./health-checker-details"));
exports.HealthCheckerDetails = HealthCheckerDetails.HealthCheckerDetails;
const Hostname = __importStar(require("./hostname"));
exports.Hostname = Hostname.Hostname;
const HostnameDetails = __importStar(require("./hostname-details"));
exports.HostnameDetails = HostnameDetails.HostnameDetails;
const IpAddress = __importStar(require("./ip-address"));
exports.IpAddress = IpAddress.IpAddress;
const LBCookieSessionPersistenceConfigurationDetails = __importStar(require("./l-bcookie-session-persistence-configuration-details"));
exports.LBCookieSessionPersistenceConfigurationDetails = LBCookieSessionPersistenceConfigurationDetails.LBCookieSessionPersistenceConfigurationDetails;
const Listener = __importStar(require("./listener"));
exports.Listener = Listener.Listener;
const ListenerDetails = __importStar(require("./listener-details"));
exports.ListenerDetails = ListenerDetails.ListenerDetails;
const ListenerRuleSummary = __importStar(require("./listener-rule-summary"));
exports.ListenerRuleSummary = ListenerRuleSummary.ListenerRuleSummary;
const LoadBalancer = __importStar(require("./load-balancer"));
exports.LoadBalancer = LoadBalancer.LoadBalancer;
const LoadBalancerHealth = __importStar(require("./load-balancer-health"));
exports.LoadBalancerHealth = LoadBalancerHealth.LoadBalancerHealth;
const LoadBalancerHealthSummary = __importStar(require("./load-balancer-health-summary"));
exports.LoadBalancerHealthSummary = LoadBalancerHealthSummary.LoadBalancerHealthSummary;
const LoadBalancerPolicy = __importStar(require("./load-balancer-policy"));
exports.LoadBalancerPolicy = LoadBalancerPolicy.LoadBalancerPolicy;
const LoadBalancerProtocol = __importStar(require("./load-balancer-protocol"));
exports.LoadBalancerProtocol = LoadBalancerProtocol.LoadBalancerProtocol;
const LoadBalancerShape = __importStar(require("./load-balancer-shape"));
exports.LoadBalancerShape = LoadBalancerShape.LoadBalancerShape;
const PathMatchType = __importStar(require("./path-match-type"));
exports.PathMatchType = PathMatchType.PathMatchType;
const PathRoute = __importStar(require("./path-route"));
exports.PathRoute = PathRoute.PathRoute;
const PathRouteSet = __importStar(require("./path-route-set"));
exports.PathRouteSet = PathRouteSet.PathRouteSet;
const PathRouteSetDetails = __importStar(require("./path-route-set-details"));
exports.PathRouteSetDetails = PathRouteSetDetails.PathRouteSetDetails;
const RedirectUri = __importStar(require("./redirect-uri"));
exports.RedirectUri = RedirectUri.RedirectUri;
const ReservedIP = __importStar(require("./reserved-ip"));
exports.ReservedIP = ReservedIP.ReservedIP;
const RoutingPolicy = __importStar(require("./routing-policy"));
exports.RoutingPolicy = RoutingPolicy.RoutingPolicy;
const RoutingPolicyDetails = __importStar(require("./routing-policy-details"));
exports.RoutingPolicyDetails = RoutingPolicyDetails.RoutingPolicyDetails;
const RoutingRule = __importStar(require("./routing-rule"));
exports.RoutingRule = RoutingRule.RoutingRule;
const Rule = __importStar(require("./rule"));
exports.Rule = Rule.Rule;
const RuleCondition = __importStar(require("./rule-condition"));
exports.RuleCondition = RuleCondition.RuleCondition;
const RuleSet = __importStar(require("./rule-set"));
exports.RuleSet = RuleSet.RuleSet;
const RuleSetDetails = __importStar(require("./rule-set-details"));
exports.RuleSetDetails = RuleSetDetails.RuleSetDetails;
const SSLCipherSuite = __importStar(require("./s-sl-cipher-suite"));
exports.SSLCipherSuite = SSLCipherSuite.SSLCipherSuite;
const SSLCipherSuiteDetails = __importStar(require("./s-sl-cipher-suite-details"));
exports.SSLCipherSuiteDetails = SSLCipherSuiteDetails.SSLCipherSuiteDetails;
const SSLConfiguration = __importStar(require("./s-sl-configuration"));
exports.SSLConfiguration = SSLConfiguration.SSLConfiguration;
const SSLConfigurationDetails = __importStar(require("./s-sl-configuration-details"));
exports.SSLConfigurationDetails = SSLConfigurationDetails.SSLConfigurationDetails;
const SessionPersistenceConfigurationDetails = __importStar(require("./session-persistence-configuration-details"));
exports.SessionPersistenceConfigurationDetails = SessionPersistenceConfigurationDetails.SessionPersistenceConfigurationDetails;
const ShapeDetails = __importStar(require("./shape-details"));
exports.ShapeDetails = ShapeDetails.ShapeDetails;
const UpdateBackendDetails = __importStar(require("./update-backend-details"));
exports.UpdateBackendDetails = UpdateBackendDetails.UpdateBackendDetails;
const UpdateBackendSetDetails = __importStar(require("./update-backend-set-details"));
exports.UpdateBackendSetDetails = UpdateBackendSetDetails.UpdateBackendSetDetails;
const UpdateHealthCheckerDetails = __importStar(require("./update-health-checker-details"));
exports.UpdateHealthCheckerDetails = UpdateHealthCheckerDetails.UpdateHealthCheckerDetails;
const UpdateHostnameDetails = __importStar(require("./update-hostname-details"));
exports.UpdateHostnameDetails = UpdateHostnameDetails.UpdateHostnameDetails;
const UpdateListenerDetails = __importStar(require("./update-listener-details"));
exports.UpdateListenerDetails = UpdateListenerDetails.UpdateListenerDetails;
const UpdateLoadBalancerDetails = __importStar(require("./update-load-balancer-details"));
exports.UpdateLoadBalancerDetails = UpdateLoadBalancerDetails.UpdateLoadBalancerDetails;
const UpdateLoadBalancerShapeDetails = __importStar(require("./update-load-balancer-shape-details"));
exports.UpdateLoadBalancerShapeDetails = UpdateLoadBalancerShapeDetails.UpdateLoadBalancerShapeDetails;
const UpdateNetworkSecurityGroupsDetails = __importStar(require("./update-network-security-groups-details"));
exports.UpdateNetworkSecurityGroupsDetails = UpdateNetworkSecurityGroupsDetails.UpdateNetworkSecurityGroupsDetails;
const UpdatePathRouteSetDetails = __importStar(require("./update-path-route-set-details"));
exports.UpdatePathRouteSetDetails = UpdatePathRouteSetDetails.UpdatePathRouteSetDetails;
const UpdateRoutingPolicyDetails = __importStar(require("./update-routing-policy-details"));
exports.UpdateRoutingPolicyDetails = UpdateRoutingPolicyDetails.UpdateRoutingPolicyDetails;
const UpdateRuleSetDetails = __importStar(require("./update-rule-set-details"));
exports.UpdateRuleSetDetails = UpdateRuleSetDetails.UpdateRuleSetDetails;
const UpdateSSLCipherSuiteDetails = __importStar(require("./update-ss-lcipher-suite-details"));
exports.UpdateSSLCipherSuiteDetails = UpdateSSLCipherSuiteDetails.UpdateSSLCipherSuiteDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const AddHttpRequestHeaderRule = __importStar(require("./add-http-request-header-rule"));
exports.AddHttpRequestHeaderRule = AddHttpRequestHeaderRule.AddHttpRequestHeaderRule;
const AddHttpResponseHeaderRule = __importStar(require("./add-http-response-header-rule"));
exports.AddHttpResponseHeaderRule = AddHttpResponseHeaderRule.AddHttpResponseHeaderRule;
const AllowRule = __importStar(require("./allow-rule"));
exports.AllowRule = AllowRule.AllowRule;
const ControlAccessUsingHttpMethodsRule = __importStar(require("./control-access-using-http-methods-rule"));
exports.ControlAccessUsingHttpMethodsRule = ControlAccessUsingHttpMethodsRule.ControlAccessUsingHttpMethodsRule;
const ExtendHttpRequestHeaderValueRule = __importStar(require("./extend-http-request-header-value-rule"));
exports.ExtendHttpRequestHeaderValueRule = ExtendHttpRequestHeaderValueRule.ExtendHttpRequestHeaderValueRule;
const ExtendHttpResponseHeaderValueRule = __importStar(require("./extend-http-response-header-value-rule"));
exports.ExtendHttpResponseHeaderValueRule = ExtendHttpResponseHeaderValueRule.ExtendHttpResponseHeaderValueRule;
const ForwardToBackendSet = __importStar(require("./forward-to-backend-set"));
exports.ForwardToBackendSet = ForwardToBackendSet.ForwardToBackendSet;
const HttpHeaderRule = __importStar(require("./http-header-rule"));
exports.HttpHeaderRule = HttpHeaderRule.HttpHeaderRule;
const PathMatchCondition = __importStar(require("./path-match-condition"));
exports.PathMatchCondition = PathMatchCondition.PathMatchCondition;
const RedirectRule = __importStar(require("./redirect-rule"));
exports.RedirectRule = RedirectRule.RedirectRule;
const RemoveHttpRequestHeaderRule = __importStar(require("./remove-http-request-header-rule"));
exports.RemoveHttpRequestHeaderRule = RemoveHttpRequestHeaderRule.RemoveHttpRequestHeaderRule;
const RemoveHttpResponseHeaderRule = __importStar(require("./remove-http-response-header-rule"));
exports.RemoveHttpResponseHeaderRule = RemoveHttpResponseHeaderRule.RemoveHttpResponseHeaderRule;
const SourceIpAddressCondition = __importStar(require("./source-ip-address-condition"));
exports.SourceIpAddressCondition = SourceIpAddressCondition.SourceIpAddressCondition;
const SourceVcnIdCondition = __importStar(require("./source-vcn-id-condition"));
exports.SourceVcnIdCondition = SourceVcnIdCondition.SourceVcnIdCondition;
const SourceVcnIpAddressCondition = __importStar(require("./source-vcn-ip-address-condition"));
exports.SourceVcnIpAddressCondition = SourceVcnIpAddressCondition.SourceVcnIpAddressCondition;
