// oci-loadbalancer/lib/model/load-balancer.js
"use strict";
/**
 * Load Balancing API
 * API for the Load Balancing service. Use this API to manage load balancers, backend sets, and related items. For more
information, see [Overview of Load Balancing](/iaas/Content/Balance/Concepts/balanceoverview.htm).

 * OpenAPI spec version: 20170115
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LoadBalancer = void 0;
const model = __importStar(require("../model"));
const common = require("oci-common");
var LoadBalancer;
(function (LoadBalancer) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Failed"] = "FAILED";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = LoadBalancer.LifecycleState || (LoadBalancer.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "ipAddresses": obj.ipAddresses
                ? obj.ipAddresses.map(item => {
                    return model.IpAddress.getJsonObj(item);
                })
                : undefined,
            "shapeDetails": obj.shapeDetails
                ? model.ShapeDetails.getJsonObj(obj.shapeDetails)
                : undefined,
            "listeners": obj.listeners
                ? common.mapContainer(obj.listeners, model.Listener.getJsonObj)
                : undefined,
            "hostnames": obj.hostnames
                ? common.mapContainer(obj.hostnames, model.Hostname.getJsonObj)
                : undefined,
            "sslCipherSuites": obj.sslCipherSuites
                ? common.mapContainer(obj.sslCipherSuites, model.SSLCipherSuite.getJsonObj)
                : undefined,
            "certificates": obj.certificates
                ? common.mapContainer(obj.certificates, model.Certificate.getJsonObj)
                : undefined,
            "backendSets": obj.backendSets
                ? common.mapContainer(obj.backendSets, model.BackendSet.getJsonObj)
                : undefined,
            "pathRouteSets": obj.pathRouteSets
                ? common.mapContainer(obj.pathRouteSets, model.PathRouteSet.getJsonObj)
                : undefined,
            "ruleSets": obj.ruleSets
                ? common.mapContainer(obj.ruleSets, model.RuleSet.getJsonObj)
                : undefined,
            "routingPolicies": obj.routingPolicies
                ? common.mapContainer(obj.routingPolicies, model.RoutingPolicy.getJsonObj)
                : undefined
        });
        return jsonObj;
    }
    LoadBalancer.getJsonObj = getJsonObj;
})(LoadBalancer = exports.LoadBalancer || (exports.LoadBalancer = {}));
