// oci-loadbalancer/lib/model/path-match-type.js
"use strict";
/**
 * Load Balancing API
 * API for the Load Balancing service. Use this API to manage load balancers, backend sets, and related items. For more
information, see [Overview of Load Balancing](/iaas/Content/Balance/Concepts/balanceoverview.htm).

 * OpenAPI spec version: 20170115
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PathMatchType = void 0;
var PathMatchType;
(function (PathMatchType) {
    let MatchType;
    (function (MatchType) {
        MatchType["ExactMatch"] = "EXACT_MATCH";
        MatchType["ForceLongestPrefixMatch"] = "FORCE_LONGEST_PREFIX_MATCH";
        MatchType["PrefixMatch"] = "PREFIX_MATCH";
        MatchType["SuffixMatch"] = "SUFFIX_MATCH";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        MatchType["UnknownValue"] = "UNKNOWN_VALUE";
    })(MatchType = PathMatchType.MatchType || (PathMatchType.MatchType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    PathMatchType.getJsonObj = getJsonObj;
})(PathMatchType = exports.PathMatchType || (exports.PathMatchType = {}));
