// oci-loadbalancer/lib/model/path-match-condition.js
"use strict";
/**
 * Load Balancing API
 * API for the Load Balancing service. Use this API to manage load balancers, backend sets, and related items. For more
information, see [Overview of Load Balancing](/iaas/Content/Balance/Concepts/balanceoverview.htm).

 * OpenAPI spec version: 20170115
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PathMatchCondition = void 0;
const model = __importStar(require("../model"));
var PathMatchCondition;
(function (PathMatchCondition) {
    let Operator;
    (function (Operator) {
        Operator["ExactMatch"] = "EXACT_MATCH";
        Operator["ForceLongestPrefixMatch"] = "FORCE_LONGEST_PREFIX_MATCH";
        Operator["PrefixMatch"] = "PREFIX_MATCH";
        Operator["SuffixMatch"] = "SUFFIX_MATCH";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Operator["UnknownValue"] = "UNKNOWN_VALUE";
    })(Operator = PathMatchCondition.Operator || (PathMatchCondition.Operator = {}));
    function getJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj ? obj : model.RuleCondition.getJsonObj(obj))), {});
        return jsonObj;
    }
    PathMatchCondition.getJsonObj = getJsonObj;
    PathMatchCondition.attributeName = "PATH";
})(PathMatchCondition = exports.PathMatchCondition || (exports.PathMatchCondition = {}));
