// oci-optimizer/lib/model/index.js
"use strict";
/**
 * Cloud Advisor API
 * APIs for managing Cloud Advisor. Cloud Advisor provides recommendations that help you maximize cost savings and improve the security posture of your tenancy.
 * OpenAPI spec version: 20200606
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestActionType = exports.WorkRequest = exports.UpdateResourceActionDetails = exports.UpdateRecommendationDetails = exports.UpdateProfileDetails = exports.UpdateEnrollmentStatusDetails = exports.TargetTags = exports.TargetTag = exports.TargetCompartments = exports.TagValueType = exports.SupportedLevels = exports.SupportedLevel = exports.StrategyParameterType = exports.StrategyParameter = exports.Strategy = exports.Status = exports.SortOrder = exports.ResourceCount = exports.ResourceActionSummary = exports.ResourceActionCollection = exports.ResourceAction = exports.RecommendationSummary = exports.RecommendationStrategySummary = exports.RecommendationStrategyCollection = exports.RecommendationCount = exports.RecommendationCollection = exports.Recommendation = exports.ProfileSummary = exports.ProfileCollection = exports.Profile = exports.OptimizerEnrollmentStatus = exports.OperationType = exports.OperationStatus = exports.LifecycleState = exports.LevelsConfiguration = exports.LevelConfiguration = exports.Importance = exports.HistorySummary = exports.HistoryCollection = exports.EnrollmentStatusSummary = exports.EnrollmentStatusCollection = exports.EnrollmentStatus = exports.CreateProfileDetails = exports.CategorySummary = exports.CategoryCollection = exports.Category = exports.BulkApplyResourceAction = exports.BulkApplyRecommendationsDetails = exports.ActionType = exports.Action = void 0;
exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequestCollection = void 0;
const Action = __importStar(require("./action"));
exports.Action = Action.Action;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const BulkApplyRecommendationsDetails = __importStar(require("./bulk-apply-recommendations-details"));
exports.BulkApplyRecommendationsDetails = BulkApplyRecommendationsDetails.BulkApplyRecommendationsDetails;
const BulkApplyResourceAction = __importStar(require("./bulk-apply-resource-action"));
exports.BulkApplyResourceAction = BulkApplyResourceAction.BulkApplyResourceAction;
const Category = __importStar(require("./category"));
exports.Category = Category.Category;
const CategoryCollection = __importStar(require("./category-collection"));
exports.CategoryCollection = CategoryCollection.CategoryCollection;
const CategorySummary = __importStar(require("./category-summary"));
exports.CategorySummary = CategorySummary.CategorySummary;
const CreateProfileDetails = __importStar(require("./create-profile-details"));
exports.CreateProfileDetails = CreateProfileDetails.CreateProfileDetails;
const EnrollmentStatus = __importStar(require("./enrollment-status"));
exports.EnrollmentStatus = EnrollmentStatus.EnrollmentStatus;
const EnrollmentStatusCollection = __importStar(require("./enrollment-status-collection"));
exports.EnrollmentStatusCollection = EnrollmentStatusCollection.EnrollmentStatusCollection;
const EnrollmentStatusSummary = __importStar(require("./enrollment-status-summary"));
exports.EnrollmentStatusSummary = EnrollmentStatusSummary.EnrollmentStatusSummary;
const HistoryCollection = __importStar(require("./history-collection"));
exports.HistoryCollection = HistoryCollection.HistoryCollection;
const HistorySummary = __importStar(require("./history-summary"));
exports.HistorySummary = HistorySummary.HistorySummary;
const Importance = __importStar(require("./importance"));
exports.Importance = Importance.Importance;
const LevelConfiguration = __importStar(require("./level-configuration"));
exports.LevelConfiguration = LevelConfiguration.LevelConfiguration;
const LevelsConfiguration = __importStar(require("./levels-configuration"));
exports.LevelsConfiguration = LevelsConfiguration.LevelsConfiguration;
const LifecycleState = __importStar(require("./lifecycle-state"));
exports.LifecycleState = LifecycleState.LifecycleState;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const OptimizerEnrollmentStatus = __importStar(require("./optimizer-enrollment-status"));
exports.OptimizerEnrollmentStatus = OptimizerEnrollmentStatus.OptimizerEnrollmentStatus;
const Profile = __importStar(require("./profile"));
exports.Profile = Profile.Profile;
const ProfileCollection = __importStar(require("./profile-collection"));
exports.ProfileCollection = ProfileCollection.ProfileCollection;
const ProfileSummary = __importStar(require("./profile-summary"));
exports.ProfileSummary = ProfileSummary.ProfileSummary;
const Recommendation = __importStar(require("./recommendation"));
exports.Recommendation = Recommendation.Recommendation;
const RecommendationCollection = __importStar(require("./recommendation-collection"));
exports.RecommendationCollection = RecommendationCollection.RecommendationCollection;
const RecommendationCount = __importStar(require("./recommendation-count"));
exports.RecommendationCount = RecommendationCount.RecommendationCount;
const RecommendationStrategyCollection = __importStar(require("./recommendation-strategy-collection"));
exports.RecommendationStrategyCollection = RecommendationStrategyCollection.RecommendationStrategyCollection;
const RecommendationStrategySummary = __importStar(require("./recommendation-strategy-summary"));
exports.RecommendationStrategySummary = RecommendationStrategySummary.RecommendationStrategySummary;
const RecommendationSummary = __importStar(require("./recommendation-summary"));
exports.RecommendationSummary = RecommendationSummary.RecommendationSummary;
const ResourceAction = __importStar(require("./resource-action"));
exports.ResourceAction = ResourceAction.ResourceAction;
const ResourceActionCollection = __importStar(require("./resource-action-collection"));
exports.ResourceActionCollection = ResourceActionCollection.ResourceActionCollection;
const ResourceActionSummary = __importStar(require("./resource-action-summary"));
exports.ResourceActionSummary = ResourceActionSummary.ResourceActionSummary;
const ResourceCount = __importStar(require("./resource-count"));
exports.ResourceCount = ResourceCount.ResourceCount;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const Status = __importStar(require("./status"));
exports.Status = Status.Status;
const Strategy = __importStar(require("./strategy"));
exports.Strategy = Strategy.Strategy;
const StrategyParameter = __importStar(require("./strategy-parameter"));
exports.StrategyParameter = StrategyParameter.StrategyParameter;
const StrategyParameterType = __importStar(require("./strategy-parameter-type"));
exports.StrategyParameterType = StrategyParameterType.StrategyParameterType;
const SupportedLevel = __importStar(require("./supported-level"));
exports.SupportedLevel = SupportedLevel.SupportedLevel;
const SupportedLevels = __importStar(require("./supported-levels"));
exports.SupportedLevels = SupportedLevels.SupportedLevels;
const TagValueType = __importStar(require("./tag-value-type"));
exports.TagValueType = TagValueType.TagValueType;
const TargetCompartments = __importStar(require("./target-compartments"));
exports.TargetCompartments = TargetCompartments.TargetCompartments;
const TargetTag = __importStar(require("./target-tag"));
exports.TargetTag = TargetTag.TargetTag;
const TargetTags = __importStar(require("./target-tags"));
exports.TargetTags = TargetTags.TargetTags;
const UpdateEnrollmentStatusDetails = __importStar(require("./update-enrollment-status-details"));
exports.UpdateEnrollmentStatusDetails = UpdateEnrollmentStatusDetails.UpdateEnrollmentStatusDetails;
const UpdateProfileDetails = __importStar(require("./update-profile-details"));
exports.UpdateProfileDetails = UpdateProfileDetails.UpdateProfileDetails;
const UpdateRecommendationDetails = __importStar(require("./update-recommendation-details"));
exports.UpdateRecommendationDetails = UpdateRecommendationDetails.UpdateRecommendationDetails;
const UpdateResourceActionDetails = __importStar(require("./update-resource-action-details"));
exports.UpdateResourceActionDetails = UpdateResourceActionDetails.UpdateResourceActionDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestActionType = __importStar(require("./work-request-action-type"));
exports.WorkRequestActionType = WorkRequestActionType.WorkRequestActionType;
const WorkRequestCollection = __importStar(require("./work-request-collection"));
exports.WorkRequestCollection = WorkRequestCollection.WorkRequestCollection;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
