// oci-optimizer/lib/model/lifecycle-state.js
"use strict";
/**
 * Cloud Advisor API
 * APIs for managing Cloud Advisor. Cloud Advisor provides recommendations that help you maximize cost savings and improve the security posture of your tenancy.
 * OpenAPI spec version: 20200606
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LifecycleState = void 0;
/**
 * Possible lifecycle states.
 **/
var LifecycleState;
(function (LifecycleState) {
    LifecycleState["Active"] = "ACTIVE";
    LifecycleState["Failed"] = "FAILED";
    LifecycleState["Inactive"] = "INACTIVE";
    LifecycleState["Attaching"] = "ATTACHING";
    LifecycleState["Detaching"] = "DETACHING";
    LifecycleState["Deleting"] = "DELETING";
    LifecycleState["Deleted"] = "DELETED";
    LifecycleState["Updating"] = "UPDATING";
    LifecycleState["Creating"] = "CREATING";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(LifecycleState = exports.LifecycleState || (exports.LifecycleState = {}));
(function (LifecycleState) {
    function getJsonObj(obj) {
        return { obj };
    }
    LifecycleState.getJsonObj = getJsonObj;
})(LifecycleState = exports.LifecycleState || (exports.LifecycleState = {}));
