// oci-optimizer/lib/model/optimizer-enrollment-status.js
"use strict";
/**
 * Cloud Advisor API
 * APIs for managing Cloud Advisor. Cloud Advisor provides recommendations that help you maximize cost savings and improve the security posture of your tenancy.
 * OpenAPI spec version: 20200606
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OptimizerEnrollmentStatus = void 0;
/**
 * Possible enrollment statuses.
 **/
var OptimizerEnrollmentStatus;
(function (OptimizerEnrollmentStatus) {
    OptimizerEnrollmentStatus["Active"] = "ACTIVE";
    OptimizerEnrollmentStatus["Inactive"] = "INACTIVE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OptimizerEnrollmentStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(OptimizerEnrollmentStatus = exports.OptimizerEnrollmentStatus || (exports.OptimizerEnrollmentStatus = {}));
(function (OptimizerEnrollmentStatus) {
    function getJsonObj(obj) {
        return { obj };
    }
    OptimizerEnrollmentStatus.getJsonObj = getJsonObj;
})(OptimizerEnrollmentStatus = exports.OptimizerEnrollmentStatus || (exports.OptimizerEnrollmentStatus = {}));
