// oci-optimizer/lib/model/strategy-parameter-type.js
"use strict";
/**
 * Cloud Advisor API
 * APIs for managing Cloud Advisor. Cloud Advisor provides recommendations that help you maximize cost savings and improve the security posture of your tenancy.
 * OpenAPI spec version: 20200606
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StrategyParameterType = void 0;
/**
 * Possible strategy types.
 **/
var StrategyParameterType;
(function (StrategyParameterType) {
    StrategyParameterType["String"] = "STRING";
    StrategyParameterType["Boolean"] = "BOOLEAN";
    StrategyParameterType["Number"] = "NUMBER";
    StrategyParameterType["Datetime"] = "DATETIME";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    StrategyParameterType["UnknownValue"] = "UNKNOWN_VALUE";
})(StrategyParameterType = exports.StrategyParameterType || (exports.StrategyParameterType = {}));
(function (StrategyParameterType) {
    function getJsonObj(obj) {
        return { obj };
    }
    StrategyParameterType.getJsonObj = getJsonObj;
})(StrategyParameterType = exports.StrategyParameterType || (exports.StrategyParameterType = {}));
