// oci-optimizer/lib/model/work-request-action-type.js
"use strict";
/**
 * Cloud Advisor API
 * APIs for managing Cloud Advisor. Cloud Advisor provides recommendations that help you maximize cost savings and improve the security posture of your tenancy.
 * OpenAPI spec version: 20200606
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestActionType = void 0;
/**
 * Possible work request resource actions.
 **/
var WorkRequestActionType;
(function (WorkRequestActionType) {
    WorkRequestActionType["Created"] = "CREATED";
    WorkRequestActionType["Updated"] = "UPDATED";
    WorkRequestActionType["Deleted"] = "DELETED";
    WorkRequestActionType["InProgress"] = "IN_PROGRESS";
    WorkRequestActionType["Related"] = "RELATED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    WorkRequestActionType["UnknownValue"] = "UNKNOWN_VALUE";
})(WorkRequestActionType = exports.WorkRequestActionType || (exports.WorkRequestActionType = {}));
(function (WorkRequestActionType) {
    function getJsonObj(obj) {
        return { obj };
    }
    WorkRequestActionType.getJsonObj = getJsonObj;
})(WorkRequestActionType = exports.WorkRequestActionType || (exports.WorkRequestActionType = {}));
