// oci-dns/lib/model/steering-policy.js
"use strict";
/**
 * DNS API
 * API for the DNS service. Use this API to manage DNS zones, records, and other DNS resources.
For more information, see [Overview of the DNS Service](/iaas/Content/DNS/Concepts/dnszonemanagement.htm).

 * OpenAPI spec version: 20180115
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SteeringPolicy = void 0;
const model = __importStar(require("../model"));
var SteeringPolicy;
(function (SteeringPolicy) {
    let Template;
    (function (Template) {
        Template["Failover"] = "FAILOVER";
        Template["LoadBalance"] = "LOAD_BALANCE";
        Template["RouteByGeo"] = "ROUTE_BY_GEO";
        Template["RouteByAsn"] = "ROUTE_BY_ASN";
        Template["RouteByIp"] = "ROUTE_BY_IP";
        Template["Custom"] = "CUSTOM";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Template["UnknownValue"] = "UNKNOWN_VALUE";
    })(Template = SteeringPolicy.Template || (SteeringPolicy.Template = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Deleting"] = "DELETING";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = SteeringPolicy.LifecycleState || (SteeringPolicy.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "answers": obj.answers
                ? obj.answers.map(item => {
                    return model.SteeringPolicyAnswer.getJsonObj(item);
                })
                : undefined,
            "rules": obj.rules
                ? obj.rules.map(item => {
                    return model.SteeringPolicyRule.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    SteeringPolicy.getJsonObj = getJsonObj;
})(SteeringPolicy = exports.SteeringPolicy || (exports.SteeringPolicy = {}));
