// oci-core/lib/model/byoip-range.js
"use strict";
/**
 * Core Services API
 * API covering the [Networking](/iaas/Content/Network/Concepts/overview.htm),
[Compute](/iaas/Content/Compute/Concepts/computeoverview.htm), and
[Block Volume](/iaas/Content/Block/Concepts/overview.htm) services. Use this API
to manage resources such as virtual cloud networks (VCNs), compute instances, and
block storage volumes.

 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ByoipRange = void 0;
var ByoipRange;
(function (ByoipRange) {
    let LifecycleDetails;
    (function (LifecycleDetails) {
        LifecycleDetails["Creating"] = "CREATING";
        LifecycleDetails["Validating"] = "VALIDATING";
        LifecycleDetails["Provisioned"] = "PROVISIONED";
        LifecycleDetails["Active"] = "ACTIVE";
        LifecycleDetails["Failed"] = "FAILED";
        LifecycleDetails["Deleting"] = "DELETING";
        LifecycleDetails["Deleted"] = "DELETED";
        LifecycleDetails["Advertising"] = "ADVERTISING";
        LifecycleDetails["Withdrawing"] = "WITHDRAWING";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleDetails["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleDetails = ByoipRange.LifecycleDetails || (ByoipRange.LifecycleDetails = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Inactive"] = "INACTIVE";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = ByoipRange.LifecycleState || (ByoipRange.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ByoipRange.getJsonObj = getJsonObj;
})(ByoipRange = exports.ByoipRange || (exports.ByoipRange = {}));
