// oci-datascience/lib/request/index.js
"use strict";
/**
 * Data Science API
 * Use the Data Science APIs to organize your data science work, access data and computing resources, and build, train, deploy, and manage models on Oracle Cloud.

 * OpenAPI spec version: 20190101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWorkRequestsRequest = exports.ListProjectsRequest = exports.ListNotebookSessionsRequest = exports.ListModelsRequest = exports.ListModelDeploymentsRequest = void 0;
const ListModelDeploymentsRequest = __importStar(require("./list-model-deployments-request"));
exports.ListModelDeploymentsRequest = ListModelDeploymentsRequest.ListModelDeploymentsRequest;
const ListModelsRequest = __importStar(require("./list-models-request"));
exports.ListModelsRequest = ListModelsRequest.ListModelsRequest;
const ListNotebookSessionsRequest = __importStar(require("./list-notebook-sessions-request"));
exports.ListNotebookSessionsRequest = ListNotebookSessionsRequest.ListNotebookSessionsRequest;
const ListProjectsRequest = __importStar(require("./list-projects-request"));
exports.ListProjectsRequest = ListProjectsRequest.ListProjectsRequest;
const ListWorkRequestsRequest = __importStar(require("./list-work-requests-request"));
exports.ListWorkRequestsRequest = ListWorkRequestsRequest.ListWorkRequestsRequest;
