// oci-osmanagement/lib/request/index.js
"use strict";
/**
 * OS Management API
 * API for the OS Management service. Use these API operations for working
with Managed instances and Managed instance groups.

 * OpenAPI spec version: 20190801
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchSoftwarePackagesRequest = exports.ListWorkRequestsRequest = exports.ListWorkRequestLogsRequest = exports.ListWorkRequestErrorsRequest = exports.ListWindowsUpdatesInstalledOnManagedInstanceRequest = exports.ListWindowsUpdatesRequest = exports.ListUpcomingScheduledJobsRequest = exports.ListSoftwareSourcesRequest = exports.ListSoftwareSourcePackagesRequest = exports.ListScheduledJobsRequest = exports.ListPackagesInstalledOnManagedInstanceRequest = exports.ListManagedInstancesRequest = exports.ListManagedInstanceGroupsRequest = exports.ListManagedInstanceErrataRequest = exports.ListErrataRequest = exports.ListAvailableWindowsUpdatesForManagedInstanceRequest = exports.ListAvailableUpdatesForManagedInstanceRequest = exports.ListAvailableSoftwareSourcesForManagedInstanceRequest = exports.ListAvailablePackagesForManagedInstanceRequest = void 0;
const ListAvailablePackagesForManagedInstanceRequest = __importStar(require("./list-available-packages-for-managed-instance-request"));
exports.ListAvailablePackagesForManagedInstanceRequest = ListAvailablePackagesForManagedInstanceRequest.ListAvailablePackagesForManagedInstanceRequest;
const ListAvailableSoftwareSourcesForManagedInstanceRequest = __importStar(require("./list-available-software-sources-for-managed-instance-request"));
exports.ListAvailableSoftwareSourcesForManagedInstanceRequest = ListAvailableSoftwareSourcesForManagedInstanceRequest.ListAvailableSoftwareSourcesForManagedInstanceRequest;
const ListAvailableUpdatesForManagedInstanceRequest = __importStar(require("./list-available-updates-for-managed-instance-request"));
exports.ListAvailableUpdatesForManagedInstanceRequest = ListAvailableUpdatesForManagedInstanceRequest.ListAvailableUpdatesForManagedInstanceRequest;
const ListAvailableWindowsUpdatesForManagedInstanceRequest = __importStar(require("./list-available-windows-updates-for-managed-instance-request"));
exports.ListAvailableWindowsUpdatesForManagedInstanceRequest = ListAvailableWindowsUpdatesForManagedInstanceRequest.ListAvailableWindowsUpdatesForManagedInstanceRequest;
const ListErrataRequest = __importStar(require("./list-errata-request"));
exports.ListErrataRequest = ListErrataRequest.ListErrataRequest;
const ListManagedInstanceErrataRequest = __importStar(require("./list-managed-instance-errata-request"));
exports.ListManagedInstanceErrataRequest = ListManagedInstanceErrataRequest.ListManagedInstanceErrataRequest;
const ListManagedInstanceGroupsRequest = __importStar(require("./list-managed-instance-groups-request"));
exports.ListManagedInstanceGroupsRequest = ListManagedInstanceGroupsRequest.ListManagedInstanceGroupsRequest;
const ListManagedInstancesRequest = __importStar(require("./list-managed-instances-request"));
exports.ListManagedInstancesRequest = ListManagedInstancesRequest.ListManagedInstancesRequest;
const ListPackagesInstalledOnManagedInstanceRequest = __importStar(require("./list-packages-installed-on-managed-instance-request"));
exports.ListPackagesInstalledOnManagedInstanceRequest = ListPackagesInstalledOnManagedInstanceRequest.ListPackagesInstalledOnManagedInstanceRequest;
const ListScheduledJobsRequest = __importStar(require("./list-scheduled-jobs-request"));
exports.ListScheduledJobsRequest = ListScheduledJobsRequest.ListScheduledJobsRequest;
const ListSoftwareSourcePackagesRequest = __importStar(require("./list-software-source-packages-request"));
exports.ListSoftwareSourcePackagesRequest = ListSoftwareSourcePackagesRequest.ListSoftwareSourcePackagesRequest;
const ListSoftwareSourcesRequest = __importStar(require("./list-software-sources-request"));
exports.ListSoftwareSourcesRequest = ListSoftwareSourcesRequest.ListSoftwareSourcesRequest;
const ListUpcomingScheduledJobsRequest = __importStar(require("./list-upcoming-scheduled-jobs-request"));
exports.ListUpcomingScheduledJobsRequest = ListUpcomingScheduledJobsRequest.ListUpcomingScheduledJobsRequest;
const ListWindowsUpdatesRequest = __importStar(require("./list-windows-updates-request"));
exports.ListWindowsUpdatesRequest = ListWindowsUpdatesRequest.ListWindowsUpdatesRequest;
const ListWindowsUpdatesInstalledOnManagedInstanceRequest = __importStar(require("./list-windows-updates-installed-on-managed-instance-request"));
exports.ListWindowsUpdatesInstalledOnManagedInstanceRequest = ListWindowsUpdatesInstalledOnManagedInstanceRequest.ListWindowsUpdatesInstalledOnManagedInstanceRequest;
const ListWorkRequestErrorsRequest = __importStar(require("./list-work-request-errors-request"));
exports.ListWorkRequestErrorsRequest = ListWorkRequestErrorsRequest.ListWorkRequestErrorsRequest;
const ListWorkRequestLogsRequest = __importStar(require("./list-work-request-logs-request"));
exports.ListWorkRequestLogsRequest = ListWorkRequestLogsRequest.ListWorkRequestLogsRequest;
const ListWorkRequestsRequest = __importStar(require("./list-work-requests-request"));
exports.ListWorkRequestsRequest = ListWorkRequestsRequest.ListWorkRequestsRequest;
const SearchSoftwarePackagesRequest = __importStar(require("./search-software-packages-request"));
exports.SearchSoftwarePackagesRequest = SearchSoftwarePackagesRequest.SearchSoftwarePackagesRequest;
