// oci-osmanagement/lib/request/list-managed-instances-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190801
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListManagedInstancesRequest = void 0;
var ListManagedInstancesRequest;
(function (ListManagedInstancesRequest) {
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListManagedInstancesRequest.SortOrder || (ListManagedInstancesRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["Timecreated"] = "TIMECREATED";
        SortBy["Displayname"] = "DISPLAYNAME";
    })(SortBy = ListManagedInstancesRequest.SortBy || (ListManagedInstancesRequest.SortBy = {}));
})(ListManagedInstancesRequest = exports.ListManagedInstancesRequest || (exports.ListManagedInstancesRequest = {}));
