// oci-loganalytics/lib/model/associable-entity.js
"use strict";
/**
 * LogAnalytics API
 * The LogAnalytics API for the LogAnalytics service.

 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssociableEntity = void 0;
var AssociableEntity;
(function (AssociableEntity) {
    let EligibilityStatus;
    (function (EligibilityStatus) {
        EligibilityStatus["Eligible"] = "ELIGIBLE";
        EligibilityStatus["Ineligible"] = "INELIGIBLE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        EligibilityStatus["UnknownValue"] = "UNKNOWN_VALUE";
    })(EligibilityStatus = AssociableEntity.EligibilityStatus || (AssociableEntity.EligibilityStatus = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AssociableEntity.getJsonObj = getJsonObj;
})(AssociableEntity = exports.AssociableEntity || (exports.AssociableEntity = {}));
