// oci-loganalytics/lib/model/creation-source-type.js
"use strict";
/**
 * LogAnalytics API
 * The LogAnalytics API for the LogAnalytics service.

 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreationSourceType = void 0;
/**
 * Entities are auto-created when enterprise manager bridge is configured in logging analytics. Similarly
 * entities can be auto-created when logs are forwarded from service connector.
 *
 **/
var CreationSourceType;
(function (CreationSourceType) {
    CreationSourceType["EmBridge"] = "EM_BRIDGE";
    CreationSourceType["ServiceConnectorHub"] = "SERVICE_CONNECTOR_HUB";
    CreationSourceType["None"] = "NONE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    CreationSourceType["UnknownValue"] = "UNKNOWN_VALUE";
})(CreationSourceType = exports.CreationSourceType || (exports.CreationSourceType = {}));
(function (CreationSourceType) {
    function getJsonObj(obj) {
        return { obj };
    }
    CreationSourceType.getJsonObj = getJsonObj;
})(CreationSourceType = exports.CreationSourceType || (exports.CreationSourceType = {}));
