// oci-loganalytics/lib/model/em-bridge-lifecycle-states.js
"use strict";
/**
 * LogAnalytics API
 * The LogAnalytics API for the LogAnalytics service.

 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmBridgeLifecycleStates = void 0;
/**
 * Enterprise manager lifecycle state. Supported states are CREATING, ACTIVE, DELETED, NEEDS_ATTENTION
 *
 **/
var EmBridgeLifecycleStates;
(function (EmBridgeLifecycleStates) {
    EmBridgeLifecycleStates["Creating"] = "CREATING";
    EmBridgeLifecycleStates["Active"] = "ACTIVE";
    EmBridgeLifecycleStates["Deleted"] = "DELETED";
    EmBridgeLifecycleStates["NeedsAttention"] = "NEEDS_ATTENTION";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    EmBridgeLifecycleStates["UnknownValue"] = "UNKNOWN_VALUE";
})(EmBridgeLifecycleStates = exports.EmBridgeLifecycleStates || (exports.EmBridgeLifecycleStates = {}));
(function (EmBridgeLifecycleStates) {
    function getJsonObj(obj) {
        return { obj };
    }
    EmBridgeLifecycleStates.getJsonObj = getJsonObj;
})(EmBridgeLifecycleStates = exports.EmBridgeLifecycleStates || (exports.EmBridgeLifecycleStates = {}));
