// oci-loganalytics/lib/model/em-bridge-latest-import-processing-status.js
"use strict";
/**
 * LogAnalytics API
 * The LogAnalytics API for the LogAnalytics service.

 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmBridgeLatestImportProcessingStatus = void 0;
/**
 * Processing status of the latest upload from enterprise manager. Supported values are NOT_STARTED, SUCCESS, IN_PROGRESS, FAILED, PARTIAL_SUCCESS
 *
 **/
var EmBridgeLatestImportProcessingStatus;
(function (EmBridgeLatestImportProcessingStatus) {
    EmBridgeLatestImportProcessingStatus["NotStarted"] = "NOT_STARTED";
    EmBridgeLatestImportProcessingStatus["Success"] = "SUCCESS";
    EmBridgeLatestImportProcessingStatus["InProgress"] = "IN_PROGRESS";
    EmBridgeLatestImportProcessingStatus["Failed"] = "FAILED";
    EmBridgeLatestImportProcessingStatus["PartialSuccess"] = "PARTIAL_SUCCESS";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    EmBridgeLatestImportProcessingStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(EmBridgeLatestImportProcessingStatus = exports.EmBridgeLatestImportProcessingStatus || (exports.EmBridgeLatestImportProcessingStatus = {}));
(function (EmBridgeLatestImportProcessingStatus) {
    function getJsonObj(obj) {
        return { obj };
    }
    EmBridgeLatestImportProcessingStatus.getJsonObj = getJsonObj;
})(EmBridgeLatestImportProcessingStatus = exports.EmBridgeLatestImportProcessingStatus || (exports.EmBridgeLatestImportProcessingStatus = {}));
