// oci-loganalytics/lib/model/entity-cloud-type.js
"use strict";
/**
 * LogAnalytics API
 * The LogAnalytics API for the LogAnalytics service.

 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EntityCloudType = void 0;
/**
 * Log analytics entity type group. This can be CLOUD (OCI) or NON_CLOUD otherwise.
 *
 **/
var EntityCloudType;
(function (EntityCloudType) {
    EntityCloudType["Cloud"] = "CLOUD";
    EntityCloudType["NonCloud"] = "NON_CLOUD";
    EntityCloudType["All"] = "ALL";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    EntityCloudType["UnknownValue"] = "UNKNOWN_VALUE";
})(EntityCloudType = exports.EntityCloudType || (exports.EntityCloudType = {}));
(function (EntityCloudType) {
    function getJsonObj(obj) {
        return { obj };
    }
    EntityCloudType.getJsonObj = getJsonObj;
})(EntityCloudType = exports.EntityCloudType || (exports.EntityCloudType = {}));
