// oci-loganalytics/lib/model/entity-lifecycle-states.js
"use strict";
/**
 * LogAnalytics API
 * The LogAnalytics API for the LogAnalytics service.

 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EntityLifecycleStates = void 0;
/**
 * Log analytics entity lifecycle state. Supported states are ACTIVE, DELETED
 *
 **/
var EntityLifecycleStates;
(function (EntityLifecycleStates) {
    EntityLifecycleStates["Active"] = "ACTIVE";
    EntityLifecycleStates["Deleted"] = "DELETED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    EntityLifecycleStates["UnknownValue"] = "UNKNOWN_VALUE";
})(EntityLifecycleStates = exports.EntityLifecycleStates || (exports.EntityLifecycleStates = {}));
(function (EntityLifecycleStates) {
    function getJsonObj(obj) {
        return { obj };
    }
    EntityLifecycleStates.getJsonObj = getJsonObj;
})(EntityLifecycleStates = exports.EntityLifecycleStates || (exports.EntityLifecycleStates = {}));
