// oci-loganalytics/lib/model/filter.js
"use strict";
/**
 * LogAnalytics API
 * The LogAnalytics API for the LogAnalytics service.

 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Filter = void 0;
var Filter;
(function (Filter) {
    let Operator;
    (function (Operator) {
        Operator["Clear"] = "CLEAR";
        Operator["Replace"] = "REPLACE";
        Operator["Equals"] = "EQUALS";
        Operator["NotEquals"] = "NOT_EQUALS";
        Operator["StartsWith"] = "STARTS_WITH";
        Operator["DoesNotStartWith"] = "DOES_NOT_START_WITH";
        Operator["EndsWith"] = "ENDS_WITH";
        Operator["DoesNotEndWith"] = "DOES_NOT_END_WITH";
        Operator["Contains"] = "CONTAINS";
        Operator["DoesNotContain"] = "DOES_NOT_CONTAIN";
        Operator["IsLessThan"] = "IS_LESS_THAN";
        Operator["IsLessThanOrEqualTo"] = "IS_LESS_THAN_OR_EQUAL_TO";
        Operator["IsGreaterThan"] = "IS_GREATER_THAN";
        Operator["IsGreaterThanOrEqualTo"] = "IS_GREATER_THAN_OR_EQUAL_TO";
        Operator["IsBetween"] = "IS_BETWEEN";
        Operator["IsNotBetween"] = "IS_NOT_BETWEEN";
        Operator["AddSubquery"] = "ADD_SUBQUERY";
        Operator["ClearSubquery"] = "CLEAR_SUBQUERY";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Operator["UnknownValue"] = "UNKNOWN_VALUE";
    })(Operator = Filter.Operator || (Filter.Operator = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    Filter.getJsonObj = getJsonObj;
})(Filter = exports.Filter || (exports.Filter = {}));
