// oci-loganalytics/lib/model/job-mode-filter.js
"use strict";
/**
 * LogAnalytics API
 * The LogAnalytics API for the LogAnalytics service.

 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.JobModeFilter = void 0;
/**
 * Job execution mode
 **/
var JobModeFilter;
(function (JobModeFilter) {
    JobModeFilter["All"] = "ALL";
    JobModeFilter["Foreground"] = "FOREGROUND";
    JobModeFilter["Background"] = "BACKGROUND";
})(JobModeFilter = exports.JobModeFilter || (exports.JobModeFilter = {}));
(function (JobModeFilter) {
    function getJsonObj(obj) {
        return { obj };
    }
    JobModeFilter.getJsonObj = getJsonObj;
})(JobModeFilter = exports.JobModeFilter || (exports.JobModeFilter = {}));
