// oci-loganalytics/lib/model/log-analytics-parser-summary.js
"use strict";
/**
 * LogAnalytics API
 * The LogAnalytics API for the LogAnalytics service.

 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LogAnalyticsParserSummary = void 0;
const model = __importStar(require("../model"));
var LogAnalyticsParserSummary;
(function (LogAnalyticsParserSummary) {
    let Type;
    (function (Type) {
        Type["Xml"] = "XML";
        Type["Json"] = "JSON";
        Type["Regex"] = "REGEX";
        Type["Odl"] = "ODL";
        Type["Delimited"] = "DELIMITED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Type["UnknownValue"] = "UNKNOWN_VALUE";
    })(Type = LogAnalyticsParserSummary.Type || (LogAnalyticsParserSummary.Type = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "fieldMaps": obj.fieldMaps
                ? obj.fieldMaps.map(item => {
                    return model.LogAnalyticsParserField.getJsonObj(item);
                })
                : undefined,
            "mappedParsers": obj.mappedParsers
                ? obj.mappedParsers.map(item => {
                    return model.LogAnalyticsParser.getJsonObj(item);
                })
                : undefined,
            "parserFilter": obj.parserFilter
                ? model.LogAnalyticsParserFilter.getJsonObj(obj.parserFilter)
                : undefined,
            "parserFunctions": obj.parserFunctions
                ? obj.parserFunctions.map(item => {
                    return model.LogAnalyticsParserFunction.getJsonObj(item);
                })
                : undefined,
            "sources": obj.sources
                ? obj.sources.map(item => {
                    return model.LogAnalyticsSource.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    LogAnalyticsParserSummary.getJsonObj = getJsonObj;
})(LogAnalyticsParserSummary = exports.LogAnalyticsParserSummary || (exports.LogAnalyticsParserSummary = {}));
