// oci-loganalytics/lib/model/log-analytics-entity-type-summary.js
"use strict";
/**
 * LogAnalytics API
 * The LogAnalytics API for the LogAnalytics service.

 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LogAnalyticsEntityTypeSummary = void 0;
var LogAnalyticsEntityTypeSummary;
(function (LogAnalyticsEntityTypeSummary) {
    let ManagementAgentEligibilityStatus;
    (function (ManagementAgentEligibilityStatus) {
        ManagementAgentEligibilityStatus["Eligible"] = "ELIGIBLE";
        ManagementAgentEligibilityStatus["Ineligible"] = "INELIGIBLE";
        ManagementAgentEligibilityStatus["Unknown"] = "UNKNOWN";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ManagementAgentEligibilityStatus["UnknownValue"] = "UNKNOWN_VALUE";
    })(ManagementAgentEligibilityStatus = LogAnalyticsEntityTypeSummary.ManagementAgentEligibilityStatus || (LogAnalyticsEntityTypeSummary.ManagementAgentEligibilityStatus = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    LogAnalyticsEntityTypeSummary.getJsonObj = getJsonObj;
})(LogAnalyticsEntityTypeSummary = exports.LogAnalyticsEntityTypeSummary || (exports.LogAnalyticsEntityTypeSummary = {}));
