// oci-loganalytics/lib/model/log-analytics-operation-types.js
"use strict";
/**
 * LogAnalytics API
 * The LogAnalytics API for the LogAnalytics service.

 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LogAnalyticsOperationTypes = void 0;
/**
 * Possible operation types.
 **/
var LogAnalyticsOperationTypes;
(function (LogAnalyticsOperationTypes) {
    LogAnalyticsOperationTypes["CreateLogAnalytics"] = "CREATE_LOG_ANALYTICS";
    LogAnalyticsOperationTypes["DeleteLogAnalytics"] = "DELETE_LOG_ANALYTICS";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    LogAnalyticsOperationTypes["UnknownValue"] = "UNKNOWN_VALUE";
})(LogAnalyticsOperationTypes = exports.LogAnalyticsOperationTypes || (exports.LogAnalyticsOperationTypes = {}));
(function (LogAnalyticsOperationTypes) {
    function getJsonObj(obj) {
        return { obj };
    }
    LogAnalyticsOperationTypes.getJsonObj = getJsonObj;
})(LogAnalyticsOperationTypes = exports.LogAnalyticsOperationTypes || (exports.LogAnalyticsOperationTypes = {}));
