// oci-loganalytics/lib/model/log-analytics-source-data-filter.js
"use strict";
/**
 * LogAnalytics API
 * The LogAnalytics API for the LogAnalytics service.

 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LogAnalyticsSourceDataFilter = void 0;
var LogAnalyticsSourceDataFilter;
(function (LogAnalyticsSourceDataFilter) {
    let FilterType;
    (function (FilterType) {
        FilterType["Mask"] = "MASK";
        FilterType["HashMask"] = "HASH_MASK";
        FilterType["DropLogEntry"] = "DROP_LOG_ENTRY";
        FilterType["DropString"] = "DROP_STRING";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        FilterType["UnknownValue"] = "UNKNOWN_VALUE";
    })(FilterType = LogAnalyticsSourceDataFilter.FilterType || (LogAnalyticsSourceDataFilter.FilterType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    LogAnalyticsSourceDataFilter.getJsonObj = getJsonObj;
})(LogAnalyticsSourceDataFilter = exports.LogAnalyticsSourceDataFilter || (exports.LogAnalyticsSourceDataFilter = {}));
