// oci-loganalytics/lib/model/object-collection-rule-lifecycle-states.js
"use strict";
/**
 * LogAnalytics API
 * The LogAnalytics API for the LogAnalytics service.

 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectCollectionRuleLifecycleStates = void 0;
/**
 * Possible object collection rule life cycle states.
 *
 **/
var ObjectCollectionRuleLifecycleStates;
(function (ObjectCollectionRuleLifecycleStates) {
    ObjectCollectionRuleLifecycleStates["Active"] = "ACTIVE";
    ObjectCollectionRuleLifecycleStates["Deleted"] = "DELETED";
    ObjectCollectionRuleLifecycleStates["Inactive"] = "INACTIVE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ObjectCollectionRuleLifecycleStates["UnknownValue"] = "UNKNOWN_VALUE";
})(ObjectCollectionRuleLifecycleStates = exports.ObjectCollectionRuleLifecycleStates || (exports.ObjectCollectionRuleLifecycleStates = {}));
(function (ObjectCollectionRuleLifecycleStates) {
    function getJsonObj(obj) {
        return { obj };
    }
    ObjectCollectionRuleLifecycleStates.getJsonObj = getJsonObj;
})(ObjectCollectionRuleLifecycleStates = exports.ObjectCollectionRuleLifecycleStates || (exports.ObjectCollectionRuleLifecycleStates = {}));
