// oci-loganalytics/lib/model/query-operation-type.js
"use strict";
/**
 * LogAnalytics API
 * The LogAnalytics API for the LogAnalytics service.

 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryOperationType = void 0;
/**
 * Asynchronous operation name for a work Request
 *
 **/
var QueryOperationType;
(function (QueryOperationType) {
    QueryOperationType["ExecuteQueryJob"] = "EXECUTE_QUERY_JOB";
    QueryOperationType["ExecutePurgeJob"] = "EXECUTE_PURGE_JOB";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    QueryOperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(QueryOperationType = exports.QueryOperationType || (exports.QueryOperationType = {}));
(function (QueryOperationType) {
    function getJsonObj(obj) {
        return { obj };
    }
    QueryOperationType.getJsonObj = getJsonObj;
})(QueryOperationType = exports.QueryOperationType || (exports.QueryOperationType = {}));
