// oci-loganalytics/lib/model/scheduled-task-summary.js
"use strict";
/**
 * LogAnalytics API
 * The LogAnalytics API for the LogAnalytics service.

 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScheduledTaskSummary = void 0;
var ScheduledTaskSummary;
(function (ScheduledTaskSummary) {
    let TaskStatus;
    (function (TaskStatus) {
        TaskStatus["Ready"] = "READY";
        TaskStatus["Paused"] = "PAUSED";
        TaskStatus["Completed"] = "COMPLETED";
        TaskStatus["Blocked"] = "BLOCKED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        TaskStatus["UnknownValue"] = "UNKNOWN_VALUE";
    })(TaskStatus = ScheduledTaskSummary.TaskStatus || (ScheduledTaskSummary.TaskStatus = {}));
    let LastExecutionStatus;
    (function (LastExecutionStatus) {
        LastExecutionStatus["Failed"] = "FAILED";
        LastExecutionStatus["Succeeded"] = "SUCCEEDED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LastExecutionStatus["UnknownValue"] = "UNKNOWN_VALUE";
    })(LastExecutionStatus = ScheduledTaskSummary.LastExecutionStatus || (ScheduledTaskSummary.LastExecutionStatus = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ScheduledTaskSummary.getJsonObj = getJsonObj;
})(ScheduledTaskSummary = exports.ScheduledTaskSummary || (exports.ScheduledTaskSummary = {}));
