// oci-loganalytics/lib/model/task-type.js
"use strict";
/**
 * LogAnalytics API
 * The LogAnalytics API for the LogAnalytics service.

 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaskType = void 0;
/**
 * Task type
 **/
var TaskType;
(function (TaskType) {
    TaskType["SavedSearch"] = "SAVED_SEARCH";
    TaskType["Acceleration"] = "ACCELERATION";
    TaskType["Purge"] = "PURGE";
    TaskType["AccelerationMaintenance"] = "ACCELERATION_MAINTENANCE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    TaskType["UnknownValue"] = "UNKNOWN_VALUE";
})(TaskType = exports.TaskType || (exports.TaskType = {}));
(function (TaskType) {
    function getJsonObj(obj) {
        return { obj };
    }
    TaskType.getJsonObj = getJsonObj;
})(TaskType = exports.TaskType || (exports.TaskType = {}));
