// oci-loganalytics/lib/model/value-type.js
"use strict";
/**
 * LogAnalytics API
 * The LogAnalytics API for the LogAnalytics service.

 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ValueType = void 0;
/**
 * Data type for fields, columns or parameters.
 **/
var ValueType;
(function (ValueType) {
    ValueType["Boolean"] = "BOOLEAN";
    ValueType["String"] = "STRING";
    ValueType["Double"] = "DOUBLE";
    ValueType["Float"] = "FLOAT";
    ValueType["Long"] = "LONG";
    ValueType["Integer"] = "INTEGER";
    ValueType["Timestamp"] = "TIMESTAMP";
    ValueType["Facet"] = "FACET";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ValueType["UnknownValue"] = "UNKNOWN_VALUE";
})(ValueType = exports.ValueType || (exports.ValueType = {}));
(function (ValueType) {
    function getJsonObj(obj) {
        return { obj };
    }
    ValueType.getJsonObj = getJsonObj;
})(ValueType = exports.ValueType || (exports.ValueType = {}));
