// oci-loganalytics/lib/request/index.js
"use strict";
/**
 * LogAnalytics API
 * The LogAnalytics API for the LogAnalytics service.

 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ValidateAssociationParametersRequest = exports.TestParserRequest = exports.RegisterLookupRequest = exports.ListWarningsRequest = exports.ListUploadsRequest = exports.ListUploadFilesRequest = exports.ListStorageWorkRequestsRequest = exports.ListStorageWorkRequestErrorsRequest = exports.ListSourcesRequest = exports.ListSourcePatternsRequest = exports.ListSourceMetaFunctionsRequest = exports.ListSourceLabelOperatorsRequest = exports.ListSourceExtendedFieldDefinitionsRequest = exports.ListSourceEventTypesRequest = exports.ListSourceAssociationsRequest = exports.ListScheduledTasksRequest = exports.ListRecalledDataRequest = exports.ListQueryWorkRequestsRequest = exports.ListParsersRequest = exports.ListParserMetaPluginsRequest = exports.ListParserFunctionsRequest = exports.ListMetaSourceTypesRequest = exports.ListLookupsRequest = exports.ListLogSetsRequest = exports.ListLogAnalyticsObjectCollectionRulesRequest = exports.ListLogAnalyticsLogGroupsRequest = exports.ListLogAnalyticsEntityTypesRequest = exports.ListLogAnalyticsEntitiesRequest = exports.ListLogAnalyticsEmBridgesRequest = exports.ListLabelsRequest = exports.ListLabelSourceDetailsRequest = exports.ListFieldsRequest = exports.ListEntitySourceAssociationsRequest = exports.ListEntityAssociationsRequest = exports.ListConfigWorkRequestsRequest = exports.ListAutoAssociationsRequest = exports.ListAssociatedEntitiesRequest = exports.ListAssociableEntitiesRequest = exports.GetQueryResultRequest = exports.ExtractStructuredLogHeaderPathsRequest = exports.ExtractStructuredLogFieldPathsRequest = exports.BatchGetBasicInfoRequest = void 0;
const BatchGetBasicInfoRequest = __importStar(require("./batch-get-basic-info-request"));
exports.BatchGetBasicInfoRequest = BatchGetBasicInfoRequest.BatchGetBasicInfoRequest;
const ExtractStructuredLogFieldPathsRequest = __importStar(require("./extract-structured-log-field-paths-request"));
exports.ExtractStructuredLogFieldPathsRequest = ExtractStructuredLogFieldPathsRequest.ExtractStructuredLogFieldPathsRequest;
const ExtractStructuredLogHeaderPathsRequest = __importStar(require("./extract-structured-log-header-paths-request"));
exports.ExtractStructuredLogHeaderPathsRequest = ExtractStructuredLogHeaderPathsRequest.ExtractStructuredLogHeaderPathsRequest;
const GetQueryResultRequest = __importStar(require("./get-query-result-request"));
exports.GetQueryResultRequest = GetQueryResultRequest.GetQueryResultRequest;
const ListAssociableEntitiesRequest = __importStar(require("./list-associable-entities-request"));
exports.ListAssociableEntitiesRequest = ListAssociableEntitiesRequest.ListAssociableEntitiesRequest;
const ListAssociatedEntitiesRequest = __importStar(require("./list-associated-entities-request"));
exports.ListAssociatedEntitiesRequest = ListAssociatedEntitiesRequest.ListAssociatedEntitiesRequest;
const ListAutoAssociationsRequest = __importStar(require("./list-auto-associations-request"));
exports.ListAutoAssociationsRequest = ListAutoAssociationsRequest.ListAutoAssociationsRequest;
const ListConfigWorkRequestsRequest = __importStar(require("./list-config-work-requests-request"));
exports.ListConfigWorkRequestsRequest = ListConfigWorkRequestsRequest.ListConfigWorkRequestsRequest;
const ListEntityAssociationsRequest = __importStar(require("./list-entity-associations-request"));
exports.ListEntityAssociationsRequest = ListEntityAssociationsRequest.ListEntityAssociationsRequest;
const ListEntitySourceAssociationsRequest = __importStar(require("./list-entity-source-associations-request"));
exports.ListEntitySourceAssociationsRequest = ListEntitySourceAssociationsRequest.ListEntitySourceAssociationsRequest;
const ListFieldsRequest = __importStar(require("./list-fields-request"));
exports.ListFieldsRequest = ListFieldsRequest.ListFieldsRequest;
const ListLabelSourceDetailsRequest = __importStar(require("./list-label-source-details-request"));
exports.ListLabelSourceDetailsRequest = ListLabelSourceDetailsRequest.ListLabelSourceDetailsRequest;
const ListLabelsRequest = __importStar(require("./list-labels-request"));
exports.ListLabelsRequest = ListLabelsRequest.ListLabelsRequest;
const ListLogAnalyticsEmBridgesRequest = __importStar(require("./list-log-analytics-em-bridges-request"));
exports.ListLogAnalyticsEmBridgesRequest = ListLogAnalyticsEmBridgesRequest.ListLogAnalyticsEmBridgesRequest;
const ListLogAnalyticsEntitiesRequest = __importStar(require("./list-log-analytics-entities-request"));
exports.ListLogAnalyticsEntitiesRequest = ListLogAnalyticsEntitiesRequest.ListLogAnalyticsEntitiesRequest;
const ListLogAnalyticsEntityTypesRequest = __importStar(require("./list-log-analytics-entity-types-request"));
exports.ListLogAnalyticsEntityTypesRequest = ListLogAnalyticsEntityTypesRequest.ListLogAnalyticsEntityTypesRequest;
const ListLogAnalyticsLogGroupsRequest = __importStar(require("./list-log-analytics-log-groups-request"));
exports.ListLogAnalyticsLogGroupsRequest = ListLogAnalyticsLogGroupsRequest.ListLogAnalyticsLogGroupsRequest;
const ListLogAnalyticsObjectCollectionRulesRequest = __importStar(require("./list-log-analytics-object-collection-rules-request"));
exports.ListLogAnalyticsObjectCollectionRulesRequest = ListLogAnalyticsObjectCollectionRulesRequest.ListLogAnalyticsObjectCollectionRulesRequest;
const ListLogSetsRequest = __importStar(require("./list-log-sets-request"));
exports.ListLogSetsRequest = ListLogSetsRequest.ListLogSetsRequest;
const ListLookupsRequest = __importStar(require("./list-lookups-request"));
exports.ListLookupsRequest = ListLookupsRequest.ListLookupsRequest;
const ListMetaSourceTypesRequest = __importStar(require("./list-meta-source-types-request"));
exports.ListMetaSourceTypesRequest = ListMetaSourceTypesRequest.ListMetaSourceTypesRequest;
const ListParserFunctionsRequest = __importStar(require("./list-parser-functions-request"));
exports.ListParserFunctionsRequest = ListParserFunctionsRequest.ListParserFunctionsRequest;
const ListParserMetaPluginsRequest = __importStar(require("./list-parser-meta-plugins-request"));
exports.ListParserMetaPluginsRequest = ListParserMetaPluginsRequest.ListParserMetaPluginsRequest;
const ListParsersRequest = __importStar(require("./list-parsers-request"));
exports.ListParsersRequest = ListParsersRequest.ListParsersRequest;
const ListQueryWorkRequestsRequest = __importStar(require("./list-query-work-requests-request"));
exports.ListQueryWorkRequestsRequest = ListQueryWorkRequestsRequest.ListQueryWorkRequestsRequest;
const ListRecalledDataRequest = __importStar(require("./list-recalled-data-request"));
exports.ListRecalledDataRequest = ListRecalledDataRequest.ListRecalledDataRequest;
const ListScheduledTasksRequest = __importStar(require("./list-scheduled-tasks-request"));
exports.ListScheduledTasksRequest = ListScheduledTasksRequest.ListScheduledTasksRequest;
const ListSourceAssociationsRequest = __importStar(require("./list-source-associations-request"));
exports.ListSourceAssociationsRequest = ListSourceAssociationsRequest.ListSourceAssociationsRequest;
const ListSourceEventTypesRequest = __importStar(require("./list-source-event-types-request"));
exports.ListSourceEventTypesRequest = ListSourceEventTypesRequest.ListSourceEventTypesRequest;
const ListSourceExtendedFieldDefinitionsRequest = __importStar(require("./list-source-extended-field-definitions-request"));
exports.ListSourceExtendedFieldDefinitionsRequest = ListSourceExtendedFieldDefinitionsRequest.ListSourceExtendedFieldDefinitionsRequest;
const ListSourceLabelOperatorsRequest = __importStar(require("./list-source-label-operators-request"));
exports.ListSourceLabelOperatorsRequest = ListSourceLabelOperatorsRequest.ListSourceLabelOperatorsRequest;
const ListSourceMetaFunctionsRequest = __importStar(require("./list-source-meta-functions-request"));
exports.ListSourceMetaFunctionsRequest = ListSourceMetaFunctionsRequest.ListSourceMetaFunctionsRequest;
const ListSourcePatternsRequest = __importStar(require("./list-source-patterns-request"));
exports.ListSourcePatternsRequest = ListSourcePatternsRequest.ListSourcePatternsRequest;
const ListSourcesRequest = __importStar(require("./list-sources-request"));
exports.ListSourcesRequest = ListSourcesRequest.ListSourcesRequest;
const ListStorageWorkRequestErrorsRequest = __importStar(require("./list-storage-work-request-errors-request"));
exports.ListStorageWorkRequestErrorsRequest = ListStorageWorkRequestErrorsRequest.ListStorageWorkRequestErrorsRequest;
const ListStorageWorkRequestsRequest = __importStar(require("./list-storage-work-requests-request"));
exports.ListStorageWorkRequestsRequest = ListStorageWorkRequestsRequest.ListStorageWorkRequestsRequest;
const ListUploadFilesRequest = __importStar(require("./list-upload-files-request"));
exports.ListUploadFilesRequest = ListUploadFilesRequest.ListUploadFilesRequest;
const ListUploadsRequest = __importStar(require("./list-uploads-request"));
exports.ListUploadsRequest = ListUploadsRequest.ListUploadsRequest;
const ListWarningsRequest = __importStar(require("./list-warnings-request"));
exports.ListWarningsRequest = ListWarningsRequest.ListWarningsRequest;
const RegisterLookupRequest = __importStar(require("./register-lookup-request"));
exports.RegisterLookupRequest = RegisterLookupRequest.RegisterLookupRequest;
const TestParserRequest = __importStar(require("./test-parser-request"));
exports.TestParserRequest = TestParserRequest.TestParserRequest;
const ValidateAssociationParametersRequest = __importStar(require("./validate-association-parameters-request"));
exports.ValidateAssociationParametersRequest = ValidateAssociationParametersRequest.ValidateAssociationParametersRequest;
