// oci-loganalytics/lib/request/list-sources-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListSourcesRequest = void 0;
var ListSourcesRequest;
(function (ListSourcesRequest) {
    let IsSystem;
    (function (IsSystem) {
        IsSystem["All"] = "ALL";
        IsSystem["Custom"] = "CUSTOM";
        IsSystem["BuiltIn"] = "BUILT_IN";
    })(IsSystem = ListSourcesRequest.IsSystem || (ListSourcesRequest.IsSystem = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListSourcesRequest.SortOrder || (ListSourcesRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["Name"] = "name";
        SortBy["TimeUpdated"] = "timeUpdated";
        SortBy["AssociationCount"] = "associationCount";
        SortBy["SourceType"] = "sourceType";
    })(SortBy = ListSourcesRequest.SortBy || (ListSourcesRequest.SortBy = {}));
})(ListSourcesRequest = exports.ListSourcesRequest || (exports.ListSourcesRequest = {}));
