// oci-loganalytics/lib/request/list-warnings-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWarningsRequest = void 0;
var ListWarningsRequest;
(function (ListWarningsRequest) {
    let WarningState;
    (function (WarningState) {
        WarningState["All"] = "ALL";
        WarningState["Suppressed"] = "SUPPRESSED";
        WarningState["Unsuppressed"] = "UNSUPPRESSED";
    })(WarningState = ListWarningsRequest.WarningState || (ListWarningsRequest.WarningState = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListWarningsRequest.SortOrder || (ListWarningsRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["EntityType"] = "EntityType";
        SortBy["SourceName"] = "SourceName";
        SortBy["PatternText"] = "PatternText";
        SortBy["FirstReported"] = "FirstReported";
        SortBy["WarningMessage"] = "WarningMessage";
        SortBy["Host"] = "Host";
        SortBy["EntityName"] = "EntityName";
        SortBy["InitialWarningDate"] = "InitialWarningDate";
    })(SortBy = ListWarningsRequest.SortBy || (ListWarningsRequest.SortBy = {}));
})(ListWarningsRequest = exports.ListWarningsRequest || (exports.ListWarningsRequest = {}));
