// oci-loganalytics/lib/request/list-labels-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListLabelsRequest = void 0;
var ListLabelsRequest;
(function (ListLabelsRequest) {
    let IsSystem;
    (function (IsSystem) {
        IsSystem["All"] = "ALL";
        IsSystem["Custom"] = "CUSTOM";
        IsSystem["BuiltIn"] = "BUILT_IN";
    })(IsSystem = ListLabelsRequest.IsSystem || (ListLabelsRequest.IsSystem = {}));
    let LabelPriority;
    (function (LabelPriority) {
        LabelPriority["None"] = "NONE";
        LabelPriority["Low"] = "LOW";
        LabelPriority["Medium"] = "MEDIUM";
        LabelPriority["High"] = "HIGH";
    })(LabelPriority = ListLabelsRequest.LabelPriority || (ListLabelsRequest.LabelPriority = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListLabelsRequest.SortOrder || (ListLabelsRequest.SortOrder = {}));
    let LabelSortBy;
    (function (LabelSortBy) {
        LabelSortBy["Name"] = "name";
        LabelSortBy["Priority"] = "priority";
        LabelSortBy["SourceUsing"] = "sourceUsing";
    })(LabelSortBy = ListLabelsRequest.LabelSortBy || (ListLabelsRequest.LabelSortBy = {}));
})(ListLabelsRequest = exports.ListLabelsRequest || (exports.ListLabelsRequest = {}));
