// oci-loganalytics/lib/request/list-lookups-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListLookupsRequest = void 0;
var ListLookupsRequest;
(function (ListLookupsRequest) {
    let Type;
    (function (Type) {
        Type["Lookup"] = "Lookup";
        Type["Dictionary"] = "Dictionary";
    })(Type = ListLookupsRequest.Type || (ListLookupsRequest.Type = {}));
    let IsSystem;
    (function (IsSystem) {
        IsSystem["All"] = "ALL";
        IsSystem["Custom"] = "CUSTOM";
        IsSystem["BuiltIn"] = "BUILT_IN";
    })(IsSystem = ListLookupsRequest.IsSystem || (ListLookupsRequest.IsSystem = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["DisplayName"] = "displayName";
        SortBy["Status"] = "status";
        SortBy["Type"] = "type";
        SortBy["UpdatedTime"] = "updatedTime";
        SortBy["CreationType"] = "creationType";
    })(SortBy = ListLookupsRequest.SortBy || (ListLookupsRequest.SortBy = {}));
    let Status;
    (function (Status) {
        Status["All"] = "ALL";
        Status["Successful"] = "SUCCESSFUL";
        Status["Failed"] = "FAILED";
        Status["Inprogress"] = "INPROGRESS";
    })(Status = ListLookupsRequest.Status || (ListLookupsRequest.Status = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListLookupsRequest.SortOrder || (ListLookupsRequest.SortOrder = {}));
})(ListLookupsRequest = exports.ListLookupsRequest || (exports.ListLookupsRequest = {}));
