// oci-loganalytics/lib/request/list-uploads-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListUploadsRequest = void 0;
var ListUploadsRequest;
(function (ListUploadsRequest) {
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListUploadsRequest.SortOrder || (ListUploadsRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeUpdated"] = "timeUpdated";
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["Name"] = "name";
    })(SortBy = ListUploadsRequest.SortBy || (ListUploadsRequest.SortBy = {}));
    let WarningsFilter;
    (function (WarningsFilter) {
        WarningsFilter["WithWarnings"] = "WITH_WARNINGS";
        WarningsFilter["WithoutWarnings"] = "WITHOUT_WARNINGS";
        WarningsFilter["All"] = "ALL";
    })(WarningsFilter = ListUploadsRequest.WarningsFilter || (ListUploadsRequest.WarningsFilter = {}));
})(ListUploadsRequest = exports.ListUploadsRequest || (exports.ListUploadsRequest = {}));
