// oci-loganalytics/lib/request/list-associable-entities-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListAssociableEntitiesRequest = void 0;
var ListAssociableEntitiesRequest;
(function (ListAssociableEntitiesRequest) {
    let Type;
    (function (Type) {
        Type["Eligible"] = "ELIGIBLE";
        Type["Ineligible"] = "INELIGIBLE";
    })(Type = ListAssociableEntitiesRequest.Type || (ListAssociableEntitiesRequest.Type = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["EntityName"] = "entityName";
        SortBy["EntityTypeName"] = "entityTypeName";
        SortBy["Host"] = "host";
        SortBy["AgentId"] = "agentId";
    })(SortBy = ListAssociableEntitiesRequest.SortBy || (ListAssociableEntitiesRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListAssociableEntitiesRequest.SortOrder || (ListAssociableEntitiesRequest.SortOrder = {}));
})(ListAssociableEntitiesRequest = exports.ListAssociableEntitiesRequest || (exports.ListAssociableEntitiesRequest = {}));
