// oci-loganalytics/lib/request/list-upload-files-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListUploadFilesRequest = void 0;
var ListUploadFilesRequest;
(function (ListUploadFilesRequest) {
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListUploadFilesRequest.SortOrder || (ListUploadFilesRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeStarted"] = "timeStarted";
        SortBy["Name"] = "name";
        SortBy["LogGroupName"] = "logGroupName";
        SortBy["SourceName"] = "sourceName";
        SortBy["Status"] = "status";
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["FileName"] = "fileName";
        SortBy["LogGroup"] = "logGroup";
    })(SortBy = ListUploadFilesRequest.SortBy || (ListUploadFilesRequest.SortBy = {}));
    let Status;
    (function (Status) {
        Status["InProgress"] = "IN_PROGRESS";
        Status["Successful"] = "SUCCESSFUL";
        Status["Failed"] = "FAILED";
    })(Status = ListUploadFilesRequest.Status || (ListUploadFilesRequest.Status = {}));
})(ListUploadFilesRequest = exports.ListUploadFilesRequest || (exports.ListUploadFilesRequest = {}));
