// oci-loganalytics/lib/request/list-meta-source-types-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListMetaSourceTypesRequest = void 0;
var ListMetaSourceTypesRequest;
(function (ListMetaSourceTypesRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["Name"] = "name";
    })(SortBy = ListMetaSourceTypesRequest.SortBy || (ListMetaSourceTypesRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListMetaSourceTypesRequest.SortOrder || (ListMetaSourceTypesRequest.SortOrder = {}));
})(ListMetaSourceTypesRequest = exports.ListMetaSourceTypesRequest || (exports.ListMetaSourceTypesRequest = {}));
