// oci-loganalytics/lib/request/list-log-analytics-entities-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListLogAnalyticsEntitiesRequest = void 0;
var ListLogAnalyticsEntitiesRequest;
(function (ListLogAnalyticsEntitiesRequest) {
    let IsManagementAgentIdNull;
    (function (IsManagementAgentIdNull) {
        IsManagementAgentIdNull["True"] = "true";
        IsManagementAgentIdNull["False"] = "false";
    })(IsManagementAgentIdNull = ListLogAnalyticsEntitiesRequest.IsManagementAgentIdNull || (ListLogAnalyticsEntitiesRequest.IsManagementAgentIdNull = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListLogAnalyticsEntitiesRequest.SortOrder || (ListLogAnalyticsEntitiesRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["TimeUpdated"] = "timeUpdated";
        SortBy["Name"] = "name";
    })(SortBy = ListLogAnalyticsEntitiesRequest.SortBy || (ListLogAnalyticsEntitiesRequest.SortBy = {}));
})(ListLogAnalyticsEntitiesRequest = exports.ListLogAnalyticsEntitiesRequest || (exports.ListLogAnalyticsEntitiesRequest = {}));
