// oci-loganalytics/lib/request/list-associated-entities-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListAssociatedEntitiesRequest = void 0;
var ListAssociatedEntitiesRequest;
(function (ListAssociatedEntitiesRequest) {
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListAssociatedEntitiesRequest.SortOrder || (ListAssociatedEntitiesRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["EntityName"] = "entityName";
        SortBy["EntityTypeDisplayName"] = "entityTypeDisplayName";
        SortBy["AssociationCount"] = "associationCount";
    })(SortBy = ListAssociatedEntitiesRequest.SortBy || (ListAssociatedEntitiesRequest.SortBy = {}));
})(ListAssociatedEntitiesRequest = exports.ListAssociatedEntitiesRequest || (exports.ListAssociatedEntitiesRequest = {}));
