// oci-loganalytics/lib/request/list-entity-source-associations-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListEntitySourceAssociationsRequest = void 0;
var ListEntitySourceAssociationsRequest;
(function (ListEntitySourceAssociationsRequest) {
    let LifeCycleState;
    (function (LifeCycleState) {
        LifeCycleState["All"] = "ALL";
        LifeCycleState["Accepted"] = "ACCEPTED";
        LifeCycleState["InProgress"] = "IN_PROGRESS";
        LifeCycleState["Succeeded"] = "SUCCEEDED";
        LifeCycleState["Failed"] = "FAILED";
    })(LifeCycleState = ListEntitySourceAssociationsRequest.LifeCycleState || (ListEntitySourceAssociationsRequest.LifeCycleState = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListEntitySourceAssociationsRequest.SortOrder || (ListEntitySourceAssociationsRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["SourceDisplayName"] = "sourceDisplayName";
        SortBy["TimeLastAttempted"] = "timeLastAttempted";
        SortBy["Status"] = "status";
    })(SortBy = ListEntitySourceAssociationsRequest.SortBy || (ListEntitySourceAssociationsRequest.SortBy = {}));
})(ListEntitySourceAssociationsRequest = exports.ListEntitySourceAssociationsRequest || (exports.ListEntitySourceAssociationsRequest = {}));
