// oci-loganalytics/lib/request/list-config-work-requests-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListConfigWorkRequestsRequest = void 0;
var ListConfigWorkRequestsRequest;
(function (ListConfigWorkRequestsRequest) {
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListConfigWorkRequestsRequest.SortOrder || (ListConfigWorkRequestsRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeAccepted"] = "timeAccepted";
    })(SortBy = ListConfigWorkRequestsRequest.SortBy || (ListConfigWorkRequestsRequest.SortBy = {}));
})(ListConfigWorkRequestsRequest = exports.ListConfigWorkRequestsRequest || (exports.ListConfigWorkRequestsRequest = {}));
