// oci-oce/lib/model/index.js
"use strict";
/**
 * Oracle Content and Experience API
 * Oracle Content and Experience is a cloud-based content hub to drive omni-channel content management and accelerate experience delivery
 * OpenAPI spec version: 20190912
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkflowStep = exports.WorkflowMonitor = exports.WorkRequestResource = exports.WorkRequestLogEntry = exports.WorkRequestError = exports.WorkRequest = exports.UpdateOceInstanceDetails = exports.OceInstanceSummary = exports.OceInstance = exports.LicenseType = exports.IdentityStripeDetails = exports.CreateOceInstanceDetails = exports.ChangeOceInstanceCompartmentDetails = void 0;
const ChangeOceInstanceCompartmentDetails = __importStar(require("./change-oce-instance-compartment-details"));
exports.ChangeOceInstanceCompartmentDetails = ChangeOceInstanceCompartmentDetails.ChangeOceInstanceCompartmentDetails;
const CreateOceInstanceDetails = __importStar(require("./create-oce-instance-details"));
exports.CreateOceInstanceDetails = CreateOceInstanceDetails.CreateOceInstanceDetails;
const IdentityStripeDetails = __importStar(require("./identity-stripe-details"));
exports.IdentityStripeDetails = IdentityStripeDetails.IdentityStripeDetails;
const LicenseType = __importStar(require("./license-type"));
exports.LicenseType = LicenseType.LicenseType;
const OceInstance = __importStar(require("./oce-instance"));
exports.OceInstance = OceInstance.OceInstance;
const OceInstanceSummary = __importStar(require("./oce-instance-summary"));
exports.OceInstanceSummary = OceInstanceSummary.OceInstanceSummary;
const UpdateOceInstanceDetails = __importStar(require("./update-oce-instance-details"));
exports.UpdateOceInstanceDetails = UpdateOceInstanceDetails.UpdateOceInstanceDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkflowMonitor = __importStar(require("./workflow-monitor"));
exports.WorkflowMonitor = WorkflowMonitor.WorkflowMonitor;
const WorkflowStep = __importStar(require("./workflow-step"));
exports.WorkflowStep = WorkflowStep.WorkflowStep;
