// oci-oce/lib/model/oce-instance.js
"use strict";
/**
 * Oracle Content and Experience API
 * Oracle Content and Experience is a cloud-based content hub to drive omni-channel content management and accelerate experience delivery
 * OpenAPI spec version: 20190912
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OceInstance = void 0;
const model = __importStar(require("../model"));
var OceInstance;
(function (OceInstance) {
    let UpgradeSchedule;
    (function (UpgradeSchedule) {
        UpgradeSchedule["UpgradeImmediately"] = "UPGRADE_IMMEDIATELY";
        UpgradeSchedule["DelayedUpgrade"] = "DELAYED_UPGRADE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        UpgradeSchedule["UnknownValue"] = "UNKNOWN_VALUE";
    })(UpgradeSchedule = OceInstance.UpgradeSchedule || (OceInstance.UpgradeSchedule = {}));
    let InstanceUsageType;
    (function (InstanceUsageType) {
        InstanceUsageType["Primary"] = "PRIMARY";
        InstanceUsageType["Nonprimary"] = "NONPRIMARY";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        InstanceUsageType["UnknownValue"] = "UNKNOWN_VALUE";
    })(InstanceUsageType = OceInstance.InstanceUsageType || (OceInstance.InstanceUsageType = {}));
    let InstanceAccessType;
    (function (InstanceAccessType) {
        InstanceAccessType["Public"] = "PUBLIC";
        InstanceAccessType["Private"] = "PRIVATE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        InstanceAccessType["UnknownValue"] = "UNKNOWN_VALUE";
    })(InstanceAccessType = OceInstance.InstanceAccessType || (OceInstance.InstanceAccessType = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = OceInstance.LifecycleState || (OceInstance.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "identityStripe": obj.identityStripe
                ? model.IdentityStripeDetails.getJsonObj(obj.identityStripe)
                : undefined
        });
        return jsonObj;
    }
    OceInstance.getJsonObj = getJsonObj;
})(OceInstance = exports.OceInstance || (exports.OceInstance = {}));
