// oci-oce/lib/model/oce-instance-summary.js
"use strict";
/**
 * Oracle Content and Experience API
 * Oracle Content and Experience is a cloud-based content hub to drive omni-channel content management and accelerate experience delivery
 * OpenAPI spec version: 20190912
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OceInstanceSummary = void 0;
var OceInstanceSummary;
(function (OceInstanceSummary) {
    let InstanceUsageType;
    (function (InstanceUsageType) {
        InstanceUsageType["Primary"] = "PRIMARY";
        InstanceUsageType["Nonprimary"] = "NONPRIMARY";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        InstanceUsageType["UnknownValue"] = "UNKNOWN_VALUE";
    })(InstanceUsageType = OceInstanceSummary.InstanceUsageType || (OceInstanceSummary.InstanceUsageType = {}));
    let InstanceAccessType;
    (function (InstanceAccessType) {
        InstanceAccessType["Public"] = "PUBLIC";
        InstanceAccessType["Private"] = "PRIVATE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        InstanceAccessType["UnknownValue"] = "UNKNOWN_VALUE";
    })(InstanceAccessType = OceInstanceSummary.InstanceAccessType || (OceInstanceSummary.InstanceAccessType = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = OceInstanceSummary.LifecycleState || (OceInstanceSummary.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    OceInstanceSummary.getJsonObj = getJsonObj;
})(OceInstanceSummary = exports.OceInstanceSummary || (exports.OceInstanceSummary = {}));
