// oci-managementagent/lib/request/list-availability-histories-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200202
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListAvailabilityHistoriesRequest = void 0;
var ListAvailabilityHistoriesRequest;
(function (ListAvailabilityHistoriesRequest) {
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListAvailabilityHistoriesRequest.SortOrder || (ListAvailabilityHistoriesRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeAvailabilityStatusStarted"] = "timeAvailabilityStatusStarted";
    })(SortBy = ListAvailabilityHistoriesRequest.SortBy || (ListAvailabilityHistoriesRequest.SortBy = {}));
})(ListAvailabilityHistoriesRequest = exports.ListAvailabilityHistoriesRequest || (exports.ListAvailabilityHistoriesRequest = {}));
