// oci-ons/lib/model/index.js
"use strict";
/**
 * Notifications API
 * Use the Notifications API to broadcast messages to distributed components by topic, using a publish-subscribe pattern.
For information about managing topics, subscriptions, and messages, see [Notifications Overview](/iaas/Content/Notification/Concepts/notificationoverview.htm).

 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateSubscriptionDetails = exports.TopicAttributesDetails = exports.SubscriptionSummary = exports.Subscription = exports.PublishResult = exports.NotificationTopicSummary = exports.NotificationTopic = exports.MessageDetails = exports.DeliveryPolicy = exports.CreateTopicDetails = exports.CreateSubscriptionDetails = exports.ConfirmationResult = exports.ChangeCompartmentDetails = exports.BackoffRetryPolicy = void 0;
const BackoffRetryPolicy = __importStar(require("./backoff-retry-policy"));
exports.BackoffRetryPolicy = BackoffRetryPolicy.BackoffRetryPolicy;
const ChangeCompartmentDetails = __importStar(require("./change-compartment-details"));
exports.ChangeCompartmentDetails = ChangeCompartmentDetails.ChangeCompartmentDetails;
const ConfirmationResult = __importStar(require("./confirmation-result"));
exports.ConfirmationResult = ConfirmationResult.ConfirmationResult;
const CreateSubscriptionDetails = __importStar(require("./create-subscription-details"));
exports.CreateSubscriptionDetails = CreateSubscriptionDetails.CreateSubscriptionDetails;
const CreateTopicDetails = __importStar(require("./create-topic-details"));
exports.CreateTopicDetails = CreateTopicDetails.CreateTopicDetails;
const DeliveryPolicy = __importStar(require("./delivery-policy"));
exports.DeliveryPolicy = DeliveryPolicy.DeliveryPolicy;
const MessageDetails = __importStar(require("./message-details"));
exports.MessageDetails = MessageDetails.MessageDetails;
const NotificationTopic = __importStar(require("./notification-topic"));
exports.NotificationTopic = NotificationTopic.NotificationTopic;
const NotificationTopicSummary = __importStar(require("./notification-topic-summary"));
exports.NotificationTopicSummary = NotificationTopicSummary.NotificationTopicSummary;
const PublishResult = __importStar(require("./publish-result"));
exports.PublishResult = PublishResult.PublishResult;
const Subscription = __importStar(require("./subscription"));
exports.Subscription = Subscription.Subscription;
const SubscriptionSummary = __importStar(require("./subscription-summary"));
exports.SubscriptionSummary = SubscriptionSummary.SubscriptionSummary;
const TopicAttributesDetails = __importStar(require("./topic-attributes-details"));
exports.TopicAttributesDetails = TopicAttributesDetails.TopicAttributesDetails;
const UpdateSubscriptionDetails = __importStar(require("./update-subscription-details"));
exports.UpdateSubscriptionDetails = UpdateSubscriptionDetails.UpdateSubscriptionDetails;
