// oci-keymanagement/lib/request/list-keys-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: release
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListKeysRequest = void 0;
var ListKeysRequest;
(function (ListKeysRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["Timecreated"] = "TIMECREATED";
        SortBy["Displayname"] = "DISPLAYNAME";
    })(SortBy = ListKeysRequest.SortBy || (ListKeysRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListKeysRequest.SortOrder || (ListKeysRequest.SortOrder = {}));
    let ProtectionMode;
    (function (ProtectionMode) {
        ProtectionMode["Hsm"] = "HSM";
        ProtectionMode["Software"] = "SOFTWARE";
    })(ProtectionMode = ListKeysRequest.ProtectionMode || (ListKeysRequest.ProtectionMode = {}));
    let Algorithm;
    (function (Algorithm) {
        Algorithm["Aes"] = "AES";
        Algorithm["Rsa"] = "RSA";
        Algorithm["Ecdsa"] = "ECDSA";
    })(Algorithm = ListKeysRequest.Algorithm || (ListKeysRequest.Algorithm = {}));
    let CurveId;
    (function (CurveId) {
        CurveId["NistP256"] = "NIST_P256";
        CurveId["NistP384"] = "NIST_P384";
        CurveId["NistP521"] = "NIST_P521";
    })(CurveId = ListKeysRequest.CurveId || (ListKeysRequest.CurveId = {}));
})(ListKeysRequest = exports.ListKeysRequest || (exports.ListKeysRequest = {}));
