// oci-dataflow/lib/model/index.js
"use strict";
/**
 * Data Flow API
 * Use the Data Flow APIs to run any Apache Spark application at any scale without deploying or managing any infrastructure.

 * OpenAPI spec version: 20200129
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestSummary = exports.WorkRequestStatus = exports.WorkRequestResource = exports.WorkRequestOperation = exports.WorkRequestLogCollection = exports.WorkRequestLog = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequestCollection = exports.WorkRequest = exports.UpdateRunDetails = exports.UpdatePrivateEndpointDetails = exports.UpdateApplicationDetails = exports.RunSummary = exports.RunLogSummary = exports.RunLifecycleState = exports.Run = exports.PrivateEndpointSummary = exports.PrivateEndpointLifecycleState = exports.PrivateEndpointCollection = exports.PrivateEndpoint = exports.CreateRunDetails = exports.CreatePrivateEndpointDetails = exports.CreateApplicationDetails = exports.ChangeRunCompartmentDetails = exports.ChangePrivateEndpointCompartmentDetails = exports.ChangeApplicationCompartmentDetails = exports.ApplicationSummary = exports.ApplicationParameter = exports.ApplicationLifecycleState = exports.ApplicationLanguage = exports.Application = void 0;
const Application = __importStar(require("./application"));
exports.Application = Application.Application;
const ApplicationLanguage = __importStar(require("./application-language"));
exports.ApplicationLanguage = ApplicationLanguage.ApplicationLanguage;
const ApplicationLifecycleState = __importStar(require("./application-lifecycle-state"));
exports.ApplicationLifecycleState = ApplicationLifecycleState.ApplicationLifecycleState;
const ApplicationParameter = __importStar(require("./application-parameter"));
exports.ApplicationParameter = ApplicationParameter.ApplicationParameter;
const ApplicationSummary = __importStar(require("./application-summary"));
exports.ApplicationSummary = ApplicationSummary.ApplicationSummary;
const ChangeApplicationCompartmentDetails = __importStar(require("./change-application-compartment-details"));
exports.ChangeApplicationCompartmentDetails = ChangeApplicationCompartmentDetails.ChangeApplicationCompartmentDetails;
const ChangePrivateEndpointCompartmentDetails = __importStar(require("./change-private-endpoint-compartment-details"));
exports.ChangePrivateEndpointCompartmentDetails = ChangePrivateEndpointCompartmentDetails.ChangePrivateEndpointCompartmentDetails;
const ChangeRunCompartmentDetails = __importStar(require("./change-run-compartment-details"));
exports.ChangeRunCompartmentDetails = ChangeRunCompartmentDetails.ChangeRunCompartmentDetails;
const CreateApplicationDetails = __importStar(require("./create-application-details"));
exports.CreateApplicationDetails = CreateApplicationDetails.CreateApplicationDetails;
const CreatePrivateEndpointDetails = __importStar(require("./create-private-endpoint-details"));
exports.CreatePrivateEndpointDetails = CreatePrivateEndpointDetails.CreatePrivateEndpointDetails;
const CreateRunDetails = __importStar(require("./create-run-details"));
exports.CreateRunDetails = CreateRunDetails.CreateRunDetails;
const PrivateEndpoint = __importStar(require("./private-endpoint"));
exports.PrivateEndpoint = PrivateEndpoint.PrivateEndpoint;
const PrivateEndpointCollection = __importStar(require("./private-endpoint-collection"));
exports.PrivateEndpointCollection = PrivateEndpointCollection.PrivateEndpointCollection;
const PrivateEndpointLifecycleState = __importStar(require("./private-endpoint-lifecycle-state"));
exports.PrivateEndpointLifecycleState = PrivateEndpointLifecycleState.PrivateEndpointLifecycleState;
const PrivateEndpointSummary = __importStar(require("./private-endpoint-summary"));
exports.PrivateEndpointSummary = PrivateEndpointSummary.PrivateEndpointSummary;
const Run = __importStar(require("./run"));
exports.Run = Run.Run;
const RunLifecycleState = __importStar(require("./run-lifecycle-state"));
exports.RunLifecycleState = RunLifecycleState.RunLifecycleState;
const RunLogSummary = __importStar(require("./run-log-summary"));
exports.RunLogSummary = RunLogSummary.RunLogSummary;
const RunSummary = __importStar(require("./run-summary"));
exports.RunSummary = RunSummary.RunSummary;
const UpdateApplicationDetails = __importStar(require("./update-application-details"));
exports.UpdateApplicationDetails = UpdateApplicationDetails.UpdateApplicationDetails;
const UpdatePrivateEndpointDetails = __importStar(require("./update-private-endpoint-details"));
exports.UpdatePrivateEndpointDetails = UpdatePrivateEndpointDetails.UpdatePrivateEndpointDetails;
const UpdateRunDetails = __importStar(require("./update-run-details"));
exports.UpdateRunDetails = UpdateRunDetails.UpdateRunDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestCollection = __importStar(require("./work-request-collection"));
exports.WorkRequestCollection = WorkRequestCollection.WorkRequestCollection;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLog = __importStar(require("./work-request-log"));
exports.WorkRequestLog = WorkRequestLog.WorkRequestLog;
const WorkRequestLogCollection = __importStar(require("./work-request-log-collection"));
exports.WorkRequestLogCollection = WorkRequestLogCollection.WorkRequestLogCollection;
const WorkRequestOperation = __importStar(require("./work-request-operation"));
exports.WorkRequestOperation = WorkRequestOperation.WorkRequestOperation;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestStatus = __importStar(require("./work-request-status"));
exports.WorkRequestStatus = WorkRequestStatus.WorkRequestStatus;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
