// oci-dataflow/lib/model/private-endpoint-lifecycle-state.js
"use strict";
/**
 * Data Flow API
 * Use the Data Flow APIs to run any Apache Spark application at any scale without deploying or managing any infrastructure.

 * OpenAPI spec version: 20200129
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateEndpointLifecycleState = void 0;
/**
 * The lifecycle state of a private endpoint.
 *
 **/
var PrivateEndpointLifecycleState;
(function (PrivateEndpointLifecycleState) {
    PrivateEndpointLifecycleState["Creating"] = "CREATING";
    PrivateEndpointLifecycleState["Active"] = "ACTIVE";
    PrivateEndpointLifecycleState["Inactive"] = "INACTIVE";
    PrivateEndpointLifecycleState["Updating"] = "UPDATING";
    PrivateEndpointLifecycleState["Deleting"] = "DELETING";
    PrivateEndpointLifecycleState["Deleted"] = "DELETED";
    PrivateEndpointLifecycleState["Failed"] = "FAILED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    PrivateEndpointLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(PrivateEndpointLifecycleState = exports.PrivateEndpointLifecycleState || (exports.PrivateEndpointLifecycleState = {}));
(function (PrivateEndpointLifecycleState) {
    function getJsonObj(obj) {
        return { obj };
    }
    PrivateEndpointLifecycleState.getJsonObj = getJsonObj;
})(PrivateEndpointLifecycleState = exports.PrivateEndpointLifecycleState || (exports.PrivateEndpointLifecycleState = {}));
