// oci-dataflow/lib/model/run-lifecycle-state.js
"use strict";
/**
 * Data Flow API
 * Use the Data Flow APIs to run any Apache Spark application at any scale without deploying or managing any infrastructure.

 * OpenAPI spec version: 20200129
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RunLifecycleState = void 0;
/**
 * The lifecycle state of a run.
 *
 **/
var RunLifecycleState;
(function (RunLifecycleState) {
    RunLifecycleState["Accepted"] = "ACCEPTED";
    RunLifecycleState["InProgress"] = "IN_PROGRESS";
    RunLifecycleState["Canceling"] = "CANCELING";
    RunLifecycleState["Canceled"] = "CANCELED";
    RunLifecycleState["Failed"] = "FAILED";
    RunLifecycleState["Succeeded"] = "SUCCEEDED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    RunLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(RunLifecycleState = exports.RunLifecycleState || (exports.RunLifecycleState = {}));
(function (RunLifecycleState) {
    function getJsonObj(obj) {
        return { obj };
    }
    RunLifecycleState.getJsonObj = getJsonObj;
})(RunLifecycleState = exports.RunLifecycleState || (exports.RunLifecycleState = {}));
