// oci-dataflow/lib/model/run-log-summary.js
"use strict";
/**
 * Data Flow API
 * Use the Data Flow APIs to run any Apache Spark application at any scale without deploying or managing any infrastructure.

 * OpenAPI spec version: 20200129
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RunLogSummary = void 0;
var RunLogSummary;
(function (RunLogSummary) {
    let Source;
    (function (Source) {
        Source["Application"] = "APPLICATION";
        Source["Driver"] = "DRIVER";
        Source["Executor"] = "EXECUTOR";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Source["UnknownValue"] = "UNKNOWN_VALUE";
    })(Source = RunLogSummary.Source || (RunLogSummary.Source = {}));
    let Type;
    (function (Type) {
        Type["Stderr"] = "STDERR";
        Type["Stdout"] = "STDOUT";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Type["UnknownValue"] = "UNKNOWN_VALUE";
    })(Type = RunLogSummary.Type || (RunLogSummary.Type = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    RunLogSummary.getJsonObj = getJsonObj;
})(RunLogSummary = exports.RunLogSummary || (exports.RunLogSummary = {}));
