// oci-dataflow/lib/model/work-request-status.js
"use strict";
/**
 * Data Flow API
 * Use the Data Flow APIs to run any Apache Spark application at any scale without deploying or managing any infrastructure.

 * OpenAPI spec version: 20200129
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestStatus = void 0;
/**
 * The status of the work request.
 *
 **/
var WorkRequestStatus;
(function (WorkRequestStatus) {
    WorkRequestStatus["Accepted"] = "ACCEPTED";
    WorkRequestStatus["Cancelled"] = "CANCELLED";
    WorkRequestStatus["Cancelling"] = "CANCELLING";
    WorkRequestStatus["Failed"] = "FAILED";
    WorkRequestStatus["Inprogress"] = "INPROGRESS";
    WorkRequestStatus["Succeeded"] = "SUCCEEDED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    WorkRequestStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(WorkRequestStatus = exports.WorkRequestStatus || (exports.WorkRequestStatus = {}));
(function (WorkRequestStatus) {
    function getJsonObj(obj) {
        return { obj };
    }
    WorkRequestStatus.getJsonObj = getJsonObj;
})(WorkRequestStatus = exports.WorkRequestStatus || (exports.WorkRequestStatus = {}));
