// oci-datascience/lib/model/index.js
"use strict";
/**
 * Data Science API
 * Use the Data Science APIs to organize your data science work, access data and computing resources, and build, train, deploy, and manage models on Oracle Cloud.

 * OpenAPI spec version: 20190101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateSingleModelDeploymentConfigurationDetails = exports.SingleModelDeploymentConfigurationDetails = exports.FixedSizeScalingPolicy = exports.WorkRequestSummary = exports.WorkRequestStatus = exports.WorkRequestResource = exports.WorkRequestOperationType = exports.WorkRequestLogEntry = exports.WorkRequestError = exports.WorkRequest = exports.UpdateProjectDetails = exports.UpdateNotebookSessionDetails = exports.UpdateModelProvenanceDetails = exports.UpdateModelDetails = exports.UpdateModelDeploymentDetails = exports.UpdateModelDeploymentConfigurationDetails = exports.UpdateModelConfigurationDetails = exports.UpdateCategoryLogDetails = exports.ScalingPolicy = exports.ProjectSummary = exports.ProjectLifecycleState = exports.Project = exports.NotebookSessionSummary = exports.NotebookSessionShapeSummary = exports.NotebookSessionLifecycleState = exports.NotebookSessionConfigurationDetails = exports.NotebookSession = exports.ModelSummary = exports.ModelProvenance = exports.ModelLifecycleState = exports.ModelDeploymentType = exports.ModelDeploymentSummary = exports.ModelDeploymentShapeSummary = exports.ModelDeploymentLifecycleState = exports.ModelDeploymentConfigurationDetails = exports.ModelDeployment = exports.ModelConfigurationDetails = exports.Model = exports.LogDetails = exports.InstanceConfiguration = exports.CreateProjectDetails = exports.CreateNotebookSessionDetails = exports.CreateModelProvenanceDetails = exports.CreateModelDetails = exports.CreateModelDeploymentDetails = exports.ChangeProjectCompartmentDetails = exports.ChangeNotebookSessionCompartmentDetails = exports.ChangeModelDeploymentCompartmentDetails = exports.ChangeModelCompartmentDetails = exports.CategoryLogDetails = void 0;
const CategoryLogDetails = __importStar(require("./category-log-details"));
exports.CategoryLogDetails = CategoryLogDetails.CategoryLogDetails;
const ChangeModelCompartmentDetails = __importStar(require("./change-model-compartment-details"));
exports.ChangeModelCompartmentDetails = ChangeModelCompartmentDetails.ChangeModelCompartmentDetails;
const ChangeModelDeploymentCompartmentDetails = __importStar(require("./change-model-deployment-compartment-details"));
exports.ChangeModelDeploymentCompartmentDetails = ChangeModelDeploymentCompartmentDetails.ChangeModelDeploymentCompartmentDetails;
const ChangeNotebookSessionCompartmentDetails = __importStar(require("./change-notebook-session-compartment-details"));
exports.ChangeNotebookSessionCompartmentDetails = ChangeNotebookSessionCompartmentDetails.ChangeNotebookSessionCompartmentDetails;
const ChangeProjectCompartmentDetails = __importStar(require("./change-project-compartment-details"));
exports.ChangeProjectCompartmentDetails = ChangeProjectCompartmentDetails.ChangeProjectCompartmentDetails;
const CreateModelDeploymentDetails = __importStar(require("./create-model-deployment-details"));
exports.CreateModelDeploymentDetails = CreateModelDeploymentDetails.CreateModelDeploymentDetails;
const CreateModelDetails = __importStar(require("./create-model-details"));
exports.CreateModelDetails = CreateModelDetails.CreateModelDetails;
const CreateModelProvenanceDetails = __importStar(require("./create-model-provenance-details"));
exports.CreateModelProvenanceDetails = CreateModelProvenanceDetails.CreateModelProvenanceDetails;
const CreateNotebookSessionDetails = __importStar(require("./create-notebook-session-details"));
exports.CreateNotebookSessionDetails = CreateNotebookSessionDetails.CreateNotebookSessionDetails;
const CreateProjectDetails = __importStar(require("./create-project-details"));
exports.CreateProjectDetails = CreateProjectDetails.CreateProjectDetails;
const InstanceConfiguration = __importStar(require("./instance-configuration"));
exports.InstanceConfiguration = InstanceConfiguration.InstanceConfiguration;
const LogDetails = __importStar(require("./log-details"));
exports.LogDetails = LogDetails.LogDetails;
const Model = __importStar(require("./model"));
exports.Model = Model.Model;
const ModelConfigurationDetails = __importStar(require("./model-configuration-details"));
exports.ModelConfigurationDetails = ModelConfigurationDetails.ModelConfigurationDetails;
const ModelDeployment = __importStar(require("./model-deployment"));
exports.ModelDeployment = ModelDeployment.ModelDeployment;
const ModelDeploymentConfigurationDetails = __importStar(require("./model-deployment-configuration-details"));
exports.ModelDeploymentConfigurationDetails = ModelDeploymentConfigurationDetails.ModelDeploymentConfigurationDetails;
const ModelDeploymentLifecycleState = __importStar(require("./model-deployment-lifecycle-state"));
exports.ModelDeploymentLifecycleState = ModelDeploymentLifecycleState.ModelDeploymentLifecycleState;
const ModelDeploymentShapeSummary = __importStar(require("./model-deployment-shape-summary"));
exports.ModelDeploymentShapeSummary = ModelDeploymentShapeSummary.ModelDeploymentShapeSummary;
const ModelDeploymentSummary = __importStar(require("./model-deployment-summary"));
exports.ModelDeploymentSummary = ModelDeploymentSummary.ModelDeploymentSummary;
const ModelDeploymentType = __importStar(require("./model-deployment-type"));
exports.ModelDeploymentType = ModelDeploymentType.ModelDeploymentType;
const ModelLifecycleState = __importStar(require("./model-lifecycle-state"));
exports.ModelLifecycleState = ModelLifecycleState.ModelLifecycleState;
const ModelProvenance = __importStar(require("./model-provenance"));
exports.ModelProvenance = ModelProvenance.ModelProvenance;
const ModelSummary = __importStar(require("./model-summary"));
exports.ModelSummary = ModelSummary.ModelSummary;
const NotebookSession = __importStar(require("./notebook-session"));
exports.NotebookSession = NotebookSession.NotebookSession;
const NotebookSessionConfigurationDetails = __importStar(require("./notebook-session-configuration-details"));
exports.NotebookSessionConfigurationDetails = NotebookSessionConfigurationDetails.NotebookSessionConfigurationDetails;
const NotebookSessionLifecycleState = __importStar(require("./notebook-session-lifecycle-state"));
exports.NotebookSessionLifecycleState = NotebookSessionLifecycleState.NotebookSessionLifecycleState;
const NotebookSessionShapeSummary = __importStar(require("./notebook-session-shape-summary"));
exports.NotebookSessionShapeSummary = NotebookSessionShapeSummary.NotebookSessionShapeSummary;
const NotebookSessionSummary = __importStar(require("./notebook-session-summary"));
exports.NotebookSessionSummary = NotebookSessionSummary.NotebookSessionSummary;
const Project = __importStar(require("./project"));
exports.Project = Project.Project;
const ProjectLifecycleState = __importStar(require("./project-lifecycle-state"));
exports.ProjectLifecycleState = ProjectLifecycleState.ProjectLifecycleState;
const ProjectSummary = __importStar(require("./project-summary"));
exports.ProjectSummary = ProjectSummary.ProjectSummary;
const ScalingPolicy = __importStar(require("./scaling-policy"));
exports.ScalingPolicy = ScalingPolicy.ScalingPolicy;
const UpdateCategoryLogDetails = __importStar(require("./update-category-log-details"));
exports.UpdateCategoryLogDetails = UpdateCategoryLogDetails.UpdateCategoryLogDetails;
const UpdateModelConfigurationDetails = __importStar(require("./update-model-configuration-details"));
exports.UpdateModelConfigurationDetails = UpdateModelConfigurationDetails.UpdateModelConfigurationDetails;
const UpdateModelDeploymentConfigurationDetails = __importStar(require("./update-model-deployment-configuration-details"));
exports.UpdateModelDeploymentConfigurationDetails = UpdateModelDeploymentConfigurationDetails.UpdateModelDeploymentConfigurationDetails;
const UpdateModelDeploymentDetails = __importStar(require("./update-model-deployment-details"));
exports.UpdateModelDeploymentDetails = UpdateModelDeploymentDetails.UpdateModelDeploymentDetails;
const UpdateModelDetails = __importStar(require("./update-model-details"));
exports.UpdateModelDetails = UpdateModelDetails.UpdateModelDetails;
const UpdateModelProvenanceDetails = __importStar(require("./update-model-provenance-details"));
exports.UpdateModelProvenanceDetails = UpdateModelProvenanceDetails.UpdateModelProvenanceDetails;
const UpdateNotebookSessionDetails = __importStar(require("./update-notebook-session-details"));
exports.UpdateNotebookSessionDetails = UpdateNotebookSessionDetails.UpdateNotebookSessionDetails;
const UpdateProjectDetails = __importStar(require("./update-project-details"));
exports.UpdateProjectDetails = UpdateProjectDetails.UpdateProjectDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestOperationType = __importStar(require("./work-request-operation-type"));
exports.WorkRequestOperationType = WorkRequestOperationType.WorkRequestOperationType;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestStatus = __importStar(require("./work-request-status"));
exports.WorkRequestStatus = WorkRequestStatus.WorkRequestStatus;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const FixedSizeScalingPolicy = __importStar(require("./fixed-size-scaling-policy"));
exports.FixedSizeScalingPolicy = FixedSizeScalingPolicy.FixedSizeScalingPolicy;
const SingleModelDeploymentConfigurationDetails = __importStar(require("./single-model-deployment-configuration-details"));
exports.SingleModelDeploymentConfigurationDetails = SingleModelDeploymentConfigurationDetails.SingleModelDeploymentConfigurationDetails;
const UpdateSingleModelDeploymentConfigurationDetails = __importStar(require("./update-single-model-deployment-configuration-details"));
exports.UpdateSingleModelDeploymentConfigurationDetails = UpdateSingleModelDeploymentConfigurationDetails.UpdateSingleModelDeploymentConfigurationDetails;
