// oci-datascience/lib/model/model-deployment-lifecycle-state.js
"use strict";
/**
 * Data Science API
 * Use the Data Science APIs to organize your data science work, access data and computing resources, and build, train, deploy, and manage models on Oracle Cloud.

 * OpenAPI spec version: 20190101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModelDeploymentLifecycleState = void 0;
/**
 * The lifecycle state of a model deployment.
 **/
var ModelDeploymentLifecycleState;
(function (ModelDeploymentLifecycleState) {
    ModelDeploymentLifecycleState["Creating"] = "CREATING";
    ModelDeploymentLifecycleState["Active"] = "ACTIVE";
    ModelDeploymentLifecycleState["Deleting"] = "DELETING";
    ModelDeploymentLifecycleState["Failed"] = "FAILED";
    ModelDeploymentLifecycleState["Inactive"] = "INACTIVE";
    ModelDeploymentLifecycleState["Updating"] = "UPDATING";
    ModelDeploymentLifecycleState["Deleted"] = "DELETED";
    ModelDeploymentLifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ModelDeploymentLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(ModelDeploymentLifecycleState = exports.ModelDeploymentLifecycleState || (exports.ModelDeploymentLifecycleState = {}));
(function (ModelDeploymentLifecycleState) {
    function getJsonObj(obj) {
        return { obj };
    }
    ModelDeploymentLifecycleState.getJsonObj = getJsonObj;
})(ModelDeploymentLifecycleState = exports.ModelDeploymentLifecycleState || (exports.ModelDeploymentLifecycleState = {}));
