// oci-datascience/lib/model/notebook-session-lifecycle-state.js
"use strict";
/**
 * Data Science API
 * Use the Data Science APIs to organize your data science work, access data and computing resources, and build, train, deploy, and manage models on Oracle Cloud.

 * OpenAPI spec version: 20190101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotebookSessionLifecycleState = void 0;
/**
 * The lifecycle state of a notebook session.
 **/
var NotebookSessionLifecycleState;
(function (NotebookSessionLifecycleState) {
    NotebookSessionLifecycleState["Creating"] = "CREATING";
    NotebookSessionLifecycleState["Active"] = "ACTIVE";
    NotebookSessionLifecycleState["Deleting"] = "DELETING";
    NotebookSessionLifecycleState["Deleted"] = "DELETED";
    NotebookSessionLifecycleState["Failed"] = "FAILED";
    NotebookSessionLifecycleState["Inactive"] = "INACTIVE";
    NotebookSessionLifecycleState["Updating"] = "UPDATING";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    NotebookSessionLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(NotebookSessionLifecycleState = exports.NotebookSessionLifecycleState || (exports.NotebookSessionLifecycleState = {}));
(function (NotebookSessionLifecycleState) {
    function getJsonObj(obj) {
        return { obj };
    }
    NotebookSessionLifecycleState.getJsonObj = getJsonObj;
})(NotebookSessionLifecycleState = exports.NotebookSessionLifecycleState || (exports.NotebookSessionLifecycleState = {}));
