// oci-datascience/lib/model/work-request-operation-type.js
"use strict";
/**
 * Data Science API
 * Use the Data Science APIs to organize your data science work, access data and computing resources, and build, train, deploy, and manage models on Oracle Cloud.

 * OpenAPI spec version: 20190101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestOperationType = void 0;
/**
 * The types of work request operations.
 **/
var WorkRequestOperationType;
(function (WorkRequestOperationType) {
    WorkRequestOperationType["NotebookSessionCreate"] = "NOTEBOOK_SESSION_CREATE";
    WorkRequestOperationType["NotebookSessionDelete"] = "NOTEBOOK_SESSION_DELETE";
    WorkRequestOperationType["NotebookSessionActivate"] = "NOTEBOOK_SESSION_ACTIVATE";
    WorkRequestOperationType["NotebookSessionDeactivate"] = "NOTEBOOK_SESSION_DEACTIVATE";
    WorkRequestOperationType["ModelDeploymentCreate"] = "MODEL_DEPLOYMENT_CREATE";
    WorkRequestOperationType["ModelDeploymentDelete"] = "MODEL_DEPLOYMENT_DELETE";
    WorkRequestOperationType["ModelDeploymentActivate"] = "MODEL_DEPLOYMENT_ACTIVATE";
    WorkRequestOperationType["ModelDeploymentDeactivate"] = "MODEL_DEPLOYMENT_DEACTIVATE";
    WorkRequestOperationType["ModelDeploymentUpdate"] = "MODEL_DEPLOYMENT_UPDATE";
    WorkRequestOperationType["ProjectDelete"] = "PROJECT_DELETE";
    WorkRequestOperationType["WorkrequestCancel"] = "WORKREQUEST_CANCEL";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    WorkRequestOperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(WorkRequestOperationType = exports.WorkRequestOperationType || (exports.WorkRequestOperationType = {}));
(function (WorkRequestOperationType) {
    function getJsonObj(obj) {
        return { obj };
    }
    WorkRequestOperationType.getJsonObj = getJsonObj;
})(WorkRequestOperationType = exports.WorkRequestOperationType || (exports.WorkRequestOperationType = {}));
