// oci-email/lib/model/index.js
"use strict";
/**
 * Email Delivery API
 * API for the Email Delivery service. Use this API to send high-volume, application-generated
emails. For more information, see [Overview of the Email Delivery Service](/iaas/Content/Email/Concepts/overview.htm).


**Note:** Write actions (POST, UPDATE, DELETE) may take several minutes to propagate and be reflected by the API. If a subsequent read request fails to reflect your changes, wait a few minutes and try again.

 * OpenAPI spec version: 20170907
 * Contact: email-dev_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateSenderDetails = exports.SuppressionSummary = exports.Suppression = exports.SenderSummary = exports.Sender = exports.CreateSuppressionDetails = exports.CreateSenderDetails = exports.ChangeSenderCompartmentDetails = void 0;
const ChangeSenderCompartmentDetails = __importStar(require("./change-sender-compartment-details"));
exports.ChangeSenderCompartmentDetails = ChangeSenderCompartmentDetails.ChangeSenderCompartmentDetails;
const CreateSenderDetails = __importStar(require("./create-sender-details"));
exports.CreateSenderDetails = CreateSenderDetails.CreateSenderDetails;
const CreateSuppressionDetails = __importStar(require("./create-suppression-details"));
exports.CreateSuppressionDetails = CreateSuppressionDetails.CreateSuppressionDetails;
const Sender = __importStar(require("./sender"));
exports.Sender = Sender.Sender;
const SenderSummary = __importStar(require("./sender-summary"));
exports.SenderSummary = SenderSummary.SenderSummary;
const Suppression = __importStar(require("./suppression"));
exports.Suppression = Suppression.Suppression;
const SuppressionSummary = __importStar(require("./suppression-summary"));
exports.SuppressionSummary = SuppressionSummary.SuppressionSummary;
const UpdateSenderDetails = __importStar(require("./update-sender-details"));
exports.UpdateSenderDetails = UpdateSenderDetails.UpdateSenderDetails;
