// oci-logging/lib/model/index.js
"use strict";
/**
 * Logging Management API
 * Use the Logging Management API to create, read, list, update, and delete log groups, log objects, and agent configurations.
 * OpenAPI spec version: 20200531
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestError = exports.WorkRequest = exports.UpdateUnifiedAgentConfigurationDetails = exports.UpdateLogSavedSearchDetails = exports.UpdateLogGroupDetails = exports.UpdateLogDetails = exports.UpdateConfigurationDetails = exports.UnifiedAgentServiceConfigurationTypes = exports.UnifiedAgentServiceConfigurationStates = exports.UnifiedAgentServiceConfigurationDetails = exports.UnifiedAgentParser = exports.UnifiedAgentLoggingSource = exports.UnifiedAgentLoggingDestination = exports.UnifiedAgentConfigurationSummary = exports.UnifiedAgentConfigurationCollection = exports.UnifiedAgentConfiguration = exports.SourceUpdateDetails = exports.Source = exports.ServiceSummary = exports.ResourceType = exports.Parameter = exports.OperationTypes = exports.OperationStatus = exports.LogSummary = exports.LogSavedSearchSummaryCollection = exports.LogSavedSearchSummary = exports.LogSavedSearchLifecycleState = exports.LogSavedSearch = exports.LogLifecycleState = exports.LogIncludedSearchSummaryCollection = exports.LogIncludedSearchSummary = exports.LogIncludedSearch = exports.LogGroupSummary = exports.LogGroupLifecycleState = exports.LogGroup = exports.Log = exports.GroupAssociationDetails = exports.GrokPattern = exports.CreateUnifiedAgentConfigurationDetails = exports.CreateLogSavedSearchDetails = exports.CreateLogGroupDetails = exports.CreateLogDetails = exports.Configuration = exports.ChangeUnifiedAgentConfigurationCompartmentDetails = exports.ChangeLogSavedSearchCompartmentDetails = exports.ChangeLogLogGroupDetails = exports.ChangeLogGroupCompartmentDetails = exports.Category = exports.Archiving = exports.ActionTypes = void 0;
exports.UnifiedJSONParser = exports.UnifiedAgentWindowsEventSource = exports.UnifiedAgentTsvParser = exports.UnifiedAgentTailLogSource = exports.UnifiedAgentSyslogParser = exports.UnifiedAgentRegexParser = exports.UnifiedAgentNoneParser = exports.UnifiedAgentMultilineParser = exports.UnifiedAgentMultilineGrokParser = exports.UnifiedAgentMsgpackParser = exports.UnifiedAgentLoggingConfiguration = exports.UnifiedAgentGrokParser = exports.UnifiedAgentCsvParser = exports.UnifiedAgentAuditdParser = exports.UnifiedAgentApacheErrorParser = exports.UnifiedAgentApache2Parser = exports.OciService = exports.WorkRequestSummary = exports.WorkRequestResource = exports.WorkRequestLog = void 0;
const ActionTypes = __importStar(require("./action-types"));
exports.ActionTypes = ActionTypes.ActionTypes;
const Archiving = __importStar(require("./archiving"));
exports.Archiving = Archiving.Archiving;
const Category = __importStar(require("./category"));
exports.Category = Category.Category;
const ChangeLogGroupCompartmentDetails = __importStar(require("./change-log-group-compartment-details"));
exports.ChangeLogGroupCompartmentDetails = ChangeLogGroupCompartmentDetails.ChangeLogGroupCompartmentDetails;
const ChangeLogLogGroupDetails = __importStar(require("./change-log-log-group-details"));
exports.ChangeLogLogGroupDetails = ChangeLogLogGroupDetails.ChangeLogLogGroupDetails;
const ChangeLogSavedSearchCompartmentDetails = __importStar(require("./change-log-saved-search-compartment-details"));
exports.ChangeLogSavedSearchCompartmentDetails = ChangeLogSavedSearchCompartmentDetails.ChangeLogSavedSearchCompartmentDetails;
const ChangeUnifiedAgentConfigurationCompartmentDetails = __importStar(require("./change-unified-agent-configuration-compartment-details"));
exports.ChangeUnifiedAgentConfigurationCompartmentDetails = ChangeUnifiedAgentConfigurationCompartmentDetails.ChangeUnifiedAgentConfigurationCompartmentDetails;
const Configuration = __importStar(require("./configuration"));
exports.Configuration = Configuration.Configuration;
const CreateLogDetails = __importStar(require("./create-log-details"));
exports.CreateLogDetails = CreateLogDetails.CreateLogDetails;
const CreateLogGroupDetails = __importStar(require("./create-log-group-details"));
exports.CreateLogGroupDetails = CreateLogGroupDetails.CreateLogGroupDetails;
const CreateLogSavedSearchDetails = __importStar(require("./create-log-saved-search-details"));
exports.CreateLogSavedSearchDetails = CreateLogSavedSearchDetails.CreateLogSavedSearchDetails;
const CreateUnifiedAgentConfigurationDetails = __importStar(require("./create-unified-agent-configuration-details"));
exports.CreateUnifiedAgentConfigurationDetails = CreateUnifiedAgentConfigurationDetails.CreateUnifiedAgentConfigurationDetails;
const GrokPattern = __importStar(require("./grok-pattern"));
exports.GrokPattern = GrokPattern.GrokPattern;
const GroupAssociationDetails = __importStar(require("./group-association-details"));
exports.GroupAssociationDetails = GroupAssociationDetails.GroupAssociationDetails;
const Log = __importStar(require("./log"));
exports.Log = Log.Log;
const LogGroup = __importStar(require("./log-group"));
exports.LogGroup = LogGroup.LogGroup;
const LogGroupLifecycleState = __importStar(require("./log-group-lifecycle-state"));
exports.LogGroupLifecycleState = LogGroupLifecycleState.LogGroupLifecycleState;
const LogGroupSummary = __importStar(require("./log-group-summary"));
exports.LogGroupSummary = LogGroupSummary.LogGroupSummary;
const LogIncludedSearch = __importStar(require("./log-included-search"));
exports.LogIncludedSearch = LogIncludedSearch.LogIncludedSearch;
const LogIncludedSearchSummary = __importStar(require("./log-included-search-summary"));
exports.LogIncludedSearchSummary = LogIncludedSearchSummary.LogIncludedSearchSummary;
const LogIncludedSearchSummaryCollection = __importStar(require("./log-included-search-summary-collection"));
exports.LogIncludedSearchSummaryCollection = LogIncludedSearchSummaryCollection.LogIncludedSearchSummaryCollection;
const LogLifecycleState = __importStar(require("./log-lifecycle-state"));
exports.LogLifecycleState = LogLifecycleState.LogLifecycleState;
const LogSavedSearch = __importStar(require("./log-saved-search"));
exports.LogSavedSearch = LogSavedSearch.LogSavedSearch;
const LogSavedSearchLifecycleState = __importStar(require("./log-saved-search-lifecycle-state"));
exports.LogSavedSearchLifecycleState = LogSavedSearchLifecycleState.LogSavedSearchLifecycleState;
const LogSavedSearchSummary = __importStar(require("./log-saved-search-summary"));
exports.LogSavedSearchSummary = LogSavedSearchSummary.LogSavedSearchSummary;
const LogSavedSearchSummaryCollection = __importStar(require("./log-saved-search-summary-collection"));
exports.LogSavedSearchSummaryCollection = LogSavedSearchSummaryCollection.LogSavedSearchSummaryCollection;
const LogSummary = __importStar(require("./log-summary"));
exports.LogSummary = LogSummary.LogSummary;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationTypes = __importStar(require("./operation-types"));
exports.OperationTypes = OperationTypes.OperationTypes;
const Parameter = __importStar(require("./parameter"));
exports.Parameter = Parameter.Parameter;
const ResourceType = __importStar(require("./resource-type"));
exports.ResourceType = ResourceType.ResourceType;
const ServiceSummary = __importStar(require("./service-summary"));
exports.ServiceSummary = ServiceSummary.ServiceSummary;
const Source = __importStar(require("./source"));
exports.Source = Source.Source;
const SourceUpdateDetails = __importStar(require("./source-update-details"));
exports.SourceUpdateDetails = SourceUpdateDetails.SourceUpdateDetails;
const UnifiedAgentConfiguration = __importStar(require("./unified-agent-configuration"));
exports.UnifiedAgentConfiguration = UnifiedAgentConfiguration.UnifiedAgentConfiguration;
const UnifiedAgentConfigurationCollection = __importStar(require("./unified-agent-configuration-collection"));
exports.UnifiedAgentConfigurationCollection = UnifiedAgentConfigurationCollection.UnifiedAgentConfigurationCollection;
const UnifiedAgentConfigurationSummary = __importStar(require("./unified-agent-configuration-summary"));
exports.UnifiedAgentConfigurationSummary = UnifiedAgentConfigurationSummary.UnifiedAgentConfigurationSummary;
const UnifiedAgentLoggingDestination = __importStar(require("./unified-agent-logging-destination"));
exports.UnifiedAgentLoggingDestination = UnifiedAgentLoggingDestination.UnifiedAgentLoggingDestination;
const UnifiedAgentLoggingSource = __importStar(require("./unified-agent-logging-source"));
exports.UnifiedAgentLoggingSource = UnifiedAgentLoggingSource.UnifiedAgentLoggingSource;
const UnifiedAgentParser = __importStar(require("./unified-agent-parser"));
exports.UnifiedAgentParser = UnifiedAgentParser.UnifiedAgentParser;
const UnifiedAgentServiceConfigurationDetails = __importStar(require("./unified-agent-service-configuration-details"));
exports.UnifiedAgentServiceConfigurationDetails = UnifiedAgentServiceConfigurationDetails.UnifiedAgentServiceConfigurationDetails;
const UnifiedAgentServiceConfigurationStates = __importStar(require("./unified-agent-service-configuration-states"));
exports.UnifiedAgentServiceConfigurationStates = UnifiedAgentServiceConfigurationStates.UnifiedAgentServiceConfigurationStates;
const UnifiedAgentServiceConfigurationTypes = __importStar(require("./unified-agent-service-configuration-types"));
exports.UnifiedAgentServiceConfigurationTypes = UnifiedAgentServiceConfigurationTypes.UnifiedAgentServiceConfigurationTypes;
const UpdateConfigurationDetails = __importStar(require("./update-configuration-details"));
exports.UpdateConfigurationDetails = UpdateConfigurationDetails.UpdateConfigurationDetails;
const UpdateLogDetails = __importStar(require("./update-log-details"));
exports.UpdateLogDetails = UpdateLogDetails.UpdateLogDetails;
const UpdateLogGroupDetails = __importStar(require("./update-log-group-details"));
exports.UpdateLogGroupDetails = UpdateLogGroupDetails.UpdateLogGroupDetails;
const UpdateLogSavedSearchDetails = __importStar(require("./update-log-saved-search-details"));
exports.UpdateLogSavedSearchDetails = UpdateLogSavedSearchDetails.UpdateLogSavedSearchDetails;
const UpdateUnifiedAgentConfigurationDetails = __importStar(require("./update-unified-agent-configuration-details"));
exports.UpdateUnifiedAgentConfigurationDetails = UpdateUnifiedAgentConfigurationDetails.UpdateUnifiedAgentConfigurationDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestLog = __importStar(require("./work-request-log"));
exports.WorkRequestLog = WorkRequestLog.WorkRequestLog;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const OciService = __importStar(require("./oci-service"));
exports.OciService = OciService.OciService;
const UnifiedAgentApache2Parser = __importStar(require("./unified-agent-apache2-parser"));
exports.UnifiedAgentApache2Parser = UnifiedAgentApache2Parser.UnifiedAgentApache2Parser;
const UnifiedAgentApacheErrorParser = __importStar(require("./unified-agent-apache-error-parser"));
exports.UnifiedAgentApacheErrorParser = UnifiedAgentApacheErrorParser.UnifiedAgentApacheErrorParser;
const UnifiedAgentAuditdParser = __importStar(require("./unified-agent-auditd-parser"));
exports.UnifiedAgentAuditdParser = UnifiedAgentAuditdParser.UnifiedAgentAuditdParser;
const UnifiedAgentCsvParser = __importStar(require("./unified-agent-csv-parser"));
exports.UnifiedAgentCsvParser = UnifiedAgentCsvParser.UnifiedAgentCsvParser;
const UnifiedAgentGrokParser = __importStar(require("./unified-agent-grok-parser"));
exports.UnifiedAgentGrokParser = UnifiedAgentGrokParser.UnifiedAgentGrokParser;
const UnifiedAgentLoggingConfiguration = __importStar(require("./unified-agent-logging-configuration"));
exports.UnifiedAgentLoggingConfiguration = UnifiedAgentLoggingConfiguration.UnifiedAgentLoggingConfiguration;
const UnifiedAgentMsgpackParser = __importStar(require("./unified-agent-msgpack-parser"));
exports.UnifiedAgentMsgpackParser = UnifiedAgentMsgpackParser.UnifiedAgentMsgpackParser;
const UnifiedAgentMultilineGrokParser = __importStar(require("./unified-agent-multiline-grok-parser"));
exports.UnifiedAgentMultilineGrokParser = UnifiedAgentMultilineGrokParser.UnifiedAgentMultilineGrokParser;
const UnifiedAgentMultilineParser = __importStar(require("./unified-agent-multiline-parser"));
exports.UnifiedAgentMultilineParser = UnifiedAgentMultilineParser.UnifiedAgentMultilineParser;
const UnifiedAgentNoneParser = __importStar(require("./unified-agent-none-parser"));
exports.UnifiedAgentNoneParser = UnifiedAgentNoneParser.UnifiedAgentNoneParser;
const UnifiedAgentRegexParser = __importStar(require("./unified-agent-regex-parser"));
exports.UnifiedAgentRegexParser = UnifiedAgentRegexParser.UnifiedAgentRegexParser;
const UnifiedAgentSyslogParser = __importStar(require("./unified-agent-syslog-parser"));
exports.UnifiedAgentSyslogParser = UnifiedAgentSyslogParser.UnifiedAgentSyslogParser;
const UnifiedAgentTailLogSource = __importStar(require("./unified-agent-tail-log-source"));
exports.UnifiedAgentTailLogSource = UnifiedAgentTailLogSource.UnifiedAgentTailLogSource;
const UnifiedAgentTsvParser = __importStar(require("./unified-agent-tsv-parser"));
exports.UnifiedAgentTsvParser = UnifiedAgentTsvParser.UnifiedAgentTsvParser;
const UnifiedAgentWindowsEventSource = __importStar(require("./unified-agent-windows-event-source"));
exports.UnifiedAgentWindowsEventSource = UnifiedAgentWindowsEventSource.UnifiedAgentWindowsEventSource;
const UnifiedJSONParser = __importStar(require("./unified-js-on-parser"));
exports.UnifiedJSONParser = UnifiedJSONParser.UnifiedJSONParser;
