// oci-logging/lib/model/operation-types.js
"use strict";
/**
 * Logging Management API
 * Use the Logging Management API to create, read, list, update, and delete log groups, log objects, and agent configurations.
 * OpenAPI spec version: 20200531
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationTypes = void 0;
/**
 * The types of work request operations.
 **/
var OperationTypes;
(function (OperationTypes) {
    OperationTypes["CreateLog"] = "CREATE_LOG";
    OperationTypes["UpdateLog"] = "UPDATE_LOG";
    OperationTypes["DeleteLog"] = "DELETE_LOG";
    OperationTypes["MoveLog"] = "MOVE_LOG";
    OperationTypes["CreateLogGroup"] = "CREATE_LOG_GROUP";
    OperationTypes["UpdateLogGroup"] = "UPDATE_LOG_GROUP";
    OperationTypes["DeleteLogGroup"] = "DELETE_LOG_GROUP";
    OperationTypes["MoveLogGroup"] = "MOVE_LOG_GROUP";
    OperationTypes["CreateConfiguration"] = "CREATE_CONFIGURATION";
    OperationTypes["UpdateConfiguration"] = "UPDATE_CONFIGURATION";
    OperationTypes["DeleteConfiguration"] = "DELETE_CONFIGURATION";
    OperationTypes["MoveConfiguration"] = "MOVE_CONFIGURATION";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationTypes["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationTypes = exports.OperationTypes || (exports.OperationTypes = {}));
(function (OperationTypes) {
    function getJsonObj(obj) {
        return { obj };
    }
    OperationTypes.getJsonObj = getJsonObj;
})(OperationTypes = exports.OperationTypes || (exports.OperationTypes = {}));
