// oci-logging/lib/model/unified-agent-service-configuration-states.js
"use strict";
/**
 * Logging Management API
 * Use the Logging Management API to create, read, list, update, and delete log groups, log objects, and agent configurations.
 * OpenAPI spec version: 20200531
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UnifiedAgentServiceConfigurationStates = void 0;
/**
 * The state of the configuration:
 *    1. *VALID* configuration is valid and usable
 *    2. *INVALID* configuration has compilation errors and is not usable
 *
 **/
var UnifiedAgentServiceConfigurationStates;
(function (UnifiedAgentServiceConfigurationStates) {
    UnifiedAgentServiceConfigurationStates["Valid"] = "VALID";
    UnifiedAgentServiceConfigurationStates["Invalid"] = "INVALID";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    UnifiedAgentServiceConfigurationStates["UnknownValue"] = "UNKNOWN_VALUE";
})(UnifiedAgentServiceConfigurationStates = exports.UnifiedAgentServiceConfigurationStates || (exports.UnifiedAgentServiceConfigurationStates = {}));
(function (UnifiedAgentServiceConfigurationStates) {
    function getJsonObj(obj) {
        return { obj };
    }
    UnifiedAgentServiceConfigurationStates.getJsonObj = getJsonObj;
})(UnifiedAgentServiceConfigurationStates = exports.UnifiedAgentServiceConfigurationStates || (exports.UnifiedAgentServiceConfigurationStates = {}));
