// oci-managementdashboard/lib/model/index.js
"use strict";
/**
 * ManagementDashboard API
 * API for the Management Dashboard micro-service. Use this API for dashboard and saved search metadata preservation and to perform  tasks such as creating a dashboard, creating a saved search, and obtaining a list of dashboards and saved searches in a compartment.
    

 * OpenAPI spec version: 20200901
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateManagementSavedSearchDetails = exports.UpdateManagementDashboardDetails = exports.SortOrders = exports.SavedSearchTypes = exports.ManagementSavedSearchSummary = exports.ManagementSavedSearchForImportDetails = exports.ManagementSavedSearchCollection = exports.ManagementSavedSearch = exports.ManagementDashboardTileDetails = exports.ManagementDashboardSummary = exports.ManagementDashboardImportDetails = exports.ManagementDashboardForImportExportDetails = exports.ManagementDashboardExportDetails = exports.ManagementDashboardCollection = exports.ManagementDashboard = exports.LifecycleStates = exports.CreateManagementSavedSearchDetails = exports.CreateManagementDashboardDetails = exports.ChangeManagementSavedSearchesCompartmentDetails = exports.ChangeManagementDashboardsCompartmentDetails = void 0;
const ChangeManagementDashboardsCompartmentDetails = __importStar(require("./change-management-dashboards-compartment-details"));
exports.ChangeManagementDashboardsCompartmentDetails = ChangeManagementDashboardsCompartmentDetails.ChangeManagementDashboardsCompartmentDetails;
const ChangeManagementSavedSearchesCompartmentDetails = __importStar(require("./change-management-saved-searches-compartment-details"));
exports.ChangeManagementSavedSearchesCompartmentDetails = ChangeManagementSavedSearchesCompartmentDetails.ChangeManagementSavedSearchesCompartmentDetails;
const CreateManagementDashboardDetails = __importStar(require("./create-management-dashboard-details"));
exports.CreateManagementDashboardDetails = CreateManagementDashboardDetails.CreateManagementDashboardDetails;
const CreateManagementSavedSearchDetails = __importStar(require("./create-management-saved-search-details"));
exports.CreateManagementSavedSearchDetails = CreateManagementSavedSearchDetails.CreateManagementSavedSearchDetails;
const LifecycleStates = __importStar(require("./lifecycle-states"));
exports.LifecycleStates = LifecycleStates.LifecycleStates;
const ManagementDashboard = __importStar(require("./management-dashboard"));
exports.ManagementDashboard = ManagementDashboard.ManagementDashboard;
const ManagementDashboardCollection = __importStar(require("./management-dashboard-collection"));
exports.ManagementDashboardCollection = ManagementDashboardCollection.ManagementDashboardCollection;
const ManagementDashboardExportDetails = __importStar(require("./management-dashboard-export-details"));
exports.ManagementDashboardExportDetails = ManagementDashboardExportDetails.ManagementDashboardExportDetails;
const ManagementDashboardForImportExportDetails = __importStar(require("./management-dashboard-for-import-export-details"));
exports.ManagementDashboardForImportExportDetails = ManagementDashboardForImportExportDetails.ManagementDashboardForImportExportDetails;
const ManagementDashboardImportDetails = __importStar(require("./management-dashboard-import-details"));
exports.ManagementDashboardImportDetails = ManagementDashboardImportDetails.ManagementDashboardImportDetails;
const ManagementDashboardSummary = __importStar(require("./management-dashboard-summary"));
exports.ManagementDashboardSummary = ManagementDashboardSummary.ManagementDashboardSummary;
const ManagementDashboardTileDetails = __importStar(require("./management-dashboard-tile-details"));
exports.ManagementDashboardTileDetails = ManagementDashboardTileDetails.ManagementDashboardTileDetails;
const ManagementSavedSearch = __importStar(require("./management-saved-search"));
exports.ManagementSavedSearch = ManagementSavedSearch.ManagementSavedSearch;
const ManagementSavedSearchCollection = __importStar(require("./management-saved-search-collection"));
exports.ManagementSavedSearchCollection = ManagementSavedSearchCollection.ManagementSavedSearchCollection;
const ManagementSavedSearchForImportDetails = __importStar(require("./management-saved-search-for-import-details"));
exports.ManagementSavedSearchForImportDetails = ManagementSavedSearchForImportDetails.ManagementSavedSearchForImportDetails;
const ManagementSavedSearchSummary = __importStar(require("./management-saved-search-summary"));
exports.ManagementSavedSearchSummary = ManagementSavedSearchSummary.ManagementSavedSearchSummary;
const SavedSearchTypes = __importStar(require("./saved-search-types"));
exports.SavedSearchTypes = SavedSearchTypes.SavedSearchTypes;
const SortOrders = __importStar(require("./sort-orders"));
exports.SortOrders = SortOrders.SortOrders;
const UpdateManagementDashboardDetails = __importStar(require("./update-management-dashboard-details"));
exports.UpdateManagementDashboardDetails = UpdateManagementDashboardDetails.UpdateManagementDashboardDetails;
const UpdateManagementSavedSearchDetails = __importStar(require("./update-management-saved-search-details"));
exports.UpdateManagementSavedSearchDetails = UpdateManagementSavedSearchDetails.UpdateManagementSavedSearchDetails;
