// oci-managementdashboard/lib/model/saved-search-types.js
"use strict";
/**
 * ManagementDashboard API
 * API for the Management Dashboard micro-service. Use this API for dashboard and saved search metadata preservation and to perform  tasks such as creating a dashboard, creating a saved search, and obtaining a list of dashboards and saved searches in a compartment.
    

 * OpenAPI spec version: 20200901
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SavedSearchTypes = void 0;
/**
 * Determines how the saved search is displayed in a dashboard.
 **/
var SavedSearchTypes;
(function (SavedSearchTypes) {
    SavedSearchTypes["SearchShowInDashboard"] = "SEARCH_SHOW_IN_DASHBOARD";
    SavedSearchTypes["SearchDontShowInDashboard"] = "SEARCH_DONT_SHOW_IN_DASHBOARD";
    SavedSearchTypes["WidgetShowInDashboard"] = "WIDGET_SHOW_IN_DASHBOARD";
    SavedSearchTypes["WidgetDontShowInDashboard"] = "WIDGET_DONT_SHOW_IN_DASHBOARD";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    SavedSearchTypes["UnknownValue"] = "UNKNOWN_VALUE";
})(SavedSearchTypes = exports.SavedSearchTypes || (exports.SavedSearchTypes = {}));
(function (SavedSearchTypes) {
    function getJsonObj(obj) {
        return { obj };
    }
    SavedSearchTypes.getJsonObj = getJsonObj;
})(SavedSearchTypes = exports.SavedSearchTypes || (exports.SavedSearchTypes = {}));
