// oci-managementdashboard/lib/model/sort-orders.js
"use strict";
/**
 * ManagementDashboard API
 * API for the Management Dashboard micro-service. Use this API for dashboard and saved search metadata preservation and to perform  tasks such as creating a dashboard, creating a saved search, and obtaining a list of dashboards and saved searches in a compartment.
    

 * OpenAPI spec version: 20200901
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SortOrders = void 0;
/**
 * Sort orders.
 **/
var SortOrders;
(function (SortOrders) {
    SortOrders["Asc"] = "ASC";
    SortOrders["Desc"] = "DESC";
})(SortOrders = exports.SortOrders || (exports.SortOrders = {}));
(function (SortOrders) {
    function getJsonObj(obj) {
        return { obj };
    }
    SortOrders.getJsonObj = getJsonObj;
})(SortOrders = exports.SortOrders || (exports.SortOrders = {}));
