// oci-networkloadbalancer/lib/model/health-check-protocols.js
"use strict";
/**
 * NetworkLoadBalancer API
 * A description of the network load balancer API
 * OpenAPI spec version: 20200501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.HealthCheckProtocols = void 0;
/**
 * Protocols supported for the health checks of the network load balancer.
 **/
var HealthCheckProtocols;
(function (HealthCheckProtocols) {
    HealthCheckProtocols["Http"] = "HTTP";
    HealthCheckProtocols["Https"] = "HTTPS";
    HealthCheckProtocols["Tcp"] = "TCP";
    HealthCheckProtocols["Udp"] = "UDP";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    HealthCheckProtocols["UnknownValue"] = "UNKNOWN_VALUE";
})(HealthCheckProtocols = exports.HealthCheckProtocols || (exports.HealthCheckProtocols = {}));
(function (HealthCheckProtocols) {
    function getJsonObj(obj) {
        return { obj };
    }
    HealthCheckProtocols.getJsonObj = getJsonObj;
})(HealthCheckProtocols = exports.HealthCheckProtocols || (exports.HealthCheckProtocols = {}));
