// oci-networkloadbalancer/lib/model/index.js
"use strict";
/**
 * NetworkLoadBalancer API
 * A description of the network load balancer API
 * OpenAPI spec version: 20200501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestCollection = exports.WorkRequest = exports.UpdateNetworkSecurityGroupsDetails = exports.UpdateNetworkLoadBalancerDetails = exports.UpdateListenerDetails = exports.UpdateHealthCheckerDetails = exports.UpdateBackendSetDetails = exports.UpdateBackendDetails = exports.SortOrder = exports.ReservedIP = exports.OperationType = exports.OperationStatus = exports.NetworkLoadBalancingPolicy = exports.NetworkLoadBalancersProtocolSummary = exports.NetworkLoadBalancersProtocolCollection = exports.NetworkLoadBalancersPolicySummary = exports.NetworkLoadBalancersPolicyCollection = exports.NetworkLoadBalancerSummary = exports.NetworkLoadBalancerHealthSummary = exports.NetworkLoadBalancerHealthCollection = exports.NetworkLoadBalancerHealth = exports.NetworkLoadBalancerCollection = exports.NetworkLoadBalancer = exports.ListenerSummary = exports.ListenerProtocols = exports.ListenerDetails = exports.ListenerCollection = exports.Listener = exports.LifecycleState = exports.IpAddress = exports.HealthCheckerDetails = exports.HealthChecker = exports.HealthCheckResult = exports.HealthCheckProtocols = exports.CreateNetworkLoadBalancerDetails = exports.CreateListenerDetails = exports.CreateBackendSetDetails = exports.CreateBackendDetails = exports.ChangeNetworkLoadBalancerCompartmentDetails = exports.BackendSummary = exports.BackendSetSummary = exports.BackendSetHealth = exports.BackendSetDetails = exports.BackendSetCollection = exports.BackendSet = exports.BackendHealth = exports.BackendDetails = exports.BackendCollection = exports.Backend = exports.ActionType = void 0;
exports.WorkRequestSummary = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const Backend = __importStar(require("./backend"));
exports.Backend = Backend.Backend;
const BackendCollection = __importStar(require("./backend-collection"));
exports.BackendCollection = BackendCollection.BackendCollection;
const BackendDetails = __importStar(require("./backend-details"));
exports.BackendDetails = BackendDetails.BackendDetails;
const BackendHealth = __importStar(require("./backend-health"));
exports.BackendHealth = BackendHealth.BackendHealth;
const BackendSet = __importStar(require("./backend-set"));
exports.BackendSet = BackendSet.BackendSet;
const BackendSetCollection = __importStar(require("./backend-set-collection"));
exports.BackendSetCollection = BackendSetCollection.BackendSetCollection;
const BackendSetDetails = __importStar(require("./backend-set-details"));
exports.BackendSetDetails = BackendSetDetails.BackendSetDetails;
const BackendSetHealth = __importStar(require("./backend-set-health"));
exports.BackendSetHealth = BackendSetHealth.BackendSetHealth;
const BackendSetSummary = __importStar(require("./backend-set-summary"));
exports.BackendSetSummary = BackendSetSummary.BackendSetSummary;
const BackendSummary = __importStar(require("./backend-summary"));
exports.BackendSummary = BackendSummary.BackendSummary;
const ChangeNetworkLoadBalancerCompartmentDetails = __importStar(require("./change-network-load-balancer-compartment-details"));
exports.ChangeNetworkLoadBalancerCompartmentDetails = ChangeNetworkLoadBalancerCompartmentDetails.ChangeNetworkLoadBalancerCompartmentDetails;
const CreateBackendDetails = __importStar(require("./create-backend-details"));
exports.CreateBackendDetails = CreateBackendDetails.CreateBackendDetails;
const CreateBackendSetDetails = __importStar(require("./create-backend-set-details"));
exports.CreateBackendSetDetails = CreateBackendSetDetails.CreateBackendSetDetails;
const CreateListenerDetails = __importStar(require("./create-listener-details"));
exports.CreateListenerDetails = CreateListenerDetails.CreateListenerDetails;
const CreateNetworkLoadBalancerDetails = __importStar(require("./create-network-load-balancer-details"));
exports.CreateNetworkLoadBalancerDetails = CreateNetworkLoadBalancerDetails.CreateNetworkLoadBalancerDetails;
const HealthCheckProtocols = __importStar(require("./health-check-protocols"));
exports.HealthCheckProtocols = HealthCheckProtocols.HealthCheckProtocols;
const HealthCheckResult = __importStar(require("./health-check-result"));
exports.HealthCheckResult = HealthCheckResult.HealthCheckResult;
const HealthChecker = __importStar(require("./health-checker"));
exports.HealthChecker = HealthChecker.HealthChecker;
const HealthCheckerDetails = __importStar(require("./health-checker-details"));
exports.HealthCheckerDetails = HealthCheckerDetails.HealthCheckerDetails;
const IpAddress = __importStar(require("./ip-address"));
exports.IpAddress = IpAddress.IpAddress;
const LifecycleState = __importStar(require("./lifecycle-state"));
exports.LifecycleState = LifecycleState.LifecycleState;
const Listener = __importStar(require("./listener"));
exports.Listener = Listener.Listener;
const ListenerCollection = __importStar(require("./listener-collection"));
exports.ListenerCollection = ListenerCollection.ListenerCollection;
const ListenerDetails = __importStar(require("./listener-details"));
exports.ListenerDetails = ListenerDetails.ListenerDetails;
const ListenerProtocols = __importStar(require("./listener-protocols"));
exports.ListenerProtocols = ListenerProtocols.ListenerProtocols;
const ListenerSummary = __importStar(require("./listener-summary"));
exports.ListenerSummary = ListenerSummary.ListenerSummary;
const NetworkLoadBalancer = __importStar(require("./network-load-balancer"));
exports.NetworkLoadBalancer = NetworkLoadBalancer.NetworkLoadBalancer;
const NetworkLoadBalancerCollection = __importStar(require("./network-load-balancer-collection"));
exports.NetworkLoadBalancerCollection = NetworkLoadBalancerCollection.NetworkLoadBalancerCollection;
const NetworkLoadBalancerHealth = __importStar(require("./network-load-balancer-health"));
exports.NetworkLoadBalancerHealth = NetworkLoadBalancerHealth.NetworkLoadBalancerHealth;
const NetworkLoadBalancerHealthCollection = __importStar(require("./network-load-balancer-health-collection"));
exports.NetworkLoadBalancerHealthCollection = NetworkLoadBalancerHealthCollection.NetworkLoadBalancerHealthCollection;
const NetworkLoadBalancerHealthSummary = __importStar(require("./network-load-balancer-health-summary"));
exports.NetworkLoadBalancerHealthSummary = NetworkLoadBalancerHealthSummary.NetworkLoadBalancerHealthSummary;
const NetworkLoadBalancerSummary = __importStar(require("./network-load-balancer-summary"));
exports.NetworkLoadBalancerSummary = NetworkLoadBalancerSummary.NetworkLoadBalancerSummary;
const NetworkLoadBalancersPolicyCollection = __importStar(require("./network-load-balancers-policy-collection"));
exports.NetworkLoadBalancersPolicyCollection = NetworkLoadBalancersPolicyCollection.NetworkLoadBalancersPolicyCollection;
const NetworkLoadBalancersPolicySummary = __importStar(require("./network-load-balancers-policy-summary"));
exports.NetworkLoadBalancersPolicySummary = NetworkLoadBalancersPolicySummary.NetworkLoadBalancersPolicySummary;
const NetworkLoadBalancersProtocolCollection = __importStar(require("./network-load-balancers-protocol-collection"));
exports.NetworkLoadBalancersProtocolCollection = NetworkLoadBalancersProtocolCollection.NetworkLoadBalancersProtocolCollection;
const NetworkLoadBalancersProtocolSummary = __importStar(require("./network-load-balancers-protocol-summary"));
exports.NetworkLoadBalancersProtocolSummary = NetworkLoadBalancersProtocolSummary.NetworkLoadBalancersProtocolSummary;
const NetworkLoadBalancingPolicy = __importStar(require("./network-load-balancing-policy"));
exports.NetworkLoadBalancingPolicy = NetworkLoadBalancingPolicy.NetworkLoadBalancingPolicy;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const ReservedIP = __importStar(require("./reserved-ip"));
exports.ReservedIP = ReservedIP.ReservedIP;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const UpdateBackendDetails = __importStar(require("./update-backend-details"));
exports.UpdateBackendDetails = UpdateBackendDetails.UpdateBackendDetails;
const UpdateBackendSetDetails = __importStar(require("./update-backend-set-details"));
exports.UpdateBackendSetDetails = UpdateBackendSetDetails.UpdateBackendSetDetails;
const UpdateHealthCheckerDetails = __importStar(require("./update-health-checker-details"));
exports.UpdateHealthCheckerDetails = UpdateHealthCheckerDetails.UpdateHealthCheckerDetails;
const UpdateListenerDetails = __importStar(require("./update-listener-details"));
exports.UpdateListenerDetails = UpdateListenerDetails.UpdateListenerDetails;
const UpdateNetworkLoadBalancerDetails = __importStar(require("./update-network-load-balancer-details"));
exports.UpdateNetworkLoadBalancerDetails = UpdateNetworkLoadBalancerDetails.UpdateNetworkLoadBalancerDetails;
const UpdateNetworkSecurityGroupsDetails = __importStar(require("./update-network-security-groups-details"));
exports.UpdateNetworkSecurityGroupsDetails = UpdateNetworkSecurityGroupsDetails.UpdateNetworkSecurityGroupsDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestCollection = __importStar(require("./work-request-collection"));
exports.WorkRequestCollection = WorkRequestCollection.WorkRequestCollection;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
