// oci-networkloadbalancer/lib/model/network-load-balancer-health-summary.js
"use strict";
/**
 * NetworkLoadBalancer API
 * A description of the network load balancer API
 * OpenAPI spec version: 20200501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkLoadBalancerHealthSummary = void 0;
var NetworkLoadBalancerHealthSummary;
(function (NetworkLoadBalancerHealthSummary) {
    let Status;
    (function (Status) {
        Status["Ok"] = "OK";
        Status["Warning"] = "WARNING";
        Status["Critical"] = "CRITICAL";
        Status["Unknown"] = "UNKNOWN";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Status["UnknownValue"] = "UNKNOWN_VALUE";
    })(Status = NetworkLoadBalancerHealthSummary.Status || (NetworkLoadBalancerHealthSummary.Status = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    NetworkLoadBalancerHealthSummary.getJsonObj = getJsonObj;
})(NetworkLoadBalancerHealthSummary = exports.NetworkLoadBalancerHealthSummary || (exports.NetworkLoadBalancerHealthSummary = {}));
