// oci-networkloadbalancer/lib/model/operation-type.js
"use strict";
/**
 * NetworkLoadBalancer API
 * A description of the network load balancer API
 * OpenAPI spec version: 20200501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationType = void 0;
/**
 * Possible operations.
 **/
var OperationType;
(function (OperationType) {
    OperationType["CreateNetworkLoadBalancer"] = "CREATE_NETWORK_LOAD_BALANCER";
    OperationType["UpdateNetworkLoadBalancer"] = "UPDATE_NETWORK_LOAD_BALANCER";
    OperationType["DeleteNetworkLoadBalancer"] = "DELETE_NETWORK_LOAD_BALANCER";
    OperationType["CreateBackend"] = "CREATE_BACKEND";
    OperationType["UpdateBackend"] = "UPDATE_BACKEND";
    OperationType["DeleteBackend"] = "DELETE_BACKEND";
    OperationType["CreateListener"] = "CREATE_LISTENER";
    OperationType["UpdateListener"] = "UPDATE_LISTENER";
    OperationType["DeleteListener"] = "DELETE_LISTENER";
    OperationType["CreateBackendset"] = "CREATE_BACKENDSET";
    OperationType["UpdateBackendset"] = "UPDATE_BACKENDSET";
    OperationType["DeleteBackendset"] = "DELETE_BACKENDSET";
    OperationType["UpdateNsgs"] = "UPDATE_NSGS";
    OperationType["UpdateHealthChecker"] = "UPDATE_HEALTH_CHECKER";
    OperationType["ChangeCompartment"] = "CHANGE_COMPARTMENT";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationType = exports.OperationType || (exports.OperationType = {}));
(function (OperationType) {
    function getJsonObj(obj) {
        return { obj };
    }
    OperationType.getJsonObj = getJsonObj;
})(OperationType = exports.OperationType || (exports.OperationType = {}));
