// oci-rover/lib/model/index.js
"use strict";
/**
 * RoverCloudService API
 * A description of the RoverCloudService API.
 * OpenAPI spec version: 20201210
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateRoverNodeDetails = exports.UpdateRoverEntitlementDetails = exports.UpdateRoverClusterDetails = exports.SortOrders = exports.ShippingAddress = exports.RoverWorkload = exports.RoverNodeSummary = exports.RoverNodeSetKey = exports.RoverNodeGetRpt = exports.RoverNodeEncryptionKey = exports.RoverNodeCollection = exports.RoverNodeCertificate = exports.RoverNodeActionSetKeyDetails = exports.RoverNode = exports.RoverEntitlementSummary = exports.RoverEntitlementCollection = exports.RoverEntitlement = exports.RoverClusterSummary = exports.RoverClusterCollection = exports.RoverClusterCertificate = exports.RoverCluster = exports.NodeType = exports.LifecycleState = exports.EnclosureType = exports.CreateRoverNodeDetails = exports.CreateRoverEntitlementDetails = exports.CreateRoverClusterDetails = exports.ChangeRoverNodeCompartmentDetails = exports.ChangeRoverEntitlementCompartmentDetails = exports.ChangeRoverClusterCompartmentDetails = void 0;
const ChangeRoverClusterCompartmentDetails = __importStar(require("./change-rover-cluster-compartment-details"));
exports.ChangeRoverClusterCompartmentDetails = ChangeRoverClusterCompartmentDetails.ChangeRoverClusterCompartmentDetails;
const ChangeRoverEntitlementCompartmentDetails = __importStar(require("./change-rover-entitlement-compartment-details"));
exports.ChangeRoverEntitlementCompartmentDetails = ChangeRoverEntitlementCompartmentDetails.ChangeRoverEntitlementCompartmentDetails;
const ChangeRoverNodeCompartmentDetails = __importStar(require("./change-rover-node-compartment-details"));
exports.ChangeRoverNodeCompartmentDetails = ChangeRoverNodeCompartmentDetails.ChangeRoverNodeCompartmentDetails;
const CreateRoverClusterDetails = __importStar(require("./create-rover-cluster-details"));
exports.CreateRoverClusterDetails = CreateRoverClusterDetails.CreateRoverClusterDetails;
const CreateRoverEntitlementDetails = __importStar(require("./create-rover-entitlement-details"));
exports.CreateRoverEntitlementDetails = CreateRoverEntitlementDetails.CreateRoverEntitlementDetails;
const CreateRoverNodeDetails = __importStar(require("./create-rover-node-details"));
exports.CreateRoverNodeDetails = CreateRoverNodeDetails.CreateRoverNodeDetails;
const EnclosureType = __importStar(require("./enclosure-type"));
exports.EnclosureType = EnclosureType.EnclosureType;
const LifecycleState = __importStar(require("./lifecycle-state"));
exports.LifecycleState = LifecycleState.LifecycleState;
const NodeType = __importStar(require("./node-type"));
exports.NodeType = NodeType.NodeType;
const RoverCluster = __importStar(require("./rover-cluster"));
exports.RoverCluster = RoverCluster.RoverCluster;
const RoverClusterCertificate = __importStar(require("./rover-cluster-certificate"));
exports.RoverClusterCertificate = RoverClusterCertificate.RoverClusterCertificate;
const RoverClusterCollection = __importStar(require("./rover-cluster-collection"));
exports.RoverClusterCollection = RoverClusterCollection.RoverClusterCollection;
const RoverClusterSummary = __importStar(require("./rover-cluster-summary"));
exports.RoverClusterSummary = RoverClusterSummary.RoverClusterSummary;
const RoverEntitlement = __importStar(require("./rover-entitlement"));
exports.RoverEntitlement = RoverEntitlement.RoverEntitlement;
const RoverEntitlementCollection = __importStar(require("./rover-entitlement-collection"));
exports.RoverEntitlementCollection = RoverEntitlementCollection.RoverEntitlementCollection;
const RoverEntitlementSummary = __importStar(require("./rover-entitlement-summary"));
exports.RoverEntitlementSummary = RoverEntitlementSummary.RoverEntitlementSummary;
const RoverNode = __importStar(require("./rover-node"));
exports.RoverNode = RoverNode.RoverNode;
const RoverNodeActionSetKeyDetails = __importStar(require("./rover-node-action-set-key-details"));
exports.RoverNodeActionSetKeyDetails = RoverNodeActionSetKeyDetails.RoverNodeActionSetKeyDetails;
const RoverNodeCertificate = __importStar(require("./rover-node-certificate"));
exports.RoverNodeCertificate = RoverNodeCertificate.RoverNodeCertificate;
const RoverNodeCollection = __importStar(require("./rover-node-collection"));
exports.RoverNodeCollection = RoverNodeCollection.RoverNodeCollection;
const RoverNodeEncryptionKey = __importStar(require("./rover-node-encryption-key"));
exports.RoverNodeEncryptionKey = RoverNodeEncryptionKey.RoverNodeEncryptionKey;
const RoverNodeGetRpt = __importStar(require("./rover-node-get-rpt"));
exports.RoverNodeGetRpt = RoverNodeGetRpt.RoverNodeGetRpt;
const RoverNodeSetKey = __importStar(require("./rover-node-set-key"));
exports.RoverNodeSetKey = RoverNodeSetKey.RoverNodeSetKey;
const RoverNodeSummary = __importStar(require("./rover-node-summary"));
exports.RoverNodeSummary = RoverNodeSummary.RoverNodeSummary;
const RoverWorkload = __importStar(require("./rover-workload"));
exports.RoverWorkload = RoverWorkload.RoverWorkload;
const ShippingAddress = __importStar(require("./shipping-address"));
exports.ShippingAddress = ShippingAddress.ShippingAddress;
const SortOrders = __importStar(require("./sort-orders"));
exports.SortOrders = SortOrders.SortOrders;
const UpdateRoverClusterDetails = __importStar(require("./update-rover-cluster-details"));
exports.UpdateRoverClusterDetails = UpdateRoverClusterDetails.UpdateRoverClusterDetails;
const UpdateRoverEntitlementDetails = __importStar(require("./update-rover-entitlement-details"));
exports.UpdateRoverEntitlementDetails = UpdateRoverEntitlementDetails.UpdateRoverEntitlementDetails;
const UpdateRoverNodeDetails = __importStar(require("./update-rover-node-details"));
exports.UpdateRoverNodeDetails = UpdateRoverNodeDetails.UpdateRoverNodeDetails;
