// oci-rover/lib/model/node-type.js
"use strict";
/**
 * RoverCloudService API
 * A description of the RoverCloudService API.
 * OpenAPI spec version: 20201210
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NodeType = void 0;
/**
 * Possible rover node types.
 **/
var NodeType;
(function (NodeType) {
    NodeType["Standalone"] = "STANDALONE";
    NodeType["Clustered"] = "CLUSTERED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    NodeType["UnknownValue"] = "UNKNOWN_VALUE";
})(NodeType = exports.NodeType || (exports.NodeType = {}));
(function (NodeType) {
    function getJsonObj(obj) {
        return { obj };
    }
    NodeType.getJsonObj = getJsonObj;
})(NodeType = exports.NodeType || (exports.NodeType = {}));
