// oci-keymanagement/lib/model/exported-key-data.js
"use strict";
/**
 * Vault Service Key Management API
 * API for managing and performing operations with keys and vaults. (For the API for managing secrets, see the Vault Service
Secret Management API. For the API for retrieving secrets, see the Vault Service Secret Retrieval API.)

 * OpenAPI spec version: release
 * Contact: sparta_kms_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExportedKeyData = void 0;
var ExportedKeyData;
(function (ExportedKeyData) {
    let Algorithm;
    (function (Algorithm) {
        Algorithm["RsaOaepAesSha256"] = "RSA_OAEP_AES_SHA256";
        Algorithm["RsaOaepSha256"] = "RSA_OAEP_SHA256";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Algorithm["UnknownValue"] = "UNKNOWN_VALUE";
    })(Algorithm = ExportedKeyData.Algorithm || (ExportedKeyData.Algorithm = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ExportedKeyData.getJsonObj = getJsonObj;
})(ExportedKeyData = exports.ExportedKeyData || (exports.ExportedKeyData = {}));
