// oci-keymanagement/lib/model/index.js
"use strict";
/**
 * Vault Service Key Management API
 * API for managing and performing operations with keys and vaults. (For the API for managing secrets, see the Vault Service
Secret Management API. For the API for retrieving secrets, see the Vault Service Secret Retrieval API.)

 * OpenAPI spec version: release
 * Contact: sparta_kms_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BackupLocationURI = exports.BackupLocationBucket = exports.WrappingKey = exports.WrappedImportKey = exports.VerifyDataDetails = exports.VerifiedData = exports.VaultUsage = exports.VaultSummary = exports.VaultReplicaSummary = exports.VaultReplicaDetails = exports.Vault = exports.UpdateVaultDetails = exports.UpdateKeyDetails = exports.SignedData = exports.SignDataDetails = exports.ScheduleVaultDeletionDetails = exports.ScheduleKeyVersionDeletionDetails = exports.ScheduleKeyDeletionDetails = exports.RestoreVaultFromObjectStoreDetails = exports.RestoreKeyFromObjectStoreDetails = exports.ReplicationStatusDetails = exports.ReplicaDetails = exports.KeyVersionSummary = exports.KeyVersionReplicaDetails = exports.KeyVersion = exports.KeySummary = exports.KeyShape = exports.KeyReplicaDetails = exports.Key = exports.ImportKeyVersionDetails = exports.ImportKeyDetails = exports.GeneratedKey = exports.GenerateKeyDetails = exports.ExportedKeyData = exports.ExportKeyDetails = exports.EncryptedData = exports.EncryptDataDetails = exports.DeleteVaultReplicaDetails = exports.DecryptedData = exports.DecryptDataDetails = exports.CreateVaultReplicaDetails = exports.CreateVaultDetails = exports.CreateKeyDetails = exports.ChangeVaultCompartmentDetails = exports.ChangeKeyCompartmentDetails = exports.BackupVaultDetails = exports.BackupLocation = exports.BackupKeyDetails = void 0;
const BackupKeyDetails = __importStar(require("./backup-key-details"));
exports.BackupKeyDetails = BackupKeyDetails.BackupKeyDetails;
const BackupLocation = __importStar(require("./backup-location"));
exports.BackupLocation = BackupLocation.BackupLocation;
const BackupVaultDetails = __importStar(require("./backup-vault-details"));
exports.BackupVaultDetails = BackupVaultDetails.BackupVaultDetails;
const ChangeKeyCompartmentDetails = __importStar(require("./change-key-compartment-details"));
exports.ChangeKeyCompartmentDetails = ChangeKeyCompartmentDetails.ChangeKeyCompartmentDetails;
const ChangeVaultCompartmentDetails = __importStar(require("./change-vault-compartment-details"));
exports.ChangeVaultCompartmentDetails = ChangeVaultCompartmentDetails.ChangeVaultCompartmentDetails;
const CreateKeyDetails = __importStar(require("./create-key-details"));
exports.CreateKeyDetails = CreateKeyDetails.CreateKeyDetails;
const CreateVaultDetails = __importStar(require("./create-vault-details"));
exports.CreateVaultDetails = CreateVaultDetails.CreateVaultDetails;
const CreateVaultReplicaDetails = __importStar(require("./create-vault-replica-details"));
exports.CreateVaultReplicaDetails = CreateVaultReplicaDetails.CreateVaultReplicaDetails;
const DecryptDataDetails = __importStar(require("./decrypt-data-details"));
exports.DecryptDataDetails = DecryptDataDetails.DecryptDataDetails;
const DecryptedData = __importStar(require("./decrypted-data"));
exports.DecryptedData = DecryptedData.DecryptedData;
const DeleteVaultReplicaDetails = __importStar(require("./delete-vault-replica-details"));
exports.DeleteVaultReplicaDetails = DeleteVaultReplicaDetails.DeleteVaultReplicaDetails;
const EncryptDataDetails = __importStar(require("./encrypt-data-details"));
exports.EncryptDataDetails = EncryptDataDetails.EncryptDataDetails;
const EncryptedData = __importStar(require("./encrypted-data"));
exports.EncryptedData = EncryptedData.EncryptedData;
const ExportKeyDetails = __importStar(require("./export-key-details"));
exports.ExportKeyDetails = ExportKeyDetails.ExportKeyDetails;
const ExportedKeyData = __importStar(require("./exported-key-data"));
exports.ExportedKeyData = ExportedKeyData.ExportedKeyData;
const GenerateKeyDetails = __importStar(require("./generate-key-details"));
exports.GenerateKeyDetails = GenerateKeyDetails.GenerateKeyDetails;
const GeneratedKey = __importStar(require("./generated-key"));
exports.GeneratedKey = GeneratedKey.GeneratedKey;
const ImportKeyDetails = __importStar(require("./import-key-details"));
exports.ImportKeyDetails = ImportKeyDetails.ImportKeyDetails;
const ImportKeyVersionDetails = __importStar(require("./import-key-version-details"));
exports.ImportKeyVersionDetails = ImportKeyVersionDetails.ImportKeyVersionDetails;
const Key = __importStar(require("./key"));
exports.Key = Key.Key;
const KeyReplicaDetails = __importStar(require("./key-replica-details"));
exports.KeyReplicaDetails = KeyReplicaDetails.KeyReplicaDetails;
const KeyShape = __importStar(require("./key-shape"));
exports.KeyShape = KeyShape.KeyShape;
const KeySummary = __importStar(require("./key-summary"));
exports.KeySummary = KeySummary.KeySummary;
const KeyVersion = __importStar(require("./key-version"));
exports.KeyVersion = KeyVersion.KeyVersion;
const KeyVersionReplicaDetails = __importStar(require("./key-version-replica-details"));
exports.KeyVersionReplicaDetails = KeyVersionReplicaDetails.KeyVersionReplicaDetails;
const KeyVersionSummary = __importStar(require("./key-version-summary"));
exports.KeyVersionSummary = KeyVersionSummary.KeyVersionSummary;
const ReplicaDetails = __importStar(require("./replica-details"));
exports.ReplicaDetails = ReplicaDetails.ReplicaDetails;
const ReplicationStatusDetails = __importStar(require("./replication-status-details"));
exports.ReplicationStatusDetails = ReplicationStatusDetails.ReplicationStatusDetails;
const RestoreKeyFromObjectStoreDetails = __importStar(require("./restore-key-from-object-store-details"));
exports.RestoreKeyFromObjectStoreDetails = RestoreKeyFromObjectStoreDetails.RestoreKeyFromObjectStoreDetails;
const RestoreVaultFromObjectStoreDetails = __importStar(require("./restore-vault-from-object-store-details"));
exports.RestoreVaultFromObjectStoreDetails = RestoreVaultFromObjectStoreDetails.RestoreVaultFromObjectStoreDetails;
const ScheduleKeyDeletionDetails = __importStar(require("./schedule-key-deletion-details"));
exports.ScheduleKeyDeletionDetails = ScheduleKeyDeletionDetails.ScheduleKeyDeletionDetails;
const ScheduleKeyVersionDeletionDetails = __importStar(require("./schedule-key-version-deletion-details"));
exports.ScheduleKeyVersionDeletionDetails = ScheduleKeyVersionDeletionDetails.ScheduleKeyVersionDeletionDetails;
const ScheduleVaultDeletionDetails = __importStar(require("./schedule-vault-deletion-details"));
exports.ScheduleVaultDeletionDetails = ScheduleVaultDeletionDetails.ScheduleVaultDeletionDetails;
const SignDataDetails = __importStar(require("./sign-data-details"));
exports.SignDataDetails = SignDataDetails.SignDataDetails;
const SignedData = __importStar(require("./signed-data"));
exports.SignedData = SignedData.SignedData;
const UpdateKeyDetails = __importStar(require("./update-key-details"));
exports.UpdateKeyDetails = UpdateKeyDetails.UpdateKeyDetails;
const UpdateVaultDetails = __importStar(require("./update-vault-details"));
exports.UpdateVaultDetails = UpdateVaultDetails.UpdateVaultDetails;
const Vault = __importStar(require("./vault"));
exports.Vault = Vault.Vault;
const VaultReplicaDetails = __importStar(require("./vault-replica-details"));
exports.VaultReplicaDetails = VaultReplicaDetails.VaultReplicaDetails;
const VaultReplicaSummary = __importStar(require("./vault-replica-summary"));
exports.VaultReplicaSummary = VaultReplicaSummary.VaultReplicaSummary;
const VaultSummary = __importStar(require("./vault-summary"));
exports.VaultSummary = VaultSummary.VaultSummary;
const VaultUsage = __importStar(require("./vault-usage"));
exports.VaultUsage = VaultUsage.VaultUsage;
const VerifiedData = __importStar(require("./verified-data"));
exports.VerifiedData = VerifiedData.VerifiedData;
const VerifyDataDetails = __importStar(require("./verify-data-details"));
exports.VerifyDataDetails = VerifyDataDetails.VerifyDataDetails;
const WrappedImportKey = __importStar(require("./wrapped-import-key"));
exports.WrappedImportKey = WrappedImportKey.WrappedImportKey;
const WrappingKey = __importStar(require("./wrapping-key"));
exports.WrappingKey = WrappingKey.WrappingKey;
const BackupLocationBucket = __importStar(require("./backup-location-bucket"));
exports.BackupLocationBucket = BackupLocationBucket.BackupLocationBucket;
const BackupLocationURI = __importStar(require("./backup-location-ur-i"));
exports.BackupLocationURI = BackupLocationURI.BackupLocationURI;
