// oci-integration/lib/model/index.js
"use strict";
/**
 * Oracle Integration API
 * Oracle Integration API.
 * OpenAPI spec version: 20190131
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicEndpointDetails = exports.WorkRequestSummary = exports.WorkRequestResource = exports.WorkRequestLogEntry = exports.WorkRequestError = exports.WorkRequest = exports.VirtualCloudNetwork = exports.UpdateIntegrationInstanceDetails = exports.UpdateCustomEndpointDetails = exports.NetworkEndpointType = exports.NetworkEndpointDetails = exports.IntegrationInstanceSummary = exports.IntegrationInstance = exports.CustomEndpointDetails = exports.CreateIntegrationInstanceDetails = exports.CreateCustomEndpointDetails = exports.ChangeIntegrationInstanceNetworkEndpointDetails = exports.ChangeIntegrationInstanceCompartmentDetails = void 0;
const ChangeIntegrationInstanceCompartmentDetails = __importStar(require("./change-integration-instance-compartment-details"));
exports.ChangeIntegrationInstanceCompartmentDetails = ChangeIntegrationInstanceCompartmentDetails.ChangeIntegrationInstanceCompartmentDetails;
const ChangeIntegrationInstanceNetworkEndpointDetails = __importStar(require("./change-integration-instance-network-endpoint-details"));
exports.ChangeIntegrationInstanceNetworkEndpointDetails = ChangeIntegrationInstanceNetworkEndpointDetails.ChangeIntegrationInstanceNetworkEndpointDetails;
const CreateCustomEndpointDetails = __importStar(require("./create-custom-endpoint-details"));
exports.CreateCustomEndpointDetails = CreateCustomEndpointDetails.CreateCustomEndpointDetails;
const CreateIntegrationInstanceDetails = __importStar(require("./create-integration-instance-details"));
exports.CreateIntegrationInstanceDetails = CreateIntegrationInstanceDetails.CreateIntegrationInstanceDetails;
const CustomEndpointDetails = __importStar(require("./custom-endpoint-details"));
exports.CustomEndpointDetails = CustomEndpointDetails.CustomEndpointDetails;
const IntegrationInstance = __importStar(require("./integration-instance"));
exports.IntegrationInstance = IntegrationInstance.IntegrationInstance;
const IntegrationInstanceSummary = __importStar(require("./integration-instance-summary"));
exports.IntegrationInstanceSummary = IntegrationInstanceSummary.IntegrationInstanceSummary;
const NetworkEndpointDetails = __importStar(require("./network-endpoint-details"));
exports.NetworkEndpointDetails = NetworkEndpointDetails.NetworkEndpointDetails;
const NetworkEndpointType = __importStar(require("./network-endpoint-type"));
exports.NetworkEndpointType = NetworkEndpointType.NetworkEndpointType;
const UpdateCustomEndpointDetails = __importStar(require("./update-custom-endpoint-details"));
exports.UpdateCustomEndpointDetails = UpdateCustomEndpointDetails.UpdateCustomEndpointDetails;
const UpdateIntegrationInstanceDetails = __importStar(require("./update-integration-instance-details"));
exports.UpdateIntegrationInstanceDetails = UpdateIntegrationInstanceDetails.UpdateIntegrationInstanceDetails;
const VirtualCloudNetwork = __importStar(require("./virtual-cloud-network"));
exports.VirtualCloudNetwork = VirtualCloudNetwork.VirtualCloudNetwork;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const PublicEndpointDetails = __importStar(require("./public-endpoint-details"));
exports.PublicEndpointDetails = PublicEndpointDetails.PublicEndpointDetails;
