// oci-integration/lib/model/network-endpoint-type.js
"use strict";
/**
 * Oracle Integration API
 * Oracle Integration API.
 * OpenAPI spec version: 20190131
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkEndpointType = void 0;
/**
 * Public endpoint access type.
 *
 **/
var NetworkEndpointType;
(function (NetworkEndpointType) {
    NetworkEndpointType["Public"] = "PUBLIC";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    NetworkEndpointType["UnknownValue"] = "UNKNOWN_VALUE";
})(NetworkEndpointType = exports.NetworkEndpointType || (exports.NetworkEndpointType = {}));
(function (NetworkEndpointType) {
    function getJsonObj(obj) {
        return { obj };
    }
    NetworkEndpointType.getJsonObj = getJsonObj;
})(NetworkEndpointType = exports.NetworkEndpointType || (exports.NetworkEndpointType = {}));
