// oci-filestorage/lib/model/client-options.js
"use strict";
/**
 * File Storage API
 * API for the File Storage service. Use this API to manage file systems, mount targets, and snapshots. For more information, see [Overview of File Storage](/iaas/Content/File/Concepts/filestorageoverview.htm).

 * OpenAPI spec version: 20171215
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ClientOptions = void 0;
var ClientOptions;
(function (ClientOptions) {
    let Access;
    (function (Access) {
        Access["ReadWrite"] = "READ_WRITE";
        Access["ReadOnly"] = "READ_ONLY";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Access["UnknownValue"] = "UNKNOWN_VALUE";
    })(Access = ClientOptions.Access || (ClientOptions.Access = {}));
    let IdentitySquash;
    (function (IdentitySquash) {
        IdentitySquash["None"] = "NONE";
        IdentitySquash["Root"] = "ROOT";
        IdentitySquash["All"] = "ALL";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        IdentitySquash["UnknownValue"] = "UNKNOWN_VALUE";
    })(IdentitySquash = ClientOptions.IdentitySquash || (ClientOptions.IdentitySquash = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ClientOptions.getJsonObj = getJsonObj;
})(ClientOptions = exports.ClientOptions || (exports.ClientOptions = {}));
