// oci-filestorage/lib/model/index.js
"use strict";
/**
 * File Storage API
 * API for the File Storage service. Use this API to manage file systems, mount targets, and snapshots. For more information, see [Overview of File Storage](/iaas/Content/File/Concepts/filestorageoverview.htm).

 * OpenAPI spec version: 20171215
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateSnapshotDetails = exports.UpdateMountTargetDetails = exports.UpdateFileSystemDetails = exports.UpdateExportSetDetails = exports.UpdateExportDetails = exports.SourceDetails = exports.SnapshotSummary = exports.Snapshot = exports.MountTargetSummary = exports.MountTarget = exports.FileSystemSummary = exports.FileSystem = exports.ExportSummary = exports.ExportSetSummary = exports.ExportSet = exports.Export = exports.CreateSnapshotDetails = exports.CreateMountTargetDetails = exports.CreateFileSystemDetails = exports.CreateExportDetails = exports.ClientOptions = exports.ChangeMountTargetCompartmentDetails = exports.ChangeFileSystemCompartmentDetails = void 0;
const ChangeFileSystemCompartmentDetails = __importStar(require("./change-file-system-compartment-details"));
exports.ChangeFileSystemCompartmentDetails = ChangeFileSystemCompartmentDetails.ChangeFileSystemCompartmentDetails;
const ChangeMountTargetCompartmentDetails = __importStar(require("./change-mount-target-compartment-details"));
exports.ChangeMountTargetCompartmentDetails = ChangeMountTargetCompartmentDetails.ChangeMountTargetCompartmentDetails;
const ClientOptions = __importStar(require("./client-options"));
exports.ClientOptions = ClientOptions.ClientOptions;
const CreateExportDetails = __importStar(require("./create-export-details"));
exports.CreateExportDetails = CreateExportDetails.CreateExportDetails;
const CreateFileSystemDetails = __importStar(require("./create-file-system-details"));
exports.CreateFileSystemDetails = CreateFileSystemDetails.CreateFileSystemDetails;
const CreateMountTargetDetails = __importStar(require("./create-mount-target-details"));
exports.CreateMountTargetDetails = CreateMountTargetDetails.CreateMountTargetDetails;
const CreateSnapshotDetails = __importStar(require("./create-snapshot-details"));
exports.CreateSnapshotDetails = CreateSnapshotDetails.CreateSnapshotDetails;
const Export = __importStar(require("./export"));
exports.Export = Export.Export;
const ExportSet = __importStar(require("./export-set"));
exports.ExportSet = ExportSet.ExportSet;
const ExportSetSummary = __importStar(require("./export-set-summary"));
exports.ExportSetSummary = ExportSetSummary.ExportSetSummary;
const ExportSummary = __importStar(require("./export-summary"));
exports.ExportSummary = ExportSummary.ExportSummary;
const FileSystem = __importStar(require("./file-system"));
exports.FileSystem = FileSystem.FileSystem;
const FileSystemSummary = __importStar(require("./file-system-summary"));
exports.FileSystemSummary = FileSystemSummary.FileSystemSummary;
const MountTarget = __importStar(require("./mount-target"));
exports.MountTarget = MountTarget.MountTarget;
const MountTargetSummary = __importStar(require("./mount-target-summary"));
exports.MountTargetSummary = MountTargetSummary.MountTargetSummary;
const Snapshot = __importStar(require("./snapshot"));
exports.Snapshot = Snapshot.Snapshot;
const SnapshotSummary = __importStar(require("./snapshot-summary"));
exports.SnapshotSummary = SnapshotSummary.SnapshotSummary;
const SourceDetails = __importStar(require("./source-details"));
exports.SourceDetails = SourceDetails.SourceDetails;
const UpdateExportDetails = __importStar(require("./update-export-details"));
exports.UpdateExportDetails = UpdateExportDetails.UpdateExportDetails;
const UpdateExportSetDetails = __importStar(require("./update-export-set-details"));
exports.UpdateExportSetDetails = UpdateExportSetDetails.UpdateExportSetDetails;
const UpdateFileSystemDetails = __importStar(require("./update-file-system-details"));
exports.UpdateFileSystemDetails = UpdateFileSystemDetails.UpdateFileSystemDetails;
const UpdateMountTargetDetails = __importStar(require("./update-mount-target-details"));
exports.UpdateMountTargetDetails = UpdateMountTargetDetails.UpdateMountTargetDetails;
const UpdateSnapshotDetails = __importStar(require("./update-snapshot-details"));
exports.UpdateSnapshotDetails = UpdateSnapshotDetails.UpdateSnapshotDetails;
