// oci-cloudguard/lib/request/index.js
"use strict";
/**
 * Cloud Guard APIs
 * A description of the Cloud Guard APIs
 * OpenAPI spec version: 20200131
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2021, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RequestSummarizedTrendResponderExecutionsRequest = exports.RequestSummarizedTrendProblemsRequest = exports.RequestSummarizedResponderExecutionsRequest = exports.RequestSummarizedProblemsRequest = exports.RequestSummarizedActivityProblemsRequest = exports.ListTargetsRequest = exports.ListTargetResponderRecipesRequest = exports.ListTargetResponderRecipeResponderRulesRequest = exports.ListTargetDetectorRecipesRequest = exports.ListTargetDetectorRecipeDetectorRulesRequest = exports.ListResponderRulesRequest = exports.ListResponderRecipesRequest = exports.ListResponderRecipeResponderRulesRequest = exports.ListResponderExecutionsRequest = exports.ListResponderActivitiesRequest = exports.ListResourceTypesRequest = exports.ListRecommendationsRequest = exports.ListProblemsRequest = exports.ListProblemHistoriesRequest = exports.ListPoliciesRequest = exports.ListManagedListsRequest = exports.ListManagedListTypesRequest = exports.ListImpactedResourcesRequest = exports.ListDetectorsRequest = exports.ListDetectorRulesRequest = exports.ListDetectorRecipesRequest = exports.ListDetectorRecipeDetectorRulesRequest = exports.ListDataMaskRulesRequest = exports.ListConditionMetadataTypesRequest = void 0;
const ListConditionMetadataTypesRequest = __importStar(require("./list-condition-metadata-types-request"));
exports.ListConditionMetadataTypesRequest = ListConditionMetadataTypesRequest.ListConditionMetadataTypesRequest;
const ListDataMaskRulesRequest = __importStar(require("./list-data-mask-rules-request"));
exports.ListDataMaskRulesRequest = ListDataMaskRulesRequest.ListDataMaskRulesRequest;
const ListDetectorRecipeDetectorRulesRequest = __importStar(require("./list-detector-recipe-detector-rules-request"));
exports.ListDetectorRecipeDetectorRulesRequest = ListDetectorRecipeDetectorRulesRequest.ListDetectorRecipeDetectorRulesRequest;
const ListDetectorRecipesRequest = __importStar(require("./list-detector-recipes-request"));
exports.ListDetectorRecipesRequest = ListDetectorRecipesRequest.ListDetectorRecipesRequest;
const ListDetectorRulesRequest = __importStar(require("./list-detector-rules-request"));
exports.ListDetectorRulesRequest = ListDetectorRulesRequest.ListDetectorRulesRequest;
const ListDetectorsRequest = __importStar(require("./list-detectors-request"));
exports.ListDetectorsRequest = ListDetectorsRequest.ListDetectorsRequest;
const ListImpactedResourcesRequest = __importStar(require("./list-impacted-resources-request"));
exports.ListImpactedResourcesRequest = ListImpactedResourcesRequest.ListImpactedResourcesRequest;
const ListManagedListTypesRequest = __importStar(require("./list-managed-list-types-request"));
exports.ListManagedListTypesRequest = ListManagedListTypesRequest.ListManagedListTypesRequest;
const ListManagedListsRequest = __importStar(require("./list-managed-lists-request"));
exports.ListManagedListsRequest = ListManagedListsRequest.ListManagedListsRequest;
const ListPoliciesRequest = __importStar(require("./list-policies-request"));
exports.ListPoliciesRequest = ListPoliciesRequest.ListPoliciesRequest;
const ListProblemHistoriesRequest = __importStar(require("./list-problem-histories-request"));
exports.ListProblemHistoriesRequest = ListProblemHistoriesRequest.ListProblemHistoriesRequest;
const ListProblemsRequest = __importStar(require("./list-problems-request"));
exports.ListProblemsRequest = ListProblemsRequest.ListProblemsRequest;
const ListRecommendationsRequest = __importStar(require("./list-recommendations-request"));
exports.ListRecommendationsRequest = ListRecommendationsRequest.ListRecommendationsRequest;
const ListResourceTypesRequest = __importStar(require("./list-resource-types-request"));
exports.ListResourceTypesRequest = ListResourceTypesRequest.ListResourceTypesRequest;
const ListResponderActivitiesRequest = __importStar(require("./list-responder-activities-request"));
exports.ListResponderActivitiesRequest = ListResponderActivitiesRequest.ListResponderActivitiesRequest;
const ListResponderExecutionsRequest = __importStar(require("./list-responder-executions-request"));
exports.ListResponderExecutionsRequest = ListResponderExecutionsRequest.ListResponderExecutionsRequest;
const ListResponderRecipeResponderRulesRequest = __importStar(require("./list-responder-recipe-responder-rules-request"));
exports.ListResponderRecipeResponderRulesRequest = ListResponderRecipeResponderRulesRequest.ListResponderRecipeResponderRulesRequest;
const ListResponderRecipesRequest = __importStar(require("./list-responder-recipes-request"));
exports.ListResponderRecipesRequest = ListResponderRecipesRequest.ListResponderRecipesRequest;
const ListResponderRulesRequest = __importStar(require("./list-responder-rules-request"));
exports.ListResponderRulesRequest = ListResponderRulesRequest.ListResponderRulesRequest;
const ListTargetDetectorRecipeDetectorRulesRequest = __importStar(require("./list-target-detector-recipe-detector-rules-request"));
exports.ListTargetDetectorRecipeDetectorRulesRequest = ListTargetDetectorRecipeDetectorRulesRequest.ListTargetDetectorRecipeDetectorRulesRequest;
const ListTargetDetectorRecipesRequest = __importStar(require("./list-target-detector-recipes-request"));
exports.ListTargetDetectorRecipesRequest = ListTargetDetectorRecipesRequest.ListTargetDetectorRecipesRequest;
const ListTargetResponderRecipeResponderRulesRequest = __importStar(require("./list-target-responder-recipe-responder-rules-request"));
exports.ListTargetResponderRecipeResponderRulesRequest = ListTargetResponderRecipeResponderRulesRequest.ListTargetResponderRecipeResponderRulesRequest;
const ListTargetResponderRecipesRequest = __importStar(require("./list-target-responder-recipes-request"));
exports.ListTargetResponderRecipesRequest = ListTargetResponderRecipesRequest.ListTargetResponderRecipesRequest;
const ListTargetsRequest = __importStar(require("./list-targets-request"));
exports.ListTargetsRequest = ListTargetsRequest.ListTargetsRequest;
const RequestSummarizedActivityProblemsRequest = __importStar(require("./request-summarized-activity-problems-request"));
exports.RequestSummarizedActivityProblemsRequest = RequestSummarizedActivityProblemsRequest.RequestSummarizedActivityProblemsRequest;
const RequestSummarizedProblemsRequest = __importStar(require("./request-summarized-problems-request"));
exports.RequestSummarizedProblemsRequest = RequestSummarizedProblemsRequest.RequestSummarizedProblemsRequest;
const RequestSummarizedResponderExecutionsRequest = __importStar(require("./request-summarized-responder-executions-request"));
exports.RequestSummarizedResponderExecutionsRequest = RequestSummarizedResponderExecutionsRequest.RequestSummarizedResponderExecutionsRequest;
const RequestSummarizedTrendProblemsRequest = __importStar(require("./request-summarized-trend-problems-request"));
exports.RequestSummarizedTrendProblemsRequest = RequestSummarizedTrendProblemsRequest.RequestSummarizedTrendProblemsRequest;
const RequestSummarizedTrendResponderExecutionsRequest = __importStar(require("./request-summarized-trend-responder-executions-request"));
exports.RequestSummarizedTrendResponderExecutionsRequest = RequestSummarizedTrendResponderExecutionsRequest.RequestSummarizedTrendResponderExecutionsRequest;
